<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 9 Association Mapping. | Quantitative Methods in Plant Breeding: The Digital Tutorials.</title>
  <meta name="description" content="This is the online tutorial booklet for the QMPB course run at NIAB. The online book was formed using R bookdown." />
  <meta name="generator" content="bookdown 0.21 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 9 Association Mapping. | Quantitative Methods in Plant Breeding: The Digital Tutorials." />
  <meta property="og:type" content="book" />
  
  <meta property="og:image" content="images/cover.jpg" />
  <meta property="og:description" content="This is the online tutorial booklet for the QMPB course run at NIAB. The online book was formed using R bookdown." />
  <meta name="github-repo" content="rstudio/bookdown" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 9 Association Mapping. | Quantitative Methods in Plant Breeding: The Digital Tutorials." />
  
  <meta name="twitter:description" content="This is the online tutorial booklet for the QMPB course run at NIAB. The online book was formed using R bookdown." />
  <meta name="twitter:image" content="images/cover.jpg" />

<meta name="author" content="NIAB 2021" />


<meta name="date" content="2021-03-01" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="Imputation.html"/>
<link rel="next" href="Mapping-Markers.html"/>
<script src="libs/header-attrs-2.6/header-attrs.js"></script>
<script src="libs/jquery-2.2.3/jquery.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />











<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">The QMPB digital tutorials</a></li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> Introduction</a></li>
<li class="chapter" data-level="2" data-path="Acknow.html"><a href="Acknow.html"><i class="fa fa-check"></i><b>2</b> Acknowledgements and Contents</a>
<ul>
<li class="chapter" data-level="2.1" data-path="Acknow.html"><a href="Acknow.html#acknowledgements"><i class="fa fa-check"></i><b>2.1</b> Acknowledgements</a></li>
<li class="chapter" data-level="2.2" data-path="Acknow.html"><a href="Acknow.html#digital-tutorial-chapters"><i class="fa fa-check"></i><b>2.2</b> Digital Tutorial Chapters:</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="R-and-stats.html"><a href="R-and-stats.html"><i class="fa fa-check"></i><b>3</b> R and Rstudio: a general introduction with statistics.</a>
<ul>
<li class="chapter" data-level="3.1" data-path="R-and-stats.html"><a href="R-and-stats.html#introduction-section-on-r"><i class="fa fa-check"></i><b>3.1</b> Introduction section on R</a>
<ul>
<li class="chapter" data-level="3.1.1" data-path="R-and-stats.html"><a href="R-and-stats.html#getting-the-data-ready"><i class="fa fa-check"></i><b>3.1.1</b> Getting the data ready</a></li>
</ul></li>
<li class="chapter" data-level="3.2" data-path="R-and-stats.html"><a href="R-and-stats.html#the-rstudio-layout-workspace-and-environment."><i class="fa fa-check"></i><b>3.2</b> The Rstudio layout, workspace and environment.</a>
<ul>
<li class="chapter" data-level="3.2.1" data-path="R-and-stats.html"><a href="R-and-stats.html#working-directory"><i class="fa fa-check"></i><b>3.2.1</b> Working directory</a></li>
</ul></li>
<li class="chapter" data-level="3.3" data-path="R-and-stats.html"><a href="R-and-stats.html#basic-r-syntx"><i class="fa fa-check"></i><b>3.3</b> Basic R Syntx</a>
<ul>
<li class="chapter" data-level="3.3.1" data-path="R-and-stats.html"><a href="R-and-stats.html#brackets"><i class="fa fa-check"></i><b>3.3.1</b> Brackets</a></li>
<li class="chapter" data-level="3.3.2" data-path="R-and-stats.html"><a href="R-and-stats.html#equals-symbols"><i class="fa fa-check"></i><b>3.3.2</b> Equals symbols</a></li>
<li class="chapter" data-level="3.3.3" data-path="R-and-stats.html"><a href="R-and-stats.html#continuation-character."><i class="fa fa-check"></i><b>3.3.3</b> Continuation character.</a></li>
<li class="chapter" data-level="3.3.4" data-path="R-and-stats.html"><a href="R-and-stats.html#text-and-numbers"><i class="fa fa-check"></i><b>3.3.4</b> Text and numbers</a></li>
<li class="chapter" data-level="3.3.5" data-path="R-and-stats.html"><a href="R-and-stats.html#summary-of-syntax"><i class="fa fa-check"></i><b>3.3.5</b> Summary of syntax</a></li>
</ul></li>
<li class="chapter" data-level="3.4" data-path="R-and-stats.html"><a href="R-and-stats.html#reading-in-data"><i class="fa fa-check"></i><b>3.4</b> Reading in data</a></li>
<li class="chapter" data-level="3.5" data-path="R-and-stats.html"><a href="R-and-stats.html#understanding-data-types-and-r-objects"><i class="fa fa-check"></i><b>3.5</b> Understanding data types and R objects</a>
<ul>
<li class="chapter" data-level="3.5.1" data-path="R-and-stats.html"><a href="R-and-stats.html#summary"><i class="fa fa-check"></i><b>3.5.1</b> Summary</a></li>
</ul></li>
<li class="chapter" data-level="3.6" data-path="R-and-stats.html"><a href="R-and-stats.html#summarising-and-displaying-data"><i class="fa fa-check"></i><b>3.6</b> Summarising and displaying data</a></li>
<li class="chapter" data-level="3.7" data-path="R-and-stats.html"><a href="R-and-stats.html#calculations-and-data-manipulation"><i class="fa fa-check"></i><b>3.7</b> Calculations and data manipulation</a>
<ul>
<li class="chapter" data-level="3.7.1" data-path="R-and-stats.html"><a href="R-and-stats.html#sorting-data"><i class="fa fa-check"></i><b>3.7.1</b> Sorting data</a></li>
<li class="chapter" data-level="3.7.2" data-path="R-and-stats.html"><a href="R-and-stats.html#relationships-between-variates"><i class="fa fa-check"></i><b>3.7.2</b> Relationships between variates</a></li>
<li class="chapter" data-level="3.7.3" data-path="R-and-stats.html"><a href="R-and-stats.html#exporting-your-script-and-workspace"><i class="fa fa-check"></i><b>3.7.3</b> Exporting your script and workspace</a></li>
<li class="chapter" data-level="3.7.4" data-path="R-and-stats.html"><a href="R-and-stats.html#saving-a-data-table"><i class="fa fa-check"></i><b>3.7.4</b> Saving a data table</a></li>
</ul></li>
<li class="chapter" data-level="3.8" data-path="R-and-stats.html"><a href="R-and-stats.html#basic-statistical-analysis"><i class="fa fa-check"></i><b>3.8</b> Basic statistical analysis</a>
<ul>
<li class="chapter" data-level="3.8.1" data-path="R-and-stats.html"><a href="R-and-stats.html#the-t-test"><i class="fa fa-check"></i><b>3.8.1</b> The t-test</a></li>
<li class="chapter" data-level="3.8.2" data-path="R-and-stats.html"><a href="R-and-stats.html#linear-regression"><i class="fa fa-check"></i><b>3.8.2</b> Linear Regression</a></li>
<li class="chapter" data-level="3.8.3" data-path="R-and-stats.html"><a href="R-and-stats.html#multiple-regression"><i class="fa fa-check"></i><b>3.8.3</b> Multiple regression</a></li>
<li class="chapter" data-level="3.8.4" data-path="R-and-stats.html"><a href="R-and-stats.html#the-analysis-of-variance"><i class="fa fa-check"></i><b>3.8.4</b> The analysis of variance</a></li>
</ul></li>
<li class="chapter" data-level="3.9" data-path="R-and-stats.html"><a href="R-and-stats.html#categorical-data---the-chi-squared-test"><i class="fa fa-check"></i><b>3.9</b> Categorical data - the <em>chi</em>-squared test</a></li>
<li class="chapter" data-level="3.10" data-path="R-and-stats.html"><a href="R-and-stats.html#more-useful-information"><i class="fa fa-check"></i><b>3.10</b> More useful information</a>
<ul>
<li class="chapter" data-level="3.10.1" data-path="R-and-stats.html"><a href="R-and-stats.html#plotting-graphs"><i class="fa fa-check"></i><b>3.10.1</b> Plotting Graphs</a></li>
<li class="chapter" data-level="3.10.2" data-path="R-and-stats.html"><a href="R-and-stats.html#probability-distributions"><i class="fa fa-check"></i><b>3.10.2</b> Probability distributions</a></li>
<li class="chapter" data-level="3.10.3" data-path="R-and-stats.html"><a href="R-and-stats.html#random-number-generation"><i class="fa fa-check"></i><b>3.10.3</b> Random number generation</a></li>
<li class="chapter" data-level="3.10.4" data-path="R-and-stats.html"><a href="R-and-stats.html#loops-second-example"><i class="fa fa-check"></i><b>3.10.4</b> Loops: <em>second example</em></a></li>
<li class="chapter" data-level="3.10.5" data-path="R-and-stats.html"><a href="R-and-stats.html#miscellany"><i class="fa fa-check"></i><b>3.10.5</b> Miscellany</a></li>
<li class="chapter" data-level="3.10.6" data-path="R-and-stats.html"><a href="R-and-stats.html#my-number-one-piece-of-advice"><i class="fa fa-check"></i><b>3.10.6</b> My number one piece of advice</a></li>
<li class="chapter" data-level="3.10.7" data-path="R-and-stats.html"><a href="R-and-stats.html#learn-more"><i class="fa fa-check"></i><b>3.10.7</b> Learn more</a></li>
<li class="chapter" data-level="3.10.8" data-path="R-and-stats.html"><a href="R-and-stats.html#packages"><i class="fa fa-check"></i><b>3.10.8</b> Packages</a></li>
<li class="chapter" data-level="3.10.9" data-path="R-and-stats.html"><a href="R-and-stats.html#the-tidyverse"><i class="fa fa-check"></i><b>3.10.9</b> The Tidyverse</a></li>
<li class="chapter" data-level="3.10.10" data-path="R-and-stats.html"><a href="R-and-stats.html#other-resources"><i class="fa fa-check"></i><b>3.10.10</b> Other resources</a></li>
</ul></li>
<li class="chapter" data-level="3.11" data-path="R-and-stats.html"><a href="R-and-stats.html#list-of-commands-mainly-described-in-this-guide."><i class="fa fa-check"></i><b>3.11</b> List of commands, mainly described in this guide.</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="Trial-design.html"><a href="Trial-design.html"><i class="fa fa-check"></i><b>4</b> Trial Design.</a>
<ul>
<li class="chapter" data-level="4.1" data-path="Trial-design.html"><a href="Trial-design.html#randomised-complete-block-designs"><i class="fa fa-check"></i><b>4.1</b> Randomised complete block designs</a></li>
<li class="chapter" data-level="4.2" data-path="Trial-design.html"><a href="Trial-design.html#trial-design-in-r"><i class="fa fa-check"></i><b>4.2</b> Trial design in R</a>
<ul>
<li class="chapter" data-level="4.2.1" data-path="Trial-design.html"><a href="Trial-design.html#partially-replicated-designs"><i class="fa fa-check"></i><b>4.2.1</b> Partially replicated designs</a></li>
<li class="chapter" data-level="4.2.2" data-path="Trial-design.html"><a href="Trial-design.html#two-dimensional-designs"><i class="fa fa-check"></i><b>4.2.2</b> Two-dimensional designs</a></li>
</ul></li>
<li class="chapter" data-level="4.3" data-path="Trial-design.html"><a href="Trial-design.html#farm-scale-experiments"><i class="fa fa-check"></i><b>4.3</b> Farm scale experiments</a></li>
<li class="chapter" data-level="4.4" data-path="Trial-design.html"><a href="Trial-design.html#alpha-designs."><i class="fa fa-check"></i><b>4.4</b> Alpha designs.</a>
<ul>
<li class="chapter" data-level="4.4.1" data-path="Trial-design.html"><a href="Trial-design.html#creation-of-augmented-and-p-rep-designs-from-2-replicate-alpha-designs"><i class="fa fa-check"></i><b>4.4.1</b> Creation of augmented and p-rep designs from 2-replicate alpha designs</a></li>
<li class="chapter" data-level="4.4.2" data-path="Trial-design.html"><a href="Trial-design.html#final-message"><i class="fa fa-check"></i><b>4.4.2</b> Final message</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="5" data-path="Trial-analysis.html"><a href="Trial-analysis.html"><i class="fa fa-check"></i><b>5</b> Trial analysis (sugar beet example).</a>
<ul>
<li class="chapter" data-level="5.1" data-path="Trial-analysis.html"><a href="Trial-analysis.html#the-layout-of-the-sugar-beet-trial"><i class="fa fa-check"></i><b>5.1</b> The layout of the sugar beet trial</a></li>
<li class="chapter" data-level="5.2" data-path="Trial-analysis.html"><a href="Trial-analysis.html#fixed-effects-random-effects-and-reml"><i class="fa fa-check"></i><b>5.2</b> Fixed effects, random effects and REML</a>
<ul>
<li class="chapter" data-level="5.2.1" data-path="Trial-analysis.html"><a href="Trial-analysis.html#preparing-the-data"><i class="fa fa-check"></i><b>5.2.1</b> Preparing the data</a></li>
<li class="chapter" data-level="5.2.2" data-path="Trial-analysis.html"><a href="Trial-analysis.html#randomised-complete-block-design"><i class="fa fa-check"></i><b>5.2.2</b> Randomised complete block design</a></li>
<li class="chapter" data-level="5.2.3" data-path="Trial-analysis.html"><a href="Trial-analysis.html#randomised-compelete-block-design---with-mixed-models"><i class="fa fa-check"></i><b>5.2.3</b> Randomised compelete block design - with mixed models</a></li>
<li class="chapter" data-level="5.2.4" data-path="Trial-analysis.html"><a href="Trial-analysis.html#plotting-residuals-against-field-structure."><i class="fa fa-check"></i><b>5.2.4</b> Plotting residuals against field structure.</a></li>
<li class="chapter" data-level="5.2.5" data-path="Trial-analysis.html"><a href="Trial-analysis.html#alpha-design-include-block-structure"><i class="fa fa-check"></i><b>5.2.5</b> Alpha design (include block structure)</a></li>
<li class="chapter" data-level="5.2.6" data-path="Trial-analysis.html"><a href="Trial-analysis.html#rows-and-column-analysis"><i class="fa fa-check"></i><b>5.2.6</b> Rows and column analysis</a></li>
</ul></li>
<li class="chapter" data-level="5.3" data-path="Trial-analysis.html"><a href="Trial-analysis.html#spatial-analysis"><i class="fa fa-check"></i><b>5.3</b> Spatial analysis</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="Cross-site-analysis.html"><a href="Cross-site-analysis.html"><i class="fa fa-check"></i><b>6</b> Cross sites analysis in R.</a>
<ul>
<li class="chapter" data-level="6.1" data-path="Cross-site-analysis.html"><a href="Cross-site-analysis.html#blues-blups-and-one-stage-analysis"><i class="fa fa-check"></i><b>6.1</b> BLUEs, BLUPs and one stage analysis</a>
<ul>
<li class="chapter" data-level="6.1.1" data-path="Cross-site-analysis.html"><a href="Cross-site-analysis.html#one-stage-analysis-with-weights"><i class="fa fa-check"></i><b>6.1.1</b> One stage analysis with weights</a></li>
</ul></li>
<li class="chapter" data-level="6.2" data-path="Cross-site-analysis.html"><a href="Cross-site-analysis.html#weighted-two-stage-analysis"><i class="fa fa-check"></i><b>6.2</b> Weighted two-stage analysis</a></li>
<li class="chapter" data-level="6.3" data-path="Cross-site-analysis.html"><a href="Cross-site-analysis.html#ammi"><i class="fa fa-check"></i><b>6.3</b> AMMI</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="Population-genetics.html"><a href="Population-genetics.html"><i class="fa fa-check"></i><b>7</b> Population genetics.</a>
<ul>
<li class="chapter" data-level="7.1" data-path="Population-genetics.html"><a href="Population-genetics.html#hardy-weinberg-equilibrium-and-population-structure"><i class="fa fa-check"></i><b>7.1</b> Hardy-Weinberg equilibrium and population structure</a></li>
<li class="chapter" data-level="7.2" data-path="Population-genetics.html"><a href="Population-genetics.html#other-ways-of-analysing-population-structure"><i class="fa fa-check"></i><b>7.2</b> Other ways of analysing population structure</a>
<ul>
<li class="chapter" data-level="7.2.1" data-path="Population-genetics.html"><a href="Population-genetics.html#distance-trees"><i class="fa fa-check"></i><b>7.2.1</b> Distance Trees</a></li>
</ul></li>
<li class="chapter" data-level="7.3" data-path="Population-genetics.html"><a href="Population-genetics.html#structure"><i class="fa fa-check"></i><b>7.3</b> Structure</a>
<ul>
<li class="chapter" data-level="7.3.1" data-path="Population-genetics.html"><a href="Population-genetics.html#what-structure-does"><i class="fa fa-check"></i><b>7.3.1</b> What STRUCTURE does</a></li>
<li class="chapter" data-level="7.3.2" data-path="Population-genetics.html"><a href="Population-genetics.html#limitations-of-structure"><i class="fa fa-check"></i><b>7.3.2</b> Limitations of STRUCTURE</a></li>
<li class="chapter" data-level="7.3.3" data-path="Population-genetics.html"><a href="Population-genetics.html#using-structure"><i class="fa fa-check"></i><b>7.3.3</b> Using Structure</a></li>
<li class="chapter" data-level="7.3.4" data-path="Population-genetics.html"><a href="Population-genetics.html#the-exercise"><i class="fa fa-check"></i><b>7.3.4</b> The exercise</a></li>
</ul></li>
<li class="chapter" data-level="7.4" data-path="Population-genetics.html"><a href="Population-genetics.html#linkage-disequilibrium-analysis"><i class="fa fa-check"></i><b>7.4</b> Linkage disequilibrium analysis</a>
<ul>
<li class="chapter" data-level="7.4.1" data-path="Population-genetics.html"><a href="Population-genetics.html#linkage-disequilibrium-in-the-teosinte-and-barley-data-sets"><i class="fa fa-check"></i><b>7.4.1</b> Linkage disequilibrium in the teosinte and barley data sets</a></li>
</ul></li>
<li class="chapter" data-level="7.5" data-path="Population-genetics.html"><a href="Population-genetics.html#recombination-and-linkage-disequilibrium-applications"><i class="fa fa-check"></i><b>7.5</b> Recombination and linkage disequilibrium: applications</a></li>
<li class="chapter" data-level="7.6" data-path="Population-genetics.html"><a href="Population-genetics.html#pedigree-analysis"><i class="fa fa-check"></i><b>7.6</b> Pedigree analysis</a></li>
</ul></li>
<li class="chapter" data-level="8" data-path="Imputation.html"><a href="Imputation.html"><i class="fa fa-check"></i><b>8</b> Imputation.</a>
<ul>
<li class="chapter" data-level="8.1" data-path="Imputation.html"><a href="Imputation.html#introduction"><i class="fa fa-check"></i><b>8.1</b> Introduction</a></li>
<li class="chapter" data-level="8.2" data-path="Imputation.html"><a href="Imputation.html#the-exercise-1"><i class="fa fa-check"></i><b>8.2</b> The exercise</a></li>
<li class="chapter" data-level="8.3" data-path="Imputation.html"><a href="Imputation.html#data-preparation-for-all-methods"><i class="fa fa-check"></i><b>8.3</b> Data preparation (for all methods)</a></li>
<li class="chapter" data-level="8.4" data-path="Imputation.html"><a href="Imputation.html#svd-imputation"><i class="fa fa-check"></i><b>8.4</b> SVD imputation</a></li>
<li class="chapter" data-level="8.5" data-path="Imputation.html"><a href="Imputation.html#mean-imputation"><i class="fa fa-check"></i><b>8.5</b> Mean imputation</a></li>
<li class="chapter" data-level="8.6" data-path="Imputation.html"><a href="Imputation.html#random-forest-imputation"><i class="fa fa-check"></i><b>8.6</b> Random Forest Imputation</a></li>
<li class="chapter" data-level="8.7" data-path="Imputation.html"><a href="Imputation.html#summary-1"><i class="fa fa-check"></i><b>8.7</b> Summary</a></li>
<li class="chapter" data-level="8.8" data-path="Imputation.html"><a href="Imputation.html#conclusions-and-comments"><i class="fa fa-check"></i><b>8.8</b> Conclusions and comments</a></li>
</ul></li>
<li class="chapter" data-level="9" data-path="Association-mapping.html"><a href="Association-mapping.html"><i class="fa fa-check"></i><b>9</b> Association Mapping.</a>
<ul>
<li class="chapter" data-level="9.1" data-path="Association-mapping.html"><a href="Association-mapping.html#introduction-1"><i class="fa fa-check"></i><b>9.1</b> Introduction</a></li>
<li class="chapter" data-level="9.2" data-path="Association-mapping.html"><a href="Association-mapping.html#preparing-the-data-set"><i class="fa fa-check"></i><b>9.2</b> Preparing the data set</a></li>
<li class="chapter" data-level="9.3" data-path="Association-mapping.html"><a href="Association-mapping.html#running-gwaspoly"><i class="fa fa-check"></i><b>9.3</b> Running GWASpoly</a></li>
<li class="chapter" data-level="9.4" data-path="Association-mapping.html"><a href="Association-mapping.html#the-exercise-2"><i class="fa fa-check"></i><b>9.4</b> The exercise</a>
<ul>
<li class="chapter" data-level="9.4.1" data-path="Association-mapping.html"><a href="Association-mapping.html#calculation-and-comparison-of-kinship"><i class="fa fa-check"></i><b>9.4.1</b> Calculation and comparison of kinship</a></li>
<li class="chapter" data-level="9.4.2" data-path="Association-mapping.html"><a href="Association-mapping.html#gwas"><i class="fa fa-check"></i><b>9.4.2</b> GWAS</a></li>
<li class="chapter" data-level="9.4.3" data-path="Association-mapping.html"><a href="Association-mapping.html#analysis-with-covariates"><i class="fa fa-check"></i><b>9.4.3</b> Analysis with covariates</a></li>
<li class="chapter" data-level="9.4.4" data-path="Association-mapping.html"><a href="Association-mapping.html#accounting-for-population-structure"><i class="fa fa-check"></i><b>9.4.4</b> Accounting for population structure</a></li>
<li class="chapter" data-level="9.4.5" data-path="Association-mapping.html"><a href="Association-mapping.html#further-analysis"><i class="fa fa-check"></i><b>9.4.5</b> Further analysis</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="10" data-path="Mapping-Markers.html"><a href="Mapping-Markers.html"><i class="fa fa-check"></i><b>10</b> Mapping Markers and Quantitative Trait Loci.</a>
<ul>
<li class="chapter" data-level="10.1" data-path="Mapping-Markers.html"><a href="Mapping-Markers.html#introduction-2"><i class="fa fa-check"></i><b>10.1</b> Introduction</a></li>
<li class="chapter" data-level="10.2" data-path="Mapping-Markers.html"><a href="Mapping-Markers.html#constructing-a-genetic-map"><i class="fa fa-check"></i><b>10.2</b> Constructing a genetic map</a></li>
<li class="chapter" data-level="10.3" data-path="Mapping-Markers.html"><a href="Mapping-Markers.html#formatting-and-reading-in-the-data"><i class="fa fa-check"></i><b>10.3</b> Formatting and reading in the data</a></li>
<li class="chapter" data-level="10.4" data-path="Mapping-Markers.html"><a href="Mapping-Markers.html#a-quick-map"><i class="fa fa-check"></i><b>10.4</b> A quick map!</a></li>
<li class="chapter" data-level="10.5" data-path="Mapping-Markers.html"><a href="Mapping-Markers.html#estimating-recombination-fractions"><i class="fa fa-check"></i><b>10.5</b> Estimating recombination fractions</a></li>
<li class="chapter" data-level="10.6" data-path="Mapping-Markers.html"><a href="Mapping-Markers.html#forming-linkage-groups"><i class="fa fa-check"></i><b>10.6</b> Forming Linkage groups</a></li>
<li class="chapter" data-level="10.7" data-path="Mapping-Markers.html"><a href="Mapping-Markers.html#marker-ordering"><i class="fa fa-check"></i><b>10.7</b> Marker ordering</a></li>
<li class="chapter" data-level="10.8" data-path="Mapping-Markers.html"><a href="Mapping-Markers.html#improving-the-map"><i class="fa fa-check"></i><b>10.8</b> Improving the map</a></li>
<li class="chapter" data-level="10.9" data-path="Mapping-Markers.html"><a href="Mapping-Markers.html#quality-control-commands-for-marker-qc-and-mapping"><i class="fa fa-check"></i><b>10.9</b> Quality Control commands for marker QC and mapping</a></li>
<li class="chapter" data-level="10.10" data-path="Mapping-Markers.html"><a href="Mapping-Markers.html#asmap"><i class="fa fa-check"></i><b>10.10</b> ASMap</a></li>
</ul></li>
<li class="chapter" data-level="11" data-path="Data-Visualization.html"><a href="Data-Visualization.html"><i class="fa fa-check"></i><b>11</b> Data Visualization.</a>
<ul>
<li class="chapter" data-level="11.1" data-path="Data-Visualization.html"><a href="Data-Visualization.html#flapjack"><i class="fa fa-check"></i><b>11.1</b> Flapjack</a>
<ul>
<li class="chapter" data-level="11.1.1" data-path="Data-Visualization.html"><a href="Data-Visualization.html#genotype-data-.geno-file."><i class="fa fa-check"></i><b>11.1.1</b> Genotype data: “.geno” file.</a></li>
<li class="chapter" data-level="11.1.2" data-path="Data-Visualization.html"><a href="Data-Visualization.html#map-data-.map-file."><i class="fa fa-check"></i><b>11.1.2</b> Map data : “.MAP” file.</a></li>
<li class="chapter" data-level="11.1.3" data-path="Data-Visualization.html"><a href="Data-Visualization.html#trait-data"><i class="fa fa-check"></i><b>11.1.3</b> Trait data</a></li>
<li class="chapter" data-level="11.1.4" data-path="Data-Visualization.html"><a href="Data-Visualization.html#the-exercise-3"><i class="fa fa-check"></i><b>11.1.4</b> The Exercise</a></li>
</ul></li>
<li class="chapter" data-level="11.2" data-path="Data-Visualization.html"><a href="Data-Visualization.html#helium-pedigree-visualization-software"><i class="fa fa-check"></i><b>11.2</b> Helium pedigree visualization software</a>
<ul>
<li class="chapter" data-level="11.2.1" data-path="Data-Visualization.html"><a href="Data-Visualization.html#introduction-3"><i class="fa fa-check"></i><b>11.2.1</b> Introduction</a></li>
<li class="chapter" data-level="11.2.2" data-path="Data-Visualization.html"><a href="Data-Visualization.html#the-exercise-4"><i class="fa fa-check"></i><b>11.2.2</b> The exercise</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="12" data-path="selection-theory.html"><a href="selection-theory.html"><i class="fa fa-check"></i><b>12</b> Selection theory and breeding program design.</a></li>
<li class="chapter" data-level="13" data-path="Genomic-prediction.html"><a href="Genomic-prediction.html"><i class="fa fa-check"></i><b>13</b> Genomic prediction.</a>
<ul>
<li class="chapter" data-level="13.1" data-path="Genomic-prediction.html"><a href="Genomic-prediction.html#ridge-regression-and-genomic-prediction-in-excel"><i class="fa fa-check"></i><b>13.1</b> Ridge regression and Genomic Prediction in Excel</a>
<ul>
<li class="chapter" data-level="13.1.1" data-path="Genomic-prediction.html"><a href="Genomic-prediction.html#introduction-4"><i class="fa fa-check"></i><b>13.1.1</b> Introduction</a></li>
<li class="chapter" data-level="13.1.2" data-path="Genomic-prediction.html"><a href="Genomic-prediction.html#the-exercise-5"><i class="fa fa-check"></i><b>13.1.2</b> The Exercise</a></li>
</ul></li>
<li class="chapter" data-level="13.2" data-path="Genomic-prediction.html"><a href="Genomic-prediction.html#genomic-prediction-with-rrblup"><i class="fa fa-check"></i><b>13.2</b> Genomic Prediction with rrBLUP</a>
<ul>
<li class="chapter" data-level="13.2.1" data-path="Genomic-prediction.html"><a href="Genomic-prediction.html#introduction-5"><i class="fa fa-check"></i><b>13.2.1</b> Introduction</a></li>
<li class="chapter" data-level="13.2.2" data-path="Genomic-prediction.html"><a href="Genomic-prediction.html#the-dataset"><i class="fa fa-check"></i><b>13.2.2</b> The dataset</a></li>
<li class="chapter" data-level="13.2.3" data-path="Genomic-prediction.html"><a href="Genomic-prediction.html#loading-and-preparing-the-data"><i class="fa fa-check"></i><b>13.2.3</b> Loading and preparing the data</a></li>
<li class="chapter" data-level="13.2.4" data-path="Genomic-prediction.html"><a href="Genomic-prediction.html#examining-the-data-structure"><i class="fa fa-check"></i><b>13.2.4</b> Examining the data structure</a></li>
<li class="chapter" data-level="13.2.5" data-path="Genomic-prediction.html"><a href="Genomic-prediction.html#ordinary-least-squares"><i class="fa fa-check"></i><b>13.2.5</b> Ordinary least squares</a></li>
<li class="chapter" data-level="13.2.6" data-path="Genomic-prediction.html"><a href="Genomic-prediction.html#rrblup"><i class="fa fa-check"></i><b>13.2.6</b> rrBLUP</a></li>
<li class="chapter" data-level="13.2.7" data-path="Genomic-prediction.html"><a href="Genomic-prediction.html#prediction-forward-in-time"><i class="fa fa-check"></i><b>13.2.7</b> Prediction forward in time</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="14" data-path="Breeding-simulations.html"><a href="Breeding-simulations.html"><i class="fa fa-check"></i><b>14</b> Breeding simulations.</a></li>
<li class="chapter" data-level="" data-path="references.html"><a href="references.html"><i class="fa fa-check"></i>References</a></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Quantitative Methods in Plant Breeding: The Digital Tutorials.</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="Association-mapping" class="section level1" number="9">
<h1><span class="header-section-number">Chapter 9</span> Association Mapping.</h1>
<div id="introduction-1" class="section level2" number="9.1">
<h2><span class="header-section-number">9.1</span> Introduction</h2>
<p><em>Association Mapping with GWASpoly</em></p>
<p>We are going to use the R package <code>GWASpoly</code> to carry out association mapping using the mixed model. We shall analyse the <a href="http://www.triticeaegenome.eu">TriticeaeGenome data set</a> which we have used previously in this course. To recap: it was a European collaborative project which developed a panel of 384 UK, French and German winter wheat varieties. Here we are going to use the variety means across trials for a range of phenotypes together with a genome-wide set of DArT markers for association mapping.</p>
<p><code>GWASpoly</code> is an R package from the Endelman group in the University of Wisconsin. The group is more noted for the well-established package <code>rrBLUP</code> for genomic prediction. You can see the group’s software on their <a href="http://potatobreeding.cals.wisc.edu/software/">website</a>. The installation process for GWASpoly has recently changed. You use to be able to download from their website. However, the package is now hosted on <a href="https://github.com/jendelman/GWASpoly">GitHub</a>. If you carry on working with <code>R</code> you will find that a number of packages are hosted on GitHub. The only real difference between a package hosted on Github and through CRAN that we need to worry about for now is the installation process. <code>install.packages()</code> will not work in this case. Instead we will use the <code>devtools</code> package to install from Github:</p>
<div class="sourceCode" id="cb465"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb465-1"><a href="Association-mapping.html#cb465-1" aria-hidden="true" tabindex="-1"></a><span class="co">#install devtools</span></span>
<span id="cb465-2"><a href="Association-mapping.html#cb465-2" aria-hidden="true" tabindex="-1"></a><span class="fu">install.packages</span>(<span class="st">&quot;devtools&quot;</span>) </span>
<span id="cb465-3"><a href="Association-mapping.html#cb465-3" aria-hidden="true" tabindex="-1"></a><span class="co">#use devtools function to install from Github</span></span>
<span id="cb465-4"><a href="Association-mapping.html#cb465-4" aria-hidden="true" tabindex="-1"></a><span class="co"># you may be asked to update some packages, just follow the promts in the console</span></span>
<span id="cb465-5"><a href="Association-mapping.html#cb465-5" aria-hidden="true" tabindex="-1"></a>devtools<span class="sc">::</span><span class="fu">install_github</span>(<span class="st">&quot;jendelman/GWASpoly&quot;</span>)</span></code></pre></div>
<p>That should of installed <code>GWASpoly</code> and all the required associated packages. We should load the package and test it’s installed ok (there should be no error messages):</p>
<div class="sourceCode" id="cb466"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb466-1"><a href="Association-mapping.html#cb466-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(GWASpoly)</span></code></pre></div>
<p>GWASpoly is relatively straightforward to use. Its unique selling point is that it works on autopolyploid crops too. For these, the form of the kinship matrix changes, and this is accounted for. The main weakness of <code>GWASpoly</code> is that is can only handle bi-allelic markers, thus making it less useful for haplotype analysis. Several other GWAS packages are available for use in plants, notably <a href="https://www.maizegenetics.net/tassel,%20which%20can%20deal%20with%20multiple%20alleles">TASSEL</a> and GAPIT (<a href="http://www.zzlab.net/GAPIT/" class="uri">http://www.zzlab.net/GAPIT/</a>), both originally developed in the Buckler lab at Cornell. More recently, the authors of GAPIT have released an improvement over GAPIT called <a href="http://www.zzlab.net/FarmCPU">FarmCPU</a>, which incorporates recent methodological advances to reduce the number of false negatives which occur as a result of the adjustment for kinship and population structure. <code>GAPIT</code> and <code>FarmCPU</code> are both R packages.</p>
<p>For <code>GWASpoly</code>, there is a vignette available <a href="www.jendelman.github.io/GWASpoly/GWASpoly.html">here</a>, which is worth reading at some stage.</p>
</div>
<div id="preparing-the-data-set" class="section level2" number="9.2">
<h2><span class="header-section-number">9.2</span> Preparing the data set</h2>
<p>The complete TG data set is found in the file “TG data for GWAS.poly.xlsx.” For reading into <code>GWASpoly</code>, two data files are required: a phenotype file and a genotype file which includes map co-ordinates. Here, phenotype data are in the file “TG_phens.csv.” The first column must be the accession names. The next set of columns contains data on the traits to be analysed. There is then an optional final set of columns which gives covariates to be included in the analysis. In the TriticeaeGenome data there are six covariates:</p>
<ol style="list-style-type: decimal">
<li>PCR marker for the Rht2 dwarfing allele</li>
<li>PCR marker for the Ppd flowering time insensitivity allele</li>
<li>Year of registration of the variety</li>
<li>Country of variety origin (DEU, FRA, GBR)</li>
<li>2 columns for country 0/1 “DEU,” “FRA”</li>
</ol>
<p>Genotype data for the GWAS are in the file “TGgeno.csv.” Data here are DArT markers, coded 0/1(this is a data set of inbred lines). The data are transposed compared to the phenotypic data: variety names are in the first row rather than the first column. The first column here is the marker name, the second the chromosome number and the third the position of the marker in cM. As this is wheat, the chromosomes are numbered consecutively 1…21 for chromosomes 1A, 1B, 1D,2A, 2B…, 7B, 7D , Chromosome 22 is of SNPs provided by BioGemma. Chromosome 23 is of unmapped DArT markers, and chromosome 24 is a small number of diagnostic markers for known QTL. Subsequent columns contain the marker data.</p>
<p>In preparing the data for analysis, markers with a minor allele frequency (maf) &lt;4% were deleted. Most association mapping studies delete loci with low maf as they are unlikely, on their own, to detect any association, even if they are functional. The threshold is somewhat subjective. In the current case, &lt;4% means less than 15 copies of the rare allele, which seems reasonable: think of doing a t-test where one group has fewer than 15 samples. The package itself, however, sets the minimum maf for inclusion in the analysis to 0.05, so we shall work with that. The maf threshold can be changed, however. Markers with &gt;10% missing data were also deleted, as they are likely to be of poor quality.</p>
<p>The sheet “TGkin” contains a subset of the markers to use for kinship calculation. Markers are frequently unevenly distributed across plant genomes. If kinship matrices are dominated by dense clusters of markers, then any operations upon these matrices, such as PCO and tree construction, will be biased towards these markers. So a tree, for example, will not necessarily reflect average relationships over the genome, but the relationships among varieties in the genomic regions which have the highest number of markers. This is particularly a problem if, as is often the case, the dense clusters of markers are also in high LD. Such clusters are frequently found in crop species where non-recombining genomic regions from related species have been introgressed by plant breeders. For this reason, it is good practice to thin the markers to give a more uniform distribution over the whole genome. One way of doing this is to calculate the correlation coefficient among all markers and delete one marker of each pair in high LD (say r&gt;0.9). Another is to delete markers if they are too closely linked. There is a fine balancing act in thinning markers. If markers are not uniformly distributed over the genome, then marker-trait associations will be over-corrected in some parts of the genome and under-corrected in others. If too few markers are left, then kinship will not be accurately estimated and adjusted for. For our data, we shall use only the mapped markers for kinship estimation, deleting all but one of any markers which map to the same position. This left 1,044 markers (from 2686 before thinning) selected for kinship calculation.</p>
<p>For some reason, no packages we are aware of provide this option of thinning markers selectively by correlation coefficient or by map position, despite the necessity of doing so. There are exceptions: <span class="citation"><a href="references.html#ref-speed2012" role="doc-biblioref">Speed et al.</a> (<a href="references.html#ref-speed2012" role="doc-biblioref">2012</a>)</span> developed an approach to calculating a kinship matrix which takes into account the clustering of markers without the need for thinning markers.</p>
</div>
<div id="running-gwaspoly" class="section level2" number="9.3">
<h2><span class="header-section-number">9.3</span> Running GWASpoly</h2>

<div class="rmdnote">
A script has been provided with this tutorial. You can access it from the data folder for this Chapter. Open the script, load the GWASpoly package and set the working directory to the directory containing the tutorial data. You can also copy from the tutorial as we’ve done in the other tutorials.
</div>
<p>The command to read in the data is <code>read.GWASpoly()</code>. Various data formats are available. Markers can be coded in AB format (common in linkage analysis), as ACGT nucleotides, or numeric; 0, 1, 2, 3, 4 for a tetraploid for instance. As this is an R package, missing data are coded as <code>NA</code>. GWASpoly only works with SNPs or binary markers. To analyse SSRs or haplotype data, each allele could be tested for association independently, possibly with some amalgamation of the results at the end, but this is an unnecessary complication for demonstration purposes. If you have a haplotype data set, using TASSEL is a better option.</p>
<p>Read in the data as follows. In my current working directory I have a subfolder called data, where all the data files are kept. Therefore, I start each file name with “data/,” you will need to take this out if you copy the below:</p>
<div class="sourceCode" id="cb467"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb467-1"><a href="Association-mapping.html#cb467-1" aria-hidden="true" tabindex="-1"></a>TGall<span class="ot">&lt;-</span><span class="fu">read.GWASpoly</span>(<span class="at">ploidy=</span><span class="dv">1</span>,<span class="st">&quot;data/TGphens.csv&quot;</span>,<span class="st">&quot;data/TGgeno.csv&quot;</span>,<span class="st">&quot;numeric&quot;</span>,<span class="at">n.traits=</span><span class="dv">18</span>,<span class="at">delim=</span><span class="st">&quot;,&quot;</span>)</span></code></pre></div>
<pre><code>## Number of polymorphic markers: 2686 
## Missing marker data imputed with population mode 
## N = 376 individuals with phenotypic and genotypic information 
## Detected following fixed effects:
## Rht2_400
## PpdD1_297
## YEAR
## COUNTRY
## FRA
## DEU
## Detected following traits:
## YLD_ALL_BLUP
## CALLOW_2011
## FRANCE_2010
## FRANCE_2011
## LGE_2010
## LGE_2011
## NIAB_2011
## FT
## HT
## AWNS
## EARS_M2
## LODG
## MAT
## PROT
## SP_WT
## TGW
## TILLERS
## WINT_KILL</code></pre>
<p>Once again, we are treating inbred wheat lines as haploids here as also reflected in the numerically coded data, where we have 0 or 1 copy of an allele. Note that we had to declare the number of traits. The extra columns in the input file will be considered (correctly) as covariates in the analysis by GWASpoly. Finally we have declared the field separator to be a ‘,’ as we are reading a csv file. For other options use <code>help</code>.</p>
<p>On reading in the data you will see a summary of the data structure, now all stored in <code>TGall</code>. Notice that the output says <code>Missing marker data imputed with population mode</code>. We had some missing marker data but GWASpoly requires complete data so has imputed the missing values on data input using “population mode.” This substitutes missing data by the most frequent genotype class at each marker. This is not the best method. Ideally, we would impute missing data by a more accurate method prior to analysis (see the imputation tutorial).</p>
<p>We also need to read in the version of the data with just the markers required for kinship calculation.</p>
<div class="sourceCode" id="cb469"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb469-1"><a href="Association-mapping.html#cb469-1" aria-hidden="true" tabindex="-1"></a><span class="co">#take out the &quot;data/&quot; if you copy</span></span>
<span id="cb469-2"><a href="Association-mapping.html#cb469-2" aria-hidden="true" tabindex="-1"></a>TGkin<span class="ot">&lt;-</span><span class="fu">read.GWASpoly</span>(<span class="at">ploidy=</span><span class="dv">1</span>,<span class="st">&quot;data/TGphens.csv&quot;</span>,<span class="st">&quot;data/TGkin.csv&quot;</span>,<span class="st">&quot;numeric&quot;</span>,<span class="at">n.traits=</span><span class="dv">18</span>,<span class="at">delim=</span><span class="st">&quot;,&quot;</span>)</span></code></pre></div>
<p>Note, we have 1044 markers in the kinship dataset.</p>
</div>
<div id="the-exercise-2" class="section level2" number="9.4">
<h2><span class="header-section-number">9.4</span> The exercise</h2>
<p>The aim of the exercise is to familiarise yourselves with the package and compare the results obtained when:</p>
<ol style="list-style-type: lower-alpha">
<li>Using different kinship adjustments, or without including kinship at all</li>
<li>Including known trait covariates (e.g. Rht, Ppd loci)</li>
<li>Using different population structure adjustments, or without including population structure at all.</li>
</ol>
<div id="calculation-and-comparison-of-kinship" class="section level3" number="9.4.1">
<h3><span class="header-section-number">9.4.1</span> Calculation and comparison of kinship</h3>
<p>To estimate kinship in GWASpoly, we simply type:</p>
<div class="sourceCode" id="cb470"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb470-1"><a href="Association-mapping.html#cb470-1" aria-hidden="true" tabindex="-1"></a>TGall<span class="ot">&lt;-</span><span class="fu">set.K</span>(TGall)  </span>
<span id="cb470-2"><a href="Association-mapping.html#cb470-2" aria-hidden="true" tabindex="-1"></a>TGkin<span class="ot">&lt;-</span><span class="fu">set.K</span>(TGkin) </span></code></pre></div>
<p>Each version of the data has now had a kinship matrix calculated from its associated marker set. We’ll see later how to specify different kinship matrices to be used. There are multiple forms of kinship matrices. GWASpoly centres each marker (so the mean of each marker is zero and the variance is one). If the centred data are in a matrix M, the kinship matrix is then MMt. The kinship matrix is also scaled so that the mean of the diagonal elements are one. This is the VanRaden kinship matrix <span class="citation">(<a href="references.html#ref-vanraden_efficient_2008" role="doc-biblioref">VanRaden 2008</a>)</span>, which is in common use. Alternatively, you can supply your own matrix (we’ll see how shortly). MtM/n would be the variance/covariance matrix among the n markers if M was centred, and would be the correlation matrix if each marker was initially adjusted to have a variance of one too.</p>
<p>The kinship matrix is stored in a ‘slot’ in the GWASpoly object storing the data. Slots are accessed using <code>@</code> just as named vectors of data accessed using <code>$</code> Thus:</p>
<div class="sourceCode" id="cb471"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb471-1"><a href="Association-mapping.html#cb471-1" aria-hidden="true" tabindex="-1"></a>TGall<span class="sc">@</span>K[<span class="dv">1</span><span class="sc">:</span><span class="dv">5</span>,<span class="dv">1</span><span class="sc">:</span><span class="dv">5</span>]</span></code></pre></div>
<pre><code>##               AARDEN    AARDVARK       ABELE         ABO     ACCESS
## AARDEN    0.89425094  0.06820056 -0.08325033 -0.17401122  0.1904552
## AARDVARK  0.06820056  0.98608975  0.04025471 -0.03429426 -0.0102782
## ABELE    -0.08325033  0.04025471  1.16491365 -0.03983785  0.1856692
## ABO      -0.17401122 -0.03429426 -0.03983785  1.30763030 -0.1575344
## ACCESS    0.19045520 -0.01027820  0.18566924 -0.15753443  1.0314240</code></pre>
<p>This shows the top left hand corner of the matrix made from all markers. Compare to <code>TGkin</code>.</p>
<p>These kinship matrices are essentially measuring the genetic distance between varieties using different marker sets. So, as you now know from the Population Genetics exercise, we can now visualise the genetic distances in a PCoA – principal coordinate analysis. The standard R function for spectral decomposition of a matrix is <code>eigen()</code>, which produces eigenvalues and vectors. We can plot the eigenvalues for both data sets:</p>
<div class="sourceCode" id="cb473"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb473-1"><a href="Association-mapping.html#cb473-1" aria-hidden="true" tabindex="-1"></a><span class="fu">plot</span>(<span class="fu">eigen</span>(TGall<span class="sc">@</span>K)<span class="sc">$</span>values)          </span></code></pre></div>
<p><img src="bookdown-demo_files/figure-html/unnamed-chunk-351-1.png" width="672" /></p>
<div class="sourceCode" id="cb474"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb474-1"><a href="Association-mapping.html#cb474-1" aria-hidden="true" tabindex="-1"></a><span class="fu">plot</span>(<span class="fu">eigen</span>(TGkin<span class="sc">@</span>K)<span class="sc">$</span>values)</span></code></pre></div>
<p><img src="bookdown-demo_files/figure-html/unnamed-chunk-351-2.png" width="672" /></p>
<p>You should see that in both cases, the first few eigenvalues are much larger than the remainder: they therefore account for most of the variation in their respective matrices. We can now plot the largest and second largest eigenvectors against each other for both data sets:</p>
<div class="sourceCode" id="cb475"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb475-1"><a href="Association-mapping.html#cb475-1" aria-hidden="true" tabindex="-1"></a><span class="fu">plot</span>(<span class="fu">eigen</span>(TGall<span class="sc">@</span>K)<span class="sc">$</span>vectors[,<span class="dv">1</span><span class="sc">:</span><span class="dv">2</span>]) </span></code></pre></div>
<p><img src="bookdown-demo_files/figure-html/unnamed-chunk-352-1.png" width="672" /></p>
<div class="sourceCode" id="cb476"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb476-1"><a href="Association-mapping.html#cb476-1" aria-hidden="true" tabindex="-1"></a><span class="fu">plot</span>(<span class="fu">eigen</span>(TGkin<span class="sc">@</span>K)<span class="sc">$</span>vectors[,<span class="dv">1</span><span class="sc">:</span><span class="dv">2</span>]) </span></code></pre></div>
<p><img src="bookdown-demo_files/figure-html/unnamed-chunk-352-2.png" width="672" /></p>

<div class="rmdquiz">
<p><strong>1</strong></p>
The plots for all the data, and for the thinned markers alone are very different. What is the explanation for this? On the basis of this, which kinship matrix is most representative of relationships over the whole genome?
</div>
<p>We now want to make two data sets for the GWAS analysis. The GWAS analysis itself will be conducted with all the genotype data, but we want one data set with kinship calculated from all markers:</p>
<div class="sourceCode" id="cb477"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb477-1"><a href="Association-mapping.html#cb477-1" aria-hidden="true" tabindex="-1"></a>TGanalysisUnthinned<span class="ot">&lt;-</span>TGall </span></code></pre></div>
<p>… and one data set with kinship calculated from only the thinned markers. This is remarkably easy to create: we just substitute in the kinship matrix calculate on the thinned markers:</p>
<div class="sourceCode" id="cb478"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb478-1"><a href="Association-mapping.html#cb478-1" aria-hidden="true" tabindex="-1"></a>TGanalysisThinned<span class="ot">&lt;-</span>TGall</span>
<span id="cb478-2"><a href="Association-mapping.html#cb478-2" aria-hidden="true" tabindex="-1"></a>TGanalysisThinned<span class="sc">@</span>K<span class="ot">&lt;-</span>TGkin<span class="sc">@</span>K </span></code></pre></div>
<p>Then check:</p>
<div class="sourceCode" id="cb479"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb479-1"><a href="Association-mapping.html#cb479-1" aria-hidden="true" tabindex="-1"></a>TGanalysisUnthinned<span class="sc">@</span>K[<span class="dv">1</span><span class="sc">:</span><span class="dv">5</span>,<span class="dv">1</span><span class="sc">:</span><span class="dv">5</span>] </span></code></pre></div>
<pre><code>##               AARDEN    AARDVARK       ABELE         ABO     ACCESS
## AARDEN    0.89425094  0.06820056 -0.08325033 -0.17401122  0.1904552
## AARDVARK  0.06820056  0.98608975  0.04025471 -0.03429426 -0.0102782
## ABELE    -0.08325033  0.04025471  1.16491365 -0.03983785  0.1856692
## ABO      -0.17401122 -0.03429426 -0.03983785  1.30763030 -0.1575344
## ACCESS    0.19045520 -0.01027820  0.18566924 -0.15753443  1.0314240</code></pre>
<div class="sourceCode" id="cb481"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb481-1"><a href="Association-mapping.html#cb481-1" aria-hidden="true" tabindex="-1"></a>TGanalysisThinned<span class="sc">@</span>K[<span class="dv">1</span><span class="sc">:</span><span class="dv">5</span>,<span class="dv">1</span><span class="sc">:</span><span class="dv">5</span>] </span></code></pre></div>
<pre><code>##               AARDEN    AARDVARK       ABELE         ABO      ACCESS
## AARDEN    0.94938532  0.07464131 -0.07579459 -0.17446488  0.24676194
## AARDVARK  0.07464131  0.92078798  0.07606171 -0.03447680  0.03070368
## ABELE    -0.07579459  0.07606171  1.04717181  0.02871512  0.08796148
## ABO      -0.17446488 -0.03447680  0.02871512  1.30082327 -0.16499556
## ACCESS    0.24676194  0.03070368  0.08796148 -0.16499556  1.04546733</code></pre>
</div>
<div id="gwas" class="section level3" number="9.4.2">
<h3><span class="header-section-number">9.4.2</span> GWAS</h3>
<p>Strictly, in using the mixed model for GWAS, genetic variances and covariances should be estimated separately for every marker, since the estimates will vary depending on the magnitude of the SNP effect and the distribution of the marker alleles over individuals. However, given that most markers have very little effect, the variances change very little.
Therefore GWASpoly follows an approximate approach outlined by <span class="citation"><a href="references.html#ref-kang_variance_2010" role="doc-biblioref">Kang et al.</a> (<a href="references.html#ref-kang_variance_2010" role="doc-biblioref">2010</a>)</span> and <span class="citation"><a href="references.html#ref-zhang_mixed_2010" role="doc-biblioref">Zhang et al.</a> (<a href="references.html#ref-zhang_mixed_2010" role="doc-biblioref">2010</a>)</span> in which the variances are estimated only once, in a model without any SNP effect. This greatly speeds up the analysis with little loss of accuracy or bias. To analyse all traits for the thinned data set:</p>
<div class="sourceCode" id="cb483"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb483-1"><a href="Association-mapping.html#cb483-1" aria-hidden="true" tabindex="-1"></a>TG.GWAS.Thinned<span class="ot">&lt;-</span><span class="fu">GWASpoly</span>(TGanalysisThinned,<span class="at">models=</span><span class="st">&quot;additive&quot;</span>) </span></code></pre></div>
<p>A choice must be made of what model of marker effects to fit. For pseudo-haploid data, as here, the choice makes no difference. In general a simple additive model is best. Other options (see <code>help(GWASpoly()</code> ) are targeted at analyses with polyploids, where the multiple heterozygous classes can be pooled in various ways. <code>models ="general"</code> would fit separate additive and a dominance terms in a diploid species.</p>
<p>You can specify multiple models in the same analysis. More usefully for us, you can restrict the analysis to certain traits only. We will now focus on analysing only two traits, height and flowering time:</p>
<div class="sourceCode" id="cb484"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb484-1"><a href="Association-mapping.html#cb484-1" aria-hidden="true" tabindex="-1"></a>TG.HT.FT.Thinned<span class="ot">&lt;-</span><span class="fu">GWASpoly</span>(TGanalysisThinned, <span class="at">models=</span><span class="st">&quot;additive&quot;</span>, <span class="at">traits=</span><span class="fu">c</span>(<span class="st">&quot;HT&quot;</span>,<span class="st">&quot;FT&quot;</span>)) </span></code></pre></div>
<pre><code>## Analyzing trait: HT 
## P3D approach: Estimating variance components...Completed 
## Testing markers for model: additive 
## Analyzing trait: FT 
## P3D approach: Estimating variance components...Completed 
## Testing markers for model: additive</code></pre>
<p>We now need to look at QQ plots:</p>
<div class="sourceCode" id="cb486"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb486-1"><a href="Association-mapping.html#cb486-1" aria-hidden="true" tabindex="-1"></a><span class="fu">qq.plot</span>(TG.HT.FT.Thinned,<span class="at">trait=</span><span class="st">&quot;FT&quot;</span>,<span class="at">model=</span><span class="st">&quot;additive&quot;</span>)</span></code></pre></div>
<p><img src="bookdown-demo_files/figure-html/unnamed-chunk-359-1.png" width="672" /></p>
<div class="sourceCode" id="cb487"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb487-1"><a href="Association-mapping.html#cb487-1" aria-hidden="true" tabindex="-1"></a><span class="fu">qq.plot</span>(TG.HT.FT.Thinned,<span class="at">trait=</span><span class="st">&quot;HT&quot;</span>,<span class="at">model=</span><span class="st">&quot;additive&quot;</span>)</span></code></pre></div>
<p><img src="bookdown-demo_files/figure-html/unnamed-chunk-359-2.png" width="672" /></p>
<p>These look OK. If the lower values did not fall close to the line, we would need to reassess our corrections for kinship and/or take into account other covariates (such as population structure, as below).</p>
<p>What QTL have we found? First we need to specify a significance threshold:</p>
<div class="sourceCode" id="cb488"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb488-1"><a href="Association-mapping.html#cb488-1" aria-hidden="true" tabindex="-1"></a>TG.HT.FT.Thinned.Bonf<span class="ot">&lt;-</span><span class="fu">set.threshold</span>(TG.HT.FT.Thinned,<span class="at">method=</span><span class="st">&quot;Bonferroni&quot;</span>,<span class="at">level=</span><span class="fl">0.05</span>)</span></code></pre></div>
<pre><code>## Thresholds
##    additive
## HT     4.71
## FT     4.71</code></pre>
<p>You have three choices for setting the threshold: <code>“Bonferroni”</code>, <code>“FDR”</code> (False Discovery Rate) and <code>“permute”</code> (permutation test). The permutation test is relatively quick but to do the default 1000 permutations would take too long for this exercise! I will show the results from this in the class exercise. For now, you can run the <code>"FDR"</code> threshold for comparison:</p>
<div class="sourceCode" id="cb490"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb490-1"><a href="Association-mapping.html#cb490-1" aria-hidden="true" tabindex="-1"></a>TG.HT.FT.Thinned.fdr<span class="ot">&lt;-</span><span class="fu">set.threshold</span>(TG.HT.FT.Thinned,<span class="at">method=</span><span class="st">&quot;FDR&quot;</span>,<span class="at">level=</span><span class="fl">0.05</span>) </span></code></pre></div>
<pre><code>## Thresholds
##    additive
## HT     3.47
## FT     3.73</code></pre>
<p>Having set the threshold, you will find it is plotted on any Manhattan plots you create provided there is something that exceeds the threshold. We can plot Manhattan plots from our analyses as:</p>
<div class="sourceCode" id="cb492"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb492-1"><a href="Association-mapping.html#cb492-1" aria-hidden="true" tabindex="-1"></a><span class="fu">manhattan.plot</span>(TG.HT.FT.Thinned.Bonf,<span class="at">trait=</span><span class="fu">c</span>(<span class="st">&quot;HT&quot;</span>),<span class="at">model=</span><span class="st">&quot;additive&quot;</span>) </span></code></pre></div>
<p><img src="bookdown-demo_files/figure-html/unnamed-chunk-362-1.png" width="672" /></p>
<p>You must specify a single trait and a single model.</p>
<p>We can examine the significant results directly:</p>
<div class="sourceCode" id="cb493"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb493-1"><a href="Association-mapping.html#cb493-1" aria-hidden="true" tabindex="-1"></a><span class="fu">get.QTL</span>(TG.HT.FT.Thinned.Bonf)</span></code></pre></div>
<pre><code>##      Trait    Model Threshold    Marker Chrom Position Ref Alt Score    Effect
## 1363    HT additive      4.71 DArT_1448    16  48.7860   0   1  5.72  4.87e+00
## 1364    HT additive      4.71 DArT_1449    16  48.7860   0   1  5.72  4.87e+00
## 1365    HT additive      4.71 DArT_1451    16  49.1736   0   1  5.27  4.70e+00
## 1366    HT additive      4.71 DArT_1452    16  49.1736   0   1  5.72  4.87e+00
## 1367    HT additive      4.71 DArT_1453    16  49.1736   0   1  5.72  4.87e+00
## 2042    HT additive      4.71   SNP_168    22 167.0000   0   1  6.03  3.32e+00
## 2150    HT additive      4.71   SNP_278    22 277.0000   0   1  5.00 -3.03e+00
## 2253    HT additive      4.71 DArT_2072    23  59.0000   0   1  5.41 -3.18e+00
## 2684    HT additive      4.71  Rht2_400    24   9.0000   0   1 12.72 -5.51e+00
## 654     FT additive      4.71  DArT_678     6  84.5399   0   1  5.18  1.66e+00
## 2065    FT additive      4.71   SNP_191    22 190.0000   0   1 12.42 -4.22e+00
## 2682    FT additive      4.71 PpdD1_297    24   7.0000   0   1 40.74 -4.81e+00</code></pre>
<p>“Score” is the –log10(p-value) from the analyses, to be compared with the Threshold – which is also a –log10(p-value). “Ref and “Alt” give the reference allele and the alternative allele. “Effect” gives the gene effect for the alternative allele. In the example, The alternative allele (1) for PpdD1_297 has an effect of -4.81 on flowering time. (i.e. it reduces flowering time by nearly 5 days). We can confirm this by extracting and plotting individual trait – marker combinations:</p>
<div class="sourceCode" id="cb495"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb495-1"><a href="Association-mapping.html#cb495-1" aria-hidden="true" tabindex="-1"></a><span class="fu">boxplot</span>(TG.HT.FT.Thinned.Bonf<span class="sc">@</span>pheno<span class="sc">$</span>FT<span class="sc">~</span><span class="fu">as.data.frame</span>(TG.HT.FT.Thinned<span class="sc">@</span>geno)<span class="sc">$</span>PpdD1_297,<span class="at">varwidth=</span>T) </span></code></pre></div>
<p><img src="bookdown-demo_files/figure-html/unnamed-chunk-364-1.png" width="672" /></p>
<p>Similarly, the top hit for height is the unmapped Rht2 gene (one of the ‘green revolution’ dwarfing loci), where the reduced height allele has an effect size of about 5.5 cm relative to the reference allele.</p>
<div class="sourceCode" id="cb496"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb496-1"><a href="Association-mapping.html#cb496-1" aria-hidden="true" tabindex="-1"></a><span class="fu">boxplot</span>(TG.HT.FT.Thinned.Bonf<span class="sc">@</span>pheno<span class="sc">$</span>HT<span class="sc">~</span><span class="fu">as.data.frame</span>(TG.HT.FT.Thinned<span class="sc">@</span>geno)<span class="sc">$</span>Rht2_400,<span class="at">varwidth=</span>T)</span></code></pre></div>
<p><img src="bookdown-demo_files/figure-html/unnamed-chunk-365-1.png" width="672" /></p>

<div class="rmdquiz">
<p><strong>2</strong></p>
<p>a). Compare the results for “HT” and “FT” between the ‘Bonferroni’ and ‘FDR,’ and, if possible, ‘permutation’ thresholds. What do you conclude?</p>
b). Rerun the analysis using the Non-thinned data. What differences might you expect to see? What do you observe and what are the implications?
</div>
<p>We will now run the analysis unadjusted for kinship. The unadjusted analysis is simply linear regression, or a t-test, on each marker in turn. We could do this by putting <code>lm()</code> or <code>t.test()</code> inside a loop in R. If you have time, this is a good exercise to try. However, we can get the same result directly from GWASpoly by analysing the data with a kinship matrix which has ‘1’ down the leading diagonal and ‘0’ everywhere else – all the lines are unrelated. Strictly, it isn’t that the lines are unrelated, it is that they have equal relationships: we should get the same answer if you used, say, 0.25, for the off- diagonal elements (another good exercise to try).</p>
<p>First take a copy of the data and then create the diagonal matrix:</p>
<div class="sourceCode" id="cb497"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb497-1"><a href="Association-mapping.html#cb497-1" aria-hidden="true" tabindex="-1"></a>TGdiag<span class="ot">&lt;-</span>TGanalysisThinned</span>
<span id="cb497-2"><a href="Association-mapping.html#cb497-2" aria-hidden="true" tabindex="-1"></a><span class="co">#There is a command which will create diagonal matrices in R: diag().  </span></span>
<span id="cb497-3"><a href="Association-mapping.html#cb497-3" aria-hidden="true" tabindex="-1"></a>TGdiag<span class="sc">@</span>K <span class="ot">&lt;-</span> <span class="fu">diag</span>(<span class="dv">1</span>,<span class="fu">length</span>(<span class="fu">rownames</span>(TGanalysisThinned<span class="sc">@</span>geno)))</span></code></pre></div>
<p>We’ve also picked up the required dimensions of the kinship matrix using <code>length(rownames (....))</code>. The new kinship matrix must also have row names and column names which match the names of the original matrix:</p>
<div class="sourceCode" id="cb498"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb498-1"><a href="Association-mapping.html#cb498-1" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(TGdiag<span class="sc">@</span>K)<span class="ot">&lt;-</span>(<span class="fu">rownames</span>(TGanalysisThinned<span class="sc">@</span>geno))</span>
<span id="cb498-2"><a href="Association-mapping.html#cb498-2" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(TGdiag<span class="sc">@</span>K)<span class="ot">&lt;-</span>(<span class="fu">rownames</span>(TGanalysisThinned<span class="sc">@</span>geno))</span></code></pre></div>
<p>Check all looks ok:</p>
<div class="sourceCode" id="cb499"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb499-1"><a href="Association-mapping.html#cb499-1" aria-hidden="true" tabindex="-1"></a>TGdiag<span class="sc">@</span>K[<span class="dv">1</span><span class="sc">:</span><span class="dv">5</span>,<span class="dv">1</span><span class="sc">:</span><span class="dv">5</span>]</span></code></pre></div>
<pre><code>##          AARDEN AARDVARK ABELE ABO ACCESS
## AARDEN        1        0     0   0      0
## AARDVARK      0        1     0   0      0
## ABELE         0        0     1   0      0
## ABO           0        0     0   1      0
## ACCESS        0        0     0   0      1</code></pre>
<p>And run the analyses, just as before:</p>
<div class="sourceCode" id="cb501"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb501-1"><a href="Association-mapping.html#cb501-1" aria-hidden="true" tabindex="-1"></a>TGdiag <span class="ot">&lt;-</span><span class="fu">GWASpoly</span>(TGdiag,<span class="at">models=</span><span class="st">&quot;additive&quot;</span>) </span>
<span id="cb501-2"><a href="Association-mapping.html#cb501-2" aria-hidden="true" tabindex="-1"></a>TGdiag.fdr <span class="ot">&lt;-</span><span class="fu">set.threshold</span>(TGdiag,<span class="at">method=</span><span class="st">&quot;FDR&quot;</span>,<span class="at">level=</span><span class="fl">0.05</span>) </span>
<span id="cb501-3"><a href="Association-mapping.html#cb501-3" aria-hidden="true" tabindex="-1"></a><span class="fu">get.QTL</span>(TGdiag.fdr) </span>
<span id="cb501-4"><a href="Association-mapping.html#cb501-4" aria-hidden="true" tabindex="-1"></a><span class="fu">manhattan.plot</span>(TGdiag.fdr,<span class="at">trait=</span><span class="fu">c</span>(<span class="st">&quot;HT&quot;</span>),<span class="at">model=</span><span class="st">&quot;additive&quot;</span>) </span></code></pre></div>

<div class="rmdquiz">
<p><strong>3</strong></p>
How do the results compare with the results of the mixed model which we ran previously? Are there many significant results arising from not considering kinship? Was the mixed model analysis necessary?
</div>
</div>
<div id="analysis-with-covariates" class="section level3" number="9.4.3">
<h3><span class="header-section-number">9.4.3</span> Analysis with covariates</h3>
<p>For height, in the unadjusted results, you should find Rht2 is the most significant marker and has the largest effect. It is the most important dwarfing gene employed in Europe, so this is not surprising. In the adjusted results, you will still find Rht2 on top, but the p-value is much reduced, as is the estimate of its effect. Height is a major contributor to population structure in wheat: German varieties tend to rely less on dwarfing genes for example and so are taller. Adjusting for kinship has the effect of leaving less variation to be accounted for by genuine marker relationships.</p>
<p>Perhaps Rht2 is dominating the results. If we include Rht2 as a covariate in the analysis, will other QTL for reduced height be revealed? For this we need to include Rht2 in the analysis as a covariate. Recall that the phenotypic data input included some additional columns, to be used as covariates. These have been stored in a slot: <code>@fixed</code>:</p>
<div class="sourceCode" id="cb502"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb502-1"><a href="Association-mapping.html#cb502-1" aria-hidden="true" tabindex="-1"></a><span class="fu">summary</span>(TGanalysisThinned<span class="sc">@</span>fixed) </span></code></pre></div>
<pre><code>##     Rht2_400        PpdD1_297           YEAR        COUNTRY         
##  Min.   :0.0000   Min.   :0.0000   Min.   :1946   Length:376        
##  1st Qu.:0.0000   1st Qu.:0.0000   1st Qu.:1988   Class :character  
##  Median :1.0000   Median :0.0000   Median :1997   Mode  :character  
##  Mean   :0.5455   Mean   :0.2253   Mean   :1994                     
##  3rd Qu.:1.0000   3rd Qu.:0.0000   3rd Qu.:2003                     
##  Max.   :1.0000   Max.   :1.0000   Max.   :2007                     
##       FRA              DEU        
##  Min.   :0.0000   Min.   :0.0000  
##  1st Qu.:0.0000   1st Qu.:0.0000  
##  Median :1.0000   Median :0.0000  
##  Mean   :0.5656   Mean   :0.2385  
##  3rd Qu.:1.0000   3rd Qu.:0.0000  
##  Max.   :1.0000   Max.   :1.0000</code></pre>
<p>I’ve included Rht2 as one of these covariates. To select other SNPs as covariates, I expect you could add or edit <code>TGanalysisThinned@fixed</code> to include the desired SNPs, but I’ve not tried this myself. It’s another exercise in R and GWAS for you to try! To add a covariate to the mixed model, first provide their names and their data type. The covariates can be numeric or factors, such as country of origin.</p>
<div class="sourceCode" id="cb504"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb504-1"><a href="Association-mapping.html#cb504-1" aria-hidden="true" tabindex="-1"></a>Rht2<span class="ot">&lt;-</span> <span class="fu">set.params</span>(<span class="at">fixed=</span><span class="fu">c</span>(<span class="st">&quot;Rht2_400&quot;</span>),<span class="at">fixed.type=</span><span class="fu">c</span>(<span class="st">&quot;numeric&quot;</span>))</span></code></pre></div>
<p>We could include multiple covariates in the analysis (by using <code>c(cov1,cov2...)</code>. In the interests of speed, we’ll just use Rht2:</p>
<div class="sourceCode" id="cb505"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb505-1"><a href="Association-mapping.html#cb505-1" aria-hidden="true" tabindex="-1"></a>TG.HT.Rht2<span class="ot">&lt;-</span><span class="fu">GWASpoly</span>(TGanalysisThinned,<span class="at">models=</span><span class="st">&quot;additive&quot;</span>,<span class="at">trait=</span><span class="fu">c</span>(<span class="st">&quot;HT&quot;</span>),<span class="at">params=</span>Rht2)</span>
<span id="cb505-2"><a href="Association-mapping.html#cb505-2" aria-hidden="true" tabindex="-1"></a>TG.HT.Rht2 <span class="ot">&lt;-</span><span class="fu">set.threshold</span>(TG.HT.Rht2,<span class="at">method=</span><span class="st">&quot;FDR&quot;</span>,<span class="at">level=</span><span class="fl">0.05</span>)</span></code></pre></div>
<p>Now compare the results on your own PC:</p>
<div class="sourceCode" id="cb506"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb506-1"><a href="Association-mapping.html#cb506-1" aria-hidden="true" tabindex="-1"></a><span class="fu">get.QTL</span>(TG.HT.Rht2,<span class="at">traits=</span><span class="fu">c</span>(<span class="st">&quot;HT&quot;</span>))</span>
<span id="cb506-2"><a href="Association-mapping.html#cb506-2" aria-hidden="true" tabindex="-1"></a><span class="fu">get.QTL</span>(TG.HT.FT.Thinned.fdr,<span class="at">traits=</span><span class="fu">c</span>(<span class="st">&quot;HT&quot;</span>)) </span></code></pre></div>

<div class="rmdquiz">
<p><strong>4</strong></p>
What is the main difference of including a covariate? Do any hits vanish? If so, why do you think this might be? Do new hits appear? If so, why do you think they were not previously detected? (For those of you who work on wheat, PpdD1 and Rht1 should be familiar, in particular what do you think is going on with Rht1?
</div>
</div>
<div id="accounting-for-population-structure" class="section level3" number="9.4.4">
<h3><span class="header-section-number">9.4.4</span> Accounting for population structure</h3>
<p>We could use as covariates the population membership as coefficients estimated in the program STRUCTURE. To do this, we would have to run STRUCTURE and import these as covariates with the rest of the trait and covariate data. Including these together with kinship coefficients is often called the “Q + K” analysis. Alternatively, as we have seen, for crop germplasm “populations” such as this one, principal coordinate analysis may be a more appropriate choice for estimating population structure. For this reason, the principal components from a PCoA may be the best choice to include as covariates. GWASpoly has options to do this automatically. Here we use the first 10 eigenvectors:</p>
<div class="sourceCode" id="cb507"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb507-1"><a href="Association-mapping.html#cb507-1" aria-hidden="true" tabindex="-1"></a>params <span class="ot">&lt;-</span> <span class="fu">set.params</span>(<span class="at">n.PC=</span><span class="dv">10</span>) </span>
<span id="cb507-2"><a href="Association-mapping.html#cb507-2" aria-hidden="true" tabindex="-1"></a>TG.PCO<span class="ot">&lt;-</span><span class="fu">GWASpoly</span>(TGanalysisThinned,<span class="at">models=</span><span class="st">&quot;additive&quot;</span>,<span class="at">traits=</span><span class="fu">c</span>(<span class="st">&quot;HT&quot;</span>,<span class="st">&quot;FT&quot;</span>),<span class="at">params=</span>params)</span>
<span id="cb507-3"><a href="Association-mapping.html#cb507-3" aria-hidden="true" tabindex="-1"></a>TG.PCO<span class="ot">&lt;-</span><span class="fu">set.threshold</span>(TG.PCO,<span class="at">method=</span><span class="st">&quot;Bonferroni&quot;</span>,<span class="at">level=</span><span class="fl">0.05</span>)</span>
<span id="cb507-4"><a href="Association-mapping.html#cb507-4" aria-hidden="true" tabindex="-1"></a><span class="fu">get.QTL</span>(TG.PCO)</span>
<span id="cb507-5"><a href="Association-mapping.html#cb507-5" aria-hidden="true" tabindex="-1"></a><span class="fu">get.QTL</span>(TG.HT.FT.Thinned.fdr)</span></code></pre></div>
<p>Then inspect he results:</p>
<div class="sourceCode" id="cb508"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb508-1"><a href="Association-mapping.html#cb508-1" aria-hidden="true" tabindex="-1"></a><span class="fu">qq.plot</span>(TG.PCO,<span class="at">trait=</span><span class="st">&quot;FT&quot;</span>,<span class="at">model=</span><span class="st">&quot;additive&quot;</span>)</span>
<span id="cb508-2"><a href="Association-mapping.html#cb508-2" aria-hidden="true" tabindex="-1"></a><span class="fu">qq.plot</span>(TG.HT.FT.Thinned,<span class="at">trait=</span><span class="st">&quot;FT&quot;</span>,<span class="at">model=</span><span class="st">&quot;additive&quot;</span>)</span></code></pre></div>

<div class="rmdquiz">
<p><strong>5</strong></p>
Does this make any difference to the results (note the subtle difference in the qq plots as well as the qtls detected)? Do you think the results might differ for a different kind of mapping population? Do you think it is a problem that we are using the kinship matrix twice?
</div>
<p>Rather than estimate population structure from the kinship matrix, we can analyse the data with country of varietal origin as a factor. To do this,</p>
<div class="sourceCode" id="cb509"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb509-1"><a href="Association-mapping.html#cb509-1" aria-hidden="true" tabindex="-1"></a>country<span class="ot">&lt;-</span><span class="fu">set.params</span>(<span class="at">fixed =</span> <span class="st">&quot;COUNTRY&quot;</span>,<span class="at">fixed.type =</span> <span class="st">&quot;factor&quot;</span>) </span>
<span id="cb509-2"><a href="Association-mapping.html#cb509-2" aria-hidden="true" tabindex="-1"></a>TG.country<span class="ot">&lt;-</span><span class="fu">GWASpoly</span>(TGanalysisThinned,<span class="at">models=</span><span class="st">&quot;additive&quot;</span>,<span class="at">traits=</span><span class="fu">c</span>(<span class="st">&quot;HT&quot;</span>,<span class="st">&quot;FT&quot;</span>), <span class="at">params=</span>country) </span>
<span id="cb509-3"><a href="Association-mapping.html#cb509-3" aria-hidden="true" tabindex="-1"></a>TG.country<span class="ot">&lt;-</span><span class="fu">set.threshold</span>(TG.country,<span class="at">method=</span><span class="st">&quot;FDR&quot;</span>,<span class="at">level=</span><span class="fl">0.05</span>)</span>
<span id="cb509-4"><a href="Association-mapping.html#cb509-4" aria-hidden="true" tabindex="-1"></a><span class="fu">get.QTL</span>(TG.country) </span>
<span id="cb509-5"><a href="Association-mapping.html#cb509-5" aria-hidden="true" tabindex="-1"></a><span class="fu">get.QTL</span>(TG.HT.FT.Thinned.fdr)</span></code></pre></div>
<p>For covariates to be included, there must be no missing data. A pseudo country has been added here for the three missing values. A better way of dealing with the unknown factor levels would be to include two columns for the three countries coded as FRA (1) and not-FRA (0) in the first column and DEU and notDEU in the second. If a country is not-FRA and not-DEU then it must be GBR: we have three countries but only two degrees of freedom so the two columns alone are adequate. Then the missing values in each column could be coded as 1/3: the average population membership over the three countries. This is an example of the equivalence of regression and the analysis of factors with multiple levels in the analysis of variance.</p>

<div class="rmdquiz">
<p><strong>6</strong></p>
Did using countries as a covariate make any difference to the results?
</div>
</div>
<div id="further-analysis" class="section level3" number="9.4.5">
<h3><span class="header-section-number">9.4.5</span> Further analysis</h3>
<p>There are other things that can be done with the <code>set.params()</code> command: the minor allele frequency for inclusion in the analysis can be set. The default is 0.05. Also, the approximate method of carrying out the mixed analysis can be turned off. See <code>help(set.params)</code> for details.</p>
<p>We have seen that there remains some art in carrying out association analyses: what kinship matrix to use, what covariates to include and so on. The best approach is probably to simulate traits from the marker data for your data set and test analysis methods on those. In simulations on UK barley and UK wheat data we concluded that the best all-round method is the mixed model, with a kinship matrix similar to the one used here and no additional covariates to account for population structure. But this wasn’t always best for all traits and datasets. If one has time, it is worth experimenting. And of course simulations are always needed to estimate the power of your analysis, and hence the data collection requirements, whichever approach you take.</p>

<div class="rmdquiz">
<p><strong>7</strong></p>
How would we run simulations to estimate the power of different analysis approaches?
</div>

</div>
</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="Imputation.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="Mapping-Markers.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": "https://github.com/rstudio/bookdown-demo/edit/master/09-Assocation_mapping.Rmd",
"text": "Edit"
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": ["bookdown-demo.pdf", "bookdown-demo.epub"],
"toc": {
"collapse": "subsection"
}
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
