<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 10 Mapping Markers and Quantitative Trait Loci. | Quantitative Methods in Plant Breeding: The Digital Tutorials.</title>
  <meta name="description" content="This is the online tutorial booklet for the QMPB course run at NIAB. The online book was formed using R bookdown." />
  <meta name="generator" content="bookdown 0.21 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 10 Mapping Markers and Quantitative Trait Loci. | Quantitative Methods in Plant Breeding: The Digital Tutorials." />
  <meta property="og:type" content="book" />
  
  <meta property="og:image" content="images/cover.jpg" />
  <meta property="og:description" content="This is the online tutorial booklet for the QMPB course run at NIAB. The online book was formed using R bookdown." />
  <meta name="github-repo" content="rstudio/bookdown" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 10 Mapping Markers and Quantitative Trait Loci. | Quantitative Methods in Plant Breeding: The Digital Tutorials." />
  
  <meta name="twitter:description" content="This is the online tutorial booklet for the QMPB course run at NIAB. The online book was formed using R bookdown." />
  <meta name="twitter:image" content="images/cover.jpg" />

<meta name="author" content="NIAB 2021" />


<meta name="date" content="2021-03-04" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="Association-mapping.html"/>
<link rel="next" href="Data-Visualization.html"/>
<script src="libs/header-attrs-2.6/header-attrs.js"></script>
<script src="libs/jquery-2.2.3/jquery.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />











<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">The QMPB digital tutorials</a></li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> Introduction</a></li>
<li class="chapter" data-level="2" data-path="Acknow.html"><a href="Acknow.html"><i class="fa fa-check"></i><b>2</b> Acknowledgements and Contents</a>
<ul>
<li class="chapter" data-level="2.1" data-path="Acknow.html"><a href="Acknow.html#acknowledgements"><i class="fa fa-check"></i><b>2.1</b> Acknowledgements</a></li>
<li class="chapter" data-level="2.2" data-path="Acknow.html"><a href="Acknow.html#digital-tutorial-chapters"><i class="fa fa-check"></i><b>2.2</b> Digital Tutorial Chapters:</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="R-and-stats.html"><a href="R-and-stats.html"><i class="fa fa-check"></i><b>3</b> R and Rstudio: a general introduction with statistics.</a>
<ul>
<li class="chapter" data-level="3.1" data-path="R-and-stats.html"><a href="R-and-stats.html#introduction-section-on-r"><i class="fa fa-check"></i><b>3.1</b> Introduction section on R</a>
<ul>
<li class="chapter" data-level="3.1.1" data-path="R-and-stats.html"><a href="R-and-stats.html#getting-the-data-ready"><i class="fa fa-check"></i><b>3.1.1</b> Getting the data ready</a></li>
</ul></li>
<li class="chapter" data-level="3.2" data-path="R-and-stats.html"><a href="R-and-stats.html#the-rstudio-layout-workspace-and-environment."><i class="fa fa-check"></i><b>3.2</b> The Rstudio layout, workspace and environment.</a>
<ul>
<li class="chapter" data-level="3.2.1" data-path="R-and-stats.html"><a href="R-and-stats.html#working-directory"><i class="fa fa-check"></i><b>3.2.1</b> Working directory</a></li>
</ul></li>
<li class="chapter" data-level="3.3" data-path="R-and-stats.html"><a href="R-and-stats.html#basic-r-syntx"><i class="fa fa-check"></i><b>3.3</b> Basic R Syntx</a>
<ul>
<li class="chapter" data-level="3.3.1" data-path="R-and-stats.html"><a href="R-and-stats.html#brackets"><i class="fa fa-check"></i><b>3.3.1</b> Brackets</a></li>
<li class="chapter" data-level="3.3.2" data-path="R-and-stats.html"><a href="R-and-stats.html#equals-symbols"><i class="fa fa-check"></i><b>3.3.2</b> Equals symbols</a></li>
<li class="chapter" data-level="3.3.3" data-path="R-and-stats.html"><a href="R-and-stats.html#continuation-character."><i class="fa fa-check"></i><b>3.3.3</b> Continuation character.</a></li>
<li class="chapter" data-level="3.3.4" data-path="R-and-stats.html"><a href="R-and-stats.html#text-and-numbers"><i class="fa fa-check"></i><b>3.3.4</b> Text and numbers</a></li>
<li class="chapter" data-level="3.3.5" data-path="R-and-stats.html"><a href="R-and-stats.html#summary-of-syntax"><i class="fa fa-check"></i><b>3.3.5</b> Summary of syntax</a></li>
</ul></li>
<li class="chapter" data-level="3.4" data-path="R-and-stats.html"><a href="R-and-stats.html#reading-in-data"><i class="fa fa-check"></i><b>3.4</b> Reading in data</a></li>
<li class="chapter" data-level="3.5" data-path="R-and-stats.html"><a href="R-and-stats.html#understanding-data-types-and-r-objects"><i class="fa fa-check"></i><b>3.5</b> Understanding data types and R objects</a>
<ul>
<li class="chapter" data-level="3.5.1" data-path="R-and-stats.html"><a href="R-and-stats.html#summary"><i class="fa fa-check"></i><b>3.5.1</b> Summary</a></li>
</ul></li>
<li class="chapter" data-level="3.6" data-path="R-and-stats.html"><a href="R-and-stats.html#summarising-and-displaying-data"><i class="fa fa-check"></i><b>3.6</b> Summarising and displaying data</a></li>
<li class="chapter" data-level="3.7" data-path="R-and-stats.html"><a href="R-and-stats.html#calculations-and-data-manipulation"><i class="fa fa-check"></i><b>3.7</b> Calculations and data manipulation</a>
<ul>
<li class="chapter" data-level="3.7.1" data-path="R-and-stats.html"><a href="R-and-stats.html#sorting-data"><i class="fa fa-check"></i><b>3.7.1</b> Sorting data</a></li>
<li class="chapter" data-level="3.7.2" data-path="R-and-stats.html"><a href="R-and-stats.html#relationships-between-variates"><i class="fa fa-check"></i><b>3.7.2</b> Relationships between variates</a></li>
<li class="chapter" data-level="3.7.3" data-path="R-and-stats.html"><a href="R-and-stats.html#exporting-your-script-and-workspace"><i class="fa fa-check"></i><b>3.7.3</b> Exporting your script and workspace</a></li>
<li class="chapter" data-level="3.7.4" data-path="R-and-stats.html"><a href="R-and-stats.html#saving-a-data-table"><i class="fa fa-check"></i><b>3.7.4</b> Saving a data table</a></li>
</ul></li>
<li class="chapter" data-level="3.8" data-path="R-and-stats.html"><a href="R-and-stats.html#basic-statistical-analysis"><i class="fa fa-check"></i><b>3.8</b> Basic statistical analysis</a>
<ul>
<li class="chapter" data-level="3.8.1" data-path="R-and-stats.html"><a href="R-and-stats.html#the-t-test"><i class="fa fa-check"></i><b>3.8.1</b> The t-test</a></li>
<li class="chapter" data-level="3.8.2" data-path="R-and-stats.html"><a href="R-and-stats.html#linear-regression"><i class="fa fa-check"></i><b>3.8.2</b> Linear Regression</a></li>
<li class="chapter" data-level="3.8.3" data-path="R-and-stats.html"><a href="R-and-stats.html#multiple-regression"><i class="fa fa-check"></i><b>3.8.3</b> Multiple regression</a></li>
<li class="chapter" data-level="3.8.4" data-path="R-and-stats.html"><a href="R-and-stats.html#the-analysis-of-variance"><i class="fa fa-check"></i><b>3.8.4</b> The analysis of variance</a></li>
</ul></li>
<li class="chapter" data-level="3.9" data-path="R-and-stats.html"><a href="R-and-stats.html#categorical-data---the-chi-squared-test"><i class="fa fa-check"></i><b>3.9</b> Categorical data - the <em>chi</em>-squared test</a></li>
<li class="chapter" data-level="3.10" data-path="R-and-stats.html"><a href="R-and-stats.html#more-useful-information"><i class="fa fa-check"></i><b>3.10</b> More useful information</a>
<ul>
<li class="chapter" data-level="3.10.1" data-path="R-and-stats.html"><a href="R-and-stats.html#plotting-graphs"><i class="fa fa-check"></i><b>3.10.1</b> Plotting Graphs</a></li>
<li class="chapter" data-level="3.10.2" data-path="R-and-stats.html"><a href="R-and-stats.html#probability-distributions"><i class="fa fa-check"></i><b>3.10.2</b> Probability distributions</a></li>
<li class="chapter" data-level="3.10.3" data-path="R-and-stats.html"><a href="R-and-stats.html#random-number-generation"><i class="fa fa-check"></i><b>3.10.3</b> Random number generation</a></li>
<li class="chapter" data-level="3.10.4" data-path="R-and-stats.html"><a href="R-and-stats.html#loops-second-example"><i class="fa fa-check"></i><b>3.10.4</b> Loops: <em>second example</em></a></li>
<li class="chapter" data-level="3.10.5" data-path="R-and-stats.html"><a href="R-and-stats.html#miscellany"><i class="fa fa-check"></i><b>3.10.5</b> Miscellany</a></li>
<li class="chapter" data-level="3.10.6" data-path="R-and-stats.html"><a href="R-and-stats.html#my-number-one-piece-of-advice"><i class="fa fa-check"></i><b>3.10.6</b> My number one piece of advice</a></li>
<li class="chapter" data-level="3.10.7" data-path="R-and-stats.html"><a href="R-and-stats.html#learn-more"><i class="fa fa-check"></i><b>3.10.7</b> Learn more</a></li>
<li class="chapter" data-level="3.10.8" data-path="R-and-stats.html"><a href="R-and-stats.html#packages"><i class="fa fa-check"></i><b>3.10.8</b> Packages</a></li>
<li class="chapter" data-level="3.10.9" data-path="R-and-stats.html"><a href="R-and-stats.html#the-tidyverse"><i class="fa fa-check"></i><b>3.10.9</b> The Tidyverse</a></li>
<li class="chapter" data-level="3.10.10" data-path="R-and-stats.html"><a href="R-and-stats.html#other-resources"><i class="fa fa-check"></i><b>3.10.10</b> Other resources</a></li>
</ul></li>
<li class="chapter" data-level="3.11" data-path="R-and-stats.html"><a href="R-and-stats.html#list-of-commands-mainly-described-in-this-guide."><i class="fa fa-check"></i><b>3.11</b> List of commands, mainly described in this guide.</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="Trial-design.html"><a href="Trial-design.html"><i class="fa fa-check"></i><b>4</b> Trial Design.</a>
<ul>
<li class="chapter" data-level="4.1" data-path="Trial-design.html"><a href="Trial-design.html#randomised-complete-block-designs"><i class="fa fa-check"></i><b>4.1</b> Randomised complete block designs</a></li>
<li class="chapter" data-level="4.2" data-path="Trial-design.html"><a href="Trial-design.html#trial-design-in-r"><i class="fa fa-check"></i><b>4.2</b> Trial design in R</a>
<ul>
<li class="chapter" data-level="4.2.1" data-path="Trial-design.html"><a href="Trial-design.html#partially-replicated-designs"><i class="fa fa-check"></i><b>4.2.1</b> Partially replicated designs</a></li>
<li class="chapter" data-level="4.2.2" data-path="Trial-design.html"><a href="Trial-design.html#two-dimensional-designs"><i class="fa fa-check"></i><b>4.2.2</b> Two-dimensional designs</a></li>
</ul></li>
<li class="chapter" data-level="4.3" data-path="Trial-design.html"><a href="Trial-design.html#farm-scale-experiments"><i class="fa fa-check"></i><b>4.3</b> Farm scale experiments</a></li>
<li class="chapter" data-level="4.4" data-path="Trial-design.html"><a href="Trial-design.html#alpha-designs."><i class="fa fa-check"></i><b>4.4</b> Alpha designs.</a>
<ul>
<li class="chapter" data-level="4.4.1" data-path="Trial-design.html"><a href="Trial-design.html#creation-of-augmented-and-p-rep-designs-from-2-replicate-alpha-designs"><i class="fa fa-check"></i><b>4.4.1</b> Creation of augmented and p-rep designs from 2-replicate alpha designs</a></li>
<li class="chapter" data-level="4.4.2" data-path="Trial-design.html"><a href="Trial-design.html#final-message"><i class="fa fa-check"></i><b>4.4.2</b> Final message</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="5" data-path="Trial-analysis.html"><a href="Trial-analysis.html"><i class="fa fa-check"></i><b>5</b> Trial analysis (single-site).</a>
<ul>
<li class="chapter" data-level="5.1" data-path="Trial-analysis.html"><a href="Trial-analysis.html#the-layout-of-the-sugar-beet-trial"><i class="fa fa-check"></i><b>5.1</b> The layout of the sugar beet trial</a></li>
<li class="chapter" data-level="5.2" data-path="Trial-analysis.html"><a href="Trial-analysis.html#fixed-effects-random-effects-and-reml"><i class="fa fa-check"></i><b>5.2</b> Fixed effects, random effects and REML</a>
<ul>
<li class="chapter" data-level="5.2.1" data-path="Trial-analysis.html"><a href="Trial-analysis.html#preparing-the-data"><i class="fa fa-check"></i><b>5.2.1</b> Preparing the data</a></li>
<li class="chapter" data-level="5.2.2" data-path="Trial-analysis.html"><a href="Trial-analysis.html#randomised-complete-block-design"><i class="fa fa-check"></i><b>5.2.2</b> Randomised complete block design</a></li>
<li class="chapter" data-level="5.2.3" data-path="Trial-analysis.html"><a href="Trial-analysis.html#randomised-compelete-block-design---with-mixed-models"><i class="fa fa-check"></i><b>5.2.3</b> Randomised compelete block design - with mixed models</a></li>
<li class="chapter" data-level="5.2.4" data-path="Trial-analysis.html"><a href="Trial-analysis.html#plotting-residuals-against-field-structure."><i class="fa fa-check"></i><b>5.2.4</b> Plotting residuals against field structure.</a></li>
<li class="chapter" data-level="5.2.5" data-path="Trial-analysis.html"><a href="Trial-analysis.html#alpha-design-include-block-structure"><i class="fa fa-check"></i><b>5.2.5</b> Alpha design (include block structure)</a></li>
<li class="chapter" data-level="5.2.6" data-path="Trial-analysis.html"><a href="Trial-analysis.html#rows-and-column-analysis"><i class="fa fa-check"></i><b>5.2.6</b> Rows and column analysis</a></li>
</ul></li>
<li class="chapter" data-level="5.3" data-path="Trial-analysis.html"><a href="Trial-analysis.html#spatial-analysis"><i class="fa fa-check"></i><b>5.3</b> Spatial analysis</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="Cross-site-analysis.html"><a href="Cross-site-analysis.html"><i class="fa fa-check"></i><b>6</b> Trial analysis (cross-sites).</a>
<ul>
<li class="chapter" data-level="6.1" data-path="Cross-site-analysis.html"><a href="Cross-site-analysis.html#blues-blups-and-one-stage-analysis"><i class="fa fa-check"></i><b>6.1</b> BLUEs, BLUPs and one stage analysis</a>
<ul>
<li class="chapter" data-level="6.1.1" data-path="Cross-site-analysis.html"><a href="Cross-site-analysis.html#one-stage-analysis-with-weights"><i class="fa fa-check"></i><b>6.1.1</b> One stage analysis with weights</a></li>
</ul></li>
<li class="chapter" data-level="6.2" data-path="Cross-site-analysis.html"><a href="Cross-site-analysis.html#weighted-two-stage-analysis"><i class="fa fa-check"></i><b>6.2</b> Weighted two-stage analysis</a></li>
<li class="chapter" data-level="6.3" data-path="Cross-site-analysis.html"><a href="Cross-site-analysis.html#ammi"><i class="fa fa-check"></i><b>6.3</b> AMMI</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="Population-genetics.html"><a href="Population-genetics.html"><i class="fa fa-check"></i><b>7</b> Population genetics.</a>
<ul>
<li class="chapter" data-level="7.1" data-path="Population-genetics.html"><a href="Population-genetics.html#hardy-weinberg-equilibrium-and-population-structure"><i class="fa fa-check"></i><b>7.1</b> Hardy-Weinberg equilibrium and population structure</a></li>
<li class="chapter" data-level="7.2" data-path="Population-genetics.html"><a href="Population-genetics.html#other-ways-of-analysing-population-structure"><i class="fa fa-check"></i><b>7.2</b> Other ways of analysing population structure</a>
<ul>
<li class="chapter" data-level="7.2.1" data-path="Population-genetics.html"><a href="Population-genetics.html#distance-trees"><i class="fa fa-check"></i><b>7.2.1</b> Distance Trees</a></li>
</ul></li>
<li class="chapter" data-level="7.3" data-path="Population-genetics.html"><a href="Population-genetics.html#structure"><i class="fa fa-check"></i><b>7.3</b> Structure</a>
<ul>
<li class="chapter" data-level="7.3.1" data-path="Population-genetics.html"><a href="Population-genetics.html#what-structure-does"><i class="fa fa-check"></i><b>7.3.1</b> What STRUCTURE does</a></li>
<li class="chapter" data-level="7.3.2" data-path="Population-genetics.html"><a href="Population-genetics.html#limitations-of-structure"><i class="fa fa-check"></i><b>7.3.2</b> Limitations of STRUCTURE</a></li>
<li class="chapter" data-level="7.3.3" data-path="Population-genetics.html"><a href="Population-genetics.html#using-structure"><i class="fa fa-check"></i><b>7.3.3</b> Using Structure</a></li>
<li class="chapter" data-level="7.3.4" data-path="Population-genetics.html"><a href="Population-genetics.html#the-exercise"><i class="fa fa-check"></i><b>7.3.4</b> The exercise</a></li>
</ul></li>
<li class="chapter" data-level="7.4" data-path="Population-genetics.html"><a href="Population-genetics.html#linkage-disequilibrium-analysis"><i class="fa fa-check"></i><b>7.4</b> Linkage disequilibrium analysis</a>
<ul>
<li class="chapter" data-level="7.4.1" data-path="Population-genetics.html"><a href="Population-genetics.html#linkage-disequilibrium-in-the-teosinte-and-barley-data-sets"><i class="fa fa-check"></i><b>7.4.1</b> Linkage disequilibrium in the teosinte and barley data sets</a></li>
</ul></li>
<li class="chapter" data-level="7.5" data-path="Population-genetics.html"><a href="Population-genetics.html#recombination-and-linkage-disequilibrium-applications"><i class="fa fa-check"></i><b>7.5</b> Recombination and linkage disequilibrium: applications</a></li>
<li class="chapter" data-level="7.6" data-path="Population-genetics.html"><a href="Population-genetics.html#pedigree-analysis"><i class="fa fa-check"></i><b>7.6</b> Pedigree analysis</a></li>
</ul></li>
<li class="chapter" data-level="8" data-path="Imputation.html"><a href="Imputation.html"><i class="fa fa-check"></i><b>8</b> Imputation.</a>
<ul>
<li class="chapter" data-level="8.1" data-path="Imputation.html"><a href="Imputation.html#introduction"><i class="fa fa-check"></i><b>8.1</b> Introduction</a></li>
<li class="chapter" data-level="8.2" data-path="Imputation.html"><a href="Imputation.html#the-exercise-1"><i class="fa fa-check"></i><b>8.2</b> The exercise</a></li>
<li class="chapter" data-level="8.3" data-path="Imputation.html"><a href="Imputation.html#data-preparation-for-all-methods"><i class="fa fa-check"></i><b>8.3</b> Data preparation (for all methods)</a></li>
<li class="chapter" data-level="8.4" data-path="Imputation.html"><a href="Imputation.html#svd-imputation"><i class="fa fa-check"></i><b>8.4</b> SVD imputation</a></li>
<li class="chapter" data-level="8.5" data-path="Imputation.html"><a href="Imputation.html#mean-imputation"><i class="fa fa-check"></i><b>8.5</b> Mean imputation</a></li>
<li class="chapter" data-level="8.6" data-path="Imputation.html"><a href="Imputation.html#random-forest-imputation"><i class="fa fa-check"></i><b>8.6</b> Random Forest Imputation</a></li>
<li class="chapter" data-level="8.7" data-path="Imputation.html"><a href="Imputation.html#summary-1"><i class="fa fa-check"></i><b>8.7</b> Summary</a></li>
<li class="chapter" data-level="8.8" data-path="Imputation.html"><a href="Imputation.html#conclusions-and-comments"><i class="fa fa-check"></i><b>8.8</b> Conclusions and comments</a></li>
</ul></li>
<li class="chapter" data-level="9" data-path="Association-mapping.html"><a href="Association-mapping.html"><i class="fa fa-check"></i><b>9</b> Association Mapping.</a>
<ul>
<li class="chapter" data-level="9.1" data-path="Association-mapping.html"><a href="Association-mapping.html#introduction-1"><i class="fa fa-check"></i><b>9.1</b> Introduction</a></li>
<li class="chapter" data-level="9.2" data-path="Association-mapping.html"><a href="Association-mapping.html#preparing-the-data-set"><i class="fa fa-check"></i><b>9.2</b> Preparing the data set</a></li>
<li class="chapter" data-level="9.3" data-path="Association-mapping.html"><a href="Association-mapping.html#running-gwaspoly"><i class="fa fa-check"></i><b>9.3</b> Running GWASpoly</a></li>
<li class="chapter" data-level="9.4" data-path="Association-mapping.html"><a href="Association-mapping.html#the-exercise-2"><i class="fa fa-check"></i><b>9.4</b> The exercise</a>
<ul>
<li class="chapter" data-level="9.4.1" data-path="Association-mapping.html"><a href="Association-mapping.html#calculation-and-comparison-of-kinship"><i class="fa fa-check"></i><b>9.4.1</b> Calculation and comparison of kinship</a></li>
<li class="chapter" data-level="9.4.2" data-path="Association-mapping.html"><a href="Association-mapping.html#gwas"><i class="fa fa-check"></i><b>9.4.2</b> GWAS</a></li>
<li class="chapter" data-level="9.4.3" data-path="Association-mapping.html"><a href="Association-mapping.html#analysis-with-covariates"><i class="fa fa-check"></i><b>9.4.3</b> Analysis with covariates</a></li>
<li class="chapter" data-level="9.4.4" data-path="Association-mapping.html"><a href="Association-mapping.html#accounting-for-population-structure"><i class="fa fa-check"></i><b>9.4.4</b> Accounting for population structure</a></li>
<li class="chapter" data-level="9.4.5" data-path="Association-mapping.html"><a href="Association-mapping.html#further-analysis"><i class="fa fa-check"></i><b>9.4.5</b> Further analysis</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="10" data-path="Mapping-Markers.html"><a href="Mapping-Markers.html"><i class="fa fa-check"></i><b>10</b> Mapping Markers and Quantitative Trait Loci.</a>
<ul>
<li class="chapter" data-level="10.1" data-path="Mapping-Markers.html"><a href="Mapping-Markers.html#introduction-2"><i class="fa fa-check"></i><b>10.1</b> Introduction</a></li>
<li class="chapter" data-level="10.2" data-path="Mapping-Markers.html"><a href="Mapping-Markers.html#constructing-a-genetic-map"><i class="fa fa-check"></i><b>10.2</b> Constructing a genetic map</a>
<ul>
<li class="chapter" data-level="10.2.1" data-path="Mapping-Markers.html"><a href="Mapping-Markers.html#formatting-and-reading-in-the-data"><i class="fa fa-check"></i><b>10.2.1</b> Formatting and reading in the data</a></li>
<li class="chapter" data-level="10.2.2" data-path="Mapping-Markers.html"><a href="Mapping-Markers.html#a-quick-map"><i class="fa fa-check"></i><b>10.2.2</b> A quick map!</a></li>
<li class="chapter" data-level="10.2.3" data-path="Mapping-Markers.html"><a href="Mapping-Markers.html#estimating-recombination-fractions"><i class="fa fa-check"></i><b>10.2.3</b> Estimating recombination fractions</a></li>
<li class="chapter" data-level="10.2.4" data-path="Mapping-Markers.html"><a href="Mapping-Markers.html#forming-linkage-groups"><i class="fa fa-check"></i><b>10.2.4</b> Forming Linkage groups</a></li>
<li class="chapter" data-level="10.2.5" data-path="Mapping-Markers.html"><a href="Mapping-Markers.html#marker-ordering"><i class="fa fa-check"></i><b>10.2.5</b> Marker ordering</a></li>
<li class="chapter" data-level="10.2.6" data-path="Mapping-Markers.html"><a href="Mapping-Markers.html#improving-the-map"><i class="fa fa-check"></i><b>10.2.6</b> Improving the map</a></li>
<li class="chapter" data-level="10.2.7" data-path="Mapping-Markers.html"><a href="Mapping-Markers.html#quality-control-commands-for-marker-qc-and-mapping"><i class="fa fa-check"></i><b>10.2.7</b> Quality Control commands for marker QC and mapping</a></li>
<li class="chapter" data-level="10.2.8" data-path="Mapping-Markers.html"><a href="Mapping-Markers.html#asmap"><i class="fa fa-check"></i><b>10.2.8</b> ASMap</a></li>
</ul></li>
<li class="chapter" data-level="10.3" data-path="Mapping-Markers.html"><a href="Mapping-Markers.html#qtl-mapping-in-rqtl"><i class="fa fa-check"></i><b>10.3</b> QTL mapping in R/qtl</a>
<ul>
<li class="chapter" data-level="10.3.1" data-path="Mapping-Markers.html"><a href="Mapping-Markers.html#genome-scan-single-qtl"><i class="fa fa-check"></i><b>10.3.1</b> Genome Scan – single QTL</a></li>
<li class="chapter" data-level="10.3.2" data-path="Mapping-Markers.html"><a href="Mapping-Markers.html#genome-scan-covariates"><i class="fa fa-check"></i><b>10.3.2</b> Genome Scan – covariates</a></li>
<li class="chapter" data-level="10.3.3" data-path="Mapping-Markers.html"><a href="Mapping-Markers.html#genome-scan-scantwo"><i class="fa fa-check"></i><b>10.3.3</b> Genome Scan – scantwo</a></li>
<li class="chapter" data-level="10.3.4" data-path="Mapping-Markers.html"><a href="Mapping-Markers.html#genome-scan-composite-interval-mapping"><i class="fa fa-check"></i><b>10.3.4</b> Genome Scan – Composite Interval Mapping</a></li>
<li class="chapter" data-level="10.3.5" data-path="Mapping-Markers.html"><a href="Mapping-Markers.html#mqm-multiple-qtl-mapping"><i class="fa fa-check"></i><b>10.3.5</b> MQM: multiple QTL mapping</a></li>
<li class="chapter" data-level="10.3.6" data-path="Mapping-Markers.html"><a href="Mapping-Markers.html#some-commands-for-mapping-in-markers-in-rqtl"><i class="fa fa-check"></i><b>10.3.6</b> Some commands for mapping in markers in R/qtl</a></li>
<li class="chapter" data-level="10.3.7" data-path="Mapping-Markers.html"><a href="Mapping-Markers.html#final-comments"><i class="fa fa-check"></i><b>10.3.7</b> Final comments</a></li>
</ul></li>
<li class="chapter" data-level="10.4" data-path="Mapping-Markers.html"><a href="Mapping-Markers.html#rqtl2"><i class="fa fa-check"></i><b>10.4</b> R/QTL2</a></li>
</ul></li>
<li class="chapter" data-level="11" data-path="Data-Visualization.html"><a href="Data-Visualization.html"><i class="fa fa-check"></i><b>11</b> Data Visualization.</a>
<ul>
<li class="chapter" data-level="11.1" data-path="Data-Visualization.html"><a href="Data-Visualization.html#flapjack"><i class="fa fa-check"></i><b>11.1</b> Flapjack</a>
<ul>
<li class="chapter" data-level="11.1.1" data-path="Data-Visualization.html"><a href="Data-Visualization.html#genotype-data-.geno-file."><i class="fa fa-check"></i><b>11.1.1</b> Genotype data: “.geno” file.</a></li>
<li class="chapter" data-level="11.1.2" data-path="Data-Visualization.html"><a href="Data-Visualization.html#map-data-.map-file."><i class="fa fa-check"></i><b>11.1.2</b> Map data : “.MAP” file.</a></li>
<li class="chapter" data-level="11.1.3" data-path="Data-Visualization.html"><a href="Data-Visualization.html#trait-data"><i class="fa fa-check"></i><b>11.1.3</b> Trait data</a></li>
<li class="chapter" data-level="11.1.4" data-path="Data-Visualization.html"><a href="Data-Visualization.html#the-exercise-3"><i class="fa fa-check"></i><b>11.1.4</b> The Exercise</a></li>
</ul></li>
<li class="chapter" data-level="11.2" data-path="Data-Visualization.html"><a href="Data-Visualization.html#helium-pedigree-visualization-software"><i class="fa fa-check"></i><b>11.2</b> Helium pedigree visualization software</a>
<ul>
<li class="chapter" data-level="11.2.1" data-path="Data-Visualization.html"><a href="Data-Visualization.html#introduction-3"><i class="fa fa-check"></i><b>11.2.1</b> Introduction</a></li>
<li class="chapter" data-level="11.2.2" data-path="Data-Visualization.html"><a href="Data-Visualization.html#the-exercise-4"><i class="fa fa-check"></i><b>11.2.2</b> The exercise</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="12" data-path="Quantitative-genetics.html"><a href="Quantitative-genetics.html"><i class="fa fa-check"></i><b>12</b> Quantitative genetics and breeding program design.</a>
<ul>
<li class="chapter" data-level="12.1" data-path="Quantitative-genetics.html"><a href="Quantitative-genetics.html#means-and-variances-in-some-simple-models"><i class="fa fa-check"></i><b>12.1</b> Means and variances in some simple models</a>
<ul>
<li class="chapter" data-level="12.1.1" data-path="Quantitative-genetics.html"><a href="Quantitative-genetics.html#additive-dominance-model-of-means-in-a-tomato-cross"><i class="fa fa-check"></i><b>12.1.1</b> Additive-dominance model of means in a tomato cross</a></li>
<li class="chapter" data-level="12.1.2" data-path="Quantitative-genetics.html"><a href="Quantitative-genetics.html#variances-in-a-barley-nam-population"><i class="fa fa-check"></i><b>12.1.2</b> Variances in a barley NAM population</a></li>
</ul></li>
<li class="chapter" data-level="12.2" data-path="Quantitative-genetics.html"><a href="Quantitative-genetics.html#selecting-in-backcrosses"><i class="fa fa-check"></i><b>12.2</b> Selecting in backcrosses</a></li>
<li class="chapter" data-level="12.3" data-path="Quantitative-genetics.html"><a href="Quantitative-genetics.html#size-of-breeding-programme-cycle-time-and-recurrent-selection-schemes."><i class="fa fa-check"></i><b>12.3</b> Size of breeding programme, cycle time and recurrent selection schemes.</a>
<ul>
<li class="chapter" data-level="12.3.1" data-path="Quantitative-genetics.html"><a href="Quantitative-genetics.html#size-of-breeding-programme"><i class="fa fa-check"></i><b>12.3.1</b> Size of breeding programme</a></li>
<li class="chapter" data-level="12.3.2" data-path="Quantitative-genetics.html"><a href="Quantitative-genetics.html#speed-of-selection"><i class="fa fa-check"></i><b>12.3.2</b> Speed of selection</a></li>
<li class="chapter" data-level="12.3.3" data-path="Quantitative-genetics.html"><a href="Quantitative-genetics.html#heritability"><i class="fa fa-check"></i><b>12.3.3</b> Heritability</a></li>
<li class="chapter" data-level="12.3.4" data-path="Quantitative-genetics.html"><a href="Quantitative-genetics.html#sequential-selection"><i class="fa fa-check"></i><b>12.3.4</b> Sequential Selection</a></li>
<li class="chapter" data-level="12.3.5" data-path="Quantitative-genetics.html"><a href="Quantitative-genetics.html#optimal-contribution-theory"><i class="fa fa-check"></i><b>12.3.5</b> Optimal Contribution Theory</a></li>
</ul></li>
<li class="chapter" data-level="12.4" data-path="Quantitative-genetics.html"><a href="Quantitative-genetics.html#simulations-using-alphasim"><i class="fa fa-check"></i><b>12.4</b> Simulations using AlphaSim</a></li>
</ul></li>
<li class="chapter" data-level="13" data-path="Genomic-prediction.html"><a href="Genomic-prediction.html"><i class="fa fa-check"></i><b>13</b> Genomic prediction.</a>
<ul>
<li class="chapter" data-level="13.1" data-path="Genomic-prediction.html"><a href="Genomic-prediction.html#ridge-regression-and-genomic-prediction-in-excel"><i class="fa fa-check"></i><b>13.1</b> Ridge regression and Genomic Prediction in Excel</a>
<ul>
<li class="chapter" data-level="13.1.1" data-path="Genomic-prediction.html"><a href="Genomic-prediction.html#introduction-4"><i class="fa fa-check"></i><b>13.1.1</b> Introduction</a></li>
<li class="chapter" data-level="13.1.2" data-path="Genomic-prediction.html"><a href="Genomic-prediction.html#the-exercise-5"><i class="fa fa-check"></i><b>13.1.2</b> The Exercise</a></li>
</ul></li>
<li class="chapter" data-level="13.2" data-path="Genomic-prediction.html"><a href="Genomic-prediction.html#genomic-prediction-with-rrblup"><i class="fa fa-check"></i><b>13.2</b> Genomic Prediction with rrBLUP</a>
<ul>
<li class="chapter" data-level="13.2.1" data-path="Genomic-prediction.html"><a href="Genomic-prediction.html#introduction-5"><i class="fa fa-check"></i><b>13.2.1</b> Introduction</a></li>
<li class="chapter" data-level="13.2.2" data-path="Genomic-prediction.html"><a href="Genomic-prediction.html#the-dataset"><i class="fa fa-check"></i><b>13.2.2</b> The dataset</a></li>
<li class="chapter" data-level="13.2.3" data-path="Genomic-prediction.html"><a href="Genomic-prediction.html#loading-and-preparing-the-data"><i class="fa fa-check"></i><b>13.2.3</b> Loading and preparing the data</a></li>
<li class="chapter" data-level="13.2.4" data-path="Genomic-prediction.html"><a href="Genomic-prediction.html#examining-the-data-structure"><i class="fa fa-check"></i><b>13.2.4</b> Examining the data structure</a></li>
<li class="chapter" data-level="13.2.5" data-path="Genomic-prediction.html"><a href="Genomic-prediction.html#ordinary-least-squares"><i class="fa fa-check"></i><b>13.2.5</b> Ordinary least squares</a></li>
<li class="chapter" data-level="13.2.6" data-path="Genomic-prediction.html"><a href="Genomic-prediction.html#rrblup"><i class="fa fa-check"></i><b>13.2.6</b> rrBLUP</a></li>
<li class="chapter" data-level="13.2.7" data-path="Genomic-prediction.html"><a href="Genomic-prediction.html#prediction-forward-in-time"><i class="fa fa-check"></i><b>13.2.7</b> Prediction forward in time</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="14" data-path="Breeding-simulations.html"><a href="Breeding-simulations.html"><i class="fa fa-check"></i><b>14</b> Breeding simulations.</a>
<ul>
<li class="chapter" data-level="14.1" data-path="Breeding-simulations.html"><a href="Breeding-simulations.html#introduction-6"><i class="fa fa-check"></i><b>14.1</b> Introduction</a></li>
<li class="chapter" data-level="14.2" data-path="Breeding-simulations.html"><a href="Breeding-simulations.html#exercise"><i class="fa fa-check"></i><b>14.2</b> Exercise</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="references.html"><a href="references.html"><i class="fa fa-check"></i>References</a></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Quantitative Methods in Plant Breeding: The Digital Tutorials.</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="Mapping-Markers" class="section level1" number="10">
<h1><span class="header-section-number">Chapter 10</span> Mapping Markers and Quantitative Trait Loci.</h1>
<p><em>Mapping Markers and Quantitative Trait Loci with R/qtl (and ASMap)</em></p>
<div id="introduction-2" class="section level2" number="10.1">
<h2><span class="header-section-number">10.1</span> Introduction</h2>
<p>There are several excellent mapping programmes available. A list of some, suitable for plant breeders is found on the R/qtl <a href="http://www.rqtl.org/">homepage.</a> Software for
creating maps of markers is less common than software for mapping QTL and most has
specific limitations in terms of speed, population types, handling of missing data and so
on. The gold standard is <a href="https://www.kyazma.nl/">Joinmap,</a> but this is expensive. In fact
it’s so expensive that they don’t put the price on their web site.</p>
<p>For this course we are going to use R/qtl. This is a popular, easy-to-use and well supported package and there is a useful book on QTL mapping associated with it: “A
Guide to QTL Mapping with R/qtl.” Details are available <a href="http://www.rqtl.org/book/">here.</a> There is also an R/qtl <a href="http://www.rqtl.org/">web site,</a> which
includes much useful information, including tutorials and there is a <a href="http://groups.google.com/group/Rqtl-disc">discussion group.</a> A quick tour of R/qtl can be found <a href="http://www.rqtl.org/tutorials/rqtltour.pdf">here</a>. R/qtl was originally written to map QTL and not to create linkage maps of markers. However, it will create marker maps, although it can be slow compared to alternative programs. However, in this tutorial we shall use R/qtl both to create a linkage map and then to map QTL onto that map.</p>
<p>A rewritten version, R/qtl2 is now available, which is faster and better able to handle
large data sets and complex cross-designs, including MAGIC. It also offers an automated
marker thinning option for kinship matrix estimation (hooray!) and includes options to
go beyond QTL detection to searching genomic databases for candidate SNP variants. It
is available from <a href="http://kbroman.org/qtl2/">here</a> but not yet on CRAN. We are currently trying it out in our work at NIAB.</p>
<p>Also relatively new, and available through CRAN, is the R package ASMap. This creates marker maps for basic biparental populations, but you would have to use R/qtl to map traits. The description of it is <a href="https://www.jstatsoft.org/article/view/v079i06">here</a>. The R manual for the package can be found <a href="https://cran.r-project.org/web/packages/ASMap/ASMap.pdf">here</a>. Its main benefit over R/qtl is in utilising a more advanced, much faster algorithm for clustering and marker ordering. We shall also have a brief look at this too.</p>

<div class="rmdnote">
A R script for this tutorial has been provided with the data, either work from that or copy and paste the chunks you wish to run from the book.
</div>
</div>
<div id="constructing-a-genetic-map" class="section level2" number="10.2">
<h2><span class="header-section-number">10.2</span> Constructing a genetic map</h2>
<p>Most (all?) statistical packages for mapping markers start with a square matrix of recombination fractions. From this two-dimensional matrix, markers are partitioned into linkage groups by some form of clustering. Hopefully each cluster or linkage group belongs uniquely to a single chromosome. However, stringent significant thresholds must be employed in this clustering to reduce the number of falsely linked markers. As a result, a numbers of markers are often left unlinked or in clusters with only a very small number of members. In practice, therefore, it is common to identify more linkage groups than chromosomes. With modern genomic resources, these linkage groups can often be subsequently linked to specific chromosomes.</p>
<p>Following clustering, for each linkage group in turn, markers are projected from the two-dimensional array of recombination fractions into a linear order. There are various methods for carrying out this linearization. Usually, as a final stage, the linear order is tested by some form of perturbation of the initial order to see if a new order can be found which is judged better: reducing the total number of recombinations required to create the observed set of chromosomes for example.</p>
<p>The data we are going to analyse are from a DH mapping population in UK winter wheat: Avalon x Cadenza. Data were produced as part of an experiment to map wheat soil borne mosaic virus.</p>
<div id="formatting-and-reading-in-the-data" class="section level3" number="10.2.1">
<h3><span class="header-section-number">10.2.1</span> Formatting and reading in the data</h3>
<p>Data are in the file: “AxC_R.xlsx.” Note:</p>
<p>Row 1: in separate cells, column names for ID, phenotypes, markers.</p>
<p>Row 2: in the cells immediately under the marker names, the chromosome identifier for
that marker. Here that identifier is 50 for all markers (we could have used anything except x – reserved for sex chromosomes. If we had a pre-existing map we wished to use, obviously this identifier would be for the chromosome or linkage group.</p>
<p>Row 3: is missing here as it is not required. But if we were using a pre-existing map, then on this line, under its marker name, would be the position in cM for that marker (see
the QTL mapping tutorial). Data are read in Haldane map units. You will see this after
we have created a map and exported our data with the new map to a file.</p>
<p>Coding for marker data follows that used by Mapmaker, one of the original pieces of
mapping software (slow by current standards and hard to use):</p>
<pre><code>genotype    default coding  
AA            A               homozygous from parent A 
Aa            H               heterozygous 
aa            B               homozygous from parent B 
A_            D               for dominant markers, homozygous in parent A 
a_            C               for dominant markers, homozygous in parent B</code></pre>
<p>This is the basic coding. It has been expanded to deal with more complex cross types which R/qtl can also handle.</p>
<p>Should we drop any markers? Should we drop any individuals?</p>
<p>The dataset we are using has already been through a QC process which has removed really bad markers and individuals. This can be done without a map by looking at marker failure rates and segregation distortion – over markers and over individuals. Some of these checks can be done by the mapping software, but they are easy to do in Excel too:</p>
<p>To calculate a chi-sq with 1 df to test for 1:1 segregation you can use the quick formula:</p>
<p><span class="math display">\[(A-B)^2/(A+B)\]</span></p>
<p>This is just a simplification of the usual <span class="math inline">\(Σ(O-E)^2/E\)</span> (O= observed numbers, E = expected numbers) for a chi-sq test. It only applies when there are two classes and the expected are in the ratio 1:1. I prefer to do this prior to running any mapping programs but it is possible within R/qtl.</p>
<p>After you are happy with the data, then we read it into R. R cannot read these data directly from Excel or from the clipboard. First you must save the data as a csv file. This has already been done, to create the file AxC_R.csv.</p>
<p>Within R Studio, set the directory and load the package <code>qtl</code>. You will need to install from CRAN if you don’t have it already.</p>
<div class="sourceCode" id="cb524"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb524-1"><a href="Mapping-Markers.html#cb524-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(qtl)</span></code></pre></div>
<p>We can now read data in using the command <code>read.cross</code>.</p>
<div class="sourceCode" id="cb525"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb525-1"><a href="Mapping-Markers.html#cb525-1" aria-hidden="true" tabindex="-1"></a>AxC <span class="ot">&lt;-</span> <span class="fu">read.cross</span>(<span class="st">&quot;csv&quot;</span>, <span class="at">dir =</span> <span class="st">&quot;data&quot;</span>, <span class="st">&quot;AxC_R.csv&quot;</span>, <span class="at">genotypes =</span> <span class="fu">c</span>(<span class="st">&quot;A&quot;</span>, </span>
<span id="cb525-2"><a href="Mapping-Markers.html#cb525-2" aria-hidden="true" tabindex="-1"></a>    <span class="st">&quot;B&quot;</span>), <span class="at">estimate.map =</span> F)</span></code></pre></div>
<pre><code>##  --Read the following data:
##   78  individuals
##   186  markers
##   10  phenotypes
##  --Cross type: bc</code></pre>
<p>An option to note here is <code>estimate.map=F</code>. This is required to declare that we are not inputting a map with the data, You need the <code>dir="data"</code> in read.cross to declare the path to the file. However, you won’t have the directory ‘data’ on your computers. If you just want to call data from your working directory, like most of us do, just write this as <code>dir=""</code>. You will want to be working in the same directory as the data, as we have done for previous tutorials.</p>
<p>Take a look at the output from reading the file in. The last line indicates that the cross has been identified as a backcross from the data format: because only two marker classes were found at each locus. In fact, the cross type is a doubled haploid population. For marker and trait mapping purposes this makes absolutely no difference. Do you know why not? Nevertheless, we can change the cross to dh as below:</p>
<div class="sourceCode" id="cb527"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb527-1"><a href="Mapping-Markers.html#cb527-1" aria-hidden="true" tabindex="-1"></a><span class="fu">class</span>(AxC)[<span class="dv">1</span>] <span class="ot">&lt;-</span> <span class="st">&quot;dh&quot;</span></span></code></pre></div>
<p>If you read in a file which already has a map with some of the markers mapped to the same position you will get a warning message. To avoid problems in analysing the data, R/qtl uses the command <code>jittermap()</code> to add a tiny fraction to every interval on chromosomes where this occurs. We would need to overwrite the original data, or otherwise save the output (e.g. <code>AxC&lt;-jittermap(AxC)</code>) but there is no need to do that here.</p>
</div>
<div id="a-quick-map" class="section level3" number="10.2.2">
<h3><span class="header-section-number">10.2.2</span> A quick map!</h3>
<p>Remember we have just read the data into the data table <code>AxC</code>. As with many R objects, the commands summary and plot give object specific details of the data contained within:</p>
<div class="sourceCode" id="cb528"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb528-1"><a href="Mapping-Markers.html#cb528-1" aria-hidden="true" tabindex="-1"></a><span class="fu">summary</span>(AxC)</span></code></pre></div>
<pre><code>##     Doubled haploids
## 
##     No. individuals:    78 
## 
##     No. phenotypes:     10 
##     Percent phenotyped: 100 94.9 100 100 100 100 100 100 100 100 
## 
##     No. chromosomes:    1 
##         Autosomes:      50 
## 
##     Total markers:      186 
##     No. markers:        186 
##     Percent genotyped:  96.1 
##     Genotypes (%):      AA:50.2  BB:49.8</code></pre>
<div class="sourceCode" id="cb530"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb530-1"><a href="Mapping-Markers.html#cb530-1" aria-hidden="true" tabindex="-1"></a><span class="fu">plot</span>(AxC)</span></code></pre></div>
<p><img src="bookdown-demo_files/figure-html/unnamed-chunk-403-1.png" width="672" /></p>
<p>Before we go into detail about creating a map, we shall first do so very quickly to show that this can be done simply and with only a few commands:</p>
<div class="sourceCode" id="cb531"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb531-1"><a href="Mapping-Markers.html#cb531-1" aria-hidden="true" tabindex="-1"></a>quick.map <span class="ot">&lt;-</span> <span class="fu">est.rf</span>(AxC)</span>
<span id="cb531-2"><a href="Mapping-Markers.html#cb531-2" aria-hidden="true" tabindex="-1"></a>quick.map <span class="ot">&lt;-</span> <span class="fu">formLinkageGroups</span>(quick.map, <span class="at">reorgMarkers =</span> T)</span>
<span id="cb531-3"><a href="Mapping-Markers.html#cb531-3" aria-hidden="true" tabindex="-1"></a>quick.map <span class="ot">&lt;-</span> <span class="fu">orderMarkers</span>(quick.map)</span>
<span id="cb531-4"><a href="Mapping-Markers.html#cb531-4" aria-hidden="true" tabindex="-1"></a><span class="fu">plotMap</span>(quick.map)</span></code></pre></div>
<p><img src="bookdown-demo_files/figure-html/unnamed-chunk-404-1.png" width="672" /></p>
<p>This should produce the map shown above, which doesn’t look bad, though we may wonder about the length of linkage group one, seemingly with some very large inter marker distances. While most mapping packages can probably produce maps as quickly and easily as this, it doesn’t mean that they are correct. It is worth understanding how and why they work so that you can better judge the quality of the map and seek improvements to it.</p>
</div>
<div id="estimating-recombination-fractions" class="section level3" number="10.2.3">
<h3><span class="header-section-number">10.2.3</span> Estimating recombination fractions</h3>
<p>First delete quick.map</p>
<div class="sourceCode" id="cb532"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb532-1"><a href="Mapping-Markers.html#cb532-1" aria-hidden="true" tabindex="-1"></a><span class="fu">rm</span>(quick.map)</span></code></pre></div>
<p>Now, the first thing to do is estimate the recombination frequencies among all pairs of loci:</p>
<div class="sourceCode" id="cb533"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb533-1"><a href="Mapping-Markers.html#cb533-1" aria-hidden="true" tabindex="-1"></a>AxC <span class="ot">&lt;-</span> <span class="fu">est.rf</span>(AxC)</span></code></pre></div>
<p>Note that we have directed the output of <code>est.rf</code> back to <code>AxC</code> to which it is added.</p>
<p>We can inspect the matrix of recombination frequencies. However, this is a large table, so we’ll just look at the top left hand corner:</p>
<div class="sourceCode" id="cb534"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb534-1"><a href="Mapping-Markers.html#cb534-1" aria-hidden="true" tabindex="-1"></a><span class="fu">pull.rf</span>(AxC)[<span class="dv">1</span><span class="sc">:</span><span class="dv">5</span>, <span class="dv">1</span><span class="sc">:</span><span class="dv">5</span>]</span></code></pre></div>
<pre><code>##              Aflp100.175     AFLP24 AFLP24.320 AFLP43-160Bp Aflp57.165
## Aflp100.175           NA 0.44594595 0.46052632   0.01351351 0.46666667
## AFLP24        0.44594595         NA 0.01369863   0.45833333 0.00000000
## AFLP24.320    0.46052632 0.01369863         NA   0.47945205 0.01351351
## AFLP43-160Bp  0.01351351 0.45833333 0.47945205           NA 0.46575342
## Aflp57.165    0.46666667 0.00000000 0.01351351   0.46575342         NA</code></pre>
<p>Rather fortuitously, you can see some very low recombination fractions and some indicating clear lack of linkage. Since we are working in R, we have access to all its other features. For instance, we can look at the distribution of recombination fractions:</p>
<div class="sourceCode" id="cb536"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb536-1"><a href="Mapping-Markers.html#cb536-1" aria-hidden="true" tabindex="-1"></a><span class="fu">hist</span>(<span class="fu">pull.rf</span>(AxC), <span class="at">breaks =</span> <span class="dv">40</span>)</span></code></pre></div>
<p><img src="bookdown-demo_files/figure-html/unnamed-chunk-408-1.png" width="672" /></p>
<p>Does this help you put a threshold on recombination fraction at which you are willing to declare a pair of markers as linked? Generally, packages use a threshold on both recombination fraction and significance level. We can also extract the latter, as a LOD score:</p>
<div class="sourceCode" id="cb537"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb537-1"><a href="Mapping-Markers.html#cb537-1" aria-hidden="true" tabindex="-1"></a><span class="fu">hist</span>(<span class="fu">pull.rf</span>(AxC, <span class="at">what =</span> <span class="st">&quot;lod&quot;</span>))</span></code></pre></div>
<p><img src="bookdown-demo_files/figure-html/unnamed-chunk-409-1.png" width="672" /></p>
<p>Most marker pairs show no evidence of linkage and have a low LOD score. If we truncate the y axis the plot is a little more clear:</p>
<div class="sourceCode" id="cb538"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb538-1"><a href="Mapping-Markers.html#cb538-1" aria-hidden="true" tabindex="-1"></a><span class="fu">hist</span>(<span class="fu">pull.rf</span>(AxC, <span class="at">what =</span> <span class="st">&quot;lod&quot;</span>), <span class="at">ylim =</span> <span class="fu">c</span>(<span class="dv">0</span>, <span class="dv">200</span>), <span class="at">breaks =</span> <span class="dv">40</span>)</span></code></pre></div>
<p><img src="bookdown-demo_files/figure-html/unnamed-chunk-410-1.png" width="672" /></p>
<p>Are the recombination fractions and LOD scores telling us the same thing? After a bit of trial and error, we can plot one against the other:</p>
<div class="sourceCode" id="cb539"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb539-1"><a href="Mapping-Markers.html#cb539-1" aria-hidden="true" tabindex="-1"></a><span class="fu">plot</span>(<span class="fu">pull.rf</span>(AxC, <span class="at">what =</span> <span class="st">&quot;rf&quot;</span>)[, ], <span class="fu">pull.rf</span>(AxC, <span class="at">what =</span> <span class="st">&quot;lod&quot;</span>)[, </span>
<span id="cb539-2"><a href="Mapping-Markers.html#cb539-2" aria-hidden="true" tabindex="-1"></a>    ], <span class="at">xlab =</span> <span class="st">&quot;rf&quot;</span>, <span class="at">ylab =</span> <span class="st">&quot;lod&quot;</span>)</span></code></pre></div>
<p><img src="bookdown-demo_files/figure-html/unnamed-chunk-411-1.png" width="672" /></p>

<div class="rmdquiz">
<p><strong>1</strong></p>
What does this plot tell us? What do you think would be a good combination of cut-off parameters, and why?
</div>
</div>
<div id="forming-linkage-groups" class="section level3" number="10.2.4">
<h3><span class="header-section-number">10.2.4</span> Forming Linkage groups</h3>
<p>Next we want to allocate our markers to linkage groups. The command for this is <code>formLinkageGroups</code>. It has some options that we could experiment with:</p>
<p><code>max.rf</code> = Maximum recombination fraction for placing two markers in the same linkage group. The default is 0.25.</p>
<p><code>min.lod</code> = Minimum LOD score for placing two markers in the same linkage group. The default is 3.</p>
<p><code>reorgMarkers</code> = If TRUE, the markers are organized into the inferred linkage groups. If FALSE, the output is a table indicating the initial chromosome assignments and the inferred linkage group partitions.</p>
<p>We want to add our chromosome assignments to <code>AxC</code> so <code>reorgMarkers</code> should be set to <code>T</code>.</p>
<div class="sourceCode" id="cb540"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb540-1"><a href="Mapping-Markers.html#cb540-1" aria-hidden="true" tabindex="-1"></a>test.linkage.groups <span class="ot">&lt;-</span> <span class="fu">formLinkageGroups</span>(AxC, <span class="at">reorgMarkers =</span> T, </span>
<span id="cb540-2"><a href="Mapping-Markers.html#cb540-2" aria-hidden="true" tabindex="-1"></a>    <span class="at">verbose =</span> T, <span class="at">min.lod =</span> <span class="dv">3</span>, <span class="at">max.rf =</span> <span class="fl">0.25</span>)</span></code></pre></div>
<pre><code>## 10 of 186 
## 20 of 186 
## 30 of 186 
## 40 of 186 
## 50 of 186 
## 60 of 186 
## 70 of 186 
## 80 of 186 
## 90 of 186 
## 100 of 186 
## 110 of 186 
## 120 of 186 
## 130 of 186 
## 140 of 186 
## 150 of 186 
## 160 of 186 
## 170 of 186 
## 180 of 186</code></pre>
<p>An easy way to look at the linkage groups created in <code>test.linkage.groups</code> is to use the command</p>
<div class="sourceCode" id="cb542"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb542-1"><a href="Mapping-Markers.html#cb542-1" aria-hidden="true" tabindex="-1"></a><span class="fu">plotRF</span>(test.linkage.groups)</span></code></pre></div>
<p><img src="bookdown-demo_files/figure-html/unnamed-chunk-414-1.png" width="672" /></p>
<p>Compare this with the unordered markers set in <code>AxC</code></p>
<div class="sourceCode" id="cb543"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb543-1"><a href="Mapping-Markers.html#cb543-1" aria-hidden="true" tabindex="-1"></a><span class="fu">plotRF</span>(AxC)</span></code></pre></div>
<img src="bookdown-demo_files/figure-html/unnamed-chunk-415-1.png" width="672" />

<div class="rmdquiz">
<p><strong>2</strong></p>
Alter the min.lod and max.rf thresholds (try higher and lower) to see what difference this makes to the plots. Do the default values seem reasonable?
</div>
<p>Once you have decided on a threshold, you can save the linkage groups back to <code>AxC</code> using your own thresholds.</p>
<p><code>AxC &lt;-formLinkageGroups(AxC,reorgMarkers=T, verbose=T,min.lod=??,max.rf=??)</code></p>
</div>
<div id="marker-ordering" class="section level3" number="10.2.5">
<h3><span class="header-section-number">10.2.5</span> Marker ordering</h3>
<p>Next we must order markers within these linkage groups using the command orderMarkers(). This command also has some options we should experiment with:</p>
<p><code>use.ripple</code> = If TRUE (the default), the initial order is refined by a call to the function ripple.</p>
<p><code>window</code> = Indicates the number of markers to include in the sliding window of permuted markers for ripple (default is 7). Ripple can take a long time, especially with large number of markers in the window.</p>
<p><code>error.prob</code> = Assumed genotyping error rate used in the final estimated map. Default is 0.0001. This can make a large difference: a double recombination between closely linked markers is more likely to result from genotyping errors.</p>
<p>First. Let’s try without rippling;</p>
<div class="sourceCode" id="cb544"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb544-1"><a href="Mapping-Markers.html#cb544-1" aria-hidden="true" tabindex="-1"></a>test <span class="ot">&lt;-</span> <span class="fu">orderMarkers</span>(AxC, <span class="at">use.ripple =</span> F)</span></code></pre></div>
<p>Plot the recombination fractions of the ordered markers and compare with the unordered (in AxC).</p>
<div class="sourceCode" id="cb545"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb545-1"><a href="Mapping-Markers.html#cb545-1" aria-hidden="true" tabindex="-1"></a><span class="fu">plotRF</span>(test)</span></code></pre></div>
<p><img src="bookdown-demo_files/figure-html/unnamed-chunk-419-1.png" width="672" /></p>
<div class="sourceCode" id="cb546"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb546-1"><a href="Mapping-Markers.html#cb546-1" aria-hidden="true" tabindex="-1"></a><span class="fu">plotRF</span>(AxC)</span></code></pre></div>
<p><img src="bookdown-demo_files/figure-html/unnamed-chunk-419-2.png" width="672" /></p>
<p>We can also plot a simple cartoon of our mapped markers:</p>
<div class="sourceCode" id="cb547"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb547-1"><a href="Mapping-Markers.html#cb547-1" aria-hidden="true" tabindex="-1"></a><span class="fu">plot.map</span>(test)</span></code></pre></div>
<p><img src="bookdown-demo_files/figure-html/unnamed-chunk-420-1.png" width="672" /></p>

<div class="rmdquiz">
<p><strong>3</strong></p>
What do you think of the chromosome lengths and numbers in this map?
</div>
</div>
<div id="improving-the-map" class="section level3" number="10.2.6">
<h3><span class="header-section-number">10.2.6</span> Improving the map</h3>
<p>There are three things we can do to try to improve the map:</p>
<ol style="list-style-type: decimal">
<li><p>Ripple; where the marker order within a defined window is perturbed to see if improvements are possible.</p></li>
<li><p>Take into account the probability of genotype errors directly in the ordering process. This may be worthwhile for this data set, where we know the quality of the genotyping was poor.</p></li>
<li><p>Clean the data of probable errors – eliminate improbable double recombinants.</p></li>
</ol>
<p>We’ll have a look at ripple first. Only one chromosome at a time can be rippled, and the window size should not be too large since this can take a long time. Here are three different versions of the command:</p>
<div class="sourceCode" id="cb548"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb548-1"><a href="Mapping-Markers.html#cb548-1" aria-hidden="true" tabindex="-1"></a>chr1.ripple4 <span class="ot">&lt;-</span> <span class="fu">ripple</span>(AxC, <span class="at">chr =</span> <span class="dv">1</span>)  <span class="co"># ripple chr 1 with defaults (window=4)</span></span></code></pre></div>
<pre><code>##    276 total orders</code></pre>
<div class="sourceCode" id="cb550"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb550-1"><a href="Mapping-Markers.html#cb550-1" aria-hidden="true" tabindex="-1"></a>chr1.ripple7 <span class="ot">&lt;-</span> <span class="fu">ripple</span>(AxC, <span class="at">chr =</span> <span class="dv">1</span>, <span class="at">window =</span> <span class="dv">7</span>)  <span class="co"># increase window size </span></span></code></pre></div>
<pre><code>##    52560 total orders</code></pre>
<div class="sourceCode" id="cb552"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb552-1"><a href="Mapping-Markers.html#cb552-1" aria-hidden="true" tabindex="-1"></a>chr1.ripple4.like <span class="ot">&lt;-</span> <span class="fu">ripple</span>(AxC, <span class="at">chr =</span> <span class="dv">1</span>, <span class="at">method =</span> <span class="st">&quot;likelihood&quot;</span>)  <span class="co"># use likelihood</span></span></code></pre></div>
<pre><code>##    276 total orders
##     --Order 10 
##     --Order 20 
##     --Order 30 
##     --Order 40 
##     --Order 50 
##     --Order 60 
##     --Order 70 
##     --Order 80 
##     --Order 90 
##     --Order 100 
##     --Order 110 
##     --Order 120 
##     --Order 130 
##     --Order 140 
##     --Order 150 
##     --Order 160 
##     --Order 170 
##     --Order 180 
##     --Order 190 
##     --Order 200 
##     --Order 210 
##     --Order 220 
##     --Order 230 
##     --Order 240 
##     --Order 250 
##     --Order 260 
##     --Order 270</code></pre>
<p>The default method searches for orders which reduce the total number of recombinations. Using likelihood is better but takes longer.</p>
<p>You can examine the output directly but using summary is easier:</p>
<p><code>summary(chr1.ripple4)</code></p>

<div class="rmdquiz">
<p><strong>4</strong></p>
Compare the three versions you have just made for chromosome 1. Has any version of ripple made an improvement? If so, which has worked best?
</div>
<p>Once we have identified an improved order, we can substitute it for the existing order using the command <code>switch.order</code>. Here is a worked example:</p>
<div class="sourceCode" id="cb554"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb554-1"><a href="Mapping-Markers.html#cb554-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Existing map</span></span>
<span id="cb554-2"><a href="Mapping-Markers.html#cb554-2" aria-hidden="true" tabindex="-1"></a><span class="fu">pull.map</span>(AxC, <span class="at">chr =</span> <span class="dv">1</span>)</span></code></pre></div>
<pre><code>##   BArc142_5B   BArc176_7B    BArc74_5B   dUpw115_5B   dUpw398_7B  eAACmCCG171 
##            0           10           20           30           40           50 
##  eAACmCCG184  eAACmCCG408  eAACmCGA270 eAACmeCCC249  eACGmCGG220  eACTmCGG200 
##           60           70           80           90          100          110 
##  eAGCmCTA340  eAGGmCTA120  eATTmCGG230       gwm213    gwm234_5B    wmc517_7B 
##          120          130          140          150          160          170</code></pre>
<div class="sourceCode" id="cb556"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb556-1"><a href="Mapping-Markers.html#cb556-1" aria-hidden="true" tabindex="-1"></a><span class="co"># switch order (make sure you choose the correct row)</span></span>
<span id="cb556-2"><a href="Mapping-Markers.html#cb556-2" aria-hidden="true" tabindex="-1"></a>AxC.v2 <span class="ot">&lt;-</span> <span class="fu">switch.order</span>(AxC, <span class="at">chr =</span> <span class="dv">1</span>, <span class="at">order =</span> chr1.ripple7[<span class="dv">2</span>, </span>
<span id="cb556-3"><a href="Mapping-Markers.html#cb556-3" aria-hidden="true" tabindex="-1"></a>    ])</span>
<span id="cb556-4"><a href="Mapping-Markers.html#cb556-4" aria-hidden="true" tabindex="-1"></a><span class="co"># Compare new order with the initial order</span></span>
<span id="cb556-5"><a href="Mapping-Markers.html#cb556-5" aria-hidden="true" tabindex="-1"></a><span class="fu">pull.map</span>(AxC.v2, <span class="at">chr =</span> <span class="dv">1</span>)</span></code></pre></div>
<pre><code>##   BArc142_5B   BArc176_7B    BArc74_5B   dUpw115_5B   dUpw398_7B  eAACmCCG408 
##      0.00000     77.79435     80.52102    109.50827    183.34461    187.46286 
##  eAACmCCG171  eACGmCGG220 eAACmeCCC249  eAACmCGA270  eACTmCGG200  eAACmCCG184 
##    190.09164    226.04541    254.62921    264.18371    265.56842    282.12582 
##  eAGCmCTA340  eAGGmCTA120  eATTmCGG230       gwm213    gwm234_5B    wmc517_7B 
##    287.50267    322.18080    359.48948    394.33618    432.34606    487.79182</code></pre>
<div class="sourceCode" id="cb558"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb558-1"><a href="Mapping-Markers.html#cb558-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Tidy up</span></span>
<span id="cb558-2"><a href="Mapping-Markers.html#cb558-2" aria-hidden="true" tabindex="-1"></a><span class="fu">rm</span>(AxC.v2)</span></code></pre></div>
<p>We don’t have to type in the new order, we can select the desired order as a row from the ripple output. We can also return the new order to our cross directly, which is what we want to do here:</p>
<div class="sourceCode" id="cb559"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb559-1"><a href="Mapping-Markers.html#cb559-1" aria-hidden="true" tabindex="-1"></a>test <span class="ot">&lt;-</span> <span class="fu">switch.order</span>(test, <span class="at">chr =</span> <span class="dv">1</span>, <span class="at">order =</span> <span class="fu">summary</span>(chr1.ripple7)[<span class="dv">2</span>, </span>
<span id="cb559-2"><a href="Mapping-Markers.html#cb559-2" aria-hidden="true" tabindex="-1"></a>    ])</span></code></pre></div>
<p>We could go through every linkage group in turn like this. However, it is also possible to use the option <code>use.ripple =T</code> directly within orderMarkers which would be more efficient in general use.</p>
<p>Now we can examine the effect of different assumed error rates on the data. The default in R/qtl is 0.0001 which is probably too low for most genotype platforms; at NIAB we achieved this only by extensive manual curation of SNP array data. 0.001 would be a good target to aim for, but 0.01 is probably more realistic for this data set, where it still represents an average of less than one error per marker over all lines. From your knowledge of genotyping, what do you think would be appropriate?</p>
<p>To examine the effect of a higher assumed error rate, we will reorder our markers in “test” using different error rates :</p>
<div class="sourceCode" id="cb560"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb560-1"><a href="Mapping-Markers.html#cb560-1" aria-hidden="true" tabindex="-1"></a>err.pr.order <span class="ot">&lt;-</span> <span class="fu">orderMarkers</span>(test, <span class="at">use.ripple =</span> F, <span class="at">error.prob =</span> <span class="fl">0.01</span>)</span></code></pre></div>
<p>There is an easy graphical way of comparing the original and new map:</p>
<div class="sourceCode" id="cb561"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb561-1"><a href="Mapping-Markers.html#cb561-1" aria-hidden="true" tabindex="-1"></a><span class="fu">plotMap</span>(err.pr.order, test)</span></code></pre></div>
<p><img src="bookdown-demo_files/figure-html/unnamed-chunk-427-1.png" width="672" /></p>
<p>Though it is easier to examine a chromosome at a time.</p>
<div class="sourceCode" id="cb562"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb562-1"><a href="Mapping-Markers.html#cb562-1" aria-hidden="true" tabindex="-1"></a><span class="fu">plotMap</span>(err.pr.order, test, <span class="at">chr =</span> <span class="dv">1</span>)</span></code></pre></div>
<p><img src="bookdown-demo_files/figure-html/unnamed-chunk-428-1.png" width="672" /></p>
<p>This has changed the order substantially and has also shrunk the chromosome length. However, there is a danger of over interpreting this map reduction. The elimination of genuine errors will generally reduce map length, but the assumption of too high an error rate will give a falsely short map and incorrect marker order.</p>

<div class="rmdquiz">
<p><strong>5</strong></p>
Compare error rates of 0.0001, 0.01, 0.1 and 0.5. What do you notice about the inter-marker distances on the new map?
</div>
<p>Now, before doing further QC, we will make a basic map using the default rippling parameters (<code>window=4</code>) for all chromosomes. We will start from the first step again, so we know we are starting from a fresh data set, and as a reminder of the key steps required in making a map:</p>
<div class="sourceCode" id="cb563"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb563-1"><a href="Mapping-Markers.html#cb563-1" aria-hidden="true" tabindex="-1"></a>AxC <span class="ot">&lt;-</span> <span class="fu">read.cross</span>(<span class="st">&quot;csv&quot;</span>, <span class="at">dir =</span> <span class="st">&quot;data&quot;</span>, <span class="st">&quot;AxC_R.csv&quot;</span>, <span class="at">genotypes =</span> <span class="fu">c</span>(<span class="st">&quot;A&quot;</span>, </span>
<span id="cb563-2"><a href="Mapping-Markers.html#cb563-2" aria-hidden="true" tabindex="-1"></a>    <span class="st">&quot;B&quot;</span>), <span class="at">estimate.map =</span> F)</span></code></pre></div>
<pre><code>##  --Read the following data:
##   78  individuals
##   186  markers
##   10  phenotypes
##  --Cross type: bc</code></pre>
<div class="sourceCode" id="cb565"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb565-1"><a href="Mapping-Markers.html#cb565-1" aria-hidden="true" tabindex="-1"></a>AxC <span class="ot">&lt;-</span> <span class="fu">est.rf</span>(AxC)</span>
<span id="cb565-2"><a href="Mapping-Markers.html#cb565-2" aria-hidden="true" tabindex="-1"></a>AxC <span class="ot">&lt;-</span> <span class="fu">formLinkageGroups</span>(AxC, <span class="at">reorgMarkers =</span> T)</span>
<span id="cb565-3"><a href="Mapping-Markers.html#cb565-3" aria-hidden="true" tabindex="-1"></a>AxC <span class="ot">&lt;-</span> <span class="fu">orderMarkers</span>(AxC, <span class="at">use.ripple =</span> T)</span></code></pre></div>
<p>Note we’ve set <code>ripple = T</code>.</p>
<p>We can also export our data so as to save a copy for later work using <code>write.cross</code>, which has a simple to understand format:</p>
<div class="sourceCode" id="cb566"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb566-1"><a href="Mapping-Markers.html#cb566-1" aria-hidden="true" tabindex="-1"></a><span class="fu">write.cross</span>(AxC, <span class="st">&quot;csv&quot;</span>, <span class="at">filestem =</span> <span class="st">&quot;AxCmapped&quot;</span>)</span></code></pre></div>
<p>To highlight probable errors, R/qtl can calculate a LOD associated with p(single genotype is wrong). This is not an absolute test, but intended as a QC tool. It is particularly useful for highlighting improbable double recombinants. The advice is to use a LOD of 4 as a threshold. The results will be added to the cross (AxC). In this case, our data looks pretty good:</p>
<div class="sourceCode" id="cb567"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb567-1"><a href="Mapping-Markers.html#cb567-1" aria-hidden="true" tabindex="-1"></a>AxC <span class="ot">&lt;-</span> <span class="fu">calc.errorlod</span>(AxC)</span>
<span id="cb567-2"><a href="Mapping-Markers.html#cb567-2" aria-hidden="true" tabindex="-1"></a><span class="fu">plotErrorlod</span>(AxC)</span></code></pre></div>
<p><img src="bookdown-demo_files/figure-html/unnamed-chunk-432-1.png" width="672" /></p>
<p>But, from another data set, here is a non-so-clean version:</p>
<p><img src="images/dirty_data.png" /></p>
<ul>
<li>LOD &lt;= 2 in white;</li>
<li>2 &lt; LOD &lt;= 3 in grey;</li>
<li>3 &lt; LOD &lt;= 4.5 in pink;</li>
<li>LOD &gt; 4.5 in purple.</li>
</ul>
<p>You can restrict the plot to selected chromosomes if desired.</p>
<div class="sourceCode" id="cb568"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb568-1"><a href="Mapping-Markers.html#cb568-1" aria-hidden="true" tabindex="-1"></a><span class="fu">plotErrorlod</span>(AxC, <span class="at">chr =</span> <span class="dv">2</span>)</span></code></pre></div>
<p><img src="bookdown-demo_files/figure-html/unnamed-chunk-433-1.png" width="672" /></p>
<p>And take a look at the high error LOD markers:</p>
<div class="sourceCode" id="cb569"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb569-1"><a href="Mapping-Markers.html#cb569-1" aria-hidden="true" tabindex="-1"></a><span class="fu">top.errorlod</span>(AxC, <span class="dv">2</span>, <span class="dv">2</span>)</span></code></pre></div>
<pre><code>##   chr id      marker errorlod
## 1   2 78 eACTmCAC172 3.739982
## 2   2  5 eACTmCAC320 2.827777
## 3   2 54 eAACmCCC176 2.671936
## 4   2  8 eACTmCAC148 2.406375
## 5   2 37 eACTmCAC148 2.406375
## 6   2 75 eACTmCAC148 2.186916</code></pre>
<p>With two markers given with the same map location (ignoring the effect of jitter) any observed recombination will generate a big LOD. Taking a look at the map:</p>
<div class="sourceCode" id="cb571"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb571-1"><a href="Mapping-Markers.html#cb571-1" aria-hidden="true" tabindex="-1"></a><span class="fu">pull.map</span>(AxC, <span class="dv">2</span>)</span></code></pre></div>
<pre><code>## eAACmCCC272 B598mCGT390    gwm18_1B eACTmCGG365 eAACmCCC176 eACTmCAC148 
##     0.00000    28.32601    29.86730    31.17554    33.75115    42.03577 
## eACTmCAC320 eAACmCCC270 eACTmCAC172 eACTmCAC170  BArc240_1B eATTmCGG110 
##    46.78551    49.91895    52.08530    52.92547    55.57076    56.86974 
## eACGmCTG185 eAGCmCTA234 eACTmCAC180 eAGGmCTA130  eAACmCCG96 eAGGmCTA290 
##    63.73034    72.04487    88.72929    96.42972   105.66208   135.40817 
## eACTmCAC115    wmc44_1B 
##   166.09210   177.82376</code></pre>
<p>That isn’t a problem here, though you can see the high error markers are all very close to adjacent markers, and presumably have some recombination between them. We can see this in more detail with this very useful plot:</p>
<div class="sourceCode" id="cb573"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb573-1"><a href="Mapping-Markers.html#cb573-1" aria-hidden="true" tabindex="-1"></a><span class="fu">plotGeno</span>(AxC, <span class="dv">2</span>)</span></code></pre></div>
<p><img src="bookdown-demo_files/figure-html/unnamed-chunk-436-1.png" width="672" /></p>
<p>White and black circles indicate markers from each of the two parents. Blue crosses indicate cross-overs. Red squares (none here) highlight genotypes with LODs &gt; the error LOD threshold (default 4). A single double recombination event can however clearly be seen around the highest error LOD marker eACTmCAC172 (around 60 or 125cM depending on the orientation of the chromosome) for the individual at the bottom of the plot, despite the gaps to the neighbouring markers being 0.8 and 1.2cM respectively. This maybe a genotype error. Multiple double recombinant individuals in such a narrow interval would be a sure sign of a mapping or genotype error, suggesting the marker should be dropped.
Other useful quality control tools are listed in the table below. The graphical tools are particularly helpful for detecting regions of the map that require further work. For example <code>plotRF(AxC)</code> plots pairwise recombination frequencies across all chromosomes or a restricted set: <code>plotRF(AxC,chr=1:5)</code>. In our case, chromosome 1 looks very well ordered.</p>
<div class="sourceCode" id="cb574"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb574-1"><a href="Mapping-Markers.html#cb574-1" aria-hidden="true" tabindex="-1"></a><span class="fu">plotRF</span>(AxC, <span class="at">chr =</span> <span class="dv">1</span>)</span></code></pre></div>
<p><img src="bookdown-demo_files/figure-html/unnamed-chunk-437-1.png" width="672" /></p>
<p>However, with the same dataset we have observed issues with this ordering before, take a look at the image below, it can clearly be seen that the ordering of this Chromosome 1 example could still be improved:</p>
<p><img src="images/Chrome_1_error.png" /></p>

<div class="rmdquiz">
<p><strong>6</strong></p>
In the image above. The block of markers 1:3 appear to be more closely linked to markers 11:14. In this case, we could swap their position as a block and see if it made an improvement. This anomaly was observed in the same dataset we are using and may help explain some of the very long inter-marker distances we observed at the start of chromosome 1, that cause this chromosome to be so much longer than all the rest. What do you think might be going on here in the image above?
</div>
<p>Overall we still have some way to go in improving our map! In reality, several cycles of improvement and QC will be involved in creating a final map; this tutorial has given a quick overview of the main approaches. It can be seen that R/qtl has excellent tools for detecting and displaying potential genotype errors: a table of key Quality Control commands is given below. There is plenty of good documentation on how to map markers in r/qtl, including the book <span class="citation"><a href="references.html#ref-broman_guide_2009" role="doc-biblioref">Broman and Sen</a> (<a href="references.html#ref-broman_guide_2009" role="doc-biblioref">2009</a>)</span>, though it is not free, and the discussion page mentioned at the start of the tutorial.</p>
</div>
<div id="quality-control-commands-for-marker-qc-and-mapping" class="section level3" number="10.2.7">
<h3><span class="header-section-number">10.2.7</span> Quality Control commands for marker QC and mapping</h3>
<p><code>plot(AxC)</code> = provides a neat graphical summary of phenotypes and genotypes</p>
<p><code>summary(AxC)</code>= worth running to check the data have been read in correctly</p>
<p><code>summary.map(AxC)</code> = summarises chromosome lengths, marker nos. and spacing</p>
<p><code>geno.table(AxC)</code> = summarises alleles and tests for segregation distortion</p>
<p><code>plot(AxC,pheno.col=x)</code> = plots histogram for phenotype in column x or of a named phenotype - in which case the name must be in quotes. The phenotype data are stored in a data frame <code>AxC$pheno</code> which can be accessed and manipulated by column number or “name.” You might want to transform the data. eg:</p>
<ul>
<li><code>hist(AxC$pheno[,1])</code></li>
<li><code>AxC$pheno[,1]&lt;-AxC$pheno[,1]^0.5 hist(AxC$pheno[,1])</code></li>
<li><code>AxC$pheno[,1]&lt;-AxC$pheno[,1]^2</code></li>
</ul>
<p><code>plot.map(AxC,chr=1:2)</code> plots map for chrs 1 and 2. If the <code>chr</code> argument is missing,
then plots all chromosomes. Can also be used to plot comparisons between maps – see <code>help(plot.map)</code>. This provides an easy way to look at map expansion/contraction between the supplied map and the estimated map in R/qtl (using the same map order). For example (using a different data set):</p>
<p><code>plot.map(est.map(AxC),AxC)</code> = comparison of maps</p>
<p><code>plotRF(AxC,chr=1)</code> = Plots pairwise recombination frequencies</p>
<p><code>subset(AxC,chrs,inds)</code> = Selects a range of markers and individuals. eg<code>subset(AxC,5,2:10)</code></p>
<p><code>pull.map(AxC,chr=5)</code> = Extract the existing map</p>
<p><code>est.map(subset(AxC,2))</code> = Estimate map distances but keeps the order the same. Useful to study map expansion / contraction. The example re-estimates for chr 2 only</p>
<p><code>jittermap(AxC)</code> = Gets rid of the problem of having markers at the same location in the supplied map.</p>
<p><code>AxC&lt;-calc.errorlod(AxC)</code> = Calculates a LOD associated with p(single genotype is wrong). Add results to the cross (AxC here). Should be used to highlight probable errors, not as an absolute test. Advice is to use a LOD of 4 as a threshold.</p>
<p><code>plotErrorlod(AxC)</code> = Plots possible genotype errors.</p>
</div>
<div id="asmap" class="section level3" number="10.2.8">
<h3><span class="header-section-number">10.2.8</span> ASMap</h3>
<p>Before using R/qtl to map QTL, we’ll have a brief look at <code>ASMap</code> as an alternative for mapping markers, which is definitely worth considering if you have a larger number of markers and R/qtl is too slow.</p>
<p>As ever, the problem with all computer packages is getting the data into them. Once that is achieved, then getting them to do something useful is generally straightforward. <code>ASmap</code> should work directly on pre-existing R/qtl objects, but after a lot of experimentation, this fails:</p>
<p><code>test&lt;-mstmap.cross(AxC) Error in mstmap.cross(AxC)</code> :</p>
<p><em>The unique identifier for the genotypes, “Genotype,” cannot be found in the object</em></p>
<p><code>ASMap</code> will, however, work on dataframes, if correctly formatted. There is a text version of the AxC marker data in ASmap.txt. If you look at this, you will notice that; (1) there is no phenotype data; (2) missing data are represented by “U” and not “NA”; (3) the data are transposed compared to the initial AxC data file, with markers running across rows and individuals down columns. This is better for populations with large numbers of markers.</p>
<p>Load the package, you will need to install if you don’t have it on your computers:</p>
<div class="sourceCode" id="cb575"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb575-1"><a href="Mapping-Markers.html#cb575-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ASMap)</span></code></pre></div>
<div class="sourceCode" id="cb576"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb576-1"><a href="Mapping-Markers.html#cb576-1" aria-hidden="true" tabindex="-1"></a><span class="co"># delete the &#39;data/&#39;</span></span>
<span id="cb576-2"><a href="Mapping-Markers.html#cb576-2" aria-hidden="true" tabindex="-1"></a>AxC.mst <span class="ot">&lt;-</span> <span class="fu">read.table</span>(<span class="st">&quot;data/AxC_ASmap.txt&quot;</span>, <span class="at">header =</span> T, <span class="at">colClasses =</span> <span class="st">&quot;character&quot;</span>)</span></code></pre></div>
<p>Note the added option <code>colClasses = "character"</code>. This is critical. Without it, the data might be read in as factors rather than text, and this is not an obvious error to pick up.</p>
<div class="sourceCode" id="cb577"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb577-1"><a href="Mapping-Markers.html#cb577-1" aria-hidden="true" tabindex="-1"></a>AxC.mst.map <span class="ot">&lt;-</span> <span class="fu">mstmap.data.frame</span>(AxC.mst)</span></code></pre></div>
<p>As the name of the command suggests, the markers are mapped (very rapidly) by means of a minimum spanning tree: finding the shortest route between points defined by the recombination fractions between all marker pairs. A minimum spanning tree is a solution to what is called “the traveling salesman problem”: what is the shortest route a salesman should take to visit all the towns in which he wishes to knock on doors? See the paper and documentation for more information.</p>
<p>We are advised that the clustering of markers into linkage groups is very sensitive to the p-value used. The default is 1e-06 and as more markers are added, this should be increased. Try this:</p>
<div class="sourceCode" id="cb578"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb578-1"><a href="Mapping-Markers.html#cb578-1" aria-hidden="true" tabindex="-1"></a><span class="fu">plot.map</span>(<span class="fu">mstmap.data.frame</span>(AxC.mst, <span class="at">p.value =</span> <span class="fl">0.000001</span>))</span></code></pre></div>
<pre><code>## Number of linkage groups: 45
## The size of the linkage groups are: 6    8   10  5   11  1   1   7   5   19  6   6   6   3   6   18  3   3   3   3   2   4   3   1   5   1   6   3   1   3   2   1   5   3   1   1   1   1   2   4   1   1   2   1   1   
## The number of bins in each linkage group: 6  5   10  3   10  1   1   7   5   18  4   4   5   3   5   18  2   3   3   3   2   4   3   1   5   1   6   3   1   3   2   1   5   3   1   1   1   1   2   4   1   1   2   1   1   </code></pre>
<p><img src="bookdown-demo_files/figure-html/unnamed-chunk-442-1.png" width="672" /></p>

<div class="rmdquiz">
<p><strong>7</strong></p>
Now try altering the p.value up and down to see what effect this has on the map. What do you observe? (remember that the target number of chromosomes for wheat is 21). How might we decide what is best?
</div>
<p>Note that in all the ASMap maps, we still observe one very long chromosome, just as when we used R/qtl to create the map. As with R/qtl, this observation emphasizes that errors, or chromosomal arrangements in the data will cause the same problems, no matter what the analytical approach. We will return now to using the default p.value.</p>
<div class="sourceCode" id="cb580"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb580-1"><a href="Mapping-Markers.html#cb580-1" aria-hidden="true" tabindex="-1"></a>AxC.mst.map <span class="ot">&lt;-</span> <span class="fu">mstmap.data.frame</span>(AxC.mst, <span class="at">p.value =</span> <span class="fl">0.000001</span>)</span></code></pre></div>
<p>The commands available in R/qtl should all work on the output file we have created (I haven’t tested this fully), but there are some useful additional commands available:</p>
<div class="sourceCode" id="cb581"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb581-1"><a href="Mapping-Markers.html#cb581-1" aria-hidden="true" tabindex="-1"></a><span class="fu">heatMap</span>(AxC.mst.map)  <span class="co"># LOD at the bottom rf at the top </span></span></code></pre></div>
<pre><code>## Warning in heatMap(AxC.mst.map): Running est.rf.</code></pre>
<p><img src="bookdown-demo_files/figure-html/unnamed-chunk-445-1.png" width="672" /></p>
<p>gives better looking output than</p>
<p><code>plotRF(AxC.mst.map)</code></p>
<p>Note that LG 16 (the long one) is still badly ordered in ASMap.</p>
<p>Also note these useful diagnostic commands</p>
<div class="sourceCode" id="cb583"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb583-1"><a href="Mapping-Markers.html#cb583-1" aria-hidden="true" tabindex="-1"></a><span class="fu">profileGen</span>(AxC.mst.map, <span class="at">chr =</span> <span class="st">&quot;L3&quot;</span>)</span></code></pre></div>
<p><img src="bookdown-demo_files/figure-html/unnamed-chunk-446-1.png" width="672" /></p>
<div class="sourceCode" id="cb584"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb584-1"><a href="Mapping-Markers.html#cb584-1" aria-hidden="true" tabindex="-1"></a><span class="fu">profileMark</span>(AxC.mst.map, <span class="at">chr =</span> <span class="st">&quot;L3&quot;</span>)</span></code></pre></div>
<p><img src="bookdown-demo_files/figure-html/unnamed-chunk-446-2.png" width="672" /></p>
<p>There are other options to audit and edit the map too: see the package documentation for more information.</p>
</div>
</div>
<div id="qtl-mapping-in-rqtl" class="section level2" number="10.3">
<h2><span class="header-section-number">10.3</span> QTL mapping in R/qtl</h2>
<p>The first step in carrying out interval mapping is to calculate the genotype probabilities at set intervals (here we use every 2cM) and add them to AxC. You need the output for all your genome scans. This can take a while to run:</p>
<div class="sourceCode" id="cb585"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb585-1"><a href="Mapping-Markers.html#cb585-1" aria-hidden="true" tabindex="-1"></a>AxC <span class="ot">&lt;-</span> <span class="fu">calc.genoprob</span>(AxC, <span class="dv">2</span>)</span></code></pre></div>
<p>An interesting feature is that you can include a genotyping error rate. If you think the data are flaky, or have an independent estimate of this (from replicate genotypes) you could alter the default value. Could one maximise LODs for QTL while treating this as a parameter?</p>
<div id="genome-scan-single-qtl" class="section level3" number="10.3.1">
<h3><span class="header-section-number">10.3.1</span> Genome Scan – single QTL</h3>
<p><code>Scanone</code> is the command to carry out a genome scan for single QTLs:</p>
<div class="sourceCode" id="cb586"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb586-1"><a href="Mapping-Markers.html#cb586-1" aria-hidden="true" tabindex="-1"></a>ResultPhen3 <span class="ot">&lt;-</span> <span class="fu">scanone</span>(AxC, <span class="at">pheno.col =</span> <span class="dv">3</span>)</span>
<span id="cb586-2"><a href="Mapping-Markers.html#cb586-2" aria-hidden="true" tabindex="-1"></a><span class="fu">plot</span>(ResultPhen3)</span></code></pre></div>
<p><img src="bookdown-demo_files/figure-html/unnamed-chunk-448-1.png" width="672" /></p>
<p>This will carry out a basic genome scan for single QTLs on the third trait.</p>
<p>However, there are many options. See help(scanone):</p>
<pre><code>method =    hk    Least squares interval mapping   
            em    Interval mapping by maximum likelihood (slower)   
            mr    Marker regression (ie marker by marker analysis) 

model =     normal  Trait is assumed normally distributed                   
            binary  Could be useful for +/- disease scores
            np      A non-parametric test 
            2part   For distributions with a spike at the high or low end.   

n.perm = n    Permutes the phenotype n times and reports the maximum LOD for each permutation. </code></pre>
<p>An example where you might have a <code>2part</code> distribution is if you have a large resistant class, “0,” and then the rest of the data has an assumed normal distribution – measuring strength of tolerance maybe. You need to declare whether your spike is the phenotype coded with the highest or lowest value using the option <code>upper  = FALSE</code> We can try this with the AxC dataset by declaring some lines as definitely resistant (or susceptible) and using the elisa values for the others</p>
<p>The permutations provides empirical significance values. Test with a small number first before running in anger because it could take a while.</p>
<p>It is possible to include covariates in the analysis too, but at the moment these must be numeric, so a factor with n levels would have to be recoded into n-1 columns for inclusion. It is also possible to have covariates which interact with the QTL.</p>
<p>As ever, there are commands to summarise and display the results. First, it is worthwhile to save your result to prevent having to re-run scanone all the time. For example,</p>
<div class="sourceCode" id="cb588"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb588-1"><a href="Mapping-Markers.html#cb588-1" aria-hidden="true" tabindex="-1"></a>scanone.result <span class="ot">&lt;-</span> <span class="fu">scanone</span>(AxC, <span class="at">pheno.col =</span> <span class="dv">3</span>, <span class="at">model =</span> <span class="st">&quot;normal&quot;</span>, </span>
<span id="cb588-2"><a href="Mapping-Markers.html#cb588-2" aria-hidden="true" tabindex="-1"></a>    <span class="at">method =</span> <span class="st">&quot;hk&quot;</span>)</span></code></pre></div>
<p>will scan the whole genome for the phenotype in the third column of the data matrix using Haley-Knott regression – i.e. interval mapping by least squares.</p>
<div class="sourceCode" id="cb589"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb589-1"><a href="Mapping-Markers.html#cb589-1" aria-hidden="true" tabindex="-1"></a><span class="fu">summary</span>(scanone.result)</span></code></pre></div>
<pre><code>##                 chr    pos      lod
## c1.loc160         1 160.00  1.31513
## eAACmCCG96        2 105.66  1.18097
## gwm617.2_6A       3  27.40  1.25221
## BArc20_7B         4  88.49  0.44606
## TAglUt_1A         5  13.92  1.08510
## SBm1-RotHpAtH5D   6  14.37 13.50803
## c7.loc58          7  58.00  0.10049
## WMC489_5A         8  74.18  0.27425
## AFLP43-160Bp      9   1.31  5.26837
## eAACmCCC236      10   0.00  0.54297
## gwm614.2_2B      11   2.60  0.07646
## c12.loc4         12   4.00  0.17256
## eAGGmCTA350      13  48.71  0.44250
## gdm72_3D         14   9.82  0.52445
## c15.loc32        15  32.00  0.31542
## c16.loc2         16   2.00  0.44217
## c17.loc22        17  22.00  1.17265
## c18.loc26        18  26.00  0.60374
## gwm614.1_2A      19  12.50  0.00948
## eAGCmCTA240      20  13.11  0.69652
## BArc170_4A       21   3.73  0.14507
## eACTmCAC375      22  10.25  0.34791
## gwm631_7A        23   5.49  0.53412
## BArc97_7D        24   0.00  0.04197
## c25.loc2         25   2.00  0.02293
## c26.loc10        26  10.00  0.05409
## c27.loc58        27  58.00  0.57581
## B598mCGT205      28   0.00  0.22441
## gwm325_6D        29  10.00  0.08418
## gwm437_7D        30   0.00  0.10709
## c31.loc-1        31  -1.00  0.14200</code></pre>
<p>That outputs the highest peak in each chromosome. We can also plot the LOD scores across all chromosomes and print the QTL with the highest LOD:</p>
<div class="sourceCode" id="cb591"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb591-1"><a href="Mapping-Markers.html#cb591-1" aria-hidden="true" tabindex="-1"></a><span class="fu">plot</span>(scanone.result)</span></code></pre></div>
<p><img src="bookdown-demo_files/figure-html/unnamed-chunk-451-1.png" width="672" /></p>
<div class="sourceCode" id="cb592"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb592-1"><a href="Mapping-Markers.html#cb592-1" aria-hidden="true" tabindex="-1"></a><span class="fu">max</span>(scanone.result)</span></code></pre></div>
<pre><code>##                 chr  pos  lod
## SBm1-RotHpAtH5D   6 14.4 13.5</code></pre>
<p>We then then find the closest marker to a specified position. (You may get a different marker returned – use <code>pull.map</code> to discover why):</p>
<div class="sourceCode" id="cb594"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb594-1"><a href="Mapping-Markers.html#cb594-1" aria-hidden="true" tabindex="-1"></a><span class="fu">find.marker</span>(AxC, <span class="at">chr =</span> <span class="dv">6</span>, <span class="at">pos =</span> <span class="fl">14.1</span>)</span></code></pre></div>
<pre><code>## [1] &quot;AFLP24&quot;</code></pre>
<p>If we run:</p>
<div class="sourceCode" id="cb596"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb596-1"><a href="Mapping-Markers.html#cb596-1" aria-hidden="true" tabindex="-1"></a><span class="fu">plotPXG</span>(AxC, <span class="at">pheno.col =</span> <span class="dv">3</span>, <span class="st">&quot;AFLP24&quot;</span>)</span></code></pre></div>
<p><img src="bookdown-demo_files/figure-html/unnamed-chunk-453-1.png" width="672" /></p>
<p>we plot the specified phenotype and genotype. Red points are missing data, inferred from linked markers. As mentioned, permutation tests for scanone are available:</p>
<div class="sourceCode" id="cb597"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb597-1"><a href="Mapping-Markers.html#cb597-1" aria-hidden="true" tabindex="-1"></a>scanone.perm <span class="ot">&lt;-</span> <span class="fu">scanone</span>(AxC, <span class="at">pheno.col =</span> <span class="dv">3</span>, <span class="at">model =</span> <span class="st">&quot;normal&quot;</span>, </span>
<span id="cb597-2"><a href="Mapping-Markers.html#cb597-2" aria-hidden="true" tabindex="-1"></a>    <span class="at">method =</span> <span class="st">&quot;hk&quot;</span>, <span class="at">n.perm =</span> <span class="dv">1000</span>)</span></code></pre></div>
<pre><code>## Doing permutation in batch mode ...</code></pre>
<div class="sourceCode" id="cb599"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb599-1"><a href="Mapping-Markers.html#cb599-1" aria-hidden="true" tabindex="-1"></a><span class="fu">summary</span>(scanone.perm)</span></code></pre></div>
<pre><code>## LOD thresholds (1000 permutations)
##      lod
## 5%  2.95
## 10% 2.60</code></pre>
<p>1,000 permutations are too few. Running permutations, say 100,000, will take time. For this reason, it is best to use <code>method ="hk”</code> for permutations and also for the original analysis: the method used should be the same for both.</p>
<p>An estimate of QTL heritability (for scanone only, this doesn’t work for multiple QTL mapping) is</p>
<p><span class="math display">\[1-10 ^{-2LOD/n}\]</span></p>
<p>where n is the number of lines. This is only true for single QTL models. For more complex cases you must use <code>fitqtl()</code>: see the manual or FAQ.</p>
</div>
<div id="genome-scan-covariates" class="section level3" number="10.3.2">
<h3><span class="header-section-number">10.3.2</span> Genome Scan – covariates</h3>
<p><code>pull.geno(AxC)[,"marker"]</code> will pull out a marker so it can be included as a covariate. These can be included as covariates in <code>scanone</code> or <code>scantwo</code> (below). E.g.</p>
<div class="sourceCode" id="cb601"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb601-1"><a href="Mapping-Markers.html#cb601-1" aria-hidden="true" tabindex="-1"></a><span class="fu">max</span>(<span class="fu">scanone</span>(AxC, <span class="at">chr =</span> <span class="dv">6</span>, <span class="at">pheno.col =</span> <span class="dv">3</span>))</span></code></pre></div>
<pre><code>##        chr  pos  lod
## AFLP24   6 14.4 13.5</code></pre>
<div class="sourceCode" id="cb603"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb603-1"><a href="Mapping-Markers.html#cb603-1" aria-hidden="true" tabindex="-1"></a>covariate_1 <span class="ot">&lt;-</span> <span class="fu">pull.geno</span>(AxC)[, <span class="st">&quot;AFLP24&quot;</span>]</span>
<span id="cb603-2"><a href="Mapping-Markers.html#cb603-2" aria-hidden="true" tabindex="-1"></a>scan.result.cov <span class="ot">&lt;-</span> <span class="fu">scanone</span>(AxC, <span class="at">pheno.col =</span> <span class="dv">3</span>, <span class="at">model =</span> <span class="st">&quot;normal&quot;</span>, </span>
<span id="cb603-3"><a href="Mapping-Markers.html#cb603-3" aria-hidden="true" tabindex="-1"></a>    <span class="at">method =</span> <span class="st">&quot;hk&quot;</span>, <span class="at">addcov =</span> covariate_1)</span></code></pre></div>

<div class="rmdquiz">
<strong>8</strong>
Has this made any difference compared to what we found before? (Hint - use <code>plot</code>, <code>summary</code> and <code>max</code>).
</div>
</div>
<div id="genome-scan-scantwo" class="section level3" number="10.3.3">
<h3><span class="header-section-number">10.3.3</span> Genome Scan – scantwo</h3>
<p><code>scantwo(AxC,pheno.col=3)</code> scans the genome for a pair of QTL. Options are essentially the same as for <code>scanone</code>. Use <code>method="hk"</code> for speed of calculation. <code>n.perm</code> may take longer. The output is longer and LOD scores require more interpretation (see below, the course notes, and R/qtl help).</p>
<p>To carry out the genome wide two-dimensional analysis and save the result:</p>
<div class="sourceCode" id="cb604"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb604-1"><a href="Mapping-Markers.html#cb604-1" aria-hidden="true" tabindex="-1"></a>scan2.res <span class="ot">&lt;-</span> <span class="fu">scantwo</span>(AxC, <span class="at">pheno.col =</span> <span class="dv">3</span>, <span class="at">method =</span> <span class="st">&quot;hk&quot;</span>)</span>
<span id="cb604-2"><a href="Mapping-Markers.html#cb604-2" aria-hidden="true" tabindex="-1"></a><span class="fu">summary</span>(scan2.res)</span></code></pre></div>
<p>This produces a lot of output, inspect on your own computers. All these LODs require some explanation.
Looking at rows first – results are given for the peak values on each pair of chromosomes and on each individual chromosome. With n chromosomes, n(n+1)/2 results would be given. This is the default – to print the best results on all pairs of chromosomes. This can be modified (see below) to print results which exceed supplied thresholds. The LOD scores first.</p>
<ul>
<li><code>lod.full</code> is the 3df LOD for pos 1 + pos 2 + their interaction</li>
<li><code>lod.fvl</code> is the 2df LOD for one locus plus its interaction with the second locus. It is therefore evidence for a second locus including its interaction with the first locus.</li>
<li><code>lod.int</code> is the 1df LOD for the interaction only</li>
<li><code>lod.add</code> is the 2df LOD for a joint additive effect at pos 1 and pos 2</li>
<li><code>lod.av1</code> is the 1df LOD for an additive effect at the second locus after the first locus is fitted.
<code>lod.full</code> and <code>lod.add</code> can be maximised at separate locations, so we have two pairs of positions – pos1f, pos2f and pos1a and pos2a for each chromosome pair. There is far too much information here. Worse, significance thresholds for each LOD need to be computed. This takes a long time.</li>
</ul>
<p>The command below carries out only 10 permutations and will still take time</p>
<div class="sourceCode" id="cb605"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb605-1"><a href="Mapping-Markers.html#cb605-1" aria-hidden="true" tabindex="-1"></a>scantwo_perm <span class="ot">&lt;-</span> (<span class="fu">scantwo</span>(AxC, <span class="at">pheno.col =</span> <span class="dv">3</span>, <span class="at">method =</span> <span class="st">&quot;hk&quot;</span>, <span class="at">n.perm =</span> <span class="dv">10</span>, </span>
<span id="cb605-2"><a href="Mapping-Markers.html#cb605-2" aria-hidden="true" tabindex="-1"></a>    <span class="at">chr =</span> <span class="fu">c</span>(<span class="dv">1</span><span class="sc">:</span><span class="dv">30</span>)))</span></code></pre></div>
<pre><code>## Doing permutation in batch mode ...</code></pre>
<p>We have selected only the first 30 chromosomes to include in the permutation test: <code>chr=c(1:30)</code>. Without this, we get a cryptic error message. For the map we are using, the 31st linkage group consists of only one marker. This does not cause analysis of the actual data to fail, but does cause problems to the permutation test. We should have spent more time creating a good genetic map – which would not include singleton markers!</p>
<p>My results are from 100 permutations are summarised as:</p>
<div class="sourceCode" id="cb607"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb607-1"><a href="Mapping-Markers.html#cb607-1" aria-hidden="true" tabindex="-1"></a><span class="fu">summary</span>(scantwo_perm)</span></code></pre></div>
<pre><code>## elisa_all (100 permutations)
##     full  fv1  int  add  av1  one
## 5%  6.29 4.69 4.37 5.10 2.43 2.78
## 10% 5.81 4.47 4.05 4.45 2.24 2.58</code></pre>
<p>The maximum interaction LOD in my 100 permutations was:</p>
<div class="sourceCode" id="cb609"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb609-1"><a href="Mapping-Markers.html#cb609-1" aria-hidden="true" tabindex="-1"></a><span class="fu">max</span>(scantwo_perm<span class="sc">$</span>int)</span></code></pre></div>
<pre><code>## [1] 5.323228</code></pre>
<p>Output from multiple permutations can be combined and then summarised. In fact there is a command within the package, <code>c.scantwoperm</code>, which will do this but you’d have to have access to the multiple outputs from the same R session.</p>
<p><code>max(scantwo.res)</code> finds the maximum LOD which can be compared with these empirical significance levels.</p>
<div class="sourceCode" id="cb611"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb611-1"><a href="Mapping-Markers.html#cb611-1" aria-hidden="true" tabindex="-1"></a><span class="fu">max</span>(scan2.res)</span></code></pre></div>
<pre><code>##       pos1f pos2f lod.full lod.fv1 lod.int     pos1a pos2a lod.add lod.av1
## c6:c9    14     2     32.5    19.1    7.97        14     2    24.5    11.1</code></pre>
<p>Even from only 100 permutations, it would seem our interaction is genuine, but we should carry out more.</p>
<p><code>help(max.scantwo)</code> will give options on selection of LODs to search for the maximum.</p>
<p>The most useful of the scantwo commands is plot:</p>
<div class="sourceCode" id="cb613"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb613-1"><a href="Mapping-Markers.html#cb613-1" aria-hidden="true" tabindex="-1"></a><span class="fu">plot</span>(scan2.res)</span></code></pre></div>
<p><img src="bookdown-demo_files/figure-html/unnamed-chunk-464-1.png" width="672" /></p>
<p>The upper left of the plot contains interaction LOD scores. The lower right contains the LOD scores for the full model. These defaults can be altered – see <code>help(plot.scantwo)</code>.</p>
<p>This plot gives good visual evidence of the interaction between the QTLs on chromosome 6 and 9 (Interaction LOD = 7.77).</p>
</div>
<div id="genome-scan-composite-interval-mapping" class="section level3" number="10.3.4">
<h3><span class="header-section-number">10.3.4</span> Genome Scan – Composite Interval Mapping</h3>
<p><code>scantwo</code> is R/qtl’s version of multiple interval mapping. An automated version of composite interval mapping is given “by one of the schemes used in QTL cartographer.”</p>
<div class="sourceCode" id="cb614"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb614-1"><a href="Mapping-Markers.html#cb614-1" aria-hidden="true" tabindex="-1"></a>cim_out <span class="ot">&lt;-</span> <span class="fu">cim</span>(AxC, <span class="at">pheno.col =</span> <span class="dv">3</span>, <span class="at">n.marcovar =</span> <span class="dv">5</span>, <span class="at">window =</span> <span class="dv">10</span>, </span>
<span id="cb614-2"><a href="Mapping-Markers.html#cb614-2" aria-hidden="true" tabindex="-1"></a>    <span class="at">method =</span> <span class="st">&quot;hk&quot;</span>)</span></code></pre></div>
<p><code>n.marcovar</code> is the number of marker covariates to be included. <code>window=10</code> specifies that any markers within a 10cM window of the test locus will be dropped.</p>
<p>Markers are selected by “forward selection” – the largest single LODs are selected. There is no automatic inclusion of flanking markers (see course notes).</p>
<div class="sourceCode" id="cb615"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb615-1"><a href="Mapping-Markers.html#cb615-1" aria-hidden="true" tabindex="-1"></a><span class="fu">plot</span>(cim_out)</span></code></pre></div>
<p><img src="bookdown-demo_files/figure-html/unnamed-chunk-466-1.png" width="672" /></p>
<div class="sourceCode" id="cb616"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb616-1"><a href="Mapping-Markers.html#cb616-1" aria-hidden="true" tabindex="-1"></a><span class="fu">plot</span>(cim_out, <span class="at">chr =</span> <span class="fu">c</span>(<span class="dv">6</span>, <span class="dv">9</span>))</span></code></pre></div>
<p><img src="bookdown-demo_files/figure-html/unnamed-chunk-467-1.png" width="672" /></p>
<p>To summarise some cim commands:</p>
<ul>
<li><code>n.marcovar = 0</code> gives the same results as scanone.</li>
<li><code>plot(cim_out)</code> Plots results</li>
<li><code>add.cim.covar(cim_out)</code> Highlights the covariates on the plot</li>
<li><code>attributes(cim_out)</code> Lists the covariate markers</li>
<li><code>summary(cim_out)</code> Summarises results</li>
</ul>
<p>It is worth experimenting. It seems to get rid of a lot of background weak peaks. Permuted significance values are possible again:</p>
<div class="sourceCode" id="cb617"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb617-1"><a href="Mapping-Markers.html#cb617-1" aria-hidden="true" tabindex="-1"></a>cim_perm <span class="ot">&lt;-</span> <span class="fu">cim</span>(AxC, <span class="at">pheno.col =</span> <span class="dv">3</span>, <span class="at">n.marcovar =</span> <span class="dv">5</span>, <span class="at">window =</span> <span class="dv">10</span>, </span>
<span id="cb617-2"><a href="Mapping-Markers.html#cb617-2" aria-hidden="true" tabindex="-1"></a>    <span class="at">method =</span> <span class="st">&quot;hk&quot;</span>, <span class="at">n.perm =</span> <span class="dv">100</span>)</span>
<span id="cb617-3"><a href="Mapping-Markers.html#cb617-3" aria-hidden="true" tabindex="-1"></a><span class="fu">summary</span>(cim_perm)</span></code></pre></div>
<pre><code>## LOD thresholds (100 permutations)
##     [,1]
## 5%  4.62
## 10% 4.23</code></pre>
<div class="sourceCode" id="cb619"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb619-1"><a href="Mapping-Markers.html#cb619-1" aria-hidden="true" tabindex="-1"></a><span class="fu">max</span>(cim_perm)</span></code></pre></div>
<pre><code>## [1] 5.180333</code></pre>
<p>There is good evidence that the two peaks on linkage groups 6 and 9 are genuine, but note how high the empirical significance threshold LOD scores are!</p>
</div>
<div id="mqm-multiple-qtl-mapping" class="section level3" number="10.3.5">
<h3><span class="header-section-number">10.3.5</span> MQM: multiple QTL mapping</h3>
<p>Multiple QTL mapping is available in R/qtl. Prior to this, there was no free software available which implemented the method. It is preferred to composite interval mapping, for reasons which are not totally clear to me, by the originator of R/qtl (Karl Broman) and also by Ritsert Jansen who developed the method. Within R/qtl, it exists as a separate mini-package, with a set of commands all beginning “mqm….” MQM has a separate tutorial pdf too in which more detailed information than given here can be found, Unfortunately, we do not have the time to go into more detail here.</p>
<p>Take a look at Example 5 in “A brief tour of R/qtl” to see the full extent of multiple QTL modeling that is available.</p>
</div>
<div id="some-commands-for-mapping-in-markers-in-rqtl" class="section level3" number="10.3.6">
<h3><span class="header-section-number">10.3.6</span> Some commands for mapping in markers in R/qtl</h3>
<p><code>allchrsplits</code> = Test all possible splits of a chromosome into two pieces.</p>
<p><code>calc.errorlod</code> = Calculates a LOD associated with p(single genotype is wrong).</p>
<p><code>class(cross name)[1]&lt;-"dh"</code> = Change cross type.</p>
<p><code>cleanGeno</code> = Delete probable errors. Only works for backcrosses.</p>
<p><code>compareorder</code> = Compare the likelihood of two orderings of markers.</p>
<p><code>droponemarker</code> = Drop one marker at a time and determine effect on genetic map I find the output hard to interpret.</p>
<p><code>est.map</code> = Estimate map distances but keeps the same order.</p>
<p><code>est.rf</code> = Estimate all pair-wise recombination fractions.</p>
<p><code>formLinkageGroups</code> = Cluster markers into linkage groups.</p>
<p><code>geno.image</code> = Plot grid of genotype data.</p>
<p><code>geno.table</code> = Tabulate genotype classes and test for segregation distortion.</p>
<p><code>locateXO</code> = Estimate locations of crossovers.</p>
<p><code>movemarker</code>= Move a marker to a new chromosome.</p>
<p><code>nchr</code> = Determine the number of chromosomes.</p>
<p><code>nmar</code> = Determine the number of markers per chromosome.</p>
<p><code>orderMarkers</code> = Order markers within linkage groups.</p>
<p><code>plot(your cross)</code> = Plots a summary of the cross data.</p>
<p><code>plotGeno</code> = Plot observed genotypes, flagging likely errors.</p>
<p><code>plotErrorlod</code> = Plots possible genotype errors.</p>
<p><code>plotMap</code> = Plot linkage groups.</p>
<p><code>plotRF</code> = Produces a heat map of recombination fractions.</p>
<p><code>pull.rf</code> = Extract the matrix of recombination fractions from a cross.</p>
<p><code>pull.map</code> = Extract the current map from a cross.</p>
<p><code>jittermap</code> = Gets rid of the problem of having markers at the same location.</p>
<p><code>read.cross</code> = Read in genotype and phenotype data.</p>
<p><code>summary</code> = Works on much R/qtl output. Recent name changes may cause confusion. For example <code>summary.cross</code> = used to summarise the genotype and phenotype data in you a cross. This command is no longer recognised. <code>summary</code> on its own now does this. Older names are still present in books, online resources and R help.</p>
<p><code>ripple</code> = Search for improved marker order by permuting order within a small window.</p>
<p><code>switch.order</code> = Replace the order of markers in a linkage group with a new order.</p>
<p><code>write.cross</code> = Export your cross object</p>
</div>
<div id="final-comments" class="section level3" number="10.3.7">
<h3><span class="header-section-number">10.3.7</span> Final comments</h3>
<p>If you think mapping markers and QTL is more of an art than a science, you would be right. Remember all maps are approximations. Even when supplied with data and an existing map it is worth examining it for errors. Best, of course, is to produce high quality genotype data in the first place.</p>
</div>
</div>
<div id="rqtl2" class="section level2" number="10.4">
<h2><span class="header-section-number">10.4</span> R/QTL2</h2>
<p><a href="https://kbroman.org/qtl2/">R/QTL2</a> is an updated implementation of R/QTL. It has better ability to handle complex crosses, so we use it here to demonstrate analysis of an 8-founder MAGIC population. In addition, it can handle larger data sets and has the ability to link to genomic SNP databases, to allow smooth QTL mapping from phenotypic data through to candidate locus selection.</p>
<p>The data we will analyse are flowering time data from two years (2012 and 2013) in 643 lines from NIAB’s “MAGIC Elite” population, founded by 8-diverse elite (at the time of population construction) wheat varieties, containing c. 80% of polymorphisms found in UK wheat germplasm, as a whole <span class="citation">(<a href="references.html#ref-Gardner_2016" role="doc-biblioref">Gardner, Wittern, and Mackay 2016b</a>)</span>.</p>
<p>R/QTL2 has a completely novel data import system, based on creation of a .yaml file containing descriptive information on the population, and the names of the data files. The .yaml files and the data files all need to be in the same working directory. More details can be found on the R/QTL2 <a href="https://kbroman.org/qtl2/assets/vignettes/user_guide.html">website</a>.</p>
<p>The yaml file contains the following information.</p>
<pre><code>crosstype: riself8      
geno:  finalfileskim.csv

founder_geno:  founderfileskim.csv
pheno:  phenofile.csv

gmap:  gmapfileskim.csv
alleles:
  - ALC
  - BRO
  - CLA
  - HER
  - RIA
  - ROB
  - SOI
  - XI9
genotypes:

  AA: 1
  AB: 2
  BB: 3
cross_info: pedigree.csv

na.strings:
  - &#39;-&#39;
  - NA</code></pre>
<p>Filenames (included in your data folder) end with “.csv.” Note that we have 2 genotype files (founders and RILs), a genetic map file, a phenotype file and a pedigree file. The latter is necessary to analyse MAGIC populations as there were several rounds of crossing involved in the population construction. The 8 alleles are abbreviations of the names of the founder lines. Unlike in R/QTL, covariates need to be included in a separate covariates file in R/QTL2, although we are not going to use any in this analysis today. Finally, note that the map and genotype data here has been skimmed to remove 100% correlated markers, but not skimmed any further than that. We will investigate the effect of marker density on QTL mapping below.</p>

<div class="rmdnote">
There is a separate script for this section called: “MAGIC_QTL_mapping_script” in the Chapter 10 data folder.
</div>
<p>Install R/QTL and load your data:</p>
<div class="sourceCode" id="cb622"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb622-1"><a href="Mapping-Markers.html#cb622-1" aria-hidden="true" tabindex="-1"></a><span class="fu">install.packages</span>(<span class="st">&quot;qtl2&quot;</span>)</span>
<span id="cb622-2"><a href="Mapping-Markers.html#cb622-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(qtl2)</span>
<span id="cb622-3"><a href="Mapping-Markers.html#cb622-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb622-4"><a href="Mapping-Markers.html#cb622-4" aria-hidden="true" tabindex="-1"></a><span class="co"># read in the data</span></span>
<span id="cb622-5"><a href="Mapping-Markers.html#cb622-5" aria-hidden="true" tabindex="-1"></a>NIAB_MAGIC <span class="ot">&lt;-</span> <span class="fu">read_cross2</span>(<span class="st">&quot;MAGIC.yaml&quot;</span>)</span></code></pre></div>
<p>Firstly, we create pseudomarkers – markers at 1cM intervals (<code>step=1</code>) throughout the genome. These will help with QTL mapping in low density regions but also with a new way of estimating kinship to avoid marker density bias.</p>
<div class="sourceCode" id="cb623"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb623-1"><a href="Mapping-Markers.html#cb623-1" aria-hidden="true" tabindex="-1"></a><span class="co"># insert pseudomarkers (to estimate parentage at 1cM</span></span>
<span id="cb623-2"><a href="Mapping-Markers.html#cb623-2" aria-hidden="true" tabindex="-1"></a><span class="co"># intervals)</span></span>
<span id="cb623-3"><a href="Mapping-Markers.html#cb623-3" aria-hidden="true" tabindex="-1"></a>map <span class="ot">&lt;-</span> <span class="fu">insert_pseudomarkers</span>(NIAB_MAGIC<span class="sc">$</span>gmap, <span class="at">step =</span> <span class="dv">1</span>)</span></code></pre></div>
<p>Next, we need to calculate the genotype probabilities along the chromosome at both marker and pseudomarker positions, and then convert these to allele probabilities. Unlike in bi-parental mapping, where there are only two probabilities (parent 1 and parent 2), here there are 8 possible alleles to estimate.</p>
<div class="sourceCode" id="cb624"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb624-1"><a href="Mapping-Markers.html#cb624-1" aria-hidden="true" tabindex="-1"></a><span class="co"># calculate genotype probabilities and convert to allele</span></span>
<span id="cb624-2"><a href="Mapping-Markers.html#cb624-2" aria-hidden="true" tabindex="-1"></a><span class="co"># probabilties</span></span>
<span id="cb624-3"><a href="Mapping-Markers.html#cb624-3" aria-hidden="true" tabindex="-1"></a>pr <span class="ot">&lt;-</span> <span class="fu">calc_genoprob</span>(NIAB_MAGIC, map, <span class="at">error_prob =</span> <span class="fl">0.002</span>)</span>
<span id="cb624-4"><a href="Mapping-Markers.html#cb624-4" aria-hidden="true" tabindex="-1"></a>apr <span class="ot">&lt;-</span> <span class="fu">genoprob_to_alleleprob</span>(pr)</span></code></pre></div>

<div class="rmdquiz">
<strong>9</strong>
What might be the statistical consequences of QTL mapping with 8 alleles, compared to only 2?
</div>
<p>In R/QTL2, there are two main options for QTL analysis: Haley-Knott regression and a linear mixed model (LMM) approach requiring a kinship matrix in a similar fashion to association mapping. For multi-parental populations, achieving a perfectly balanced population is unfeasible, so the <code>LMM</code> approach is used. There are several options for estimating the kinship matrix:</p>
<ol style="list-style-type: decimal">
<li><p>Use all markers.</p></li>
<li><p>Scan each chromosome using a kinship matrix that is calculated using data from all other chromosomes – the “leave one chromosome out” (LOCO) method.</p></li>
<li><p>Using only the pseudomarkers so that we have an even density of markers across the recombination landscape (assuming our map is correct!). This approach is a novel way of estimating unbiased kinship, compared to our previous approach of strong marker thinning.</p></li>
</ol>
<p>Here we will compare approaches 2 and 3.</p>
<div class="sourceCode" id="cb625"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb625-1"><a href="Mapping-Markers.html#cb625-1" aria-hidden="true" tabindex="-1"></a><span class="co"># calculate kinship using only the evenly spread</span></span>
<span id="cb625-2"><a href="Mapping-Markers.html#cb625-2" aria-hidden="true" tabindex="-1"></a><span class="co"># pseudomarkers</span></span>
<span id="cb625-3"><a href="Mapping-Markers.html#cb625-3" aria-hidden="true" tabindex="-1"></a>grid <span class="ot">&lt;-</span> <span class="fu">calc_grid</span>(NIAB_MAGIC<span class="sc">$</span>gmap, <span class="at">step =</span> <span class="dv">1</span>)</span>
<span id="cb625-4"><a href="Mapping-Markers.html#cb625-4" aria-hidden="true" tabindex="-1"></a>pr_grid <span class="ot">&lt;-</span> <span class="fu">probs_to_grid</span>(pr, grid)</span>
<span id="cb625-5"><a href="Mapping-Markers.html#cb625-5" aria-hidden="true" tabindex="-1"></a>kinship_grid <span class="ot">&lt;-</span> <span class="fu">calc_kinship</span>(pr_grid)</span>
<span id="cb625-6"><a href="Mapping-Markers.html#cb625-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb625-7"><a href="Mapping-Markers.html#cb625-7" aria-hidden="true" tabindex="-1"></a><span class="co"># alternative kinship calculation with loco</span></span>
<span id="cb625-8"><a href="Mapping-Markers.html#cb625-8" aria-hidden="true" tabindex="-1"></a>kinship_loco <span class="ot">&lt;-</span> <span class="fu">calc_kinship</span>(apr, <span class="st">&quot;loco&quot;</span>)</span></code></pre></div>
<p>Now we are ready to perform a genome scan using the R/QTL2 version of <code>scan1</code>.</p>
<div class="sourceCode" id="cb626"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb626-1"><a href="Mapping-Markers.html#cb626-1" aria-hidden="true" tabindex="-1"></a><span class="co"># carry out a genome scan</span></span>
<span id="cb626-2"><a href="Mapping-Markers.html#cb626-2" aria-hidden="true" tabindex="-1"></a>out_grid <span class="ot">&lt;-</span> <span class="fu">scan1</span>(apr, NIAB_MAGIC<span class="sc">$</span>pheno, kinship_grid)</span>
<span id="cb626-3"><a href="Mapping-Markers.html#cb626-3" aria-hidden="true" tabindex="-1"></a>out_loco <span class="ot">&lt;-</span> <span class="fu">scan1</span>(apr, NIAB_MAGIC<span class="sc">$</span>pheno, kinship_loco)</span></code></pre></div>
<p>To obtain significance thresholds, we again run a permutation test (beware, this may take a few minutes).</p>
<div class="sourceCode" id="cb627"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb627-1"><a href="Mapping-Markers.html#cb627-1" aria-hidden="true" tabindex="-1"></a><span class="co"># carry our permutation tests</span></span>
<span id="cb627-2"><a href="Mapping-Markers.html#cb627-2" aria-hidden="true" tabindex="-1"></a>permtest <span class="ot">&lt;-</span> <span class="fu">scan1perm</span>(pr, NIAB_MAGIC<span class="sc">$</span>pheno, <span class="at">n_perm =</span> <span class="dv">1000</span>)</span>
<span id="cb627-3"><a href="Mapping-Markers.html#cb627-3" aria-hidden="true" tabindex="-1"></a><span class="fu">summary</span>(permtest, <span class="at">alpha =</span> <span class="fu">c</span>(<span class="fl">0.2</span>, <span class="fl">0.05</span>))</span></code></pre></div>
<pre><code>## LOD thresholds (1000 permutations)
##      GSJune11_2012 GS55T2013
## 0.2           5.84      5.91
## 0.05          6.85      6.76</code></pre>
<p>We can now plot our results. Firstly, we will compare the results between 2012 and 2013 for the LOCO approach.</p>
<div class="sourceCode" id="cb629"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb629-1"><a href="Mapping-Markers.html#cb629-1" aria-hidden="true" tabindex="-1"></a><span class="co"># plot QTL results for &#39;loco&#39; method</span></span>
<span id="cb629-2"><a href="Mapping-Markers.html#cb629-2" aria-hidden="true" tabindex="-1"></a><span class="fu">par</span>(<span class="at">mar =</span> <span class="fu">c</span>(<span class="fl">4.1</span>, <span class="fl">4.1</span>, <span class="fl">0.6</span>, <span class="fl">0.6</span>))</span>
<span id="cb629-3"><a href="Mapping-Markers.html#cb629-3" aria-hidden="true" tabindex="-1"></a>ymx <span class="ot">&lt;-</span> <span class="fu">maxlod</span>(out_loco)</span>
<span id="cb629-4"><a href="Mapping-Markers.html#cb629-4" aria-hidden="true" tabindex="-1"></a><span class="fu">par</span>(<span class="at">mfrow =</span> <span class="fu">c</span>(<span class="dv">2</span>, <span class="dv">1</span>))</span>
<span id="cb629-5"><a href="Mapping-Markers.html#cb629-5" aria-hidden="true" tabindex="-1"></a><span class="fu">plot</span>(out_loco, NIAB_MAGIC<span class="sc">$</span>gmap, <span class="at">lodcolumn =</span> <span class="dv">1</span>, <span class="at">ylim =</span> <span class="fu">c</span>(<span class="dv">0</span>, ymx <span class="sc">*</span> </span>
<span id="cb629-6"><a href="Mapping-Markers.html#cb629-6" aria-hidden="true" tabindex="-1"></a>    <span class="fl">1.02</span>))</span>
<span id="cb629-7"><a href="Mapping-Markers.html#cb629-7" aria-hidden="true" tabindex="-1"></a><span class="fu">abline</span>(<span class="at">h =</span> <span class="fl">6.68</span>, <span class="at">lty =</span> <span class="dv">2</span>)</span>
<span id="cb629-8"><a href="Mapping-Markers.html#cb629-8" aria-hidden="true" tabindex="-1"></a><span class="fu">legend</span>(<span class="st">&quot;topright&quot;</span>, <span class="at">lwd =</span> <span class="dv">2</span>, <span class="fu">c</span>(<span class="st">&quot;2012&quot;</span>), <span class="at">bg =</span> <span class="st">&quot;gray90&quot;</span>)</span>
<span id="cb629-9"><a href="Mapping-Markers.html#cb629-9" aria-hidden="true" tabindex="-1"></a><span class="fu">plot</span>(out_loco, NIAB_MAGIC<span class="sc">$</span>gmap, <span class="at">lodcolumn =</span> <span class="dv">2</span>, <span class="at">ylim =</span> <span class="fu">c</span>(<span class="dv">0</span>, ymx <span class="sc">*</span> </span>
<span id="cb629-10"><a href="Mapping-Markers.html#cb629-10" aria-hidden="true" tabindex="-1"></a>    <span class="fl">1.02</span>))</span>
<span id="cb629-11"><a href="Mapping-Markers.html#cb629-11" aria-hidden="true" tabindex="-1"></a><span class="fu">abline</span>(<span class="at">h =</span> <span class="fl">6.69</span>, <span class="at">lty =</span> <span class="dv">2</span>)</span>
<span id="cb629-12"><a href="Mapping-Markers.html#cb629-12" aria-hidden="true" tabindex="-1"></a><span class="fu">legend</span>(<span class="st">&quot;topright&quot;</span>, <span class="at">lwd =</span> <span class="dv">2</span>, <span class="fu">c</span>(<span class="st">&quot;2013&quot;</span>), <span class="at">bg =</span> <span class="st">&quot;gray90&quot;</span>)</span></code></pre></div>
<img src="bookdown-demo_files/figure-html/unnamed-chunk-479-1.png" width="672" />

<div class="rmdquiz">
<strong>10</strong>
Are QTL consistent between years? Do you think this trait may have high or low heritability?
</div>
<p>You will notice that there is one very strong QTL on chromosome 6. This is the PpdD1 locus for daylength sensitivity, which is segregating in this population, and is known to have a major effect on flowering time in wheat. If we had more time, we would use this allele as a covariate and re-run the analysis, as we did in the R/QTL exercise. We can also plot the two years in the same graph. This may give a better idea of between years presence/absence differences.</p>
<div class="sourceCode" id="cb630"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb630-1"><a href="Mapping-Markers.html#cb630-1" aria-hidden="true" tabindex="-1"></a><span class="co"># plot on same figure</span></span>
<span id="cb630-2"><a href="Mapping-Markers.html#cb630-2" aria-hidden="true" tabindex="-1"></a><span class="fu">par</span>(<span class="at">mfrow =</span> <span class="fu">c</span>(<span class="dv">1</span>, <span class="dv">1</span>))</span>
<span id="cb630-3"><a href="Mapping-Markers.html#cb630-3" aria-hidden="true" tabindex="-1"></a><span class="fu">plot</span>(out_loco, NIAB_MAGIC<span class="sc">$</span>gmap, <span class="at">lodcolumn =</span> <span class="dv">1</span>, <span class="at">col =</span> <span class="st">&quot;blue&quot;</span>, </span>
<span id="cb630-4"><a href="Mapping-Markers.html#cb630-4" aria-hidden="true" tabindex="-1"></a>    <span class="at">ylim =</span> <span class="fu">c</span>(<span class="dv">0</span>, ymx <span class="sc">*</span> <span class="fl">1.02</span>))</span>
<span id="cb630-5"><a href="Mapping-Markers.html#cb630-5" aria-hidden="true" tabindex="-1"></a><span class="fu">plot</span>(out_loco, NIAB_MAGIC<span class="sc">$</span>gmap, <span class="at">lodcolumn =</span> <span class="dv">2</span>, <span class="at">col =</span> <span class="st">&quot;green&quot;</span>, </span>
<span id="cb630-6"><a href="Mapping-Markers.html#cb630-6" aria-hidden="true" tabindex="-1"></a>    <span class="at">add =</span> <span class="cn">TRUE</span>)</span>
<span id="cb630-7"><a href="Mapping-Markers.html#cb630-7" aria-hidden="true" tabindex="-1"></a><span class="fu">abline</span>(<span class="at">h =</span> <span class="fl">6.68</span>, <span class="at">lty =</span> <span class="dv">2</span>)</span>
<span id="cb630-8"><a href="Mapping-Markers.html#cb630-8" aria-hidden="true" tabindex="-1"></a><span class="fu">legend</span>(<span class="st">&quot;topright&quot;</span>, <span class="at">lwd =</span> <span class="dv">2</span>, <span class="at">col =</span> <span class="fu">c</span>(<span class="st">&quot;blue&quot;</span>, <span class="st">&quot;green&quot;</span>), <span class="fu">colnames</span>(out_loco), </span>
<span id="cb630-9"><a href="Mapping-Markers.html#cb630-9" aria-hidden="true" tabindex="-1"></a>    <span class="at">bg =</span> <span class="st">&quot;gray90&quot;</span>)</span></code></pre></div>
<p><img src="bookdown-demo_files/figure-html/unnamed-chunk-481-1.png" width="672" /></p>
<p>Now we can compare the LOCO and grid kinship approaches for the 2013 dataset.</p>
<div class="sourceCode" id="cb631"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb631-1"><a href="Mapping-Markers.html#cb631-1" aria-hidden="true" tabindex="-1"></a><span class="fu">par</span>(<span class="at">mfrow =</span> <span class="fu">c</span>(<span class="dv">2</span>, <span class="dv">1</span>))</span>
<span id="cb631-2"><a href="Mapping-Markers.html#cb631-2" aria-hidden="true" tabindex="-1"></a><span class="fu">plot</span>(out_loco, NIAB_MAGIC<span class="sc">$</span>gmap, <span class="at">lodcolumn =</span> <span class="dv">2</span>, <span class="at">col =</span> <span class="st">&quot;blue&quot;</span>, </span>
<span id="cb631-3"><a href="Mapping-Markers.html#cb631-3" aria-hidden="true" tabindex="-1"></a>    <span class="at">ylim =</span> <span class="fu">c</span>(<span class="dv">0</span>, ymx <span class="sc">*</span> <span class="fl">1.02</span>))</span>
<span id="cb631-4"><a href="Mapping-Markers.html#cb631-4" aria-hidden="true" tabindex="-1"></a><span class="fu">abline</span>(<span class="at">h =</span> <span class="fl">6.69</span>, <span class="at">lty =</span> <span class="dv">2</span>)</span>
<span id="cb631-5"><a href="Mapping-Markers.html#cb631-5" aria-hidden="true" tabindex="-1"></a><span class="fu">legend</span>(<span class="st">&quot;topright&quot;</span>, <span class="at">lwd =</span> <span class="dv">2</span>, <span class="fu">c</span>(<span class="st">&quot;2013_loco&quot;</span>), <span class="at">bg =</span> <span class="st">&quot;gray90&quot;</span>)</span>
<span id="cb631-6"><a href="Mapping-Markers.html#cb631-6" aria-hidden="true" tabindex="-1"></a><span class="fu">plot</span>(out_grid, NIAB_MAGIC<span class="sc">$</span>gmap, <span class="at">lodcolumn =</span> <span class="dv">2</span>, <span class="at">col =</span> <span class="st">&quot;blue&quot;</span>, </span>
<span id="cb631-7"><a href="Mapping-Markers.html#cb631-7" aria-hidden="true" tabindex="-1"></a>    <span class="at">ylim =</span> <span class="fu">c</span>(<span class="dv">0</span>, ymx <span class="sc">*</span> <span class="fl">1.02</span>))</span>
<span id="cb631-8"><a href="Mapping-Markers.html#cb631-8" aria-hidden="true" tabindex="-1"></a><span class="fu">abline</span>(<span class="at">h =</span> <span class="fl">6.69</span>, <span class="at">lty =</span> <span class="dv">2</span>)</span>
<span id="cb631-9"><a href="Mapping-Markers.html#cb631-9" aria-hidden="true" tabindex="-1"></a><span class="fu">legend</span>(<span class="st">&quot;topright&quot;</span>, <span class="at">lwd =</span> <span class="dv">2</span>, <span class="fu">c</span>(<span class="st">&quot;2013_grid&quot;</span>), <span class="at">bg =</span> <span class="st">&quot;gray90&quot;</span>)</span></code></pre></div>
<p><img src="bookdown-demo_files/figure-html/unnamed-chunk-482-1.png" width="672" /></p>

<div class="rmdquiz">
<strong>11</strong>
What do you conclude about using the two approaches?
</div>
<p>We can now find the peaks and their confidence intervals. Two approaches are available in R/QTL2: a Bayesian and a lod based method. To separate QTL peaks, you need to specify a lod drop interval and a peak lod drop interval. The latter is to cover the lowest of the two adjacent peaks. There is something of an artform to choosing the lod drop values. A lod drop of 1.8 is equivalent to about a 1:60 odds ratio.</p>
<div class="sourceCode" id="cb632"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb632-1"><a href="Mapping-Markers.html#cb632-1" aria-hidden="true" tabindex="-1"></a><span class="co"># find the peaks and their intervals, peakdrop indicates the</span></span>
<span id="cb632-2"><a href="Mapping-Markers.html#cb632-2" aria-hidden="true" tabindex="-1"></a><span class="co"># amount that the LOD curve must drop below the lowest of two</span></span>
<span id="cb632-3"><a href="Mapping-Markers.html#cb632-3" aria-hidden="true" tabindex="-1"></a><span class="co"># adjacent peaks.</span></span>
<span id="cb632-4"><a href="Mapping-Markers.html#cb632-4" aria-hidden="true" tabindex="-1"></a><span class="fu">find_peaks</span>(out_loco, map, <span class="at">threshold =</span> <span class="fl">6.69</span>, <span class="at">peakdrop =</span> <span class="fl">1.8</span>, <span class="at">drop =</span> <span class="fl">1.5</span>)</span></code></pre></div>
<pre><code>##    lodindex     lodcolumn chr       pos       lod     ci_lo     ci_hi
## 1         1 GSJune11_2012   2 349.59230  8.301127 343.90026 351.09987
## 2         1 GSJune11_2012   6  61.00000 19.012676  55.40415  77.76853
## 3         1 GSJune11_2012  10 158.74511  7.331446 137.78485 164.01313
## 4         1 GSJune11_2012  10 172.00000  7.254170 164.01315 177.86805
## 5         2     GS55T2013   2 349.59230  7.019046 345.42322 351.09987
## 6         2     GS55T2013   6  60.67219 27.232911  55.40415  77.76853
## 7         2     GS55T2013   7 108.23122  8.107504  90.11471 114.28704
## 8         2     GS55T2013  10 139.00000  9.147522 131.22663 164.01313
## 9         2     GS55T2013  10 175.00000 11.809972 170.63993 179.37562
## 10        2     GS55T2013  11 119.35148 10.270608  95.42024 125.98272
## 11        2     GS55T2013  11 170.65828  8.617282 158.50466 179.44603
## 12        2     GS55T2013  11 183.54483  7.495636 179.44603 188.63207</code></pre>

<div class="rmdquiz">
<strong>12</strong>
Examine the selected peaks, especially on chromosomnes 10 and 11. Do you think our lod drop thresholds were sufficient or should we be more conservative?
</div>
<p>Now we can examine the allelic effects of our detected QTL. We will focus here on the QTLs on chromosomes 6 and 10.</p>
<div class="sourceCode" id="cb634"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb634-1"><a href="Mapping-Markers.html#cb634-1" aria-hidden="true" tabindex="-1"></a><span class="co"># plot allelic effects for chromosome 6 (2013 phenotype)</span></span>
<span id="cb634-2"><a href="Mapping-Markers.html#cb634-2" aria-hidden="true" tabindex="-1"></a><span class="fu">par</span>(<span class="at">mfrow =</span> <span class="fu">c</span>(<span class="dv">1</span>, <span class="dv">1</span>))</span>
<span id="cb634-3"><a href="Mapping-Markers.html#cb634-3" aria-hidden="true" tabindex="-1"></a>col <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&quot;#FFDC00&quot;</span>, <span class="st">&quot;#888888&quot;</span>, <span class="st">&quot;#F08080&quot;</span>, <span class="st">&quot;#0064C9&quot;</span>, <span class="st">&quot;#7FDBFF&quot;</span>, </span>
<span id="cb634-4"><a href="Mapping-Markers.html#cb634-4" aria-hidden="true" tabindex="-1"></a>    <span class="st">&quot;#2ECC40&quot;</span>, <span class="st">&quot;#FF4136&quot;</span>, <span class="st">&quot;#B10DC9&quot;</span>)</span>
<span id="cb634-5"><a href="Mapping-Markers.html#cb634-5" aria-hidden="true" tabindex="-1"></a>legend <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&quot;ALC&quot;</span>, <span class="st">&quot;BRO&quot;</span>, <span class="st">&quot;CLA&quot;</span>, <span class="st">&quot;HER&quot;</span>, <span class="st">&quot;RIA&quot;</span>, <span class="st">&quot;ROB&quot;</span>, <span class="st">&quot;SOI&quot;</span>, </span>
<span id="cb634-6"><a href="Mapping-Markers.html#cb634-6" aria-hidden="true" tabindex="-1"></a>    <span class="st">&quot;X19&quot;</span>)</span>
<span id="cb634-7"><a href="Mapping-Markers.html#cb634-7" aria-hidden="true" tabindex="-1"></a>coef_c2 <span class="ot">&lt;-</span> <span class="fu">scan1coef</span>(apr[, <span class="st">&quot;6&quot;</span>], NIAB_MAGIC<span class="sc">$</span>pheno[, <span class="dv">2</span>])</span>
<span id="cb634-8"><a href="Mapping-Markers.html#cb634-8" aria-hidden="true" tabindex="-1"></a><span class="fu">par</span>(<span class="at">mar =</span> <span class="fu">c</span>(<span class="fl">4.1</span>, <span class="fl">4.1</span>, <span class="fl">0.6</span>, <span class="fl">0.6</span>))</span>
<span id="cb634-9"><a href="Mapping-Markers.html#cb634-9" aria-hidden="true" tabindex="-1"></a><span class="fu">plot_coef</span>(coef_c2[, <span class="dv">1</span><span class="sc">:</span><span class="dv">8</span>], NIAB_MAGIC<span class="sc">$</span>gmap[<span class="st">&quot;6&quot;</span>], <span class="at">col =</span> col, <span class="at">scan1_output =</span> out_loco, </span>
<span id="cb634-10"><a href="Mapping-Markers.html#cb634-10" aria-hidden="true" tabindex="-1"></a>    <span class="at">bgcolor =</span> <span class="st">&quot;gray95&quot;</span>, <span class="at">legend =</span> <span class="st">&quot;bottomright&quot;</span>)</span></code></pre></div>
<p><img src="bookdown-demo_files/figure-html/unnamed-chunk-486-1.png" width="672" /></p>
<div class="sourceCode" id="cb635"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb635-1"><a href="Mapping-Markers.html#cb635-1" aria-hidden="true" tabindex="-1"></a><span class="co"># and for chromosome 10</span></span>
<span id="cb635-2"><a href="Mapping-Markers.html#cb635-2" aria-hidden="true" tabindex="-1"></a>coef_c2 <span class="ot">&lt;-</span> <span class="fu">scan1coef</span>(apr[, <span class="st">&quot;10&quot;</span>], NIAB_MAGIC<span class="sc">$</span>pheno[, <span class="dv">2</span>])</span>
<span id="cb635-3"><a href="Mapping-Markers.html#cb635-3" aria-hidden="true" tabindex="-1"></a><span class="fu">par</span>(<span class="at">mar =</span> <span class="fu">c</span>(<span class="fl">4.1</span>, <span class="fl">4.1</span>, <span class="fl">0.6</span>, <span class="fl">0.6</span>))</span>
<span id="cb635-4"><a href="Mapping-Markers.html#cb635-4" aria-hidden="true" tabindex="-1"></a><span class="fu">plot_coef</span>(coef_c2[, <span class="dv">1</span><span class="sc">:</span><span class="dv">8</span>], NIAB_MAGIC<span class="sc">$</span>gmap[<span class="st">&quot;10&quot;</span>], <span class="at">col =</span> col, <span class="at">scan1_output =</span> out_loco, </span>
<span id="cb635-5"><a href="Mapping-Markers.html#cb635-5" aria-hidden="true" tabindex="-1"></a>    <span class="at">bgcolor =</span> <span class="st">&quot;gray95&quot;</span>, <span class="at">legend =</span> <span class="st">&quot;bottomleft&quot;</span>)</span></code></pre></div>
<p><img src="bookdown-demo_files/figure-html/unnamed-chunk-486-2.png" width="672" /></p>

<div class="rmdquiz">
<strong>13</strong>
Which parental lines are carrying the biggest effects at these QTLs? How big are these effects?
</div>
<p>There is an artefact evident on chromosome 10, which adds complexity interpretation of our results. This is common in this population: we will discuss this as a class. Interpretation in MAGIC is not always easy!</p>
<p>One of the big advantages of MAGIC populations compared to bi-parental mapping populations is that the several rounds of crossing involved in their construction result in more recombination and hence finer QTL resolution, i.e., there are fewer potential causative genes underlying each QTL in MAGIC populations. A powerful feature of R/QTL2 is the ability to connect directly to SNP and gene databases to go straight to this list of potential candidates. A nice MAGIC example is provided in the R/QTL User guide. We do have complete whole genome data for all our MAGIC founder lines so if we had time, the next step in this analysis would be to pull up all the SNP variants in the QTL intervals we have detected above. We can then use our genotype probabilities and founder SNP genotypes to assign a SNP allele to every RIL in the population based on founder haplotypes, and then scan every SNP in this interval for association with the QTL. The output is a list of SNPs ordered by lod score, including the identities of any genes in which the SNPs are found, i.e. a list of candidate genes. This is a very powerful tool for gene discovery.</p>

</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="Association-mapping.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="Data-Visualization.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": "https://github.com/rstudio/bookdown-demo/edit/master/10-Mapping_markers_and_qtl.Rmd",
"text": "Edit"
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": ["bookdown-demo.pdf", "bookdown-demo.epub"],
"toc": {
"collapse": "subsection"
}
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
