<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 7 Population genetics. | Quantitative Methods in Plant Breeding: The Digital Tutorials.</title>
  <meta name="description" content="This is the online tutorial booklet for the QMPB course run at NIAB. The online book was formed using R bookdown." />
  <meta name="generator" content="bookdown 0.21 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 7 Population genetics. | Quantitative Methods in Plant Breeding: The Digital Tutorials." />
  <meta property="og:type" content="book" />
  
  <meta property="og:image" content="images/cover.jpg" />
  <meta property="og:description" content="This is the online tutorial booklet for the QMPB course run at NIAB. The online book was formed using R bookdown." />
  <meta name="github-repo" content="rstudio/bookdown" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 7 Population genetics. | Quantitative Methods in Plant Breeding: The Digital Tutorials." />
  
  <meta name="twitter:description" content="This is the online tutorial booklet for the QMPB course run at NIAB. The online book was formed using R bookdown." />
  <meta name="twitter:image" content="images/cover.jpg" />

<meta name="author" content="NIAB 2021" />


<meta name="date" content="2021-03-03" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="Cross-site-analysis.html"/>
<link rel="next" href="Imputation.html"/>
<script src="libs/header-attrs-2.6/header-attrs.js"></script>
<script src="libs/jquery-2.2.3/jquery.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />











<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">The QMPB digital tutorials</a></li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> Introduction</a></li>
<li class="chapter" data-level="2" data-path="Acknow.html"><a href="Acknow.html"><i class="fa fa-check"></i><b>2</b> Acknowledgements and Contents</a>
<ul>
<li class="chapter" data-level="2.1" data-path="Acknow.html"><a href="Acknow.html#acknowledgements"><i class="fa fa-check"></i><b>2.1</b> Acknowledgements</a></li>
<li class="chapter" data-level="2.2" data-path="Acknow.html"><a href="Acknow.html#digital-tutorial-chapters"><i class="fa fa-check"></i><b>2.2</b> Digital Tutorial Chapters:</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="R-and-stats.html"><a href="R-and-stats.html"><i class="fa fa-check"></i><b>3</b> R and Rstudio: a general introduction with statistics.</a>
<ul>
<li class="chapter" data-level="3.1" data-path="R-and-stats.html"><a href="R-and-stats.html#introduction-section-on-r"><i class="fa fa-check"></i><b>3.1</b> Introduction section on R</a>
<ul>
<li class="chapter" data-level="3.1.1" data-path="R-and-stats.html"><a href="R-and-stats.html#getting-the-data-ready"><i class="fa fa-check"></i><b>3.1.1</b> Getting the data ready</a></li>
</ul></li>
<li class="chapter" data-level="3.2" data-path="R-and-stats.html"><a href="R-and-stats.html#the-rstudio-layout-workspace-and-environment."><i class="fa fa-check"></i><b>3.2</b> The Rstudio layout, workspace and environment.</a>
<ul>
<li class="chapter" data-level="3.2.1" data-path="R-and-stats.html"><a href="R-and-stats.html#working-directory"><i class="fa fa-check"></i><b>3.2.1</b> Working directory</a></li>
</ul></li>
<li class="chapter" data-level="3.3" data-path="R-and-stats.html"><a href="R-and-stats.html#basic-r-syntx"><i class="fa fa-check"></i><b>3.3</b> Basic R Syntx</a>
<ul>
<li class="chapter" data-level="3.3.1" data-path="R-and-stats.html"><a href="R-and-stats.html#brackets"><i class="fa fa-check"></i><b>3.3.1</b> Brackets</a></li>
<li class="chapter" data-level="3.3.2" data-path="R-and-stats.html"><a href="R-and-stats.html#equals-symbols"><i class="fa fa-check"></i><b>3.3.2</b> Equals symbols</a></li>
<li class="chapter" data-level="3.3.3" data-path="R-and-stats.html"><a href="R-and-stats.html#continuation-character."><i class="fa fa-check"></i><b>3.3.3</b> Continuation character.</a></li>
<li class="chapter" data-level="3.3.4" data-path="R-and-stats.html"><a href="R-and-stats.html#text-and-numbers"><i class="fa fa-check"></i><b>3.3.4</b> Text and numbers</a></li>
<li class="chapter" data-level="3.3.5" data-path="R-and-stats.html"><a href="R-and-stats.html#summary-of-syntax"><i class="fa fa-check"></i><b>3.3.5</b> Summary of syntax</a></li>
</ul></li>
<li class="chapter" data-level="3.4" data-path="R-and-stats.html"><a href="R-and-stats.html#reading-in-data"><i class="fa fa-check"></i><b>3.4</b> Reading in data</a></li>
<li class="chapter" data-level="3.5" data-path="R-and-stats.html"><a href="R-and-stats.html#understanding-data-types-and-r-objects"><i class="fa fa-check"></i><b>3.5</b> Understanding data types and R objects</a>
<ul>
<li class="chapter" data-level="3.5.1" data-path="R-and-stats.html"><a href="R-and-stats.html#summary"><i class="fa fa-check"></i><b>3.5.1</b> Summary</a></li>
</ul></li>
<li class="chapter" data-level="3.6" data-path="R-and-stats.html"><a href="R-and-stats.html#summarising-and-displaying-data"><i class="fa fa-check"></i><b>3.6</b> Summarising and displaying data</a></li>
<li class="chapter" data-level="3.7" data-path="R-and-stats.html"><a href="R-and-stats.html#calculations-and-data-manipulation"><i class="fa fa-check"></i><b>3.7</b> Calculations and data manipulation</a>
<ul>
<li class="chapter" data-level="3.7.1" data-path="R-and-stats.html"><a href="R-and-stats.html#sorting-data"><i class="fa fa-check"></i><b>3.7.1</b> Sorting data</a></li>
<li class="chapter" data-level="3.7.2" data-path="R-and-stats.html"><a href="R-and-stats.html#relationships-between-variates"><i class="fa fa-check"></i><b>3.7.2</b> Relationships between variates</a></li>
<li class="chapter" data-level="3.7.3" data-path="R-and-stats.html"><a href="R-and-stats.html#exporting-your-script-and-workspace"><i class="fa fa-check"></i><b>3.7.3</b> Exporting your script and workspace</a></li>
<li class="chapter" data-level="3.7.4" data-path="R-and-stats.html"><a href="R-and-stats.html#saving-a-data-table"><i class="fa fa-check"></i><b>3.7.4</b> Saving a data table</a></li>
</ul></li>
<li class="chapter" data-level="3.8" data-path="R-and-stats.html"><a href="R-and-stats.html#basic-statistical-analysis"><i class="fa fa-check"></i><b>3.8</b> Basic statistical analysis</a>
<ul>
<li class="chapter" data-level="3.8.1" data-path="R-and-stats.html"><a href="R-and-stats.html#the-t-test"><i class="fa fa-check"></i><b>3.8.1</b> The t-test</a></li>
<li class="chapter" data-level="3.8.2" data-path="R-and-stats.html"><a href="R-and-stats.html#linear-regression"><i class="fa fa-check"></i><b>3.8.2</b> Linear Regression</a></li>
<li class="chapter" data-level="3.8.3" data-path="R-and-stats.html"><a href="R-and-stats.html#multiple-regression"><i class="fa fa-check"></i><b>3.8.3</b> Multiple regression</a></li>
<li class="chapter" data-level="3.8.4" data-path="R-and-stats.html"><a href="R-and-stats.html#the-analysis-of-variance"><i class="fa fa-check"></i><b>3.8.4</b> The analysis of variance</a></li>
</ul></li>
<li class="chapter" data-level="3.9" data-path="R-and-stats.html"><a href="R-and-stats.html#categorical-data---the-chi-squared-test"><i class="fa fa-check"></i><b>3.9</b> Categorical data - the <em>chi</em>-squared test</a></li>
<li class="chapter" data-level="3.10" data-path="R-and-stats.html"><a href="R-and-stats.html#more-useful-information"><i class="fa fa-check"></i><b>3.10</b> More useful information</a>
<ul>
<li class="chapter" data-level="3.10.1" data-path="R-and-stats.html"><a href="R-and-stats.html#plotting-graphs"><i class="fa fa-check"></i><b>3.10.1</b> Plotting Graphs</a></li>
<li class="chapter" data-level="3.10.2" data-path="R-and-stats.html"><a href="R-and-stats.html#probability-distributions"><i class="fa fa-check"></i><b>3.10.2</b> Probability distributions</a></li>
<li class="chapter" data-level="3.10.3" data-path="R-and-stats.html"><a href="R-and-stats.html#random-number-generation"><i class="fa fa-check"></i><b>3.10.3</b> Random number generation</a></li>
<li class="chapter" data-level="3.10.4" data-path="R-and-stats.html"><a href="R-and-stats.html#loops-second-example"><i class="fa fa-check"></i><b>3.10.4</b> Loops: <em>second example</em></a></li>
<li class="chapter" data-level="3.10.5" data-path="R-and-stats.html"><a href="R-and-stats.html#miscellany"><i class="fa fa-check"></i><b>3.10.5</b> Miscellany</a></li>
<li class="chapter" data-level="3.10.6" data-path="R-and-stats.html"><a href="R-and-stats.html#my-number-one-piece-of-advice"><i class="fa fa-check"></i><b>3.10.6</b> My number one piece of advice</a></li>
<li class="chapter" data-level="3.10.7" data-path="R-and-stats.html"><a href="R-and-stats.html#learn-more"><i class="fa fa-check"></i><b>3.10.7</b> Learn more</a></li>
<li class="chapter" data-level="3.10.8" data-path="R-and-stats.html"><a href="R-and-stats.html#packages"><i class="fa fa-check"></i><b>3.10.8</b> Packages</a></li>
<li class="chapter" data-level="3.10.9" data-path="R-and-stats.html"><a href="R-and-stats.html#the-tidyverse"><i class="fa fa-check"></i><b>3.10.9</b> The Tidyverse</a></li>
<li class="chapter" data-level="3.10.10" data-path="R-and-stats.html"><a href="R-and-stats.html#other-resources"><i class="fa fa-check"></i><b>3.10.10</b> Other resources</a></li>
</ul></li>
<li class="chapter" data-level="3.11" data-path="R-and-stats.html"><a href="R-and-stats.html#list-of-commands-mainly-described-in-this-guide."><i class="fa fa-check"></i><b>3.11</b> List of commands, mainly described in this guide.</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="Trial-design.html"><a href="Trial-design.html"><i class="fa fa-check"></i><b>4</b> Trial Design.</a>
<ul>
<li class="chapter" data-level="4.1" data-path="Trial-design.html"><a href="Trial-design.html#randomised-complete-block-designs"><i class="fa fa-check"></i><b>4.1</b> Randomised complete block designs</a></li>
<li class="chapter" data-level="4.2" data-path="Trial-design.html"><a href="Trial-design.html#trial-design-in-r"><i class="fa fa-check"></i><b>4.2</b> Trial design in R</a>
<ul>
<li class="chapter" data-level="4.2.1" data-path="Trial-design.html"><a href="Trial-design.html#partially-replicated-designs"><i class="fa fa-check"></i><b>4.2.1</b> Partially replicated designs</a></li>
<li class="chapter" data-level="4.2.2" data-path="Trial-design.html"><a href="Trial-design.html#two-dimensional-designs"><i class="fa fa-check"></i><b>4.2.2</b> Two-dimensional designs</a></li>
</ul></li>
<li class="chapter" data-level="4.3" data-path="Trial-design.html"><a href="Trial-design.html#farm-scale-experiments"><i class="fa fa-check"></i><b>4.3</b> Farm scale experiments</a></li>
<li class="chapter" data-level="4.4" data-path="Trial-design.html"><a href="Trial-design.html#alpha-designs."><i class="fa fa-check"></i><b>4.4</b> Alpha designs.</a>
<ul>
<li class="chapter" data-level="4.4.1" data-path="Trial-design.html"><a href="Trial-design.html#creation-of-augmented-and-p-rep-designs-from-2-replicate-alpha-designs"><i class="fa fa-check"></i><b>4.4.1</b> Creation of augmented and p-rep designs from 2-replicate alpha designs</a></li>
<li class="chapter" data-level="4.4.2" data-path="Trial-design.html"><a href="Trial-design.html#final-message"><i class="fa fa-check"></i><b>4.4.2</b> Final message</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="5" data-path="Trial-analysis.html"><a href="Trial-analysis.html"><i class="fa fa-check"></i><b>5</b> Trial analysis (single-site).</a>
<ul>
<li class="chapter" data-level="5.1" data-path="Trial-analysis.html"><a href="Trial-analysis.html#the-layout-of-the-sugar-beet-trial"><i class="fa fa-check"></i><b>5.1</b> The layout of the sugar beet trial</a></li>
<li class="chapter" data-level="5.2" data-path="Trial-analysis.html"><a href="Trial-analysis.html#fixed-effects-random-effects-and-reml"><i class="fa fa-check"></i><b>5.2</b> Fixed effects, random effects and REML</a>
<ul>
<li class="chapter" data-level="5.2.1" data-path="Trial-analysis.html"><a href="Trial-analysis.html#preparing-the-data"><i class="fa fa-check"></i><b>5.2.1</b> Preparing the data</a></li>
<li class="chapter" data-level="5.2.2" data-path="Trial-analysis.html"><a href="Trial-analysis.html#randomised-complete-block-design"><i class="fa fa-check"></i><b>5.2.2</b> Randomised complete block design</a></li>
<li class="chapter" data-level="5.2.3" data-path="Trial-analysis.html"><a href="Trial-analysis.html#randomised-compelete-block-design---with-mixed-models"><i class="fa fa-check"></i><b>5.2.3</b> Randomised compelete block design - with mixed models</a></li>
<li class="chapter" data-level="5.2.4" data-path="Trial-analysis.html"><a href="Trial-analysis.html#plotting-residuals-against-field-structure."><i class="fa fa-check"></i><b>5.2.4</b> Plotting residuals against field structure.</a></li>
<li class="chapter" data-level="5.2.5" data-path="Trial-analysis.html"><a href="Trial-analysis.html#alpha-design-include-block-structure"><i class="fa fa-check"></i><b>5.2.5</b> Alpha design (include block structure)</a></li>
<li class="chapter" data-level="5.2.6" data-path="Trial-analysis.html"><a href="Trial-analysis.html#rows-and-column-analysis"><i class="fa fa-check"></i><b>5.2.6</b> Rows and column analysis</a></li>
</ul></li>
<li class="chapter" data-level="5.3" data-path="Trial-analysis.html"><a href="Trial-analysis.html#spatial-analysis"><i class="fa fa-check"></i><b>5.3</b> Spatial analysis</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="Cross-site-analysis.html"><a href="Cross-site-analysis.html"><i class="fa fa-check"></i><b>6</b> Trial analysis (cross-sites).</a>
<ul>
<li class="chapter" data-level="6.1" data-path="Cross-site-analysis.html"><a href="Cross-site-analysis.html#blues-blups-and-one-stage-analysis"><i class="fa fa-check"></i><b>6.1</b> BLUEs, BLUPs and one stage analysis</a>
<ul>
<li class="chapter" data-level="6.1.1" data-path="Cross-site-analysis.html"><a href="Cross-site-analysis.html#one-stage-analysis-with-weights"><i class="fa fa-check"></i><b>6.1.1</b> One stage analysis with weights</a></li>
</ul></li>
<li class="chapter" data-level="6.2" data-path="Cross-site-analysis.html"><a href="Cross-site-analysis.html#weighted-two-stage-analysis"><i class="fa fa-check"></i><b>6.2</b> Weighted two-stage analysis</a></li>
<li class="chapter" data-level="6.3" data-path="Cross-site-analysis.html"><a href="Cross-site-analysis.html#ammi"><i class="fa fa-check"></i><b>6.3</b> AMMI</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="Population-genetics.html"><a href="Population-genetics.html"><i class="fa fa-check"></i><b>7</b> Population genetics.</a>
<ul>
<li class="chapter" data-level="7.1" data-path="Population-genetics.html"><a href="Population-genetics.html#hardy-weinberg-equilibrium-and-population-structure"><i class="fa fa-check"></i><b>7.1</b> Hardy-Weinberg equilibrium and population structure</a></li>
<li class="chapter" data-level="7.2" data-path="Population-genetics.html"><a href="Population-genetics.html#other-ways-of-analysing-population-structure"><i class="fa fa-check"></i><b>7.2</b> Other ways of analysing population structure</a>
<ul>
<li class="chapter" data-level="7.2.1" data-path="Population-genetics.html"><a href="Population-genetics.html#distance-trees"><i class="fa fa-check"></i><b>7.2.1</b> Distance Trees</a></li>
</ul></li>
<li class="chapter" data-level="7.3" data-path="Population-genetics.html"><a href="Population-genetics.html#structure"><i class="fa fa-check"></i><b>7.3</b> Structure</a>
<ul>
<li class="chapter" data-level="7.3.1" data-path="Population-genetics.html"><a href="Population-genetics.html#what-structure-does"><i class="fa fa-check"></i><b>7.3.1</b> What STRUCTURE does</a></li>
<li class="chapter" data-level="7.3.2" data-path="Population-genetics.html"><a href="Population-genetics.html#limitations-of-structure"><i class="fa fa-check"></i><b>7.3.2</b> Limitations of STRUCTURE</a></li>
<li class="chapter" data-level="7.3.3" data-path="Population-genetics.html"><a href="Population-genetics.html#using-structure"><i class="fa fa-check"></i><b>7.3.3</b> Using Structure</a></li>
<li class="chapter" data-level="7.3.4" data-path="Population-genetics.html"><a href="Population-genetics.html#the-exercise"><i class="fa fa-check"></i><b>7.3.4</b> The exercise</a></li>
</ul></li>
<li class="chapter" data-level="7.4" data-path="Population-genetics.html"><a href="Population-genetics.html#linkage-disequilibrium-analysis"><i class="fa fa-check"></i><b>7.4</b> Linkage disequilibrium analysis</a>
<ul>
<li class="chapter" data-level="7.4.1" data-path="Population-genetics.html"><a href="Population-genetics.html#linkage-disequilibrium-in-the-teosinte-and-barley-data-sets"><i class="fa fa-check"></i><b>7.4.1</b> Linkage disequilibrium in the teosinte and barley data sets</a></li>
</ul></li>
<li class="chapter" data-level="7.5" data-path="Population-genetics.html"><a href="Population-genetics.html#recombination-and-linkage-disequilibrium-applications"><i class="fa fa-check"></i><b>7.5</b> Recombination and linkage disequilibrium: applications</a></li>
<li class="chapter" data-level="7.6" data-path="Population-genetics.html"><a href="Population-genetics.html#pedigree-analysis"><i class="fa fa-check"></i><b>7.6</b> Pedigree analysis</a></li>
</ul></li>
<li class="chapter" data-level="8" data-path="Imputation.html"><a href="Imputation.html"><i class="fa fa-check"></i><b>8</b> Imputation.</a>
<ul>
<li class="chapter" data-level="8.1" data-path="Imputation.html"><a href="Imputation.html#introduction"><i class="fa fa-check"></i><b>8.1</b> Introduction</a></li>
<li class="chapter" data-level="8.2" data-path="Imputation.html"><a href="Imputation.html#the-exercise-1"><i class="fa fa-check"></i><b>8.2</b> The exercise</a></li>
<li class="chapter" data-level="8.3" data-path="Imputation.html"><a href="Imputation.html#data-preparation-for-all-methods"><i class="fa fa-check"></i><b>8.3</b> Data preparation (for all methods)</a></li>
<li class="chapter" data-level="8.4" data-path="Imputation.html"><a href="Imputation.html#svd-imputation"><i class="fa fa-check"></i><b>8.4</b> SVD imputation</a></li>
<li class="chapter" data-level="8.5" data-path="Imputation.html"><a href="Imputation.html#mean-imputation"><i class="fa fa-check"></i><b>8.5</b> Mean imputation</a></li>
<li class="chapter" data-level="8.6" data-path="Imputation.html"><a href="Imputation.html#random-forest-imputation"><i class="fa fa-check"></i><b>8.6</b> Random Forest Imputation</a></li>
<li class="chapter" data-level="8.7" data-path="Imputation.html"><a href="Imputation.html#summary-1"><i class="fa fa-check"></i><b>8.7</b> Summary</a></li>
<li class="chapter" data-level="8.8" data-path="Imputation.html"><a href="Imputation.html#conclusions-and-comments"><i class="fa fa-check"></i><b>8.8</b> Conclusions and comments</a></li>
</ul></li>
<li class="chapter" data-level="9" data-path="Association-mapping.html"><a href="Association-mapping.html"><i class="fa fa-check"></i><b>9</b> Association Mapping.</a>
<ul>
<li class="chapter" data-level="9.1" data-path="Association-mapping.html"><a href="Association-mapping.html#introduction-1"><i class="fa fa-check"></i><b>9.1</b> Introduction</a></li>
<li class="chapter" data-level="9.2" data-path="Association-mapping.html"><a href="Association-mapping.html#preparing-the-data-set"><i class="fa fa-check"></i><b>9.2</b> Preparing the data set</a></li>
<li class="chapter" data-level="9.3" data-path="Association-mapping.html"><a href="Association-mapping.html#running-gwaspoly"><i class="fa fa-check"></i><b>9.3</b> Running GWASpoly</a></li>
<li class="chapter" data-level="9.4" data-path="Association-mapping.html"><a href="Association-mapping.html#the-exercise-2"><i class="fa fa-check"></i><b>9.4</b> The exercise</a>
<ul>
<li class="chapter" data-level="9.4.1" data-path="Association-mapping.html"><a href="Association-mapping.html#calculation-and-comparison-of-kinship"><i class="fa fa-check"></i><b>9.4.1</b> Calculation and comparison of kinship</a></li>
<li class="chapter" data-level="9.4.2" data-path="Association-mapping.html"><a href="Association-mapping.html#gwas"><i class="fa fa-check"></i><b>9.4.2</b> GWAS</a></li>
<li class="chapter" data-level="9.4.3" data-path="Association-mapping.html"><a href="Association-mapping.html#analysis-with-covariates"><i class="fa fa-check"></i><b>9.4.3</b> Analysis with covariates</a></li>
<li class="chapter" data-level="9.4.4" data-path="Association-mapping.html"><a href="Association-mapping.html#accounting-for-population-structure"><i class="fa fa-check"></i><b>9.4.4</b> Accounting for population structure</a></li>
<li class="chapter" data-level="9.4.5" data-path="Association-mapping.html"><a href="Association-mapping.html#further-analysis"><i class="fa fa-check"></i><b>9.4.5</b> Further analysis</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="10" data-path="Mapping-Markers.html"><a href="Mapping-Markers.html"><i class="fa fa-check"></i><b>10</b> Mapping Markers and Quantitative Trait Loci.</a>
<ul>
<li class="chapter" data-level="10.1" data-path="Mapping-Markers.html"><a href="Mapping-Markers.html#introduction-2"><i class="fa fa-check"></i><b>10.1</b> Introduction</a></li>
<li class="chapter" data-level="10.2" data-path="Mapping-Markers.html"><a href="Mapping-Markers.html#constructing-a-genetic-map"><i class="fa fa-check"></i><b>10.2</b> Constructing a genetic map</a>
<ul>
<li class="chapter" data-level="10.2.1" data-path="Mapping-Markers.html"><a href="Mapping-Markers.html#formatting-and-reading-in-the-data"><i class="fa fa-check"></i><b>10.2.1</b> Formatting and reading in the data</a></li>
<li class="chapter" data-level="10.2.2" data-path="Mapping-Markers.html"><a href="Mapping-Markers.html#a-quick-map"><i class="fa fa-check"></i><b>10.2.2</b> A quick map!</a></li>
<li class="chapter" data-level="10.2.3" data-path="Mapping-Markers.html"><a href="Mapping-Markers.html#estimating-recombination-fractions"><i class="fa fa-check"></i><b>10.2.3</b> Estimating recombination fractions</a></li>
<li class="chapter" data-level="10.2.4" data-path="Mapping-Markers.html"><a href="Mapping-Markers.html#forming-linkage-groups"><i class="fa fa-check"></i><b>10.2.4</b> Forming Linkage groups</a></li>
<li class="chapter" data-level="10.2.5" data-path="Mapping-Markers.html"><a href="Mapping-Markers.html#marker-ordering"><i class="fa fa-check"></i><b>10.2.5</b> Marker ordering</a></li>
<li class="chapter" data-level="10.2.6" data-path="Mapping-Markers.html"><a href="Mapping-Markers.html#improving-the-map"><i class="fa fa-check"></i><b>10.2.6</b> Improving the map</a></li>
<li class="chapter" data-level="10.2.7" data-path="Mapping-Markers.html"><a href="Mapping-Markers.html#quality-control-commands-for-marker-qc-and-mapping"><i class="fa fa-check"></i><b>10.2.7</b> Quality Control commands for marker QC and mapping</a></li>
<li class="chapter" data-level="10.2.8" data-path="Mapping-Markers.html"><a href="Mapping-Markers.html#asmap"><i class="fa fa-check"></i><b>10.2.8</b> ASMap</a></li>
</ul></li>
<li class="chapter" data-level="10.3" data-path="Mapping-Markers.html"><a href="Mapping-Markers.html#qtl-mapping-in-rqtl"><i class="fa fa-check"></i><b>10.3</b> QTL mapping in R/qtl</a>
<ul>
<li class="chapter" data-level="10.3.1" data-path="Mapping-Markers.html"><a href="Mapping-Markers.html#genome-scan-single-qtl"><i class="fa fa-check"></i><b>10.3.1</b> Genome Scan – single QTL</a></li>
<li class="chapter" data-level="10.3.2" data-path="Mapping-Markers.html"><a href="Mapping-Markers.html#genome-scan-covariates"><i class="fa fa-check"></i><b>10.3.2</b> Genome Scan – covariates</a></li>
<li class="chapter" data-level="10.3.3" data-path="Mapping-Markers.html"><a href="Mapping-Markers.html#genome-scan-scantwo"><i class="fa fa-check"></i><b>10.3.3</b> Genome Scan – scantwo</a></li>
<li class="chapter" data-level="10.3.4" data-path="Mapping-Markers.html"><a href="Mapping-Markers.html#genome-scan-composite-interval-mapping"><i class="fa fa-check"></i><b>10.3.4</b> Genome Scan – Composite Interval Mapping</a></li>
<li class="chapter" data-level="10.3.5" data-path="Mapping-Markers.html"><a href="Mapping-Markers.html#mqm-multiple-qtl-mapping"><i class="fa fa-check"></i><b>10.3.5</b> MQM: multiple QTL mapping</a></li>
<li class="chapter" data-level="10.3.6" data-path="Mapping-Markers.html"><a href="Mapping-Markers.html#some-commands-for-mapping-in-markers-in-rqtl"><i class="fa fa-check"></i><b>10.3.6</b> Some commands for mapping in markers in R/qtl</a></li>
<li class="chapter" data-level="10.3.7" data-path="Mapping-Markers.html"><a href="Mapping-Markers.html#final-comments"><i class="fa fa-check"></i><b>10.3.7</b> Final comments</a></li>
</ul></li>
<li class="chapter" data-level="10.4" data-path="Mapping-Markers.html"><a href="Mapping-Markers.html#rqtl2"><i class="fa fa-check"></i><b>10.4</b> R/QTL2</a></li>
</ul></li>
<li class="chapter" data-level="11" data-path="Data-Visualization.html"><a href="Data-Visualization.html"><i class="fa fa-check"></i><b>11</b> Data Visualization.</a>
<ul>
<li class="chapter" data-level="11.1" data-path="Data-Visualization.html"><a href="Data-Visualization.html#flapjack"><i class="fa fa-check"></i><b>11.1</b> Flapjack</a>
<ul>
<li class="chapter" data-level="11.1.1" data-path="Data-Visualization.html"><a href="Data-Visualization.html#genotype-data-.geno-file."><i class="fa fa-check"></i><b>11.1.1</b> Genotype data: “.geno” file.</a></li>
<li class="chapter" data-level="11.1.2" data-path="Data-Visualization.html"><a href="Data-Visualization.html#map-data-.map-file."><i class="fa fa-check"></i><b>11.1.2</b> Map data : “.MAP” file.</a></li>
<li class="chapter" data-level="11.1.3" data-path="Data-Visualization.html"><a href="Data-Visualization.html#trait-data"><i class="fa fa-check"></i><b>11.1.3</b> Trait data</a></li>
<li class="chapter" data-level="11.1.4" data-path="Data-Visualization.html"><a href="Data-Visualization.html#the-exercise-3"><i class="fa fa-check"></i><b>11.1.4</b> The Exercise</a></li>
</ul></li>
<li class="chapter" data-level="11.2" data-path="Data-Visualization.html"><a href="Data-Visualization.html#helium-pedigree-visualization-software"><i class="fa fa-check"></i><b>11.2</b> Helium pedigree visualization software</a>
<ul>
<li class="chapter" data-level="11.2.1" data-path="Data-Visualization.html"><a href="Data-Visualization.html#introduction-3"><i class="fa fa-check"></i><b>11.2.1</b> Introduction</a></li>
<li class="chapter" data-level="11.2.2" data-path="Data-Visualization.html"><a href="Data-Visualization.html#the-exercise-4"><i class="fa fa-check"></i><b>11.2.2</b> The exercise</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="12" data-path="selection-theory.html"><a href="selection-theory.html"><i class="fa fa-check"></i><b>12</b> Selection theory and breeding program design.</a>
<ul>
<li class="chapter" data-level="12.1" data-path="selection-theory.html"><a href="selection-theory.html#selecting-in-backcrosses"><i class="fa fa-check"></i><b>12.1</b> Selecting in backcrosses</a></li>
<li class="chapter" data-level="12.2" data-path="selection-theory.html"><a href="selection-theory.html#size-of-breeding-programme-cycle-time-and-recurrent-selection-schemes."><i class="fa fa-check"></i><b>12.2</b> Size of breeding programme, cycle time and recurrent selection schemes.</a>
<ul>
<li class="chapter" data-level="12.2.1" data-path="selection-theory.html"><a href="selection-theory.html#size-of-breeding-programme"><i class="fa fa-check"></i><b>12.2.1</b> Size of breeding programme</a></li>
<li class="chapter" data-level="12.2.2" data-path="selection-theory.html"><a href="selection-theory.html#speed-of-selection"><i class="fa fa-check"></i><b>12.2.2</b> Speed of selection</a></li>
<li class="chapter" data-level="12.2.3" data-path="selection-theory.html"><a href="selection-theory.html#heritability"><i class="fa fa-check"></i><b>12.2.3</b> Heritability</a></li>
<li class="chapter" data-level="12.2.4" data-path="selection-theory.html"><a href="selection-theory.html#family-structure-if-there-is-time"><i class="fa fa-check"></i><b>12.2.4</b> Family structure (if there is time)</a></li>
<li class="chapter" data-level="12.2.5" data-path="selection-theory.html"><a href="selection-theory.html#sequential-selection"><i class="fa fa-check"></i><b>12.2.5</b> Sequential Selection</a></li>
<li class="chapter" data-level="12.2.6" data-path="selection-theory.html"><a href="selection-theory.html#optimal-contribution-theory"><i class="fa fa-check"></i><b>12.2.6</b> Optimal Contribution Theory</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="13" data-path="Genomic-prediction.html"><a href="Genomic-prediction.html"><i class="fa fa-check"></i><b>13</b> Genomic prediction.</a>
<ul>
<li class="chapter" data-level="13.1" data-path="Genomic-prediction.html"><a href="Genomic-prediction.html#ridge-regression-and-genomic-prediction-in-excel"><i class="fa fa-check"></i><b>13.1</b> Ridge regression and Genomic Prediction in Excel</a>
<ul>
<li class="chapter" data-level="13.1.1" data-path="Genomic-prediction.html"><a href="Genomic-prediction.html#introduction-4"><i class="fa fa-check"></i><b>13.1.1</b> Introduction</a></li>
<li class="chapter" data-level="13.1.2" data-path="Genomic-prediction.html"><a href="Genomic-prediction.html#the-exercise-5"><i class="fa fa-check"></i><b>13.1.2</b> The Exercise</a></li>
</ul></li>
<li class="chapter" data-level="13.2" data-path="Genomic-prediction.html"><a href="Genomic-prediction.html#genomic-prediction-with-rrblup"><i class="fa fa-check"></i><b>13.2</b> Genomic Prediction with rrBLUP</a>
<ul>
<li class="chapter" data-level="13.2.1" data-path="Genomic-prediction.html"><a href="Genomic-prediction.html#introduction-5"><i class="fa fa-check"></i><b>13.2.1</b> Introduction</a></li>
<li class="chapter" data-level="13.2.2" data-path="Genomic-prediction.html"><a href="Genomic-prediction.html#the-dataset"><i class="fa fa-check"></i><b>13.2.2</b> The dataset</a></li>
<li class="chapter" data-level="13.2.3" data-path="Genomic-prediction.html"><a href="Genomic-prediction.html#loading-and-preparing-the-data"><i class="fa fa-check"></i><b>13.2.3</b> Loading and preparing the data</a></li>
<li class="chapter" data-level="13.2.4" data-path="Genomic-prediction.html"><a href="Genomic-prediction.html#examining-the-data-structure"><i class="fa fa-check"></i><b>13.2.4</b> Examining the data structure</a></li>
<li class="chapter" data-level="13.2.5" data-path="Genomic-prediction.html"><a href="Genomic-prediction.html#ordinary-least-squares"><i class="fa fa-check"></i><b>13.2.5</b> Ordinary least squares</a></li>
<li class="chapter" data-level="13.2.6" data-path="Genomic-prediction.html"><a href="Genomic-prediction.html#rrblup"><i class="fa fa-check"></i><b>13.2.6</b> rrBLUP</a></li>
<li class="chapter" data-level="13.2.7" data-path="Genomic-prediction.html"><a href="Genomic-prediction.html#prediction-forward-in-time"><i class="fa fa-check"></i><b>13.2.7</b> Prediction forward in time</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="14" data-path="Breeding-simulations.html"><a href="Breeding-simulations.html"><i class="fa fa-check"></i><b>14</b> Breeding simulations.</a></li>
<li class="chapter" data-level="" data-path="references.html"><a href="references.html"><i class="fa fa-check"></i>References</a></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Quantitative Methods in Plant Breeding: The Digital Tutorials.</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="Population-genetics" class="section level1" number="7">
<h1><span class="header-section-number">Chapter 7</span> Population genetics.</h1>
<div id="hardy-weinberg-equilibrium-and-population-structure" class="section level2" number="7.1">
<h2><span class="header-section-number">7.1</span> Hardy-Weinberg equilibrium and population structure</h2>
<p><em>Hardy-Weinberg equilibrium frequencies and population subdivision (Fst)</em></p>
<p>In the absence of the action of evolutionary forces (selection, mutation, migration, non-random mating, genetic drift), genotypic frequencies in a population follow directly from the individual allele frequencies in the population. This is called “Hardy-Weinberg equilibrium.” However, in most populations and in particular in breeding populations, the conditions under which HW hold are not very realistic. Nevertheless, when averaged over multiple markers, the HW test provides a simple means of assessing inbreeding, selfing, or population mixing in genetic samples. In particular, it is useful for detecting population sub-structure within a population.</p>
<p>To illustrate this, we will use a subset of a Teosinte (<em>Zea mays</em> ssp. <em>parviglumis</em>) data set published by <span class="citation"><a href="references.html#ref-weber_major_2007" role="doc-biblioref">Weber et al.</a> (<a href="references.html#ref-weber_major_2007" role="doc-biblioref">2007</a>)</span>. Teosinte is the wild, outcrossing ancestral species from which maize was domesticated. Details of the entire data set can be found in the paper. Here we will use data of 592 SNP markers from 544 genotypes. The genotypes can be classified in four groups based on the geographical origin (States in Mexico).</p>
<div class="figure">
<img src="images/pop_gen1.PNG" alt="" />
<p class="caption">Teosinte distribution over the states of Mexico</p>
</div>
<p>The table below shows a SNP genotype frequency for a marker (for all individuals in the data set).</p>
<table>
<caption><span id="tab:unnamed-chunk-264">Table 7.1: </span>SNP genotype frequency for a markers.</caption>
<thead>
<tr class="header">
<th align="left">PZA00219_6</th>
<th align="right">N.individuals</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">A/A</td>
<td align="right">163</td>
</tr>
<tr class="even">
<td align="left">A/G</td>
<td align="right">158</td>
</tr>
<tr class="odd">
<td align="left">G/G</td>
<td align="right">213</td>
</tr>
</tbody>
</table>
<p>Use the information in the table to answer the following questions:</p>

<div class="rmdquiz">
<p><strong>1</strong></p>
<p>a: Based on the information in the table, give the allele frequencies, the observed (Ho) and expected (He) heterozygocity, and the fixation index (F) for the marker.</p>
<p>b: What do the results in (a) suggest in terms of population structure? Explain.</p>
c: Test whether the population is in HW equilibrium based on this marker (to do this you will need to do a Chi-square test by hand).
</div>
<p>The table below gives the same information for SNP PZA00219_6 but separated by group (defined by the Mexican state in which they were collected).</p>
<table>
<caption><span id="tab:unnamed-chunk-266">Table 7.2: </span>SNP genotype frequencies by Mexican state for a single marker</caption>
<thead>
<tr class="header">
<th align="left">PZA00219_6</th>
<th align="right">Guerrero</th>
<th align="right">Jalisco</th>
<th align="right">Mexico</th>
<th align="right">Michoacan</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">A/A</td>
<td align="right">30</td>
<td align="right">79</td>
<td align="right">18</td>
<td align="right">36</td>
</tr>
<tr class="even">
<td align="left">A/G</td>
<td align="right">61</td>
<td align="right">5</td>
<td align="right">29</td>
<td align="right">63</td>
</tr>
<tr class="odd">
<td align="left">G/G</td>
<td align="right">134</td>
<td align="right">9</td>
<td align="right">31</td>
<td align="right">39</td>
</tr>
</tbody>
</table>
<p>Use this table to address the following questions:</p>

<div class="rmdquiz">
<p><strong>2</strong></p>
<p>a: Give the Ho, He, and F for the states Guerrero and Michoacan for marker PZA00219_6.</p>
<p>b: Test whether the groups are in HW equilibrium for this marker.</p>
c: Compare the results with those in the quiz block above. How do you explain the differences?
</div>
<p>Your conclusions so far above are based on only one/two loci. We will now look across all loci using a set of three related R packages: <code>adegenet</code> <span class="citation">(<a href="references.html#ref-adegenet2008" role="doc-biblioref">Jombart 2008</a>)</span> which is a general purpose genetic data analysis package, and the population genetic packages <code>pegas</code> <span class="citation">(<a href="references.html#ref-R-pegas" role="doc-biblioref">Paradis et al. 2020</a>)</span> and <code>hierfstat</code> <span class="citation">(<a href="references.html#ref-R-hierfstat" role="doc-biblioref">Goudet and Jombart 2020</a>)</span> which use the <code>adegenet</code> data format.</p>
<p>For the next part of the tutorial, there are a large number of R commands so we will move to working from a pre-made script. You will need to install the packages <code>adegenet</code>, <code>pegas</code> and <code>hierfstat</code> to you own R environment.</p>

<div class="rmdnote">
Throughout this tutorial we start to use longer and more complicated scripts. You can copy and paste these scripts directly from the tutorial book (as we’ve been doing previously). Alternativly, if you are becoming tired of copying and pasting, we’ve added the full scripts the Chapter 7 data folder on the course website. See: “teosinte_HardyWeinberg_script.R,” “teosinte_pca_script.R,” “AG_script.R” and “LD_Script.R.”
</div>
<p>Set your home directory to a folder containing today’s data files. For the “genind” data format used in <code>adegenet</code>, it is easier to read in the genotype data and population classification of individuals from separate input files. Take a look at the datafiles – “teosinte_genos_R.txt” and “teosinte_pops_R.txt” – to see the format required. Then read in the data files “teosinte_genos_R.txt” and “teosinte_pops_R.txt,” these data files are found on the course website.</p>
<div class="sourceCode" id="cb383"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb383-1"><a href="Population-genetics.html#cb383-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Load required libraries. adegenet and its associated packages are good for population genetic analysis in R</span></span>
<span id="cb383-2"><a href="Population-genetics.html#cb383-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(adegenet)</span>
<span id="cb383-3"><a href="Population-genetics.html#cb383-3" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(pegas)</span>
<span id="cb383-4"><a href="Population-genetics.html#cb383-4" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(hierfstat)</span>
<span id="cb383-5"><a href="Population-genetics.html#cb383-5" aria-hidden="true" tabindex="-1"></a><span class="co"># read in the data for genotypes and population codes separately.</span></span>
<span id="cb383-6"><a href="Population-genetics.html#cb383-6" aria-hidden="true" tabindex="-1"></a><span class="co">#you will need to delete &quot;data/&quot; if you copy this script</span></span>
<span id="cb383-7"><a href="Population-genetics.html#cb383-7" aria-hidden="true" tabindex="-1"></a>genos<span class="ot">&lt;-</span><span class="fu">read.table</span>(<span class="st">&quot;data/teosinte_genos_R.txt&quot;</span>, <span class="at">header=</span>T)</span>
<span id="cb383-8"><a href="Population-genetics.html#cb383-8" aria-hidden="true" tabindex="-1"></a>pops<span class="ot">&lt;-</span><span class="fu">read.table</span>(<span class="st">&quot;data/teosinte_pops_R.txt&quot;</span>, <span class="at">header=</span>T)</span></code></pre></div>
<p>When loaded, convert to the “genind” format, add in the population information, and check that you have inputted the data correctly:</p>
<div class="sourceCode" id="cb384"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb384-1"><a href="Population-genetics.html#cb384-1" aria-hidden="true" tabindex="-1"></a><span class="co"># convert to genind data format</span></span>
<span id="cb384-2"><a href="Population-genetics.html#cb384-2" aria-hidden="true" tabindex="-1"></a>teosinte<span class="ot">&lt;-</span> <span class="fu">df2genind</span>(genos,<span class="at">ploidy=</span><span class="dv">2</span>,<span class="at">sep=</span><span class="st">&quot;/&quot;</span>)</span>
<span id="cb384-3"><a href="Population-genetics.html#cb384-3" aria-hidden="true" tabindex="-1"></a><span class="co"># add in the population information</span></span>
<span id="cb384-4"><a href="Population-genetics.html#cb384-4" aria-hidden="true" tabindex="-1"></a>teosinte<span class="sc">@</span>pop<span class="ot">&lt;-</span><span class="fu">as.factor</span>(pops<span class="sc">$</span>pop)</span>
<span id="cb384-5"><a href="Population-genetics.html#cb384-5" aria-hidden="true" tabindex="-1"></a><span class="co">#check that the data is as expected</span></span>
<span id="cb384-6"><a href="Population-genetics.html#cb384-6" aria-hidden="true" tabindex="-1"></a>teosinte</span></code></pre></div>
<pre><code>## /// GENIND OBJECT /////////
## 
##  // 544 individuals; 592 loci; 1,184 alleles; size: 2.8 Mb
## 
##  // Basic content
##    @tab:  544 x 1184 matrix of allele counts
##    @loc.n.all: number of alleles per locus (range: 2-2)
##    @loc.fac: locus factor for the 1184 columns of @tab
##    @all.names: list of allele names for each locus
##    @ploidy: ploidy of each individual  (range: 2-2)
##    @type:  codom
##    @call: df2genind(X = genos, sep = &quot;/&quot;, ploidy = 2)
## 
##  // Optional content
##    @pop: population of each individual (group size range: 80-227)</code></pre>
<p>First we make a summary of the data, which includes the observed and expected heterozygosities for every marker, assuming that all individuals belong to a single population.</p>
<div class="sourceCode" id="cb386"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb386-1"><a href="Population-genetics.html#cb386-1" aria-hidden="true" tabindex="-1"></a><span class="co">#make a population genetic summary of the data set considering it as a single population</span></span>
<span id="cb386-2"><a href="Population-genetics.html#cb386-2" aria-hidden="true" tabindex="-1"></a>Teo_Summary<span class="ot">&lt;-</span><span class="fu">summary</span>(teosinte)</span></code></pre></div>
<p>Inspect <code>Teo_Summary</code> on your own computer.</p>
<p>We can calculate the “fixation index” as you did earlier in the exercise, and also plot Ho-He for a subset of markers to see if there are any patterns. What do you observe?</p>
<div class="sourceCode" id="cb387"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb387-1"><a href="Population-genetics.html#cb387-1" aria-hidden="true" tabindex="-1"></a><span class="co"># plot Ho-He for the first 40 markers</span></span>
<span id="cb387-2"><a href="Population-genetics.html#cb387-2" aria-hidden="true" tabindex="-1"></a><span class="fu">barplot</span>(Teo_Summary<span class="sc">$</span>Hexp[<span class="dv">1</span><span class="sc">:</span><span class="dv">40</span>]<span class="sc">-</span>Teo_Summary<span class="sc">$</span>Hobs[<span class="dv">1</span><span class="sc">:</span><span class="dv">40</span>],<span class="at">xaxt=</span><span class="st">&#39;n&#39;</span>,<span class="at">main=</span><span class="st">&quot;He-Ho across all individuals for first 40 loci&quot;</span>, <span class="at">ylab=</span> <span class="st">&quot;He-Ho&quot;</span>, <span class="at">xlab=</span><span class="st">&quot;Markers&quot;</span>)</span></code></pre></div>
<p><img src="bookdown-demo_files/figure-html/unnamed-chunk-272-1.png" width="672" /></p>
<div class="sourceCode" id="cb388"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb388-1"><a href="Population-genetics.html#cb388-1" aria-hidden="true" tabindex="-1"></a><span class="co"># calculate F values</span></span>
<span id="cb388-2"><a href="Population-genetics.html#cb388-2" aria-hidden="true" tabindex="-1"></a>Fvalues<span class="ot">&lt;-</span><span class="dv">1</span><span class="sc">-</span>(Teo_Summary<span class="sc">$</span>Hobs<span class="sc">/</span>Teo_Summary<span class="sc">$</span>Hexp)</span></code></pre></div>
<p>Now we can test agreement with HW expectations at each locus.</p>
<div class="sourceCode" id="cb389"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb389-1"><a href="Population-genetics.html#cb389-1" aria-hidden="true" tabindex="-1"></a><span class="co">#test for Hardy-Weinberg equilibrium at each marker separately</span></span>
<span id="cb389-2"><a href="Population-genetics.html#cb389-2" aria-hidden="true" tabindex="-1"></a>hwt.all<span class="ot">&lt;-</span><span class="fu">hw.test</span>(teosinte, <span class="at">B=</span><span class="dv">0</span>)</span></code></pre></div>
<p>Inspect the first 40 markers on your own computer by running: <code>hwt.all[1:40,1:3]</code></p>
<p>We can count how many of the 592 HW tests we have done are significant at the 5% level. (In R, the <code>sum</code> function will count the number of times a condition is <code>TRUE</code> when used as in the script here).</p>
<div class="sourceCode" id="cb390"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb390-1"><a href="Population-genetics.html#cb390-1" aria-hidden="true" tabindex="-1"></a><span class="co"># count the number of markers that significantly deviate from HW equilibrium with P&lt;0.05</span></span>
<span id="cb390-2"><a href="Population-genetics.html#cb390-2" aria-hidden="true" tabindex="-1"></a><span class="fu">sum</span>(hwt.all[,<span class="dv">3</span>]<span class="sc">&lt;</span><span class="fl">0.05</span>)</span></code></pre></div>
<pre><code>## [1] 559</code></pre>

<div class="rmdquiz">
<strong>3</strong> Are more results significant than we might expect by chance?
</div>
<p>Next, calculate the average Ho, He and F across all markers:</p>
<div class="sourceCode" id="cb392"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb392-1"><a href="Population-genetics.html#cb392-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Calculate average (se) of Ho, He, F across all markers</span></span>
<span id="cb392-2"><a href="Population-genetics.html#cb392-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb392-3"><a href="Population-genetics.html#cb392-3" aria-hidden="true" tabindex="-1"></a>se <span class="ot">&lt;-</span> <span class="cf">function</span>(x) <span class="fu">sqrt</span>(<span class="fu">var</span>(x)<span class="sc">/</span><span class="fu">length</span>(x))</span>
<span id="cb392-4"><a href="Population-genetics.html#cb392-4" aria-hidden="true" tabindex="-1"></a>Ho_ave<span class="ot">&lt;-</span><span class="fu">round</span>(<span class="fu">mean</span>(Teo_Summary<span class="sc">$</span>Hobs),<span class="dv">4</span>)</span>
<span id="cb392-5"><a href="Population-genetics.html#cb392-5" aria-hidden="true" tabindex="-1"></a>Ho_se<span class="ot">&lt;-</span><span class="fu">round</span>(<span class="fu">se</span>(Teo_Summary<span class="sc">$</span>Hobs),<span class="dv">4</span>)</span>
<span id="cb392-6"><a href="Population-genetics.html#cb392-6" aria-hidden="true" tabindex="-1"></a>He_ave<span class="ot">&lt;-</span><span class="fu">round</span>(<span class="fu">mean</span>(Teo_Summary<span class="sc">$</span>Hexp),<span class="dv">4</span>)</span>
<span id="cb392-7"><a href="Population-genetics.html#cb392-7" aria-hidden="true" tabindex="-1"></a>He_se<span class="ot">&lt;-</span><span class="fu">round</span>(<span class="fu">se</span>(Teo_Summary<span class="sc">$</span>Hexp),<span class="dv">4</span>)</span>
<span id="cb392-8"><a href="Population-genetics.html#cb392-8" aria-hidden="true" tabindex="-1"></a>F_ave<span class="ot">&lt;-</span><span class="fu">round</span>(<span class="fu">mean</span>(Fvalues),<span class="dv">4</span>)</span>
<span id="cb392-9"><a href="Population-genetics.html#cb392-9" aria-hidden="true" tabindex="-1"></a>F_se<span class="ot">&lt;-</span><span class="fu">round</span>(<span class="fu">se</span>(Fvalues),<span class="dv">4</span>)</span>
<span id="cb392-10"><a href="Population-genetics.html#cb392-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb392-11"><a href="Population-genetics.html#cb392-11" aria-hidden="true" tabindex="-1"></a>Average_table<span class="ot">&lt;-</span><span class="fu">data.frame</span>(<span class="cn">NA</span>,<span class="at">nrow=</span><span class="dv">5</span>,<span class="at">ncol=</span><span class="dv">3</span>)</span>
<span id="cb392-12"><a href="Population-genetics.html#cb392-12" aria-hidden="true" tabindex="-1"></a>Average_table[<span class="dv">1</span>,<span class="dv">1</span>]<span class="ot">&lt;-</span><span class="fu">paste</span>(Ho_ave, <span class="st">&quot;(&quot;</span>, Ho_se, <span class="st">&quot;)&quot;</span>) </span>
<span id="cb392-13"><a href="Population-genetics.html#cb392-13" aria-hidden="true" tabindex="-1"></a>Average_table[<span class="dv">1</span>,<span class="dv">2</span>]<span class="ot">&lt;-</span><span class="fu">paste</span>(He_ave, <span class="st">&quot;(&quot;</span>, He_se, <span class="st">&quot;)&quot;</span>)</span>
<span id="cb392-14"><a href="Population-genetics.html#cb392-14" aria-hidden="true" tabindex="-1"></a>Average_table[<span class="dv">1</span>,<span class="dv">3</span>]<span class="ot">&lt;-</span><span class="fu">paste</span>(F_ave, <span class="st">&quot;(&quot;</span>, F_se, <span class="st">&quot;)&quot;</span>)</span>
<span id="cb392-15"><a href="Population-genetics.html#cb392-15" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(Average_table)<span class="ot">&lt;-</span><span class="fu">c</span>(<span class="st">&quot;Ho&quot;</span>, <span class="st">&quot;He&quot;</span>, <span class="st">&quot;F&quot;</span>)</span>
<span id="cb392-16"><a href="Population-genetics.html#cb392-16" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(Average_table)[<span class="dv">1</span>]<span class="ot">&lt;-</span><span class="fu">c</span>(<span class="st">&quot;All&quot;</span>)</span>
<span id="cb392-17"><a href="Population-genetics.html#cb392-17" aria-hidden="true" tabindex="-1"></a>Average_table</span></code></pre></div>
<pre><code>##                   Ho                He                 F
## All 0.2871 ( 0.004 ) 0.3747 ( 0.0043 ) 0.2327 ( 0.0059 )</code></pre>
<p>Then we can test the hypothesis that Ho-He is significant across all markers using a paired t-test, firstly checking if the Ho and He variances are equal. Is the result as expected?</p>
<div class="sourceCode" id="cb394"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb394-1"><a href="Population-genetics.html#cb394-1" aria-hidden="true" tabindex="-1"></a><span class="fu">bartlett.test</span>(<span class="fu">list</span>(Teo_Summary<span class="sc">$</span>Hexp, Teo_Summary<span class="sc">$</span>Hobs))</span></code></pre></div>
<pre><code>## 
##  Bartlett test of homogeneity of variances
## 
## data:  list(Teo_Summary$Hexp, Teo_Summary$Hobs)
## Bartlett&#39;s K-squared = 3.1346, df = 1, p-value = 0.07665</code></pre>
<p>Test the hypothesis that Hobs&gt;Hexp across all markers:</p>
<div class="sourceCode" id="cb396"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb396-1"><a href="Population-genetics.html#cb396-1" aria-hidden="true" tabindex="-1"></a><span class="fu">t.test</span>(Teo_Summary<span class="sc">$</span>Hexp, Teo_Summary<span class="sc">$</span>Hobs, <span class="at">pair=</span>T, <span class="at">var.equal=</span><span class="cn">TRUE</span>, <span class="at">alter=</span><span class="st">&quot;greater&quot;</span>)</span></code></pre></div>
<pre><code>## 
##  Paired t-test
## 
## data:  Teo_Summary$Hexp and Teo_Summary$Hobs
## t = 34.023, df = 591, p-value &lt; 2.2e-16
## alternative hypothesis: true difference in means is greater than 0
## 95 percent confidence interval:
##  0.08336362        Inf
## sample estimates:
## mean of the differences 
##              0.08760561</code></pre>
<p>We will now rerun these analyses for each state separately. The script shows how to do this for Mexico state (MEX).</p>
<div class="sourceCode" id="cb398"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb398-1"><a href="Population-genetics.html#cb398-1" aria-hidden="true" tabindex="-1"></a><span class="co">#separate into &quot;populations&quot; by state:</span></span>
<span id="cb398-2"><a href="Population-genetics.html#cb398-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb398-3"><a href="Population-genetics.html#cb398-3" aria-hidden="true" tabindex="-1"></a>teosinte_bypop<span class="ot">&lt;-</span><span class="fu">seppop</span>(teosinte)</span>
<span id="cb398-4"><a href="Population-genetics.html#cb398-4" aria-hidden="true" tabindex="-1"></a>teo.GUE<span class="ot">&lt;-</span><span class="fu">summary</span>(teosinte_bypop<span class="sc">$</span>GUE)</span>
<span id="cb398-5"><a href="Population-genetics.html#cb398-5" aria-hidden="true" tabindex="-1"></a>teo.JAL<span class="ot">&lt;-</span><span class="fu">summary</span>(teosinte_bypop<span class="sc">$</span>JAL)</span>
<span id="cb398-6"><a href="Population-genetics.html#cb398-6" aria-hidden="true" tabindex="-1"></a>teo.MEX<span class="ot">&lt;-</span><span class="fu">summary</span>(teosinte_bypop<span class="sc">$</span>MEX)</span>
<span id="cb398-7"><a href="Population-genetics.html#cb398-7" aria-hidden="true" tabindex="-1"></a>teo.MIC<span class="ot">&lt;-</span><span class="fu">summary</span>(teosinte_bypop<span class="sc">$</span>MIC)</span>
<span id="cb398-8"><a href="Population-genetics.html#cb398-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb398-9"><a href="Population-genetics.html#cb398-9" aria-hidden="true" tabindex="-1"></a><span class="co"># repeat above analysis for MEX only</span></span>
<span id="cb398-10"><a href="Population-genetics.html#cb398-10" aria-hidden="true" tabindex="-1"></a><span class="fu">barplot</span>(teo.MEX<span class="sc">$</span>Hexp[<span class="dv">1</span><span class="sc">:</span><span class="dv">40</span>]<span class="sc">-</span>teo.MEX<span class="sc">$</span>Hobs[<span class="dv">1</span><span class="sc">:</span><span class="dv">40</span>], <span class="at">xaxt=</span><span class="st">&#39;n&#39;</span>, <span class="at">main=</span><span class="st">&quot;He-Ho in MEX for first 40 loci &quot;</span>, <span class="at">ylab=</span> <span class="st">&quot;He-Ho&quot;</span>, <span class="at">xlab=</span><span class="st">&quot;Markers&quot;</span>)</span>
<span id="cb398-11"><a href="Population-genetics.html#cb398-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb398-12"><a href="Population-genetics.html#cb398-12" aria-hidden="true" tabindex="-1"></a>Fvalues<span class="ot">&lt;-</span><span class="dv">1</span><span class="sc">-</span>(teo.MEX<span class="sc">$</span>Hobs<span class="sc">/</span>teo.MEX<span class="sc">$</span>Hexp)</span>
<span id="cb398-13"><a href="Population-genetics.html#cb398-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb398-14"><a href="Population-genetics.html#cb398-14" aria-hidden="true" tabindex="-1"></a>hwt.MEX<span class="ot">&lt;-</span><span class="fu">hw.test</span>(teosinte_bypop<span class="sc">$</span>MEX, <span class="at">B=</span><span class="dv">0</span>)</span>
<span id="cb398-15"><a href="Population-genetics.html#cb398-15" aria-hidden="true" tabindex="-1"></a>hwt.MEX[<span class="dv">1</span><span class="sc">:</span><span class="dv">40</span>,<span class="dv">1</span><span class="sc">:</span><span class="dv">3</span>]</span>
<span id="cb398-16"><a href="Population-genetics.html#cb398-16" aria-hidden="true" tabindex="-1"></a><span class="fu">sum</span>(hwt.MEX[,<span class="dv">3</span>]<span class="sc">&lt;</span><span class="fl">0.05</span>)</span>
<span id="cb398-17"><a href="Population-genetics.html#cb398-17" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb398-18"><a href="Population-genetics.html#cb398-18" aria-hidden="true" tabindex="-1"></a>Ho_ave.MEX<span class="ot">&lt;-</span><span class="fu">round</span>(<span class="fu">mean</span>(teo.MEX<span class="sc">$</span>Hobs),<span class="dv">4</span>)</span>
<span id="cb398-19"><a href="Population-genetics.html#cb398-19" aria-hidden="true" tabindex="-1"></a>Ho_se.MEX<span class="ot">&lt;-</span><span class="fu">round</span>(<span class="fu">se</span>(teo.MEX<span class="sc">$</span>Hobs),<span class="dv">4</span>)</span>
<span id="cb398-20"><a href="Population-genetics.html#cb398-20" aria-hidden="true" tabindex="-1"></a>He_ave.MEX<span class="ot">&lt;-</span><span class="fu">round</span>(<span class="fu">mean</span>(teo.MEX<span class="sc">$</span>Hexp),<span class="dv">4</span>)</span>
<span id="cb398-21"><a href="Population-genetics.html#cb398-21" aria-hidden="true" tabindex="-1"></a>He_se.MEX<span class="ot">&lt;-</span><span class="fu">round</span>(<span class="fu">se</span>(teo.MEX<span class="sc">$</span>Hexp),<span class="dv">4</span>)</span>
<span id="cb398-22"><a href="Population-genetics.html#cb398-22" aria-hidden="true" tabindex="-1"></a>F_ave.MEX<span class="ot">&lt;-</span><span class="fu">round</span>(<span class="fu">mean</span>(Fvalues),<span class="dv">4</span>)</span>
<span id="cb398-23"><a href="Population-genetics.html#cb398-23" aria-hidden="true" tabindex="-1"></a>F_se.MEX<span class="ot">&lt;-</span><span class="fu">round</span>(<span class="fu">se</span>(Fvalues),<span class="dv">4</span>)</span>
<span id="cb398-24"><a href="Population-genetics.html#cb398-24" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb398-25"><a href="Population-genetics.html#cb398-25" aria-hidden="true" tabindex="-1"></a>Average_table[<span class="dv">2</span>,<span class="dv">1</span>]<span class="ot">&lt;-</span><span class="fu">paste</span>(Ho_ave.MEX, <span class="st">&quot;(&quot;</span>, Ho_se.MEX, <span class="st">&quot;)&quot;</span>) </span>
<span id="cb398-26"><a href="Population-genetics.html#cb398-26" aria-hidden="true" tabindex="-1"></a>Average_table[<span class="dv">2</span>,<span class="dv">2</span>]<span class="ot">&lt;-</span><span class="fu">paste</span>(He_ave.MEX, <span class="st">&quot;(&quot;</span>, He_se.MEX, <span class="st">&quot;)&quot;</span>)</span>
<span id="cb398-27"><a href="Population-genetics.html#cb398-27" aria-hidden="true" tabindex="-1"></a>Average_table[<span class="dv">2</span>,<span class="dv">3</span>]<span class="ot">&lt;-</span><span class="fu">paste</span>(F_ave.MEX, <span class="st">&quot;(&quot;</span>, F_se.MEX, <span class="st">&quot;)&quot;</span>)</span>
<span id="cb398-28"><a href="Population-genetics.html#cb398-28" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(Average_table)[<span class="dv">2</span>]<span class="ot">&lt;-</span><span class="fu">c</span>(<span class="st">&quot;MEX&quot;</span>)</span>
<span id="cb398-29"><a href="Population-genetics.html#cb398-29" aria-hidden="true" tabindex="-1"></a>Average_table</span>
<span id="cb398-30"><a href="Population-genetics.html#cb398-30" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb398-31"><a href="Population-genetics.html#cb398-31" aria-hidden="true" tabindex="-1"></a><span class="fu">bartlett.test</span>(<span class="fu">list</span>(teo.MEX<span class="sc">$</span>Hexp, teo.MEX<span class="sc">$</span>Hobs))</span>
<span id="cb398-32"><a href="Population-genetics.html#cb398-32" aria-hidden="true" tabindex="-1"></a><span class="fu">t.test</span>(teo.MEX<span class="sc">$</span>Hexp, teo.MEX<span class="sc">$</span>Hobs, <span class="at">pair=</span>T, <span class="at">var.equal=</span><span class="cn">TRUE</span>, <span class="at">alter=</span><span class="st">&quot;greater&quot;</span>)</span></code></pre></div>
<p>For other populations, I have compiled the data in the exercise below.</p>
<p>So for doing the same for all states, we get the following values for Ho, He and F.</p>
<table>
<caption><span id="tab:unnamed-chunk-280">Table 7.3: </span>Ho, He and F for Mexican population and states.</caption>
<thead>
<tr class="header">
<th align="left"></th>
<th align="left">Ho</th>
<th align="left">He</th>
<th align="left">F</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">Entire population</td>
<td align="left">0.287 (0.004)</td>
<td align="left">0.375 (0.004)</td>
<td align="left">0.233 (0.006)</td>
</tr>
<tr class="even">
<td align="left">Guerrero</td>
<td align="left">0.298 (0.004)</td>
<td align="left">0.375 (0.004)</td>
<td align="left">0.203 (0.007)</td>
</tr>
<tr class="odd">
<td align="left">Jalisco</td>
<td align="left">0.221 (0.005)</td>
<td align="left">0.326 (0.006)</td>
<td align="left">0.295 (0.009)</td>
</tr>
<tr class="even">
<td align="left">Mexico</td>
<td align="left">0.312 (0.005)</td>
<td align="left">0.362 (0.005)</td>
<td align="left">0.137 (0.008)</td>
</tr>
<tr class="odd">
<td align="left">Michoacan</td>
<td align="left">0.300 (0.005)</td>
<td align="left">0.357 (0.005)</td>
<td align="left">0.157 (0.007)</td>
</tr>
</tbody>
</table>

<div class="rmdquiz">
<strong>4</strong> Inspect the above table and give your conclusions: how do individual states compare to the entire population?
</div>
<p>Finally, we can calculate F statistics, considering each state as a population. This takes a few minutes so we provide the values here. Averaged over all loci , these are shown below this paragraph. <span class="math inline">\(F_{IS}\)</span> is a measure of within population inbreeding, <span class="math inline">\(F_{IT}\)</span> is a measure of total inbreeding and <span class="math inline">\(F_{ST}\)</span> is a measure of population divergence. Note that <span class="math inline">\(1-F_{IT} = (1-F_{IS})(1-F_{ST})\)</span>, as it should be.</p>
<center>
<span class="math inline">\(F_{IS} = 0.205\)</span>
</center>
<center>
<span class="math inline">\(F_{IT} = 0.246\)</span>
</center>
<center>
<span class="math inline">\(F_{ST}= 0.052\)</span>
</center>

<div class="rmdquiz">
<strong>5</strong> What can you conclude from these statistics?
</div>
<p>By now you may have some opinion on the importance of the population structure in this data set. You may also have noticed some differences among the sub-populations. We will further investigate this using principal component analysis (PCA) of the entire molecular marker matrix using R. PCA is a statistical technique which summarizes the variation observed across all markers into a smaller number of underlying component variables. Each principal component accounts for a decreasing amount of variation in the dataset, e.g. PC1 accounts for the highest amount of variability in the original data as possible, and each subsequent PC explains the highest amount of variance possible in the remainder of the data. As a result, PCA finds the number of effective dimensions in the data defined by the significant PCA axes. If two major groups are present one dimension (PC1) might be enough to describe it. If three, then an extra axis might be necessary, and so on and so forth.</p>
<p>The data for this analysis are in a file called “teosinte_imputed_R.txt.” I have <em>imputed</em> missing data in this data set – you will learn more about this later in the course (for the previous analysis, missing data were left as “NA”). We will use some simple R code to conduct a PCA on the data.</p>
<p>Let us load the data and required package <code>ggfortify</code>, you will need to install this if you don’t have it already.</p>
<div class="sourceCode" id="cb399"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb399-1"><a href="Population-genetics.html#cb399-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ggfortify)</span>
<span id="cb399-2"><a href="Population-genetics.html#cb399-2" aria-hidden="true" tabindex="-1"></a><span class="co">#load data (delete &#39;data/&#39; if you copy this!)</span></span>
<span id="cb399-3"><a href="Population-genetics.html#cb399-3" aria-hidden="true" tabindex="-1"></a>teosintepca<span class="ot">&lt;-</span><span class="fu">read.table</span>(<span class="st">&quot;data/teosinte_imputed_R.txt&quot;</span>, <span class="at">header=</span>T)</span></code></pre></div>
<p>First we fit the PCA and observe (and plot) the percentage of variance explained by each of the first 10 PCs.</p>
<div class="sourceCode" id="cb400"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb400-1"><a href="Population-genetics.html#cb400-1" aria-hidden="true" tabindex="-1"></a><span class="co">#define pca input (markers only, not state name)</span></span>
<span id="cb400-2"><a href="Population-genetics.html#cb400-2" aria-hidden="true" tabindex="-1"></a>pca.input<span class="ot">&lt;-</span>teosintepca[<span class="fu">c</span>(<span class="dv">1</span><span class="sc">:</span><span class="dv">592</span>)]</span>
<span id="cb400-3"><a href="Population-genetics.html#cb400-3" aria-hidden="true" tabindex="-1"></a><span class="co">#run PCA</span></span>
<span id="cb400-4"><a href="Population-genetics.html#cb400-4" aria-hidden="true" tabindex="-1"></a>pca.output<span class="ot">&lt;-</span><span class="fu">prcomp</span>(pca.input)</span></code></pre></div>
<p>List the first 10 PCAs on your computer by running:</p>
<div class="sourceCode" id="cb401"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb401-1"><a href="Population-genetics.html#cb401-1" aria-hidden="true" tabindex="-1"></a><span class="fu">summary</span>(pca.output)<span class="sc">$</span>importance[,<span class="dv">1</span><span class="sc">:</span><span class="dv">10</span>]</span></code></pre></div>
<p>We can also plot the proportion of variance explained by first 10 PCAs by running:</p>
<div class="sourceCode" id="cb402"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb402-1"><a href="Population-genetics.html#cb402-1" aria-hidden="true" tabindex="-1"></a><span class="fu">plot</span>(<span class="fu">summary</span>(pca.output)<span class="sc">$</span>importance[<span class="dv">2</span>,<span class="dv">1</span><span class="sc">:</span><span class="dv">10</span>], <span class="at">main=</span><span class="st">&quot;PCA analysis - Teosinte&quot;</span>, <span class="at">ylab=</span><span class="st">&quot;percentage variance explained&quot;</span>, <span class="at">ylim=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="fl">0.06</span>),<span class="at">xlab=</span><span class="st">&quot;PCA&quot;</span>, <span class="at">lty=</span><span class="fu">par</span>(<span class="st">&quot;lty&quot;</span>), <span class="at">lwd=</span><span class="fu">par</span>(<span class="st">&quot;lwd&quot;</span>))</span>
<span id="cb402-2"><a href="Population-genetics.html#cb402-2" aria-hidden="true" tabindex="-1"></a><span class="fu">lines</span>(<span class="fu">summary</span>(pca.output)<span class="sc">$</span>importance[<span class="dv">2</span>,<span class="dv">1</span><span class="sc">:</span><span class="dv">10</span>])</span></code></pre></div>
<p><img src="bookdown-demo_files/figure-html/unnamed-chunk-286-1.png" width="672" /></p>
<p>Inspect these outputs to comment on the following question:</p>

<div class="rmdquiz">
<strong>6</strong> How much of the variation in this data set is explained by the first 10 PCs? What does this suggest in terms of the underlying population structure?
</div>
<p>Now, we will plot the first two PCs against each other, colouring the points by geography (state).</p>
<div class="sourceCode" id="cb403"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb403-1"><a href="Population-genetics.html#cb403-1" aria-hidden="true" tabindex="-1"></a><span class="co">#plot first 2 PCAs</span></span>
<span id="cb403-2"><a href="Population-genetics.html#cb403-2" aria-hidden="true" tabindex="-1"></a><span class="fu">autoplot</span>(pca.output, <span class="at">data=</span>teosintepca, <span class="at">colour=</span><span class="st">&#39;State&#39;</span>)</span></code></pre></div>
<p><img src="bookdown-demo_files/figure-html/unnamed-chunk-288-1.png" width="672" /></p>
<p>Take a look at this plot and address the questions below:</p>

<div class="rmdquiz">
<p><strong>7</strong></p>
<p>a: Describe what you see in the plot. Is there (some) agreement with the classification purely on the geographical origin?</p>
b: Based on the scatter plot you have just made, what seems to be odd about the group “Jalisco?” How does your observation here correspond with your findings in the fixation index for this sate?
</div>
<p>We could have also plotted the alternative PCs (1 and 3 here). Try running this on your laptop.</p>
<div class="sourceCode" id="cb404"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb404-1"><a href="Population-genetics.html#cb404-1" aria-hidden="true" tabindex="-1"></a>pca_extra<span class="ot">=</span>pca.output</span>
<span id="cb404-2"><a href="Population-genetics.html#cb404-2" aria-hidden="true" tabindex="-1"></a>pca_extra<span class="sc">$</span>x<span class="ot">=</span>pca_extra<span class="sc">$</span>x[,<span class="fu">c</span>(<span class="dv">1</span>,<span class="dv">3</span>)]</span>
<span id="cb404-3"><a href="Population-genetics.html#cb404-3" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(pca_extra<span class="sc">$</span>x)<span class="ot">=</span><span class="fu">c</span>(<span class="st">&quot;PC1&quot;</span>,<span class="st">&quot;PC2&quot;</span>)</span>
<span id="cb404-4"><a href="Population-genetics.html#cb404-4" aria-hidden="true" tabindex="-1"></a>pca_extra<span class="sc">$</span>rotation<span class="ot">=</span>pca_extra<span class="sc">$</span>rotation[,<span class="fu">c</span>(<span class="dv">1</span>,<span class="dv">3</span>)]</span>
<span id="cb404-5"><a href="Population-genetics.html#cb404-5" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(pca_extra<span class="sc">$</span>rotation)<span class="ot">=</span><span class="fu">c</span>(<span class="st">&quot;PC1&quot;</span>,<span class="st">&quot;PC2&quot;</span>)</span>
<span id="cb404-6"><a href="Population-genetics.html#cb404-6" aria-hidden="true" tabindex="-1"></a><span class="fu">autoplot</span>(pca_extra, <span class="at">data=</span>teosintepca,<span class="at">colour=</span><span class="st">&#39;State&#39;</span>)<span class="sc">+</span><span class="fu">labs</span>(<span class="at">y=</span><span class="st">&quot;PC3&quot;</span>)</span></code></pre></div>
<p>Given what we have observed with the PCA investigation, it seems that state boundaries may not be the best way of grouping populations. Looking at the output from the PCA, it seems 3 states (Jalisco, Mexico and Michoacan) may have at least 2 groups of distinct individuals in them. A closer look at the data reveals that in all 3 states, these individuals represent distinct natural populations. If we sub-divide the samples of each of these 3 states into 2 geographic groups based on the PCA data, we now have 7 geographic groupings. We can see what difference this makes to our population statistics by recalculating the Ho, He and F values, and the F statistics using the methods from part c above. Feel free to do this (replace the original pop file with the file “teosinte_pops_subdivided_R.txt”) and explore the data. You will obtain the following summary.</p>
<table>
<caption><span id="tab:unnamed-chunk-291">Table 7.4: </span>Ho, He and F for Mexican population and sub-divided states.</caption>
<thead>
<tr class="header">
<th align="left"></th>
<th align="left">Ho</th>
<th align="left">He</th>
<th align="left">F</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">Entire population</td>
<td align="left">0.287 (0.004)</td>
<td align="left">0.375 (0.004)</td>
<td align="left">0.233 (0.006)</td>
</tr>
<tr class="even">
<td align="left">Guerrero</td>
<td align="left">0.298 (0.004)</td>
<td align="left">0.375 (0.004)</td>
<td align="left">0.203 (0.007)</td>
</tr>
<tr class="odd">
<td align="left">Jalisco 1</td>
<td align="left">0.199 (0.007)</td>
<td align="left">0.245 (0.007)</td>
<td align="left">0.161 (0.011)</td>
</tr>
<tr class="even">
<td align="left">Jalisco 2</td>
<td align="left">0.242 (0.006)</td>
<td align="left">0.329 (0.006)</td>
<td align="left">0.244 (0.010)</td>
</tr>
<tr class="odd">
<td align="left">Mexico 1</td>
<td align="left">0.318 (0.005)</td>
<td align="left">0.358 (0.005)</td>
<td align="left">0.110 (0.008)</td>
</tr>
<tr class="even">
<td align="left">Mexico 2</td>
<td align="left">0.249 (0.009)</td>
<td align="left">0.245 (0.008)</td>
<td align="left">-0.025 (0.015)</td>
</tr>
<tr class="odd">
<td align="left">Michoacan 1</td>
<td align="left">0.301 (0.005)</td>
<td align="left">0.355 (0.005)</td>
<td align="left">0.151 (0.007)</td>
</tr>
<tr class="even">
<td align="left">Michoacan 2</td>
<td align="left">0.293 (0.009)</td>
<td align="left">0.306 (0.007)</td>
<td align="left">0.020 (0.018)</td>
</tr>
</tbody>
</table>
<p><span class="math inline">\(F_{IS}\)</span>, <span class="math inline">\(F_{IT}\)</span>, <span class="math inline">\(F_{ST}\)</span> averaged over all loci:</p>
<center>
<span class="math inline">\(F_{IS}\)</span> = 0.185
</center>
<center>
<span class="math inline">\(F_{IT}\)</span> = 0.251
</center>
<center>
<span class="math inline">\(F_{ST}\)</span> = 0.081
</center>

<div class="rmdquiz">
<strong>8</strong> What conclusions can you draw from this exercise?
</div>
<p>The reassignment of populations based on the PCA here seems a bit ad hoc. Data exploration of this kind is very important in population genetics but there are more powerful methods for exploring how many “true” populations there are in a geographic data set, as well as how to assign individuals of unknown provenance to these populations. We will explore these methods more this afternoon.</p>
</div>
<div id="other-ways-of-analysing-population-structure" class="section level2" number="7.2">
<h2><span class="header-section-number">7.2</span> Other ways of analysing population structure</h2>
<div id="distance-trees" class="section level3" number="7.2.1">
<h3><span class="header-section-number">7.2.1</span> Distance Trees</h3>
<p>We have examined population structure in natural populations of Teosinte, an outbreeding species. For contrast, and to see what plant breeders are more likely to encounter, we now investigate population structure in a breeding population of a self-pollinated species (barley). The data to be used is from a UK association mapping study of barley (<a href="http://agoueb.org">AGOUEB</a>, and includes 1500 mapped SNPs for 621 genotypes (varieties). We’ll run our analysis on the barley AGOUEB data set, using SNPs from across the whole genome but thinned so that markers are &gt;2cM away from each other. This prevents results being dominated by patterns of diversity from particular chromosome regions (especially the centromere – lots of markers but very little recombination). Thinning in this manner has become standard practice. It is easily forgotten but is important. In this case, 307 markers remain to estimate relationships between the genotypes. Note that markers were scored as 0, 1 or “NA,” because we are dealing with inbred lines, so there are no heterozygotes. However, for one of the approaches below (bootstrapping) no missing data is allowed. Therefore, I have imputed the missing data. The imputed values should be clear as they have values intermediate between 0 and 1. Data are in “AG_for_R.txt,” already formatted.</p>
<p>We are going to scale the data so that each marker has mean 0 and standard deviation 1. That way they all have equal weight in the analysis.</p>
<div class="sourceCode" id="cb405"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb405-1"><a href="Population-genetics.html#cb405-1" aria-hidden="true" tabindex="-1"></a><span class="co">#load barley data, take out /data if you copy this</span></span>
<span id="cb405-2"><a href="Population-genetics.html#cb405-2" aria-hidden="true" tabindex="-1"></a>AG_data<span class="ot">&lt;-</span><span class="fu">read.table</span>(<span class="st">&quot;data/AG_for_R.txt&quot;</span>, <span class="at">header=</span>T)</span>
<span id="cb405-3"><a href="Population-genetics.html#cb405-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb405-4"><a href="Population-genetics.html#cb405-4" aria-hidden="true" tabindex="-1"></a><span class="co">#scale data so all markers have mean=0, sd=1</span></span>
<span id="cb405-5"><a href="Population-genetics.html#cb405-5" aria-hidden="true" tabindex="-1"></a>AG_scaled<span class="ot">&lt;-</span><span class="fu">as.data.frame</span>(<span class="fu">scale</span>(AG_data))</span></code></pre></div>
<p>Check that this has worked by running the following on your computer:</p>
<div class="sourceCode" id="cb406"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb406-1"><a href="Population-genetics.html#cb406-1" aria-hidden="true" tabindex="-1"></a><span class="co">#check scaling worked</span></span>
<span id="cb406-2"><a href="Population-genetics.html#cb406-2" aria-hidden="true" tabindex="-1"></a><span class="fu">mean</span>(AG_scaled<span class="sc">$</span>GM1,<span class="at">na.rm=</span>T)</span>
<span id="cb406-3"><a href="Population-genetics.html#cb406-3" aria-hidden="true" tabindex="-1"></a><span class="fu">sd</span>(AG_scaled<span class="sc">$</span>GM1,<span class="at">na.rm=</span>T)</span></code></pre></div>
<p>Then calculate a distance matrix (distance between all lines). Here we are using a simple “Euclidean” measure, based on geometric distance; other options are available.</p>
<div class="sourceCode" id="cb407"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb407-1"><a href="Population-genetics.html#cb407-1" aria-hidden="true" tabindex="-1"></a><span class="co">#calculate a distance matrix</span></span>
<span id="cb407-2"><a href="Population-genetics.html#cb407-2" aria-hidden="true" tabindex="-1"></a>dist_mat<span class="ot">&lt;-</span><span class="fu">dist</span>(AG_scaled, <span class="at">method=</span><span class="st">&#39;euclidean&#39;</span>)</span></code></pre></div>
<p>We can then use this distance matrix in two different clustering approaches. Firstly, <em>Neighbor joining</em>. A basic method of agglomerative clustering method where individuals are grouped into progressively higher order clusters. Bootstrapping is a method of obtaining confidence in each node:</p>
<div class="sourceCode" id="cb408"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb408-1"><a href="Population-genetics.html#cb408-1" aria-hidden="true" tabindex="-1"></a><span class="co">#use neighbor joining with bootstrapping</span></span>
<span id="cb408-2"><a href="Population-genetics.html#cb408-2" aria-hidden="true" tabindex="-1"></a><span class="fu">par</span>(<span class="at">cex=</span><span class="fl">0.7</span>) <span class="co">#changes font size so you can see more labels</span></span>
<span id="cb408-3"><a href="Population-genetics.html#cb408-3" aria-hidden="true" tabindex="-1"></a>tree<span class="ot">&lt;-</span><span class="fu">nj</span>(dist_mat)</span>
<span id="cb408-4"><a href="Population-genetics.html#cb408-4" aria-hidden="true" tabindex="-1"></a>boot.tree<span class="ot">&lt;-</span><span class="fu">boot.phylo</span>(tree,AG_scaled,<span class="at">FUN =</span><span class="cf">function</span>(x) <span class="fu">nj</span>(<span class="fu">dist</span>(x)),<span class="at">B=</span><span class="dv">100</span>)</span></code></pre></div>
<pre><code>## 
Running bootstraps:       100 / 100
## Calculating bootstrap values... done.</code></pre>
<p>Now we can plot the tree:</p>
<div class="sourceCode" id="cb410"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb410-1"><a href="Population-genetics.html#cb410-1" aria-hidden="true" tabindex="-1"></a><span class="fu">plot</span>(tree)</span>
<span id="cb410-2"><a href="Population-genetics.html#cb410-2" aria-hidden="true" tabindex="-1"></a><span class="fu">nodelabels</span>(boot.tree,<span class="at">frame=</span><span class="st">&quot;none&quot;</span>)</span></code></pre></div>
<p><img src="bookdown-demo_files/figure-html/unnamed-chunk-297-1.png" width="672" /></p>

<div class="rmdquiz">
<strong>9</strong> Use the zoom function to look more closely at the plot. How many higher-level groups can reliably be distinguished in this network?
</div>
<p>Our second option is Principal Coordinate Analysis. For this dataset, PCoA is probably more informative than a tree. PCoA is related to, but different from Principal Component Analysis; PCoA analysis is based on distances between pairs of individuals.</p>
<p>Install and load the package called <code>labdsv</code> for principal co-ordinate analysis (there are many alternatives available).</p>
<div class="sourceCode" id="cb411"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb411-1"><a href="Population-genetics.html#cb411-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(labdsv)</span></code></pre></div>
<p>Carry out principal co-ordinate analysis:</p>
<div class="sourceCode" id="cb412"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb412-1"><a href="Population-genetics.html#cb412-1" aria-hidden="true" tabindex="-1"></a>AG_pco<span class="ot">&lt;-</span><span class="fu">pco</span>(dist_mat,<span class="at">k=</span><span class="dv">10</span>)</span></code></pre></div>
<p>Plot the eigenvalues:</p>
<div class="sourceCode" id="cb413"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb413-1"><a href="Population-genetics.html#cb413-1" aria-hidden="true" tabindex="-1"></a><span class="fu">plot</span>(AG_pco<span class="sc">$</span>eig)</span></code></pre></div>
<p><img src="bookdown-demo_files/figure-html/unnamed-chunk-301-1.png" width="672" /></p>
<p>Tabulate percentage variation explained by top 10 eigenvalues:</p>
<div class="sourceCode" id="cb414"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb414-1"><a href="Population-genetics.html#cb414-1" aria-hidden="true" tabindex="-1"></a>pcntvar<span class="ot">&lt;-</span>AG_pco<span class="sc">$</span>eig<span class="sc">/</span><span class="fu">sum</span>(AG_pco<span class="sc">$</span>eig)</span>
<span id="cb414-2"><a href="Population-genetics.html#cb414-2" aria-hidden="true" tabindex="-1"></a>pcntvar[<span class="dv">1</span><span class="sc">:</span><span class="dv">10</span>]</span></code></pre></div>
<pre><code>##  [1] 0.18035911 0.05144023 0.02956470 0.02314040 0.02253805 0.01925869
##  [7] 0.01824109 0.01648530 0.01535072 0.01433016</code></pre>
<p>Plot the first two pco axes:</p>
<div class="sourceCode" id="cb416"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb416-1"><a href="Population-genetics.html#cb416-1" aria-hidden="true" tabindex="-1"></a><span class="fu">plot</span>(AG_pco)</span></code></pre></div>
<p><img src="bookdown-demo_files/figure-html/unnamed-chunk-303-1.png" width="672" /></p>
<p>You can also plot different pairs of pco axes using something like <code>plot(AG_pco$points[,1],AG_pco$points[,3])</code>.</p>
<p>Use your inspection of the PCoA results to answer the following questions:</p>

<div class="rmdquiz">
<p><strong>10</strong></p>
<p>a: Examine the Eigenvalues for the first 10 axes in the text output. What does this tell you about the structure in this data set?</p>
<p>b: Examine the graph for Axes 1 vs 2 but also axes 1 vs 3 and 2 vs 3. Compare your findings for the 2 approaches (NJ tree vs PCoA).</p>
c: How many “natural” groupings do you think there are in this data set?
</div>
</div>
</div>
<div id="structure" class="section level2" number="7.3">
<h2><span class="header-section-number">7.3</span> Structure</h2>
<p>This exercise is to show you how to run STRUCTURE. STRUCTURE is easy to run but has its pitfalls. This guide should not be used as a substitute for the STRUCTURE manual, which gives more detailed guidance. First, download STRUCTURE from <a href="https://web.stanford.edu/group/pritchardlab/structure_software/release_versions/v2.3.4/html/structure.html">here</a>.</p>
<div id="what-structure-does" class="section level3" number="7.3.1">
<h3><span class="header-section-number">7.3.1</span> What STRUCTURE does</h3>
<p>Imagine a population of a randomly mating diploid. Suppose we have a genome-wide set of markers, not necessarily mapped, but not closely linked to one another. Studying a sample of individuals from a single population, we would expect each marker to be in Hardy-Weinberg equilibrium. We would also expect every pair of markers to be in linkage equilibrium. Now imagine a second population of the same species. This too should be in equilibrium between and within markers. Suppose the populations have been separated for long enough that their allele frequencies have diverged and that mating within each population is at random. Now if we have a single sample drawn from both populations, but we don’t know the origin of the individuals in the sample, then we’ll find that the population is no longer in HW equilibrium nor in linkage equilibrium. However, by trial and error, we could allocate individuals to populations until we found the division which give HW proportions and no LD within each of the populations. In practice, allocating individuals by trial and error is impossible. STRUCTURE does this allocation for us by use of a model based Bayesian approach. It has proven tremendously successful. The source publication has been cited &gt;2000 times according to Google Scholar. STRUCTURE will cope with more complex scenarios than that described above such as: more than two populations; ploidy levels other than diploid; ancestry of individuals mixed between populations; modest linkage between markers. Latest versions handle dominant markers. Other related programs focus on specific scenarios such as:</p>
<ul>
<li><p><em>STRUCTURAMA</em> (available from same site as STRUCTURE). An alternative to STRUCTURE which includes calculation of the number of populations. Only works for outcrossed diploids.</p></li>
<li><p><em>INSTRUCT</em>. An alternative to STRUCTURE which works for partial selfers – estimates selfing rate simultaneously. It is available as a command line driven stand-alone or as a web based application run remotely from here: <a href="http://cbsuapps.tc.cornell.edu/InStruct.aspx" class="uri">http://cbsuapps.tc.cornell.edu/InStruct.aspx</a>.</p></li>
<li><p>More recently, <em><a href="http://software.genetics.ucla.edu/admixture/">ADMIXTURE</a></em> uses the same statistical model as STRUCTURE but a faster optimization algorithm. Furthermore, <em>ADMIXTURE</em> is now available for R and has recently been successfully used by our colleagues in this format, so is worth exploring.</p></li>
</ul>
</div>
<div id="limitations-of-structure" class="section level3" number="7.3.2">
<h3><span class="header-section-number">7.3.2</span> Limitations of STRUCTURE</h3>
<ol style="list-style-type: decimal">
<li><p>It takes time to run</p></li>
<li><p>Individuals are assumed to be random mating within populations – problems for mixed mating systems.</p></li>
<li><p>It is often not easy to decide on the number of sub-populations.</p></li>
<li><p>It is best at de-convoluting distant splits in population structure rather than more recent co-ancestry</p></li>
<li><p>Just as for PCoA, tree drawing, and most other methods of displaying genetic relationships, it is important to remember that the answer you get will depend on the distribution of markers. Once again, unlinked, or only loosely linked markers are advised for STRUCTURE. This is important. Remember to thin your markers if their distribution is patchy with clumps of markers in high LD. In our analysis, the markers have already been thinned we shall treat them as unlinked. In practice, this is what most people seem to do.</p></li>
</ol>
<p><em>A note on selfing and ploidy</em>. If species are obligate or near obligate selfers, then they should be analysed as haploids in STRUCTURE. Otherwise STRUCTURE will attempt to assign individuals to groups which are in HW equilibrium, and with a population of near-inbred lines, spurious clusters will result. For the few heterozygous markers that there may be, in something like barley for example, the simplest procedure is to switch these markers off. This is throwing out data, so as the degree of outcrossing increases it becomes more and more costly. The programme INSTRUCT should overcome these limitations; if you have a dataset for a crop with a more mixed mating system, then it is worth a go.</p>
</div>
<div id="using-structure" class="section level3" number="7.3.3">
<h3><span class="header-section-number">7.3.3</span> Using Structure</h3>
<p>We will now again use the barley data from the AGOUEB project, thinned to 307 markers. The format of the data can be seen in the file “AG for Struct.txt.” First we will learn how to use structure. Then, when we have some data, we will run through an exercise of how to interpret the data.</p>
<p>First we need to enter the data and learn how to set up a run. Structure guides you through data input process using a Windows style import wizard. To start the data import window: File, New Project…</p>
<p><strong>Step 1</strong></p>
<p><em>Enter the project name and select a directory and the input file</em>. This process can be a fiddle – sometimes you have to click once to select, and sometimes twice. Trial and error should see you through, however.</p>
<p><strong>Step 2</strong></p>
<p><em>Enter information about the data.</em> If you’ve forgotten or don’t know this, clicking on “Show data file format” may help.</p>
<p>Don’t forget to enter the “Missing Data Value.” If you don’t do this, the missing data will be treated as an extra allele. In this data file, the missing is -9 which is also the default.</p>
<p><em>Ploidy:</em> For a collection of inbred lines enter 1 here to treat the data as haploid. Number of loci – look in the dataset</p>
<p><strong>Step 3</strong></p>
<p>This formats data input for any additional columns or rows in the file.</p>
<p><em>Select the row of marker names</em> – our file includes this.</p>
<p><em>We are not using map data</em> – Linkage information can be included in the analysis for loosely linked markers. Map distances between markers can be included at the top of the input file. See the STRUCTURE manual for details. Generally, good results are obtained treating markers as unlinked. In fact for markers which are linked and in high LD (<span class="math inline">\(r^2&gt;0.9\)</span> or even less) it is worth deleting one of the pair (our marker thinning here has effectively done this).</p>
<p><em>Phase information</em>: irrelevant here because we are dealing with a haploid organism! Generally, with outbred polyploid individuals, this box would need ticking if the phase was known. See the manual for further details.</p>
<p><em>Data file stores data for individuals in a single line</em>: This also does not apply to haploids. With diploids, data may be entered on a single line as:</p>
<p>1 1 2 2 3 3 4 4 for four loci</p>
<p>or as:</p>
<pre><code>1 2 3 4 
1 2 3 4 </code></pre>
<p>The second format is the default for STRUCTURE.</p>
<p><strong>Step 4</strong></p>
<p>Individual ID for each individual: we have a unique variety code as the first column of our data, so select this.</p>
<p><em>Putative population of origin for each individual</em>: With a priori knowledge of the population of origin of each individual, this can be entered and included in the analysis. This can work extremely well if, for example, you have a set of lines of known origin, and a set of unknown or admixed individuals. The data from the known lines is effectively used as a training set to classify the unknown lines. Here, leave it blank. If you had a mix of say, spring and winter wheats, you could use this as prior population information.</p>
<p><em>USEPOPINFO selection flag</em>: Used in conjunction with the previous column to identify which individuals to use in the training set and which are to be classified without prior information. See the Structure manual for more details. Leave blank here.</p>
<p><em>Phenotype information</em>: Do not select this.</p>
<p><em>Other extra columns</em>: We don’t have any.</p>
<p>Click on Finish, Check the details are correct, then click Proceed. STRUCTURE will test the format of the data and you will be prompted to correct any errors. If the data input is successful, a spreadsheet-like display of the input data should be returned.</p>
<div class="figure">
<img src="images/pop_gen_structure1.png" alt="" />
<p class="caption">Spreadsheet-like display of the input data</p>
</div>
<p><strong>Step 5. Parameter Sets</strong></p>
<p>Before running STRUCTURE, we need to supply some input parameters. This is deceptively easy. However, the default parameters are not necessarily suitable for all datasets. See the manual for further details. One approach to selecting sensible parameter sets is to find a good publication working on a similar dataset to your own and copy from there.</p>
<p>Again there is an input wizard to guide you through the process. Select: Parameter Set, New…</p>
<p><em>Run Length</em>: STRUCTURE uses Monte Carlo Markov Chain (MCMC) methods. To run successfully, the program iterates many times. There is generally an initial “burn-in” period during which the program settles down, and then a further period in which the program runs and results are generated. The longer these periods are, the more reliable the results. In practice, the numbers selected are determined by the power of your computer and your patience. Generally, select a burn in and a run length of 100,000 at least. In this demonstration, for reasons of speed, select 10,000 for each box. In our experience, long runs are best – up to 106. One can use lower values for initial analysis and then crank up the numbers for the definitive runs.</p>
<p><em>Ancestry Model</em>: Use the Admixture model. For most crop applications, we expect there will have been some crossing between sub-populations. Otherwise stick with the defaults. See manual for more details</p>
<p><em>Allele frequency model</em>: Stick with the defaults – correlated allele frequencies. Again, although our subpopulations differ, for crops, we expect they haven’t been isolated for so long that there is no correlation in allele frequencies between them. (If you were working on indica and japonica rice, you may have another view.) See manual for more details.</p>
<p><em>Advanced</em>: “Compute probability of the data” should be ticked – we need this to work out how many subpopulations we may have. In addition, select “Print Q-hat.” This writes the membership of each individual in each of the inferred populations to a separate file: useful for subsequent analyses.</p>
<p>Finally, click OK and give your parameter set a name.</p>
<p><strong>Step 6 Running the program</strong></p>
<p>We are now finally ready to run the program. To establish how many cryptic populations we have, STRUCTURE is run multiple times, varying the population number. “K” is the number of populations. To check for stability / repeatability of the STRUCTURE run, it is advisable to replicate each run several times. We shall have only two replicates – for reasons of time - and look at the range K=1-3.</p>
<p><em>Select</em>: Project, Start a Job. Select the name of your new project. In a session running STRUCTURE you may create several different parameter sets (e.g. with different run times) and you would be prompted with a list here.</p>
<p>Remember, when running STRUCTURE on your own data, you would typically use a longer burn-in, a longer run-time, test more population numbers, and carry out more repeat runs.</p>
<p>Click Start. Structure will take a few minutes to run.</p>
<p><strong>When the runs have finished</strong>:</p>
<p><em>Select: View, Simulation Summary</em> This presents a table of summary information from each run. The most important column here is the fourth: Ln P(D).</p>
<p>Ln P(D) gives the posterior probability of the population number. Hopefully, you will see that this increases (gets less negative) as population number increases, but that values are reasonably close within replicate runs. Ideally, with more time, one would continue to increase K to find the value at which Ln P(D) was maximised. In practice, this is not always possible: runs are unstable (see below), they take too long, or K can continue to increase to improbable values. Some compromise is required. The manual describes the problems of deciding on an appropriate value of K in more detail. We will discuss this below.</p>
<p>In the left hand side of the screen, select one of your runs. The right hand side screen should now change to display the results for that run. One can select various graphical displays of the results. It is worth exploring and experimenting with these. The “Data plot” options are very useful to check on the stability of the runs. In particular the plot of Log Likelihood against the number of iterations should be seen to stabilise during the burn-in and then fluctuate around a constant value during the run. There should be no trend upwards or downwards during this period (which would indicate a longer burn-in was required). The data plot of Ln P(D) may fluctuate initially, but should settle to a constant value by the end of the run, with no increasing or decreasing trend.</p>
<p>The “Bar plot” shows population membership for each individual in the dataset. This can be sorted by maximum population membership to give a cleaner display (Sort by Q). This plot is often seen in publications using STRUCTURE.</p>
<p>The “Triangle plot” shows group membership for any pair of populations, plus the residual pooled membership for the remaining populations, all in a single graph. Generally, if the dataset has population structure, and STRUCTURE has detected it, these plots will show clusters of individuals in the corners of the triangle (they come from that particular population), with some individuals scattered along the sides and in the body of the triangle (they are admixed between two or more populations). If this pattern is not seen, then you should suspect that there is no population structure, or none has been detected. We know from the values for Ln P(D) that at least three subpopulations are present. However, 10,000 iterations is too few for this dataset, especially once K is increased much beyond 3.</p>
<p>The “Tree plot” shows the relationships between the K populations in a run.</p>
<p>We can also plot the output files in R. View the output files in explorer – xxx_q and xxx_f. They should open in notepad or wordpad. The xxx_q file will also open in excel and read into R. The code below will produce a histogram of the maximum population membership for each variety. (The run below is one of the K=3 runs). Remember to set your directory in R before running the code. These plots are often bimodal and often show that many individuals have strong membership of one or other of the populations, but other individuals have mixed ancestry.</p>
<div class="sourceCode" id="cb418"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb418-1"><a href="Population-genetics.html#cb418-1" aria-hidden="true" tabindex="-1"></a>struct<span class="ot">&lt;-</span><span class="fu">read.table</span>(“100000_run_5_q”)</span>
<span id="cb418-2"><a href="Population-genetics.html#cb418-2" aria-hidden="true" tabindex="-1"></a><span class="fu">attach</span>(struct)</span>
<span id="cb418-3"><a href="Population-genetics.html#cb418-3" aria-hidden="true" tabindex="-1"></a><span class="fu">hist</span>(<span class="fu">pmax</span>(V2,V3,V4))</span></code></pre></div>
</div>
<div id="the-exercise" class="section level3" number="7.3.4">
<h3><span class="header-section-number">7.3.4</span> The exercise</h3>
<p>I have run both the AGOUEB barely data set and the teosinte data set with 100000 run length for K=1-10 with 2 runs for every K. We are going to compare the two data sets to understand the strengths and limitations of analyses using STRUCTURE for analysing inbred crop species vs outbreeding wild species. The likelihoods and alpha values are plotted against K and the bar plots for K=3, K=5 and K=10 are shown below, together with the histograms of the maximum assignment value to a population for all individuals, for K=3, K=5 and K=10. In addition, using the <a href="http://taylor0.biology.ucla.edu/structureHarvester">Structure Harvester website</a>, <span class="citation"><a href="references.html#ref-STRUCTURE2012" role="doc-biblioref">Earl and vonHoldt</a> (<a href="references.html#ref-STRUCTURE2012" role="doc-biblioref">2012</a>)</span>, I have plotted the Delta K distribution of <span class="citation"><a href="references.html#ref-Evanno2005" role="doc-biblioref">Evanno, Regnaut, and Goudet</a> (<a href="references.html#ref-Evanno2005" role="doc-biblioref">2005</a>)</span>. This is a method for determining the most informative value of K.</p>
<div id="teosinte-data-set" class="section level4" number="7.3.4.1">
<h4><span class="header-section-number">7.3.4.1</span> Teosinte data set</h4>
<p><img src="images/pop_gen_structure2.PNG" /></p>
<p><img src="images/pop_gen_structure3.PNG" /></p>
</div>
<div id="teosinte-data-set-alternative-view-with-individuals-ordered-by-natural-populations-within-states" class="section level4" number="7.3.4.2">
<h4><span class="header-section-number">7.3.4.2</span> Teosinte data set alternative view with individuals ordered by natural populations within states</h4>
<p>States:</p>
<ol style="list-style-type: decimal">
<li><p>Guerrero</p></li>
<li><p>Jalisco</p></li>
<li><p>Mexico</p></li>
<li><p>Michoacan</p></li>
</ol>
<p><img src="images/pop_gen_structure4.PNG" /></p>
</div>
<div id="agoueb-barely-data-set" class="section level4" number="7.3.4.3">
<h4><span class="header-section-number">7.3.4.3</span> AGOUEB Barely data set</h4>
<p><img src="images/pop_gen_structure5.PNG" /></p>
<p><img src="images/pop_gen_structure6.PNG" /></p>
</div>
<div id="exercise-questions-teosinte-data" class="section level4" number="7.3.4.4">
<h4><span class="header-section-number">7.3.4.4</span> Exercise questions (teosinte data)</h4>
<p>Looking firstly at the teosinte data:</p>

<div class="rmdquiz">
<p><strong>11</strong></p>
<p>Looking at the scatter plots, what happens to the likelihood, the rate of change of the likelihood and the value of alpha as K increases?</p>
<p>What appears to be happening as we move from K=3 to K=5 and K=10 in the bar plots? Use the histograms to confirm your observations. Note that the bar plots for the teosinte data set are arranged by collection population in 4 groups for the separate states (GUE, JAL, MEX, MIC).</p>
<p>What value of K may be the most appropriate for this data set, taking into account all the information in the figures above? (hint: there is no clear answer – think of how you might try to reach a conclusion)</p>
You have now used several techniques to investigate the population structure of teosinte from field collections. As plant breeders, you may have conducted this exercise in order to select lines for a crossing programme to improve your maize breeding germplasm (say for drought or disease resistance). Say you are selecting 15 lines, which would you select? Explain the reasons for your choice and which data analyses were most useful to reaching your decision. What other (non-genetic) information might you take into account?
</div>
</div>
<div id="exercise-questions-barley-data" class="section level4" number="7.3.4.5">
<h4><span class="header-section-number">7.3.4.5</span> Exercise questions (barley data)</h4>
<p>Now looking at the barley data:</p>

<div class="rmdquiz">
<p><strong>12</strong></p>
<p>Looking at the scatter plots, what happens to the likelihood, the rate of change of the likelihood and the value of alpha as K increases – how does this differ from the teosinte data set</p>
<p>Looking at your own tree plot for K=3, what would you conclude about the relationship between the 3 inferred populations?</p>
<p>What appears to be happening as we move from K=3 to K=5 and K=10 in the bar plots? Use the histograms to confirm your observations. Is this different than the case for teosinte?</p>
<p>What value of K do you think is most appropriate for this data set, taking into account all the information in the figures above? How does this compare to your results from the PCoA and tree drawing exercise?</p>
Do you think “Structure” is a useful tool for analysing inbred crop datasets? If not, what technique should we use to account for structure in association mapping?
</div>
</div>
</div>
</div>
<div id="linkage-disequilibrium-analysis" class="section level2" number="7.4">
<h2><span class="header-section-number">7.4</span> Linkage disequilibrium analysis</h2>
<p>Linkage disequilibrium mapping underlies all marker-trait association studies in crop genetics. Studying the pattern of LD in your population is one of the first things you should do before carrying out association mapping. It will help in deciding the marker density and the population size that you require for a well-designed association mapping study. This will be discussed again later in the course. There are several R packages that will estimate and plot LD but they seem to have quite fast turnover. Here, we will use a package called <code>LDcorSV</code>, which can calculate <span class="math inline">\(R^2\)</span> values accounting for population structure or kinship. We will estimate LD for both the teosinte and barley data sets, with and without accounting for population structure.</p>
<div id="linkage-disequilibrium-in-the-teosinte-and-barley-data-sets" class="section level3" number="7.4.1">
<h3><span class="header-section-number">7.4.1</span> Linkage disequilibrium in the teosinte and barley data sets</h3>
<p>Start RStudio and load packages; you may need to first install some of these packages. <code>ggplot2</code> is a very useful graphics package to get to know in R, and part of the <a href="https://www.tidyverse.org/">Tidyverse</a>: publication-ready figures can be created very straightforwardly. <code>RcolorBrewer</code> is useful for designing your own colour schemes in heat maps. You will need to install the package <code>LDcorSV</code>.</p>
<p>We will investigate LD between loci by plotting LD against genetic distance along Teosinte chromosome 1 and barley chromosome 1 from the data sets we used earlier. For convenience I have used the imputed teosinte dataset but <code>LDcorSV</code> does accept missing data. The map positions are in “Teosintemap.txt” and “AGmap.txt.” In this exercise we will compare LD decay with and without correction for population structure so we will also load two population structure data sets for teosinte: “Teosinte_State.txt” for the Mexican states where the sample were collected, and “Teosinte_structure_k5.txt” which has the population assignments from the Structure analysis with k=5. The columns in “Teosinte_State.txt” and “Teosinte_structure_k5.txt” are the probability of belonging to each population (always 0 or 1 for states). For barley, we will account for population structure using use the results of the PCoA, using the first two principal co-ordinates (3 populations). The number of columns in the population structure data files is 1-the number of populations; the last population can be omitted as the proportion of ancestry across all populations must sum to 1.</p>
<p>The command <code>LD.measures</code> from <code>LDcorSV</code> calculates all pairwise <span class="math inline">\(R^2\)</span> values, the option <code>S=</code> allows you to incorporate population structure. The output is a table with the format: marker 1 name / marker 2 name / <span class="math inline">\(R^2\)</span> value. To plot a LD heatmap, we need to rearrange the results into a matrix, and to plot LD decay against genetic distance, we need to calculate the distance between the marker pairs. The scripts for this data manipulation are provided – we will work through them and explain the details in the class.</p>
<div id="preparing-the-analysis" class="section level4" number="7.4.1.1">
<h4><span class="header-section-number">7.4.1.1</span> Preparing the analysis</h4>
<div class="sourceCode" id="cb419"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb419-1"><a href="Population-genetics.html#cb419-1" aria-hidden="true" tabindex="-1"></a><span class="co">#load packages (you will need to install them first if you don&#39;t have them)</span></span>
<span id="cb419-2"><a href="Population-genetics.html#cb419-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(<span class="st">&quot;LDcorSV&quot;</span>)</span>
<span id="cb419-3"><a href="Population-genetics.html#cb419-3" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(<span class="st">&quot;ggplot2&quot;</span>)</span>
<span id="cb419-4"><a href="Population-genetics.html#cb419-4" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(<span class="st">&quot;RColorBrewer&quot;</span>)</span>
<span id="cb419-5"><a href="Population-genetics.html#cb419-5" aria-hidden="true" tabindex="-1"></a>colnew<span class="ot">&lt;-</span><span class="fu">colorRampPalette</span>(<span class="fu">c</span>(<span class="st">&quot;grey&quot;</span>,<span class="st">&quot;red&quot;</span>))(<span class="at">n=</span><span class="dv">99</span>)</span>
<span id="cb419-6"><a href="Population-genetics.html#cb419-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb419-7"><a href="Population-genetics.html#cb419-7" aria-hidden="true" tabindex="-1"></a><span class="co">#Read in the data (after setting working directory), delete &quot;/data&quot; if you copy this script</span></span>
<span id="cb419-8"><a href="Population-genetics.html#cb419-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb419-9"><a href="Population-genetics.html#cb419-9" aria-hidden="true" tabindex="-1"></a>Teosintemap<span class="ot">&lt;-</span><span class="fu">read.table</span>(<span class="st">&quot;data/Teosintemap.txt&quot;</span>,<span class="at">header=</span>T)</span>
<span id="cb419-10"><a href="Population-genetics.html#cb419-10" aria-hidden="true" tabindex="-1"></a>Teosinte<span class="ot">&lt;-</span><span class="fu">read.table</span>(<span class="st">&quot;data/Teosinte_Chr1.txt&quot;</span>,<span class="at">header=</span>T)</span>
<span id="cb419-11"><a href="Population-genetics.html#cb419-11" aria-hidden="true" tabindex="-1"></a>Teosinte_state<span class="ot">&lt;-</span><span class="fu">read.table</span>(<span class="st">&quot;data/Teosinte_state.txt&quot;</span>,<span class="at">header=</span>T)</span>
<span id="cb419-12"><a href="Population-genetics.html#cb419-12" aria-hidden="true" tabindex="-1"></a>Teosinte_structurek5<span class="ot">&lt;-</span><span class="fu">read.table</span>(<span class="st">&quot;data/Teosinte_structure_k5.txt&quot;</span>,<span class="at">header=</span>T)</span>
<span id="cb419-13"><a href="Population-genetics.html#cb419-13" aria-hidden="true" tabindex="-1"></a>AGmap<span class="ot">&lt;-</span><span class="fu">read.table</span>(<span class="st">&quot;data/AGmap.txt&quot;</span>,<span class="at">header=</span>T)</span>
<span id="cb419-14"><a href="Population-genetics.html#cb419-14" aria-hidden="true" tabindex="-1"></a>AG<span class="ot">&lt;-</span><span class="fu">read.table</span>(<span class="st">&quot;data/AG_Chr1.txt&quot;</span>,<span class="at">header=</span>T)</span>
<span id="cb419-15"><a href="Population-genetics.html#cb419-15" aria-hidden="true" tabindex="-1"></a>AGpcoa<span class="ot">&lt;-</span><span class="fu">read.table</span>(<span class="st">&quot;data/AG_Chr1_pcoa.txt&quot;</span>,<span class="at">header=</span>T)</span></code></pre></div>
</div>
<div id="teosinte-analysis-with-no-population-structure" class="section level4" number="7.4.1.2">
<h4><span class="header-section-number">7.4.1.2</span> Teosinte analysis with no population structure</h4>
<p>Run the script for the teosinte analysis with no population structure. You should obtain a heatmap of all pairwise r2 values and a scatter plot of the LD (values) versus the genetic distance:</p>
<div class="sourceCode" id="cb420"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb420-1"><a href="Population-genetics.html#cb420-1" aria-hidden="true" tabindex="-1"></a><span class="co">#Teosinte: calculate r^2 without taking account of population structure</span></span>
<span id="cb420-2"><a href="Population-genetics.html#cb420-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb420-3"><a href="Population-genetics.html#cb420-3" aria-hidden="true" tabindex="-1"></a>Teosinte_No_structure<span class="ot">&lt;-</span><span class="fu">LD.Measures</span>(Teosinte,<span class="at">na.presence=</span><span class="cn">TRUE</span>)</span>
<span id="cb420-4"><a href="Population-genetics.html#cb420-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb420-5"><a href="Population-genetics.html#cb420-5" aria-hidden="true" tabindex="-1"></a><span class="co">#Extract r^2 and the intermarker distances.</span></span>
<span id="cb420-6"><a href="Population-genetics.html#cb420-6" aria-hidden="true" tabindex="-1"></a>Teosinte_No_structure<span class="sc">$</span>dist<span class="ot">=</span><span class="st">&quot;NA&quot;</span></span>
<span id="cb420-7"><a href="Population-genetics.html#cb420-7" aria-hidden="true" tabindex="-1"></a>nmarkers<span class="ot">=</span><span class="dv">82</span></span>
<span id="cb420-8"><a href="Population-genetics.html#cb420-8" aria-hidden="true" tabindex="-1"></a>k<span class="ot">&lt;-</span><span class="dv">0</span></span>
<span id="cb420-9"><a href="Population-genetics.html#cb420-9" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span>(i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span> (nmarkers<span class="dv">-1</span>))</span>
<span id="cb420-10"><a href="Population-genetics.html#cb420-10" aria-hidden="true" tabindex="-1"></a>{</span>
<span id="cb420-11"><a href="Population-genetics.html#cb420-11" aria-hidden="true" tabindex="-1"></a>  <span class="cf">for</span>(j <span class="cf">in</span> (i<span class="sc">+</span><span class="dv">1</span>)<span class="sc">:</span>nmarkers)</span>
<span id="cb420-12"><a href="Population-genetics.html#cb420-12" aria-hidden="true" tabindex="-1"></a>  {</span>
<span id="cb420-13"><a href="Population-genetics.html#cb420-13" aria-hidden="true" tabindex="-1"></a>    k<span class="ot">&lt;-</span>k<span class="sc">+</span><span class="dv">1</span></span>
<span id="cb420-14"><a href="Population-genetics.html#cb420-14" aria-hidden="true" tabindex="-1"></a>    Teosinte_No_structure[k,<span class="dv">4</span>]<span class="ot">&lt;-</span><span class="fu">abs</span>(Teosintemap[i,<span class="dv">3</span>]<span class="sc">-</span>Teosintemap[j,<span class="dv">3</span>])</span>
<span id="cb420-15"><a href="Population-genetics.html#cb420-15" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb420-16"><a href="Population-genetics.html#cb420-16" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb420-17"><a href="Population-genetics.html#cb420-17" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb420-18"><a href="Population-genetics.html#cb420-18" aria-hidden="true" tabindex="-1"></a>Teosinte_No_struct_heatmap<span class="ot">&lt;-</span><span class="fu">matrix</span>(<span class="cn">NA</span>,<span class="at">nrow=</span>nmarkers<span class="dv">-1</span>,<span class="at">ncol=</span>nmarkers<span class="dv">-1</span>)</span>
<span id="cb420-19"><a href="Population-genetics.html#cb420-19" aria-hidden="true" tabindex="-1"></a>k<span class="ot">&lt;-</span><span class="dv">0</span></span>
<span id="cb420-20"><a href="Population-genetics.html#cb420-20" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span>(i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span> (nmarkers<span class="dv">-1</span>))</span>
<span id="cb420-21"><a href="Population-genetics.html#cb420-21" aria-hidden="true" tabindex="-1"></a>{</span>
<span id="cb420-22"><a href="Population-genetics.html#cb420-22" aria-hidden="true" tabindex="-1"></a>  <span class="cf">for</span>(j <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span>(nmarkers<span class="dv">-1</span>))</span>
<span id="cb420-23"><a href="Population-genetics.html#cb420-23" aria-hidden="true" tabindex="-1"></a>  {</span>
<span id="cb420-24"><a href="Population-genetics.html#cb420-24" aria-hidden="true" tabindex="-1"></a>    k<span class="ot">&lt;-</span>k<span class="sc">+</span><span class="dv">1</span></span>
<span id="cb420-25"><a href="Population-genetics.html#cb420-25" aria-hidden="true" tabindex="-1"></a>    <span class="fu">ifelse</span>(i<span class="sc">&lt;</span>(j<span class="sc">+</span><span class="dv">1</span>),Teosinte_No_struct_heatmap[i,j]<span class="ot">&lt;-</span>Teosinte_No_structure[k,<span class="dv">3</span>], k<span class="ot">&lt;-</span>k<span class="dv">-1</span>)</span>
<span id="cb420-26"><a href="Population-genetics.html#cb420-26" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb420-27"><a href="Population-genetics.html#cb420-27" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb420-28"><a href="Population-genetics.html#cb420-28" aria-hidden="true" tabindex="-1"></a>Teosinte_No_structure<span class="sc">$</span>dist<span class="ot">&lt;-</span><span class="fu">as.numeric</span>(Teosinte_No_structure<span class="sc">$</span>dist)</span>
<span id="cb420-29"><a href="Population-genetics.html#cb420-29" aria-hidden="true" tabindex="-1"></a><span class="fu">image</span>(<span class="fu">t</span>(Teosinte_No_struct_heatmap),<span class="at">col=</span>colnew)</span></code></pre></div>
<p><img src="bookdown-demo_files/figure-html/unnamed-chunk-309-1.png" width="672" /></p>
<div class="sourceCode" id="cb421"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb421-1"><a href="Population-genetics.html#cb421-1" aria-hidden="true" tabindex="-1"></a>Teosinte_No_structure_ordered<span class="ot">&lt;-</span>Teosinte_No_structure[<span class="fu">order</span>(Teosinte_No_structure<span class="sc">$</span>dist),]</span>
<span id="cb421-2"><a href="Population-genetics.html#cb421-2" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(Teosinte_No_structure_ordered, <span class="fu">aes</span>(dist,r2))<span class="sc">+</span></span>
<span id="cb421-3"><a href="Population-genetics.html#cb421-3" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_point</span>()<span class="sc">+</span></span>
<span id="cb421-4"><a href="Population-genetics.html#cb421-4" aria-hidden="true" tabindex="-1"></a>    <span class="fu">ylim</span>(<span class="dv">0</span>,<span class="dv">1</span>)<span class="sc">+</span></span>
<span id="cb421-5"><a href="Population-genetics.html#cb421-5" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_smooth</span>(<span class="at">formula=</span>Teosinte_No_structure_ordered<span class="sc">$</span>r2<span class="sc">~</span> Teosinte_No_structure_ordered<span class="sc">$</span>dist,<span class="at">se=</span>F)<span class="sc">+</span></span>
<span id="cb421-6"><a href="Population-genetics.html#cb421-6" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_hline</span>(<span class="fu">aes</span>(<span class="at">yintercept=</span><span class="fl">0.1</span>),<span class="at">col=</span><span class="st">&quot;red&quot;</span>)</span></code></pre></div>
<p><img src="bookdown-demo_files/figure-html/unnamed-chunk-309-2.png" width="672" /></p>
<p>Inspect the results and answer the following questions:</p>

<div class="rmdquiz">
<p><strong>13</strong></p>
<p>a: What is your preliminary conclusion based on the scatter plot. Does LD decay rapidly or slowly as a function of genetic distance between markers in this species?</p>
b: Now take a look at the LD matrix heatmap. What extra information is provided by this plot? Can you identify pairs of markers that show relatively high LD despite being quite far apart?
</div>
</div>
<div id="teosinte-analysis-with-population-structure-states" class="section level4" number="7.4.1.3">
<h4><span class="header-section-number">7.4.1.3</span> Teosinte analysis with population structure (states)</h4>
<p>Rerun the teosinte analysis selecting state as a subpopulation grouping (the script is provided below):</p>
<div class="sourceCode" id="cb422"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb422-1"><a href="Population-genetics.html#cb422-1" aria-hidden="true" tabindex="-1"></a><span class="co">#Teosinte: calculate r^2 taking account of population structure (states)</span></span>
<span id="cb422-2"><a href="Population-genetics.html#cb422-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb422-3"><a href="Population-genetics.html#cb422-3" aria-hidden="true" tabindex="-1"></a>Teosinte_structure<span class="ot">&lt;-</span><span class="fu">LD.Measures</span>(Teosinte,<span class="at">S=</span>Teosinte_state,<span class="at">na.presence=</span><span class="cn">TRUE</span>)</span>
<span id="cb422-4"><a href="Population-genetics.html#cb422-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb422-5"><a href="Population-genetics.html#cb422-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb422-6"><a href="Population-genetics.html#cb422-6" aria-hidden="true" tabindex="-1"></a><span class="co">#Extract r^2 and the intermarker distances.</span></span>
<span id="cb422-7"><a href="Population-genetics.html#cb422-7" aria-hidden="true" tabindex="-1"></a>Teosinte_structure<span class="sc">$</span>dist<span class="ot">=</span><span class="st">&quot;NA&quot;</span></span>
<span id="cb422-8"><a href="Population-genetics.html#cb422-8" aria-hidden="true" tabindex="-1"></a>nmarkers<span class="ot">=</span><span class="dv">82</span></span>
<span id="cb422-9"><a href="Population-genetics.html#cb422-9" aria-hidden="true" tabindex="-1"></a>k<span class="ot">&lt;-</span><span class="dv">0</span></span>
<span id="cb422-10"><a href="Population-genetics.html#cb422-10" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span>(i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span> (nmarkers<span class="dv">-1</span>))</span>
<span id="cb422-11"><a href="Population-genetics.html#cb422-11" aria-hidden="true" tabindex="-1"></a>{</span>
<span id="cb422-12"><a href="Population-genetics.html#cb422-12" aria-hidden="true" tabindex="-1"></a>  <span class="cf">for</span>(j <span class="cf">in</span> (i<span class="sc">+</span><span class="dv">1</span>)<span class="sc">:</span>nmarkers)</span>
<span id="cb422-13"><a href="Population-genetics.html#cb422-13" aria-hidden="true" tabindex="-1"></a>  {</span>
<span id="cb422-14"><a href="Population-genetics.html#cb422-14" aria-hidden="true" tabindex="-1"></a>    k<span class="ot">&lt;-</span>k<span class="sc">+</span><span class="dv">1</span></span>
<span id="cb422-15"><a href="Population-genetics.html#cb422-15" aria-hidden="true" tabindex="-1"></a>    Teosinte_structure[k,<span class="dv">5</span>]<span class="ot">&lt;-</span><span class="fu">abs</span>(Teosintemap[i,<span class="dv">3</span>]<span class="sc">-</span>Teosintemap[j,<span class="dv">3</span>])</span>
<span id="cb422-16"><a href="Population-genetics.html#cb422-16" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb422-17"><a href="Population-genetics.html#cb422-17" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb422-18"><a href="Population-genetics.html#cb422-18" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb422-19"><a href="Population-genetics.html#cb422-19" aria-hidden="true" tabindex="-1"></a>Teosinte_struct_heatmap<span class="ot">&lt;-</span><span class="fu">matrix</span>(<span class="cn">NA</span>,<span class="at">nrow=</span>nmarkers<span class="dv">-1</span>,<span class="at">ncol=</span>nmarkers<span class="dv">-1</span>)</span>
<span id="cb422-20"><a href="Population-genetics.html#cb422-20" aria-hidden="true" tabindex="-1"></a>k<span class="ot">&lt;-</span><span class="dv">0</span></span>
<span id="cb422-21"><a href="Population-genetics.html#cb422-21" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span>(i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span> (nmarkers<span class="dv">-1</span>))</span>
<span id="cb422-22"><a href="Population-genetics.html#cb422-22" aria-hidden="true" tabindex="-1"></a>{</span>
<span id="cb422-23"><a href="Population-genetics.html#cb422-23" aria-hidden="true" tabindex="-1"></a>  <span class="cf">for</span>(j <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span>(nmarkers<span class="dv">-1</span>))</span>
<span id="cb422-24"><a href="Population-genetics.html#cb422-24" aria-hidden="true" tabindex="-1"></a>  {</span>
<span id="cb422-25"><a href="Population-genetics.html#cb422-25" aria-hidden="true" tabindex="-1"></a>    k<span class="ot">&lt;-</span>k<span class="sc">+</span><span class="dv">1</span></span>
<span id="cb422-26"><a href="Population-genetics.html#cb422-26" aria-hidden="true" tabindex="-1"></a>    <span class="fu">ifelse</span>(i<span class="sc">&lt;</span>(j<span class="sc">+</span><span class="dv">1</span>),Teosinte_struct_heatmap[i,j]<span class="ot">&lt;-</span>Teosinte_structure[k,<span class="dv">4</span>], k<span class="ot">&lt;-</span>k<span class="dv">-1</span>)</span>
<span id="cb422-27"><a href="Population-genetics.html#cb422-27" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb422-28"><a href="Population-genetics.html#cb422-28" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb422-29"><a href="Population-genetics.html#cb422-29" aria-hidden="true" tabindex="-1"></a>Teosinte_structure<span class="sc">$</span>dist<span class="ot">&lt;-</span><span class="fu">as.numeric</span>(Teosinte_structure<span class="sc">$</span>dist)</span>
<span id="cb422-30"><a href="Population-genetics.html#cb422-30" aria-hidden="true" tabindex="-1"></a><span class="fu">image</span>(<span class="fu">t</span>(Teosinte_struct_heatmap),<span class="at">col=</span>colnew )</span></code></pre></div>
<p><img src="bookdown-demo_files/figure-html/unnamed-chunk-311-1.png" width="672" /></p>
<div class="sourceCode" id="cb423"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb423-1"><a href="Population-genetics.html#cb423-1" aria-hidden="true" tabindex="-1"></a>Teosinte_structure_ordered<span class="ot">&lt;-</span>Teosinte_structure[<span class="fu">order</span>(Teosinte_structure<span class="sc">$</span>dist),]</span>
<span id="cb423-2"><a href="Population-genetics.html#cb423-2" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(Teosinte_structure_ordered, <span class="fu">aes</span>(dist,r2s))<span class="sc">+</span></span>
<span id="cb423-3"><a href="Population-genetics.html#cb423-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>()<span class="sc">+</span></span>
<span id="cb423-4"><a href="Population-genetics.html#cb423-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ylim</span>(<span class="dv">0</span>,<span class="dv">1</span>)<span class="sc">+</span></span>
<span id="cb423-5"><a href="Population-genetics.html#cb423-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_smooth</span>(<span class="at">formula=</span>Teosinte_structure_ordered<span class="sc">$</span>r2s<span class="sc">~</span> Teosinte_structure_ordered<span class="sc">$</span>dist,<span class="at">se=</span>F)<span class="sc">+</span></span>
<span id="cb423-6"><a href="Population-genetics.html#cb423-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_hline</span>(<span class="fu">aes</span>(<span class="at">yintercept=</span><span class="fl">0.1</span>),<span class="at">col=</span><span class="st">&quot;red&quot;</span>)</span></code></pre></div>
<p><img src="bookdown-demo_files/figure-html/unnamed-chunk-311-2.png" width="672" /></p>
<p>Now consider the following questions:</p>

<div class="rmdquiz">
<p><strong>14</strong></p>
<p>a: Inspect the scatter plot. What is similar and what is different from the one obtained from the model that does not account for the population structure? Hint: Pay attention to the scale of the Y-axis.</p>
b: Compare the LD matrix plot with that obtained from the model that does not account for population structure.
</div>
</div>
<div id="teosinte-analysis-with-population-structure-structure-k5" class="section level4" number="7.4.1.4">
<h4><span class="header-section-number">7.4.1.4</span> Teosinte analysis with population structure (Structure k=5)</h4>
<p>You might have observed some effect from the inclusion of the groups. However, we already know from the Hardy-Weinberg analysis that “state” does not seem to very accurately represent the true population structure in this data set. Rerun the analysis again using the Structure output as the population structure.</p>
<div class="sourceCode" id="cb424"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb424-1"><a href="Population-genetics.html#cb424-1" aria-hidden="true" tabindex="-1"></a><span class="co">#Teosinte: calculate r^2 taking account of population structure (Structure k=5)</span></span>
<span id="cb424-2"><a href="Population-genetics.html#cb424-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb424-3"><a href="Population-genetics.html#cb424-3" aria-hidden="true" tabindex="-1"></a>Teosinte_structurek5<span class="ot">&lt;-</span><span class="fu">LD.Measures</span>(Teosinte,<span class="at">S=</span>Teosinte_structurek5,<span class="at">na.presence=</span><span class="cn">TRUE</span>)</span>
<span id="cb424-4"><a href="Population-genetics.html#cb424-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb424-5"><a href="Population-genetics.html#cb424-5" aria-hidden="true" tabindex="-1"></a><span class="co">#Extract r^2 and the intermarker distances.</span></span>
<span id="cb424-6"><a href="Population-genetics.html#cb424-6" aria-hidden="true" tabindex="-1"></a>Teosinte_structurek5<span class="sc">$</span>dist<span class="ot">=</span><span class="st">&quot;NA&quot;</span></span>
<span id="cb424-7"><a href="Population-genetics.html#cb424-7" aria-hidden="true" tabindex="-1"></a>nmarkers<span class="ot">=</span><span class="dv">82</span></span>
<span id="cb424-8"><a href="Population-genetics.html#cb424-8" aria-hidden="true" tabindex="-1"></a>k<span class="ot">&lt;-</span><span class="dv">0</span></span>
<span id="cb424-9"><a href="Population-genetics.html#cb424-9" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span>(i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span> (nmarkers<span class="dv">-1</span>))</span>
<span id="cb424-10"><a href="Population-genetics.html#cb424-10" aria-hidden="true" tabindex="-1"></a>{</span>
<span id="cb424-11"><a href="Population-genetics.html#cb424-11" aria-hidden="true" tabindex="-1"></a>  <span class="cf">for</span>(j <span class="cf">in</span> (i<span class="sc">+</span><span class="dv">1</span>)<span class="sc">:</span>nmarkers)</span>
<span id="cb424-12"><a href="Population-genetics.html#cb424-12" aria-hidden="true" tabindex="-1"></a>  {</span>
<span id="cb424-13"><a href="Population-genetics.html#cb424-13" aria-hidden="true" tabindex="-1"></a>    k<span class="ot">&lt;-</span>k<span class="sc">+</span><span class="dv">1</span></span>
<span id="cb424-14"><a href="Population-genetics.html#cb424-14" aria-hidden="true" tabindex="-1"></a>    Teosinte_structurek5[k,<span class="dv">5</span>]<span class="ot">&lt;-</span><span class="fu">abs</span>(Teosintemap[i,<span class="dv">3</span>]<span class="sc">-</span>Teosintemap[j,<span class="dv">3</span>])</span>
<span id="cb424-15"><a href="Population-genetics.html#cb424-15" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb424-16"><a href="Population-genetics.html#cb424-16" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb424-17"><a href="Population-genetics.html#cb424-17" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb424-18"><a href="Population-genetics.html#cb424-18" aria-hidden="true" tabindex="-1"></a>Teosinte_structk5_heatmap<span class="ot">&lt;-</span><span class="fu">matrix</span>(<span class="cn">NA</span>,<span class="at">nrow=</span>nmarkers<span class="dv">-1</span>,<span class="at">ncol=</span>nmarkers<span class="dv">-1</span>)</span>
<span id="cb424-19"><a href="Population-genetics.html#cb424-19" aria-hidden="true" tabindex="-1"></a>k<span class="ot">&lt;-</span><span class="dv">0</span></span>
<span id="cb424-20"><a href="Population-genetics.html#cb424-20" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span>(i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span> (nmarkers<span class="dv">-1</span>))</span>
<span id="cb424-21"><a href="Population-genetics.html#cb424-21" aria-hidden="true" tabindex="-1"></a>{</span>
<span id="cb424-22"><a href="Population-genetics.html#cb424-22" aria-hidden="true" tabindex="-1"></a>  <span class="cf">for</span>(j <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span>(nmarkers<span class="dv">-1</span>))</span>
<span id="cb424-23"><a href="Population-genetics.html#cb424-23" aria-hidden="true" tabindex="-1"></a>  {</span>
<span id="cb424-24"><a href="Population-genetics.html#cb424-24" aria-hidden="true" tabindex="-1"></a>    k<span class="ot">&lt;-</span>k<span class="sc">+</span><span class="dv">1</span></span>
<span id="cb424-25"><a href="Population-genetics.html#cb424-25" aria-hidden="true" tabindex="-1"></a>    <span class="fu">ifelse</span>(i<span class="sc">&lt;</span>(j<span class="sc">+</span><span class="dv">1</span>),Teosinte_structk5_heatmap[i,j]<span class="ot">&lt;-</span>Teosinte_structurek5[k,<span class="dv">4</span>], k<span class="ot">&lt;-</span>k<span class="dv">-1</span>)</span>
<span id="cb424-26"><a href="Population-genetics.html#cb424-26" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb424-27"><a href="Population-genetics.html#cb424-27" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb424-28"><a href="Population-genetics.html#cb424-28" aria-hidden="true" tabindex="-1"></a>Teosinte_structurek5<span class="sc">$</span>dist<span class="ot">&lt;-</span><span class="fu">as.numeric</span>(Teosinte_structurek5<span class="sc">$</span>dist)</span>
<span id="cb424-29"><a href="Population-genetics.html#cb424-29" aria-hidden="true" tabindex="-1"></a><span class="fu">image</span>(<span class="fu">t</span>(Teosinte_structk5_heatmap),<span class="at">col=</span>colnew )</span></code></pre></div>
<p><img src="bookdown-demo_files/figure-html/unnamed-chunk-313-1.png" width="672" /></p>
<div class="sourceCode" id="cb425"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb425-1"><a href="Population-genetics.html#cb425-1" aria-hidden="true" tabindex="-1"></a>Teosinte_structurek5_ordered<span class="ot">&lt;-</span>Teosinte_structurek5[<span class="fu">order</span>(Teosinte_structurek5<span class="sc">$</span>dist),]</span>
<span id="cb425-2"><a href="Population-genetics.html#cb425-2" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(Teosinte_structurek5_ordered, <span class="fu">aes</span>(dist,r2s))<span class="sc">+</span></span>
<span id="cb425-3"><a href="Population-genetics.html#cb425-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>()<span class="sc">+</span></span>
<span id="cb425-4"><a href="Population-genetics.html#cb425-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ylim</span>(<span class="dv">0</span>,<span class="dv">1</span>)<span class="sc">+</span></span>
<span id="cb425-5"><a href="Population-genetics.html#cb425-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_smooth</span>(<span class="at">formula=</span>Teosinte_structurek5_ordered<span class="sc">$</span>r2s<span class="sc">~</span> Teosinte_structurek5_ordered<span class="sc">$</span>dist,<span class="at">se=</span>F)<span class="sc">+</span></span>
<span id="cb425-6"><a href="Population-genetics.html#cb425-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_hline</span>(<span class="fu">aes</span>(<span class="at">yintercept=</span><span class="fl">0.1</span>),<span class="at">col=</span><span class="st">&quot;red&quot;</span>)</span></code></pre></div>
<p><img src="bookdown-demo_files/figure-html/unnamed-chunk-313-2.png" width="672" /></p>

<div class="rmdquiz">
<strong>15</strong>
Compare the results with those in the earlier parts. What do you conclude?
</div>
</div>
<div id="barley-association-mapping-panel" class="section level4" number="7.4.1.5">
<h4><span class="header-section-number">7.4.1.5</span> Barley association mapping panel</h4>
<p>Understanding the breakdown of LD and whether the pattern of LD is affected by population structure are critical components of all association mapping studies such as the barely AGOUEB study.</p>

<div class="rmdquiz">
<strong>16</strong>
Before looking at any results, what is your expectation of LD decay as function of genetic distance in the barley association mapping panel, compared to teosinte? Think of all the factors that differ between the two data sets and how they may affect LD.
</div>
</div>
<div id="barley-association-mapping-panel-no-population-structure" class="section level4" number="7.4.1.6">
<h4><span class="header-section-number">7.4.1.6</span> Barley association mapping panel (no population structure)</h4>
<p>Run the LD analysis for barley chromosome 1 without population structure.</p>
<div class="sourceCode" id="cb426"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb426-1"><a href="Population-genetics.html#cb426-1" aria-hidden="true" tabindex="-1"></a><span class="co">#Barley: calculate r^2 without taking account of population structure</span></span>
<span id="cb426-2"><a href="Population-genetics.html#cb426-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb426-3"><a href="Population-genetics.html#cb426-3" aria-hidden="true" tabindex="-1"></a>AG_No_structure<span class="ot">&lt;-</span><span class="fu">LD.Measures</span>(AG,<span class="at">na.presence=</span><span class="cn">TRUE</span>)</span>
<span id="cb426-4"><a href="Population-genetics.html#cb426-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb426-5"><a href="Population-genetics.html#cb426-5" aria-hidden="true" tabindex="-1"></a><span class="co">#Extract r^2 and the intermarker distances.</span></span>
<span id="cb426-6"><a href="Population-genetics.html#cb426-6" aria-hidden="true" tabindex="-1"></a>AG_No_structure<span class="sc">$</span>dist<span class="ot">=</span><span class="st">&quot;NA&quot;</span></span>
<span id="cb426-7"><a href="Population-genetics.html#cb426-7" aria-hidden="true" tabindex="-1"></a>nmarkers<span class="ot">=</span><span class="dv">159</span></span>
<span id="cb426-8"><a href="Population-genetics.html#cb426-8" aria-hidden="true" tabindex="-1"></a>k<span class="ot">&lt;-</span><span class="dv">0</span></span>
<span id="cb426-9"><a href="Population-genetics.html#cb426-9" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span>(i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span> (nmarkers<span class="dv">-1</span>))</span>
<span id="cb426-10"><a href="Population-genetics.html#cb426-10" aria-hidden="true" tabindex="-1"></a>{</span>
<span id="cb426-11"><a href="Population-genetics.html#cb426-11" aria-hidden="true" tabindex="-1"></a>  <span class="cf">for</span>(j <span class="cf">in</span> (i<span class="sc">+</span><span class="dv">1</span>)<span class="sc">:</span>nmarkers)</span>
<span id="cb426-12"><a href="Population-genetics.html#cb426-12" aria-hidden="true" tabindex="-1"></a>  {</span>
<span id="cb426-13"><a href="Population-genetics.html#cb426-13" aria-hidden="true" tabindex="-1"></a>    k<span class="ot">&lt;-</span>k<span class="sc">+</span><span class="dv">1</span></span>
<span id="cb426-14"><a href="Population-genetics.html#cb426-14" aria-hidden="true" tabindex="-1"></a>    AG_No_structure[k,<span class="dv">4</span>]<span class="ot">&lt;-</span><span class="fu">abs</span>(AGmap[i,<span class="dv">3</span>]<span class="sc">-</span>AGmap[j,<span class="dv">3</span>])</span>
<span id="cb426-15"><a href="Population-genetics.html#cb426-15" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb426-16"><a href="Population-genetics.html#cb426-16" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb426-17"><a href="Population-genetics.html#cb426-17" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb426-18"><a href="Population-genetics.html#cb426-18" aria-hidden="true" tabindex="-1"></a>AG_No_struct_heatmap<span class="ot">&lt;-</span><span class="fu">matrix</span>(<span class="cn">NA</span>,<span class="at">nrow=</span>nmarkers<span class="dv">-1</span>,<span class="at">ncol=</span>nmarkers<span class="dv">-1</span>)</span>
<span id="cb426-19"><a href="Population-genetics.html#cb426-19" aria-hidden="true" tabindex="-1"></a>k<span class="ot">&lt;-</span><span class="dv">0</span></span>
<span id="cb426-20"><a href="Population-genetics.html#cb426-20" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span>(i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span> (nmarkers<span class="dv">-1</span>))</span>
<span id="cb426-21"><a href="Population-genetics.html#cb426-21" aria-hidden="true" tabindex="-1"></a>{</span>
<span id="cb426-22"><a href="Population-genetics.html#cb426-22" aria-hidden="true" tabindex="-1"></a>  <span class="cf">for</span>(j <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span>(nmarkers<span class="dv">-1</span>))</span>
<span id="cb426-23"><a href="Population-genetics.html#cb426-23" aria-hidden="true" tabindex="-1"></a>  {</span>
<span id="cb426-24"><a href="Population-genetics.html#cb426-24" aria-hidden="true" tabindex="-1"></a>    k<span class="ot">&lt;-</span>k<span class="sc">+</span><span class="dv">1</span></span>
<span id="cb426-25"><a href="Population-genetics.html#cb426-25" aria-hidden="true" tabindex="-1"></a>    <span class="fu">ifelse</span>(i<span class="sc">&lt;</span>(j<span class="sc">+</span><span class="dv">1</span>),AG_No_struct_heatmap[i,j]<span class="ot">&lt;-</span>AG_No_structure[k,<span class="dv">3</span>], k<span class="ot">&lt;-</span>k<span class="dv">-1</span>)</span>
<span id="cb426-26"><a href="Population-genetics.html#cb426-26" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb426-27"><a href="Population-genetics.html#cb426-27" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb426-28"><a href="Population-genetics.html#cb426-28" aria-hidden="true" tabindex="-1"></a>AG_No_structure<span class="sc">$</span>dist<span class="ot">&lt;-</span><span class="fu">as.numeric</span>(AG_No_structure<span class="sc">$</span>dist)</span>
<span id="cb426-29"><a href="Population-genetics.html#cb426-29" aria-hidden="true" tabindex="-1"></a><span class="fu">image</span>(<span class="fu">t</span>(AG_No_struct_heatmap), <span class="at">col=</span>colnew)</span></code></pre></div>
<p><img src="bookdown-demo_files/figure-html/unnamed-chunk-316-1.png" width="672" /></p>
<div class="sourceCode" id="cb427"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb427-1"><a href="Population-genetics.html#cb427-1" aria-hidden="true" tabindex="-1"></a>AG_No_structure_ordered<span class="ot">&lt;-</span>AG_No_structure[<span class="fu">order</span>(AG_No_structure<span class="sc">$</span>dist),]</span>
<span id="cb427-2"><a href="Population-genetics.html#cb427-2" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(AG_No_structure_ordered, <span class="fu">aes</span>(dist,r2))<span class="sc">+</span></span>
<span id="cb427-3"><a href="Population-genetics.html#cb427-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>()<span class="sc">+</span></span>
<span id="cb427-4"><a href="Population-genetics.html#cb427-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ylim</span>(<span class="dv">0</span>,<span class="dv">1</span>)<span class="sc">+</span></span>
<span id="cb427-5"><a href="Population-genetics.html#cb427-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_smooth</span>(<span class="at">formula=</span>AG_No_structure_ordered<span class="sc">$</span>r2<span class="sc">~</span> AG_No_structure_ordered<span class="sc">$</span>dist,<span class="at">se=</span>F)<span class="sc">+</span></span>
<span id="cb427-6"><a href="Population-genetics.html#cb427-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_hline</span>(<span class="fu">aes</span>(<span class="at">yintercept=</span><span class="fl">0.1</span>),<span class="at">col=</span><span class="st">&quot;red&quot;</span>)</span></code></pre></div>
<p><img src="bookdown-demo_files/figure-html/unnamed-chunk-316-2.png" width="672" /></p>

<div class="rmdquiz">
<p><strong>17</strong></p>
<p>a: Inspect the scatter plot of r2 values versus genetic distance. How rapidly does LD decay in this population?</p>
b: What is your initial interpretation of both plots?
</div>
</div>
<div id="re-estimate-ld-in-the-barley-data-set-controlling-for-population-structure." class="section level4" number="7.4.1.7">
<h4><span class="header-section-number">7.4.1.7</span> Re-estimate LD in the barley data set controlling for population structure.</h4>
<div class="sourceCode" id="cb428"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb428-1"><a href="Population-genetics.html#cb428-1" aria-hidden="true" tabindex="-1"></a><span class="co">#Barley: calculate r^2 taking account of population structure (pcoas 1 and 2)</span></span>
<span id="cb428-2"><a href="Population-genetics.html#cb428-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb428-3"><a href="Population-genetics.html#cb428-3" aria-hidden="true" tabindex="-1"></a>AG_structure<span class="ot">&lt;-</span><span class="fu">LD.Measures</span>(AG,<span class="at">S=</span>AGpcoa,<span class="at">na.presence=</span><span class="cn">TRUE</span>)</span>
<span id="cb428-4"><a href="Population-genetics.html#cb428-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb428-5"><a href="Population-genetics.html#cb428-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb428-6"><a href="Population-genetics.html#cb428-6" aria-hidden="true" tabindex="-1"></a><span class="co">#Extract r^2 and the intermarker distances.</span></span>
<span id="cb428-7"><a href="Population-genetics.html#cb428-7" aria-hidden="true" tabindex="-1"></a>AG_structure<span class="sc">$</span>dist<span class="ot">=</span><span class="st">&quot;NA&quot;</span></span>
<span id="cb428-8"><a href="Population-genetics.html#cb428-8" aria-hidden="true" tabindex="-1"></a>nmarkers<span class="ot">=</span><span class="dv">159</span></span>
<span id="cb428-9"><a href="Population-genetics.html#cb428-9" aria-hidden="true" tabindex="-1"></a>k<span class="ot">&lt;-</span><span class="dv">0</span></span>
<span id="cb428-10"><a href="Population-genetics.html#cb428-10" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span>(i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span> (nmarkers<span class="dv">-1</span>))</span>
<span id="cb428-11"><a href="Population-genetics.html#cb428-11" aria-hidden="true" tabindex="-1"></a>{</span>
<span id="cb428-12"><a href="Population-genetics.html#cb428-12" aria-hidden="true" tabindex="-1"></a>  <span class="cf">for</span>(j <span class="cf">in</span> (i<span class="sc">+</span><span class="dv">1</span>)<span class="sc">:</span>nmarkers)</span>
<span id="cb428-13"><a href="Population-genetics.html#cb428-13" aria-hidden="true" tabindex="-1"></a>  {</span>
<span id="cb428-14"><a href="Population-genetics.html#cb428-14" aria-hidden="true" tabindex="-1"></a>    k<span class="ot">&lt;-</span>k<span class="sc">+</span><span class="dv">1</span></span>
<span id="cb428-15"><a href="Population-genetics.html#cb428-15" aria-hidden="true" tabindex="-1"></a>    AG_structure[k,<span class="dv">5</span>]<span class="ot">&lt;-</span><span class="fu">abs</span>(AGmap[i,<span class="dv">3</span>]<span class="sc">-</span>AGmap[j,<span class="dv">3</span>])</span>
<span id="cb428-16"><a href="Population-genetics.html#cb428-16" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb428-17"><a href="Population-genetics.html#cb428-17" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb428-18"><a href="Population-genetics.html#cb428-18" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb428-19"><a href="Population-genetics.html#cb428-19" aria-hidden="true" tabindex="-1"></a>AG_struct_heatmap<span class="ot">&lt;-</span><span class="fu">matrix</span>(<span class="cn">NA</span>,<span class="at">nrow=</span>nmarkers<span class="dv">-1</span>,<span class="at">ncol=</span>nmarkers<span class="dv">-1</span>)</span>
<span id="cb428-20"><a href="Population-genetics.html#cb428-20" aria-hidden="true" tabindex="-1"></a>k<span class="ot">&lt;-</span><span class="dv">0</span></span>
<span id="cb428-21"><a href="Population-genetics.html#cb428-21" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span>(i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span> (nmarkers<span class="dv">-1</span>))</span>
<span id="cb428-22"><a href="Population-genetics.html#cb428-22" aria-hidden="true" tabindex="-1"></a>{</span>
<span id="cb428-23"><a href="Population-genetics.html#cb428-23" aria-hidden="true" tabindex="-1"></a>  <span class="cf">for</span>(j <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span>(nmarkers<span class="dv">-1</span>))</span>
<span id="cb428-24"><a href="Population-genetics.html#cb428-24" aria-hidden="true" tabindex="-1"></a>  {</span>
<span id="cb428-25"><a href="Population-genetics.html#cb428-25" aria-hidden="true" tabindex="-1"></a>    k<span class="ot">&lt;-</span>k<span class="sc">+</span><span class="dv">1</span></span>
<span id="cb428-26"><a href="Population-genetics.html#cb428-26" aria-hidden="true" tabindex="-1"></a>    <span class="fu">ifelse</span>(i<span class="sc">&lt;</span>(j<span class="sc">+</span><span class="dv">1</span>),AG_struct_heatmap[i,j]<span class="ot">&lt;-</span>AG_structure[k,<span class="dv">4</span>], k<span class="ot">&lt;-</span>k<span class="dv">-1</span>)</span>
<span id="cb428-27"><a href="Population-genetics.html#cb428-27" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb428-28"><a href="Population-genetics.html#cb428-28" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb428-29"><a href="Population-genetics.html#cb428-29" aria-hidden="true" tabindex="-1"></a>AG_structure<span class="sc">$</span>dist<span class="ot">&lt;-</span><span class="fu">as.numeric</span>(AG_structure<span class="sc">$</span>dist)</span>
<span id="cb428-30"><a href="Population-genetics.html#cb428-30" aria-hidden="true" tabindex="-1"></a><span class="fu">image</span>(<span class="fu">t</span>(AG_struct_heatmap),<span class="at">col=</span>colnew)</span></code></pre></div>
<p><img src="bookdown-demo_files/figure-html/unnamed-chunk-318-1.png" width="672" /></p>
<div class="sourceCode" id="cb429"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb429-1"><a href="Population-genetics.html#cb429-1" aria-hidden="true" tabindex="-1"></a>AG_structure_ordered<span class="ot">&lt;-</span>AG_structure[<span class="fu">order</span>(AG_structure<span class="sc">$</span>dist),]</span>
<span id="cb429-2"><a href="Population-genetics.html#cb429-2" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(AG_structure_ordered, <span class="fu">aes</span>(dist,r2s))<span class="sc">+</span></span>
<span id="cb429-3"><a href="Population-genetics.html#cb429-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>()<span class="sc">+</span></span>
<span id="cb429-4"><a href="Population-genetics.html#cb429-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ylim</span>(<span class="dv">0</span>,<span class="dv">1</span>)<span class="sc">+</span></span>
<span id="cb429-5"><a href="Population-genetics.html#cb429-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_smooth</span>(<span class="at">formula=</span>AG_structure_ordered<span class="sc">$</span>r2s<span class="sc">~</span> AG_structure_ordered<span class="sc">$</span>dist,<span class="at">se=</span>F)<span class="sc">+</span></span>
<span id="cb429-6"><a href="Population-genetics.html#cb429-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_hline</span>(<span class="fu">aes</span>(<span class="at">yintercept=</span><span class="fl">0.1</span>),<span class="at">col=</span><span class="st">&quot;red&quot;</span>)</span></code></pre></div>
<p><img src="bookdown-demo_files/figure-html/unnamed-chunk-318-2.png" width="672" /></p>

<div class="rmdquiz">
<p><strong>18</strong></p>
<p>a: Compare the scatter plot with that obtained from the null model. Does the new plot give a clearer picture of the LD decay than before? If so, why?</p>
b: What is your conclusion regarding LD between pairs of markers in general when comparing the new LD matrix plot with that from the model that does not account for population structure? ?
</div>
</div>
<div id="comparisons-accross-datasets." class="section level4" number="7.4.1.8">
<h4><span class="header-section-number">7.4.1.8</span> Comparisons accross datasets.</h4>
<p>Compare the results observed for barley with those for Teosinte.</p>

<div class="rmdquiz">
<strong>19</strong>
What might be the explanation for the differences? How might this impact on association mapping studies?
</div>
</div>
</div>
</div>
<div id="recombination-and-linkage-disequilibrium-applications" class="section level2" number="7.5">
<h2><span class="header-section-number">7.5</span> Recombination and linkage disequilibrium: applications</h2>
<p>The following figure is from a paper <span class="citation"><a href="references.html#ref-Gardner2016" role="doc-biblioref">Gardner, Wittern, and Mackay</a> (<a href="references.html#ref-Gardner2016" role="doc-biblioref">2016a</a>)</span> which created a genetic map for wheat using an experimental MAGIC (Multiparent Advanced Generation InterCross) population. The population was created by crossing 8 distinct UK wheat varieties in all combinations over 3 generations, so that each individual has on average 1/8 of its genome from each parent. The lines are then selfed for several generations. This population has an overall high recombination rate compared to standard biparental mapping populations used for QTL detection. The top figure plots the genetic map (based on recombination frequencies) vs the physical map (in base pairs) for chromosome 3B. The genetic map is based on recombination frequencies estimated in the population. The bottom figure shows the recombination rate along the chromosome based on a comparison of the 2 maps. For mapping QTL, we would use the genetic map.</p>
<div class="figure">
<img src="images/pop_gen_magic.png" alt="" />
<p class="caption">MAGIC genetic vs. physical map positions (a). Recombination rate in MAGIC population (b).</p>
</div>

<div class="rmdquiz">
<p><strong>20</strong></p>
<p>a: It is clear that the relationship between the 2 maps is non-linear. Explain why this is so with reference to the recombination rate graph</p>
<p>b: Why does there appear to be a large region of no recombination in the centre of the chromosome?</p>
<p>c: Recombination frequencies were estimated from the linkage disequilibrium generated in this population. What would a graph of LD along the chromosome look like? Which parts of the chromosome have high and low linkage disequilibrium? What implications does this have for QTL mapping in wheat?</p>
d: As well as the centre of the chromosome, there are other shorter regions (e.g. around 610 Mb and 690 Mb) of the chromosome which have low recombination. The markers in these regions are also distinctive in being out of Hardy-Weinberg equilibrium. What could be going on here? (hint: think back to the causes of underdominance)
</div>
</div>
<div id="pedigree-analysis" class="section level2" number="7.6">
<h2><span class="header-section-number">7.6</span> Pedigree analysis</h2>
<p>Inspect the complex pedigree below.</p>

<div class="rmdquiz">
<p><em>21</em> Calculate the inbreeding coefficient of individual N in the following complex pedigree. This is difficult so we give two hints:</p>
a: there are 7 possible paths
b: two of the 7 paths have an inbred individual at the top – you must account for the fact that this individual is inbred.
</div>
<div class="figure">
<img src="images/pop_gen_pedigree.PNG" alt="" />
<p class="caption">Example of a complex pedigree</p>
</div>

</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="Cross-site-analysis.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="Imputation.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": "https://github.com/rstudio/bookdown-demo/edit/master/07-Population_genetics.Rmd",
"text": "Edit"
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": ["bookdown-demo.pdf", "bookdown-demo.epub"],
"toc": {
"collapse": "subsection"
}
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
