<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 3 R and Rstudio: a general introduction with statistics. | Quantitative Methods in Plant Breeding: The Digital Tutorials.</title>
  <meta name="description" content="This is the online tutorial booklet for the QMPB course run at NIAB. The online book was formed using R bookdown." />
  <meta name="generator" content="bookdown 0.21 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 3 R and Rstudio: a general introduction with statistics. | Quantitative Methods in Plant Breeding: The Digital Tutorials." />
  <meta property="og:type" content="book" />
  
  <meta property="og:image" content="images/cover.jpg" />
  <meta property="og:description" content="This is the online tutorial booklet for the QMPB course run at NIAB. The online book was formed using R bookdown." />
  <meta name="github-repo" content="rstudio/bookdown" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 3 R and Rstudio: a general introduction with statistics. | Quantitative Methods in Plant Breeding: The Digital Tutorials." />
  
  <meta name="twitter:description" content="This is the online tutorial booklet for the QMPB course run at NIAB. The online book was formed using R bookdown." />
  <meta name="twitter:image" content="images/cover.jpg" />

<meta name="author" content="NIAB 2021" />


<meta name="date" content="2021-02-27" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="Acknow.html"/>
<link rel="next" href="Trial-design.html"/>
<script src="libs/header-attrs-2.6/header-attrs.js"></script>
<script src="libs/jquery-2.2.3/jquery.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />











<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">The QMPB digital tutorials</a></li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> Introduction</a></li>
<li class="chapter" data-level="2" data-path="Acknow.html"><a href="Acknow.html"><i class="fa fa-check"></i><b>2</b> Acknowledgements and Contents</a>
<ul>
<li class="chapter" data-level="2.1" data-path="Acknow.html"><a href="Acknow.html#acknowledgements"><i class="fa fa-check"></i><b>2.1</b> Acknowledgements</a></li>
<li class="chapter" data-level="2.2" data-path="Acknow.html"><a href="Acknow.html#digital-tutorial-chapters"><i class="fa fa-check"></i><b>2.2</b> Digital Tutorial Chapters:</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="R-and-stats.html"><a href="R-and-stats.html"><i class="fa fa-check"></i><b>3</b> R and Rstudio: a general introduction with statistics.</a>
<ul>
<li class="chapter" data-level="3.1" data-path="R-and-stats.html"><a href="R-and-stats.html#introduction-section-on-r"><i class="fa fa-check"></i><b>3.1</b> Introduction section on R</a>
<ul>
<li class="chapter" data-level="3.1.1" data-path="R-and-stats.html"><a href="R-and-stats.html#getting-the-data-ready"><i class="fa fa-check"></i><b>3.1.1</b> Getting the data ready</a></li>
</ul></li>
<li class="chapter" data-level="3.2" data-path="R-and-stats.html"><a href="R-and-stats.html#the-rstudio-layout-workspace-and-environment."><i class="fa fa-check"></i><b>3.2</b> The Rstudio layout, workspace and environment.</a>
<ul>
<li class="chapter" data-level="3.2.1" data-path="R-and-stats.html"><a href="R-and-stats.html#working-directory"><i class="fa fa-check"></i><b>3.2.1</b> Working directory</a></li>
</ul></li>
<li class="chapter" data-level="3.3" data-path="R-and-stats.html"><a href="R-and-stats.html#basic-r-syntx"><i class="fa fa-check"></i><b>3.3</b> Basic R Syntx</a>
<ul>
<li class="chapter" data-level="3.3.1" data-path="R-and-stats.html"><a href="R-and-stats.html#brackets"><i class="fa fa-check"></i><b>3.3.1</b> Brackets</a></li>
<li class="chapter" data-level="3.3.2" data-path="R-and-stats.html"><a href="R-and-stats.html#equals-symbols"><i class="fa fa-check"></i><b>3.3.2</b> Equals symbols</a></li>
<li class="chapter" data-level="3.3.3" data-path="R-and-stats.html"><a href="R-and-stats.html#continuation-character."><i class="fa fa-check"></i><b>3.3.3</b> Continuation character.</a></li>
<li class="chapter" data-level="3.3.4" data-path="R-and-stats.html"><a href="R-and-stats.html#text-and-numbers"><i class="fa fa-check"></i><b>3.3.4</b> Text and numbers</a></li>
<li class="chapter" data-level="3.3.5" data-path="R-and-stats.html"><a href="R-and-stats.html#summary-of-syntax"><i class="fa fa-check"></i><b>3.3.5</b> Summary of syntax</a></li>
</ul></li>
<li class="chapter" data-level="3.4" data-path="R-and-stats.html"><a href="R-and-stats.html#reading-in-data"><i class="fa fa-check"></i><b>3.4</b> Reading in data</a></li>
<li class="chapter" data-level="3.5" data-path="R-and-stats.html"><a href="R-and-stats.html#understanding-data-types-and-r-objects"><i class="fa fa-check"></i><b>3.5</b> Understanding data types and R objects</a>
<ul>
<li class="chapter" data-level="3.5.1" data-path="R-and-stats.html"><a href="R-and-stats.html#summary"><i class="fa fa-check"></i><b>3.5.1</b> Summary</a></li>
</ul></li>
<li class="chapter" data-level="3.6" data-path="R-and-stats.html"><a href="R-and-stats.html#summarising-and-displaying-data"><i class="fa fa-check"></i><b>3.6</b> Summarising and displaying data</a></li>
<li class="chapter" data-level="3.7" data-path="R-and-stats.html"><a href="R-and-stats.html#calculations-and-data-manipulation"><i class="fa fa-check"></i><b>3.7</b> Calculations and data manipulation</a>
<ul>
<li class="chapter" data-level="3.7.1" data-path="R-and-stats.html"><a href="R-and-stats.html#sorting-data"><i class="fa fa-check"></i><b>3.7.1</b> Sorting data</a></li>
<li class="chapter" data-level="3.7.2" data-path="R-and-stats.html"><a href="R-and-stats.html#relationships-between-variates"><i class="fa fa-check"></i><b>3.7.2</b> Relationships between variates</a></li>
<li class="chapter" data-level="3.7.3" data-path="R-and-stats.html"><a href="R-and-stats.html#exporting-your-script-and-workspace"><i class="fa fa-check"></i><b>3.7.3</b> Exporting your script and workspace</a></li>
<li class="chapter" data-level="3.7.4" data-path="R-and-stats.html"><a href="R-and-stats.html#saving-a-data-table"><i class="fa fa-check"></i><b>3.7.4</b> Saving a data table</a></li>
</ul></li>
<li class="chapter" data-level="3.8" data-path="R-and-stats.html"><a href="R-and-stats.html#basic-statistical-analysis"><i class="fa fa-check"></i><b>3.8</b> Basic statistical analysis</a>
<ul>
<li class="chapter" data-level="3.8.1" data-path="R-and-stats.html"><a href="R-and-stats.html#the-t-test"><i class="fa fa-check"></i><b>3.8.1</b> The t-test</a></li>
<li class="chapter" data-level="3.8.2" data-path="R-and-stats.html"><a href="R-and-stats.html#linear-regression"><i class="fa fa-check"></i><b>3.8.2</b> Linear Regression</a></li>
<li class="chapter" data-level="3.8.3" data-path="R-and-stats.html"><a href="R-and-stats.html#multiple-regression"><i class="fa fa-check"></i><b>3.8.3</b> Multiple regression</a></li>
<li class="chapter" data-level="3.8.4" data-path="R-and-stats.html"><a href="R-and-stats.html#the-analysis-of-variance"><i class="fa fa-check"></i><b>3.8.4</b> The analysis of variance</a></li>
</ul></li>
<li class="chapter" data-level="3.9" data-path="R-and-stats.html"><a href="R-and-stats.html#categorical-data---the-chi-squared-test"><i class="fa fa-check"></i><b>3.9</b> Categorical data - the <em>chi</em>-squared test</a></li>
<li class="chapter" data-level="3.10" data-path="R-and-stats.html"><a href="R-and-stats.html#more-useful-information"><i class="fa fa-check"></i><b>3.10</b> More useful information</a>
<ul>
<li class="chapter" data-level="3.10.1" data-path="R-and-stats.html"><a href="R-and-stats.html#plotting-graphs"><i class="fa fa-check"></i><b>3.10.1</b> Plotting Graphs</a></li>
<li class="chapter" data-level="3.10.2" data-path="R-and-stats.html"><a href="R-and-stats.html#probability-distributions"><i class="fa fa-check"></i><b>3.10.2</b> Probability distributions</a></li>
<li class="chapter" data-level="3.10.3" data-path="R-and-stats.html"><a href="R-and-stats.html#random-number-generation"><i class="fa fa-check"></i><b>3.10.3</b> Random number generation</a></li>
<li class="chapter" data-level="3.10.4" data-path="R-and-stats.html"><a href="R-and-stats.html#loops-second-example"><i class="fa fa-check"></i><b>3.10.4</b> Loops: <em>second example</em></a></li>
<li class="chapter" data-level="3.10.5" data-path="R-and-stats.html"><a href="R-and-stats.html#miscellany"><i class="fa fa-check"></i><b>3.10.5</b> Miscellany</a></li>
<li class="chapter" data-level="3.10.6" data-path="R-and-stats.html"><a href="R-and-stats.html#my-number-one-piece-of-advice"><i class="fa fa-check"></i><b>3.10.6</b> My number one piece of advice</a></li>
<li class="chapter" data-level="3.10.7" data-path="R-and-stats.html"><a href="R-and-stats.html#learn-more"><i class="fa fa-check"></i><b>3.10.7</b> Learn more</a></li>
<li class="chapter" data-level="3.10.8" data-path="R-and-stats.html"><a href="R-and-stats.html#packages"><i class="fa fa-check"></i><b>3.10.8</b> Packages</a></li>
<li class="chapter" data-level="3.10.9" data-path="R-and-stats.html"><a href="R-and-stats.html#the-tidyverse"><i class="fa fa-check"></i><b>3.10.9</b> The Tidyverse</a></li>
<li class="chapter" data-level="3.10.10" data-path="R-and-stats.html"><a href="R-and-stats.html#other-resources"><i class="fa fa-check"></i><b>3.10.10</b> Other resources</a></li>
</ul></li>
<li class="chapter" data-level="3.11" data-path="R-and-stats.html"><a href="R-and-stats.html#list-of-commands-mainly-described-in-this-guide."><i class="fa fa-check"></i><b>3.11</b> List of commands, mainly described in this guide.</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="Trial-design.html"><a href="Trial-design.html"><i class="fa fa-check"></i><b>4</b> Trial Design.</a>
<ul>
<li class="chapter" data-level="4.1" data-path="Trial-design.html"><a href="Trial-design.html#randomised-complete-block-designs"><i class="fa fa-check"></i><b>4.1</b> Randomised complete block designs</a></li>
<li class="chapter" data-level="4.2" data-path="Trial-design.html"><a href="Trial-design.html#trial-design-in-r"><i class="fa fa-check"></i><b>4.2</b> Trial design in R</a>
<ul>
<li class="chapter" data-level="4.2.1" data-path="Trial-design.html"><a href="Trial-design.html#partially-replicated-designs"><i class="fa fa-check"></i><b>4.2.1</b> Partially replicated designs</a></li>
<li class="chapter" data-level="4.2.2" data-path="Trial-design.html"><a href="Trial-design.html#two-dimensional-designs"><i class="fa fa-check"></i><b>4.2.2</b> Two-dimensional designs</a></li>
</ul></li>
<li class="chapter" data-level="4.3" data-path="Trial-design.html"><a href="Trial-design.html#farm-scale-experiments"><i class="fa fa-check"></i><b>4.3</b> Farm scale experiments</a></li>
<li class="chapter" data-level="4.4" data-path="Trial-design.html"><a href="Trial-design.html#alpha-designs."><i class="fa fa-check"></i><b>4.4</b> Alpha designs.</a>
<ul>
<li class="chapter" data-level="4.4.1" data-path="Trial-design.html"><a href="Trial-design.html#creation-of-augmented-and-p-rep-designs-from-2-replicate-alpha-designs"><i class="fa fa-check"></i><b>4.4.1</b> Creation of augmented and p-rep designs from 2-replicate alpha designs</a></li>
<li class="chapter" data-level="4.4.2" data-path="Trial-design.html"><a href="Trial-design.html#final-message"><i class="fa fa-check"></i><b>4.4.2</b> Final message</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="5" data-path="Trial-analysis.html"><a href="Trial-analysis.html"><i class="fa fa-check"></i><b>5</b> Trial analysis (sugar beet example).</a>
<ul>
<li class="chapter" data-level="5.1" data-path="Trial-analysis.html"><a href="Trial-analysis.html#the-layout-of-the-sugar-beet-trial"><i class="fa fa-check"></i><b>5.1</b> The layout of the sugar beet trial</a></li>
<li class="chapter" data-level="5.2" data-path="Trial-analysis.html"><a href="Trial-analysis.html#fixed-effects-random-effects-and-reml"><i class="fa fa-check"></i><b>5.2</b> Fixed effects, random effects and REML</a>
<ul>
<li class="chapter" data-level="5.2.1" data-path="Trial-analysis.html"><a href="Trial-analysis.html#preparing-the-data"><i class="fa fa-check"></i><b>5.2.1</b> Preparing the data</a></li>
<li class="chapter" data-level="5.2.2" data-path="Trial-analysis.html"><a href="Trial-analysis.html#randomised-complete-block-design"><i class="fa fa-check"></i><b>5.2.2</b> Randomised complete block design</a></li>
<li class="chapter" data-level="5.2.3" data-path="Trial-analysis.html"><a href="Trial-analysis.html#randomised-compelete-block-design---with-mixed-models"><i class="fa fa-check"></i><b>5.2.3</b> Randomised compelete block design - with mixed models</a></li>
<li class="chapter" data-level="5.2.4" data-path="Trial-analysis.html"><a href="Trial-analysis.html#plotting-residuals-against-field-structure."><i class="fa fa-check"></i><b>5.2.4</b> Plotting residuals against field structure.</a></li>
<li class="chapter" data-level="5.2.5" data-path="Trial-analysis.html"><a href="Trial-analysis.html#alpha-design-include-block-structure"><i class="fa fa-check"></i><b>5.2.5</b> Alpha design (include block structure)</a></li>
<li class="chapter" data-level="5.2.6" data-path="Trial-analysis.html"><a href="Trial-analysis.html#rows-and-column-analysis"><i class="fa fa-check"></i><b>5.2.6</b> Rows and column analysis</a></li>
</ul></li>
<li class="chapter" data-level="5.3" data-path="Trial-analysis.html"><a href="Trial-analysis.html#spatial-analysis"><i class="fa fa-check"></i><b>5.3</b> Spatial analysis</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="Cross-site-analysis.html"><a href="Cross-site-analysis.html"><i class="fa fa-check"></i><b>6</b> Cross sites analysis in R.</a>
<ul>
<li class="chapter" data-level="6.1" data-path="Cross-site-analysis.html"><a href="Cross-site-analysis.html#blues-blups-and-one-stage-analysis"><i class="fa fa-check"></i><b>6.1</b> BLUEs, BLUPs and one stage analysis</a>
<ul>
<li class="chapter" data-level="6.1.1" data-path="Cross-site-analysis.html"><a href="Cross-site-analysis.html#one-stage-analysis-with-weights"><i class="fa fa-check"></i><b>6.1.1</b> One stage analysis with weights</a></li>
</ul></li>
<li class="chapter" data-level="6.2" data-path="Cross-site-analysis.html"><a href="Cross-site-analysis.html#weighted-two-stage-analysis"><i class="fa fa-check"></i><b>6.2</b> Weighted two-stage analysis</a></li>
<li class="chapter" data-level="6.3" data-path="Cross-site-analysis.html"><a href="Cross-site-analysis.html#ammi"><i class="fa fa-check"></i><b>6.3</b> AMMI</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="Population-genetics.html"><a href="Population-genetics.html"><i class="fa fa-check"></i><b>7</b> Population genetics.</a>
<ul>
<li class="chapter" data-level="7.1" data-path="Population-genetics.html"><a href="Population-genetics.html#hardy-weinberg-equilibrium-and-population-structure"><i class="fa fa-check"></i><b>7.1</b> Hardy-Weinberg equilibrium and population structure</a></li>
<li class="chapter" data-level="7.2" data-path="Population-genetics.html"><a href="Population-genetics.html#other-ways-of-analysing-population-structure"><i class="fa fa-check"></i><b>7.2</b> Other ways of analysing population structure</a>
<ul>
<li class="chapter" data-level="7.2.1" data-path="Population-genetics.html"><a href="Population-genetics.html#distance-trees"><i class="fa fa-check"></i><b>7.2.1</b> Distance Trees</a></li>
</ul></li>
<li class="chapter" data-level="7.3" data-path="Population-genetics.html"><a href="Population-genetics.html#structure"><i class="fa fa-check"></i><b>7.3</b> Structure</a>
<ul>
<li class="chapter" data-level="7.3.1" data-path="Population-genetics.html"><a href="Population-genetics.html#what-structure-does"><i class="fa fa-check"></i><b>7.3.1</b> What STRUCTURE does</a></li>
<li class="chapter" data-level="7.3.2" data-path="Population-genetics.html"><a href="Population-genetics.html#limitations-of-structure"><i class="fa fa-check"></i><b>7.3.2</b> Limitations of STRUCTURE</a></li>
<li class="chapter" data-level="7.3.3" data-path="Population-genetics.html"><a href="Population-genetics.html#using-structure"><i class="fa fa-check"></i><b>7.3.3</b> Using Structure</a></li>
<li class="chapter" data-level="7.3.4" data-path="Population-genetics.html"><a href="Population-genetics.html#the-exercise"><i class="fa fa-check"></i><b>7.3.4</b> The exercise</a></li>
</ul></li>
<li class="chapter" data-level="7.4" data-path="Population-genetics.html"><a href="Population-genetics.html#linkage-disequilibrium-analysis"><i class="fa fa-check"></i><b>7.4</b> Linkage disequilibrium analysis</a>
<ul>
<li class="chapter" data-level="7.4.1" data-path="Population-genetics.html"><a href="Population-genetics.html#linkage-disequilibrium-in-the-teosinte-and-barley-data-sets"><i class="fa fa-check"></i><b>7.4.1</b> Linkage disequilibrium in the teosinte and barley data sets</a></li>
</ul></li>
<li class="chapter" data-level="7.5" data-path="Population-genetics.html"><a href="Population-genetics.html#recombination-and-linkage-disequilibrium-applications"><i class="fa fa-check"></i><b>7.5</b> Recombination and linkage disequilibrium: applications</a></li>
<li class="chapter" data-level="7.6" data-path="Population-genetics.html"><a href="Population-genetics.html#pedigree-analysis"><i class="fa fa-check"></i><b>7.6</b> Pedigree analysis</a></li>
</ul></li>
<li class="chapter" data-level="8" data-path="Imputation.html"><a href="Imputation.html"><i class="fa fa-check"></i><b>8</b> Imputation.</a>
<ul>
<li class="chapter" data-level="8.1" data-path="Imputation.html"><a href="Imputation.html#introduction"><i class="fa fa-check"></i><b>8.1</b> Introduction</a></li>
<li class="chapter" data-level="8.2" data-path="Imputation.html"><a href="Imputation.html#the-exercise-1"><i class="fa fa-check"></i><b>8.2</b> The exercise</a></li>
<li class="chapter" data-level="8.3" data-path="Imputation.html"><a href="Imputation.html#data-preparation-for-all-methods"><i class="fa fa-check"></i><b>8.3</b> Data preparation (for all methods)</a></li>
<li class="chapter" data-level="8.4" data-path="Imputation.html"><a href="Imputation.html#svd-imputation"><i class="fa fa-check"></i><b>8.4</b> SVD imputation</a></li>
<li class="chapter" data-level="8.5" data-path="Imputation.html"><a href="Imputation.html#mean-imputation"><i class="fa fa-check"></i><b>8.5</b> Mean imputation</a></li>
<li class="chapter" data-level="8.6" data-path="Imputation.html"><a href="Imputation.html#random-forest-imputation"><i class="fa fa-check"></i><b>8.6</b> Random Forest Imputation</a></li>
<li class="chapter" data-level="8.7" data-path="Imputation.html"><a href="Imputation.html#summary-1"><i class="fa fa-check"></i><b>8.7</b> Summary</a></li>
<li class="chapter" data-level="8.8" data-path="Imputation.html"><a href="Imputation.html#conclusions-and-comments"><i class="fa fa-check"></i><b>8.8</b> Conclusions and comments</a></li>
</ul></li>
<li class="chapter" data-level="9" data-path="Association-mapping.html"><a href="Association-mapping.html"><i class="fa fa-check"></i><b>9</b> Association Mapping.</a>
<ul>
<li class="chapter" data-level="9.1" data-path="Association-mapping.html"><a href="Association-mapping.html#introduction-1"><i class="fa fa-check"></i><b>9.1</b> Introduction</a></li>
<li class="chapter" data-level="9.2" data-path="Association-mapping.html"><a href="Association-mapping.html#preparing-the-data-set"><i class="fa fa-check"></i><b>9.2</b> Preparing the data set</a></li>
<li class="chapter" data-level="9.3" data-path="Association-mapping.html"><a href="Association-mapping.html#running-gwaspoly"><i class="fa fa-check"></i><b>9.3</b> Running GWASpoly</a></li>
<li class="chapter" data-level="9.4" data-path="Association-mapping.html"><a href="Association-mapping.html#the-exercise-2"><i class="fa fa-check"></i><b>9.4</b> The exercise</a>
<ul>
<li class="chapter" data-level="9.4.1" data-path="Association-mapping.html"><a href="Association-mapping.html#calculation-and-comparison-of-kinship"><i class="fa fa-check"></i><b>9.4.1</b> Calculation and comparison of kinship</a></li>
<li class="chapter" data-level="9.4.2" data-path="Association-mapping.html"><a href="Association-mapping.html#gwas"><i class="fa fa-check"></i><b>9.4.2</b> GWAS</a></li>
<li class="chapter" data-level="9.4.3" data-path="Association-mapping.html"><a href="Association-mapping.html#analysis-with-covariates"><i class="fa fa-check"></i><b>9.4.3</b> Analysis with covariates</a></li>
<li class="chapter" data-level="9.4.4" data-path="Association-mapping.html"><a href="Association-mapping.html#accounting-for-population-structure"><i class="fa fa-check"></i><b>9.4.4</b> Accounting for population structure</a></li>
<li class="chapter" data-level="9.4.5" data-path="Association-mapping.html"><a href="Association-mapping.html#further-analysis"><i class="fa fa-check"></i><b>9.4.5</b> Further analysis</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="10" data-path="Mapping-Markers.html"><a href="Mapping-Markers.html"><i class="fa fa-check"></i><b>10</b> Mapping Markers and Quantitative Trait Loci.</a>
<ul>
<li class="chapter" data-level="10.1" data-path="Mapping-Markers.html"><a href="Mapping-Markers.html#introduction-2"><i class="fa fa-check"></i><b>10.1</b> Introduction</a></li>
<li class="chapter" data-level="10.2" data-path="Mapping-Markers.html"><a href="Mapping-Markers.html#constructing-a-genetic-map"><i class="fa fa-check"></i><b>10.2</b> Constructing a genetic map</a></li>
<li class="chapter" data-level="10.3" data-path="Mapping-Markers.html"><a href="Mapping-Markers.html#formatting-and-reading-in-the-data"><i class="fa fa-check"></i><b>10.3</b> Formatting and reading in the data</a></li>
<li class="chapter" data-level="10.4" data-path="Mapping-Markers.html"><a href="Mapping-Markers.html#a-quick-map"><i class="fa fa-check"></i><b>10.4</b> A quick map!</a></li>
<li class="chapter" data-level="10.5" data-path="Mapping-Markers.html"><a href="Mapping-Markers.html#estimating-recombination-fractions"><i class="fa fa-check"></i><b>10.5</b> Estimating recombination fractions</a></li>
<li class="chapter" data-level="10.6" data-path="Mapping-Markers.html"><a href="Mapping-Markers.html#forming-linkage-groups"><i class="fa fa-check"></i><b>10.6</b> Forming Linkage groups</a></li>
<li class="chapter" data-level="10.7" data-path="Mapping-Markers.html"><a href="Mapping-Markers.html#marker-ordering"><i class="fa fa-check"></i><b>10.7</b> Marker ordering</a></li>
<li class="chapter" data-level="10.8" data-path="Mapping-Markers.html"><a href="Mapping-Markers.html#improving-the-map"><i class="fa fa-check"></i><b>10.8</b> Improving the map</a></li>
<li class="chapter" data-level="10.9" data-path="Mapping-Markers.html"><a href="Mapping-Markers.html#quality-control-commands-for-marker-qc-and-mapping"><i class="fa fa-check"></i><b>10.9</b> Quality Control commands for marker QC and mapping</a></li>
<li class="chapter" data-level="10.10" data-path="Mapping-Markers.html"><a href="Mapping-Markers.html#asmap"><i class="fa fa-check"></i><b>10.10</b> ASMap</a></li>
</ul></li>
<li class="chapter" data-level="11" data-path="Data-Visualization.html"><a href="Data-Visualization.html"><i class="fa fa-check"></i><b>11</b> Data Visualization.</a>
<ul>
<li class="chapter" data-level="11.1" data-path="Data-Visualization.html"><a href="Data-Visualization.html#flapjack"><i class="fa fa-check"></i><b>11.1</b> Flapjack</a>
<ul>
<li class="chapter" data-level="11.1.1" data-path="Data-Visualization.html"><a href="Data-Visualization.html#genotype-data-.geno-file."><i class="fa fa-check"></i><b>11.1.1</b> Genotype data: “.geno” file.</a></li>
<li class="chapter" data-level="11.1.2" data-path="Data-Visualization.html"><a href="Data-Visualization.html#map-data-.map-file."><i class="fa fa-check"></i><b>11.1.2</b> Map data : “.MAP” file.</a></li>
<li class="chapter" data-level="11.1.3" data-path="Data-Visualization.html"><a href="Data-Visualization.html#trait-data"><i class="fa fa-check"></i><b>11.1.3</b> Trait data</a></li>
<li class="chapter" data-level="11.1.4" data-path="Data-Visualization.html"><a href="Data-Visualization.html#the-exercise-3"><i class="fa fa-check"></i><b>11.1.4</b> The Exercise</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="12" data-path="selection-theory.html"><a href="selection-theory.html"><i class="fa fa-check"></i><b>12</b> Selection theory and breeding program design.</a></li>
<li class="chapter" data-level="13" data-path="Genomic-prediction.html"><a href="Genomic-prediction.html"><i class="fa fa-check"></i><b>13</b> Genomic prediction.</a>
<ul>
<li class="chapter" data-level="13.1" data-path="Genomic-prediction.html"><a href="Genomic-prediction.html#ridge-regression-and-genomic-prediction-in-excel"><i class="fa fa-check"></i><b>13.1</b> Ridge regression and Genomic Prediction in Excel</a>
<ul>
<li class="chapter" data-level="13.1.1" data-path="Genomic-prediction.html"><a href="Genomic-prediction.html#introduction-3"><i class="fa fa-check"></i><b>13.1.1</b> Introduction</a></li>
<li class="chapter" data-level="13.1.2" data-path="Genomic-prediction.html"><a href="Genomic-prediction.html#the-exercise-4"><i class="fa fa-check"></i><b>13.1.2</b> The Exercise</a></li>
</ul></li>
<li class="chapter" data-level="13.2" data-path="Genomic-prediction.html"><a href="Genomic-prediction.html#genomic-prediction-with-rrblup"><i class="fa fa-check"></i><b>13.2</b> Genomic Prediction with rrBLUP</a>
<ul>
<li class="chapter" data-level="13.2.1" data-path="Genomic-prediction.html"><a href="Genomic-prediction.html#introduction-4"><i class="fa fa-check"></i><b>13.2.1</b> Introduction</a></li>
<li class="chapter" data-level="13.2.2" data-path="Genomic-prediction.html"><a href="Genomic-prediction.html#the-dataset"><i class="fa fa-check"></i><b>13.2.2</b> The dataset</a></li>
<li class="chapter" data-level="13.2.3" data-path="Genomic-prediction.html"><a href="Genomic-prediction.html#loading-and-preparing-the-data"><i class="fa fa-check"></i><b>13.2.3</b> Loading and preparing the data</a></li>
<li class="chapter" data-level="13.2.4" data-path="Genomic-prediction.html"><a href="Genomic-prediction.html#examining-the-data-structure"><i class="fa fa-check"></i><b>13.2.4</b> Examining the data structure</a></li>
<li class="chapter" data-level="13.2.5" data-path="Genomic-prediction.html"><a href="Genomic-prediction.html#ordinary-least-squares"><i class="fa fa-check"></i><b>13.2.5</b> Ordinary least squares</a></li>
<li class="chapter" data-level="13.2.6" data-path="Genomic-prediction.html"><a href="Genomic-prediction.html#rrblup"><i class="fa fa-check"></i><b>13.2.6</b> rrBLUP</a></li>
<li class="chapter" data-level="13.2.7" data-path="Genomic-prediction.html"><a href="Genomic-prediction.html#prediction-forward-in-time"><i class="fa fa-check"></i><b>13.2.7</b> Prediction forward in time</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="14" data-path="Breeding-simulations.html"><a href="Breeding-simulations.html"><i class="fa fa-check"></i><b>14</b> Breeding simulations.</a></li>
<li class="chapter" data-level="" data-path="references.html"><a href="references.html"><i class="fa fa-check"></i>References</a></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Quantitative Methods in Plant Breeding: The Digital Tutorials.</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="R-and-stats" class="section level1" number="3">
<h1><span class="header-section-number">Chapter 3</span> R and Rstudio: a general introduction with statistics.</h1>
<div id="introduction-section-on-r" class="section level2" number="3.1">
<h2><span class="header-section-number">3.1</span> Introduction section on R</h2>
<p>R is free software. It can be used as an overpowered calculator, to carry out many standard statistical analyses, and to develop applications carrying out complex analyses in specific subject areas. Many of these applications and much other useful information are freely available from the Comprehensive R Archive network (CRAN) <a href="http://cran.r-project.org">website</a>. In particular, the guide “An Introduction to R” is available under the documentation section and can be used to supplement the outline given here. Another excellent source is “Introductory Statistics with R” by Peter Dalgaard.</p>
<p>R is just one among several excellent statistical packages. Its major strength is that it is free and this has led to an ever expanding user base and to the development of more and better applications. R is commonly used in many commercial and academic establishments: knowledge of R is a useful and transferable skill to acquire.</p>
<p>For many users, the major disadvantage of R is that commands need to be typed at a prompt, rather than selected from a menu. However, most commands are short, and as we shall see, much repetitive typing can be avoided. Over the last few years, RStudio, an interface to R, has become so popular that it is now the standard way by which users are introduced to the package. RStudio retains the look and feel of R: it still requires commands to be typed but is more accessible. In this tutorial, we shall use Rstudio, but if you can use R you can use Rstudio, and vice-versa.</p>
<p>The intention of this guide is twofold. Firstly, it is to help the beginner to get started. It will show you how to get data into RStudio, carry out basic analyses, produce graphs, and save results. It is hoped that this will allow users to dip directly into more complete guides as required. This guide is a long way off being thorough and many of the definitions provided here are not perfectly precise. The hope is that this guide will allow you to rapidly carry out useful analyses of importance in plant breeding and provide a base from which additional knowledge can be acquired without too much extra effort.</p>
<p>Secondly, the guide is intended to act as revision of some basic statistical methods. For this purpose we shall be studying some properties of a panel of European winter wheat lines; the TriteceaeGenome panel which was collected as part of a European collaborative project in association mapping, the results of which have been described by <span class="citation">(<a href="references.html#ref-Bentley2014" role="doc-biblioref">Bentley et al. 2014</a>)</span>. A copy of the paper has been supplied in the references folder on the course <a href="https://tallywright.github.io/">website</a>. We shall use this dataset for various analyses throughout the course.</p>
<p>The subset of the TG panel data which we shall use today are in “TG data for day 1.xlsx.” Open the file and have a look and I shall describe the data. We can discuss what sorts of hypotheses we would be interested in testing with these data.</p>
<div id="getting-the-data-ready" class="section level3" number="3.1.1">
<h3><span class="header-section-number">3.1.1</span> Getting the data ready</h3>
<p>The first step of this tutorial will be to open this excel sheet (in excel) and save a copy of it as a comma-separated values (CSV) file. This can be completed in excel via the following steps:</p>
<ol style="list-style-type: decimal">
<li><p>Opening the file called: “TG data for day 1.xlsx”</p></li>
<li><p>Going to ‘save as’ and then changing the file type to: CSV (Comma delimited) (.csv).</p></li>
<li><p>Save this file in the same folder as the “.xlsx” file. You can keep the same name, as you are creating a different file type. You can only save a single excel sheet as a CSV file and that file won’t contain any formulas or formatting, only values. Now that file is converted to a CSV file, we can move on to using RStudio and come back to it later.</p></li>
</ol>

<div class="rmdcaution">
You will need to complete the above section before moving on to the rest of the tutorial.
</div>
</div>
</div>
<div id="the-rstudio-layout-workspace-and-environment." class="section level2" number="3.2">
<h2><span class="header-section-number">3.2</span> The Rstudio layout, workspace and environment.</h2>
<p>Next open Rstudio which should be installed on your computer. Typically Rstudio will open with three windows (possibly four if your script from your last session opens automatically).</p>
<p>If you only have three windows, open a new script by pressing: <em>Ctl + Shift + N</em></p>
<p>You should now see four screens:</p>
<ol style="list-style-type: decimal">
<li><p>A console or command window. This is on the bottom left in the screen-shot. In this window we type commands and the result of those commands is listed. If you were to work in R, rather than R studio, this is the only window you would see.</p></li>
<li><p>A script editing window: for writing scripts. This is shown in the top left. Commands can also be run from here.</p></li>
<li><p>An environment / workspace and history window (top right). Within the environment window you will see user defined objects (data frames, lists, vectors ect.) This workspace can be saved for later use.</p></li>
<li><p>A window for files, plots, packages and help (bottom right).</p></li>
</ol>
<p>You can re-size these windows, move about between them and essentially do everything that is possible in an ordinary R session, including developing your own programs and scripts. The idea is that this is easier and more convenient than working in R directly.</p>
<div class="figure">
<img src="images/Structure_of_Rstudio.png" alt="" />
<p class="caption">The structure if RStudio</p>
</div>
<div id="working-directory" class="section level3" number="3.2.1">
<h3><span class="header-section-number">3.2.1</span> Working directory</h3>
<p>The current working directory can be shown by the <code>getwd()</code> function. This is ultimately the folder on your computer that your current R session is working from. You can type this in the script window and then press <strong>Ctrl + Return</strong> to run, the cursor needs to be on the line you want to run. In the working directory, files will be loaded when called automatically and files will be automatically saved to this folder. You can change your working directory manually using the function <code>setwd()</code>. Use the folder path to prompt where the working directory is set. However, sometimes it’s easiest just to click on the tab at the top of the window and use the route of the drop-down menu: <em>Session / Set Working Directory / Choose Directory</em>. Which works just as well.</p>

<div class="rmdquiz">
<p><strong>1</strong>: You want to be working from a folder where the data you will be using is saved in. You could create a new folder for this tutorial and then add the CSV file “TG data for day 1.csv” that you created earlier. You can do this outside of RStudio.</p>
Next, with RStudio, change the working directory to the folder containing the CSV file “TG data for day 1.csv” that you created earlier. Then you can run the line <code>list.files()</code> in the script window to ensure that the file is present in your current working directory.
</div>
</div>
</div>
<div id="basic-r-syntx" class="section level2" number="3.3">
<h2><span class="header-section-number">3.3</span> Basic R Syntx</h2>
<p>Write your commands into the <em>script editing window</em>, to then run a command you can leave the cursor on the line you wish to run and press <em>Ctrl + Return</em>. You can also press <em>Run</em> at the top right of the window. If you want to run multiple lines at once, highlight the desired lines and click <em>Run</em> or press <em>Ctrl + Return</em>.</p>
<p>When you run the line in the <em>script editing window</em> you will see your command appear in the console and the result of what you have run.</p>
<p>For example entering 9+1 will return:</p>
<div class="sourceCode" id="cb1"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1-1"><a href="R-and-stats.html#cb1-1" aria-hidden="true" tabindex="-1"></a><span class="dv">9</span><span class="sc">+</span><span class="dv">1</span></span></code></pre></div>
<pre><code>## [1] 10</code></pre>
<p>You are now fully trained to use R as a calculator! The syntax is essentially the same as in Excel. Thus <code>^2</code> will square numbers, <code>sqrt()</code> will take the square root and so on. For example:</p>
<div class="sourceCode" id="cb3"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb3-1"><a href="R-and-stats.html#cb3-1" aria-hidden="true" tabindex="-1"></a><span class="fu">sqrt</span>(<span class="dv">2</span>)</span></code></pre></div>
<pre><code>## [1] 1.414214</code></pre>
<p>Returns the square root of 2. The <code>[1]</code> is just a line number.</p>
<p>Note that R is case sensitive, running <code>LOG10(2)</code>. Will produce an error.</p>
<p>Whereas:</p>
<div class="sourceCode" id="cb5"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb5-1"><a href="R-and-stats.html#cb5-1" aria-hidden="true" tabindex="-1"></a><span class="fu">log10</span>((<span class="dv">9</span><span class="sc">+</span><span class="dv">1</span>)<span class="sc">^</span><span class="dv">2</span>)</span></code></pre></div>
<pre><code>## [1] 2</code></pre>
<p>Does not produce any errors.</p>
<p>If you make a mistake, you can easily edit the line that contains the error in the <em>script editing window</em>. Some of the new versions of Rstudio will flag up mistakes in a line in the <em>script editing window</em>.</p>
<p>Alternatively if you are working directly in the <em>console or command window</em> and you make a mistake, it can be corrected by tapping the up arrow on the keyboard to bring back the last typed command, then moving along the command line using the left and right arrows, deleting characters using the backspace or delete key, inserting correct numbers or text and hitting the carriage return key. Multiple hits of the up arrow will bring back successively earlier commands. However, an easier way of finding and re-entering commands used previously is to click on the “history” tab at the top right. This shows the previous commands you have run. If you select one and press <em>Ret</em>, then that command will be passed to the console window. Press <em>Ret</em> again and the command will be executed. It is even possible to select several non-contiguous commands in the history window, pass them to the console and execute them all as group.</p>
<p>To avoid this slightly laborious approach, just write and run your scripts through the <em>script editing window</em>. This way you will have a complete record of your code preserved.</p>

<div class="rmdnote">
Note also that it is possible to cut from windows outside RStudio and paste into it. So R commands listed in another document, for example this one, can be copied and pasted into the <em>console or command window</em> or the <em>script editing window</em> and will run (after hitting carriage return). You will find this useful during the course.
</div>
<div id="brackets" class="section level3" number="3.3.1">
<h3><span class="header-section-number">3.3.1</span> Brackets</h3>
<p>We have seen brackets used above in simple formulae - just as in Excel formula. However, brackets are used more extensively in R: all commands include brackets. For example:</p>
<p>Running <code>quit()</code> will end your R session. However:</p>
<div class="sourceCode" id="cb7"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb7-1"><a href="R-and-stats.html#cb7-1" aria-hidden="true" tabindex="-1"></a>quit</span></code></pre></div>
<pre><code>## function (save = &quot;default&quot;, status = 0, runLast = TRUE) 
## .Internal(quit(save, status, runLast))
## &lt;bytecode: 0x000000001484c680&gt;
## &lt;environment: namespace:base&gt;</code></pre>
<p>outputs the above. Note how the use of the brackets changes the outcome. The above is the section of computer code which is run when you type <code>quit()</code>. Unless you are an enthusiast, it is not necessary to know what this means.</p>
</div>
<div id="equals-symbols" class="section level3" number="3.3.2">
<h3><span class="header-section-number">3.3.2</span> Equals symbols</h3>
<p>R has four sets of symbols all of which loosely mean “equals” but which have differences in use. This can take some getting used to.</p>
<p>The most commonly used set is</p>
<p><code>&lt;-</code></p>
<p>or sometimes</p>
<p><code>-&gt;</code></p>
<p>These symbols are typed using the dash (the minus sign) and the left or right arrow. They are used to assign results from one side of the arrow to the other. For example:</p>
<div class="sourceCode" id="cb9"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb9-1"><a href="R-and-stats.html#cb9-1" aria-hidden="true" tabindex="-1"></a>my_first_result <span class="ot">&lt;-</span> <span class="fu">sqrt</span>(<span class="fu">log10</span>((<span class="dv">9</span><span class="sc">+</span><span class="dv">1</span>)<span class="sc">^</span><span class="dv">2</span>))</span></code></pre></div>
<p>Won’t return a result and just returns the prompt in the command window. But then typing:</p>
<div class="sourceCode" id="cb10"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb10-1"><a href="R-and-stats.html#cb10-1" aria-hidden="true" tabindex="-1"></a>my_first_result</span></code></pre></div>
<pre><code>## [1] 1.414214</code></pre>
<p>Returns the desired result. We have created a variable my_first_result and assigned the result of our calculation to it, using <code>&lt;-</code>. The same result would be achieved by: <code>sqrt(log10((9+1)^2))-&gt; my_first_result</code></p>
<p>The entry stored in <code>my_first_result</code> is now available for additional manipulation. For example:</p>
<div class="sourceCode" id="cb12"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb12-1"><a href="R-and-stats.html#cb12-1" aria-hidden="true" tabindex="-1"></a>my_first_result<span class="sc">^</span><span class="dv">2</span></span></code></pre></div>
<pre><code>## [1] 2</code></pre>
<p>There are historical reasons why <code>&lt;-</code> was introduced for this purpose, but R also allows the use of “=” in addition to <code>&lt;-</code> in assigning values to variables. <code>x&lt;-3</code> and <code>x=3</code> both return the value of 3 if you run <code>x</code>. Note however that <code>2=x</code> will return an error. Because of its lack of ambiguity with the other use of <code>=</code> (see below), I prefer to stick to <code>&lt;-</code> and <code>-&gt;</code> in assigning values, though this can seem artificial and unnecessarily geeky at first.</p>
<p>The second equals symbol is the tilde <code>~</code> (above the # on my keyboard). This is used in R in statistical analysis to distinguish between x and y variables; that is to say between what is being analysed (e.g. the phenotype) and what factors and variates it is being analysed with (eg marker data and environmental factors or variates). So for example:</p>
<p><code>P~G+E</code></p>
<p>is R syntax to state that a variate P is to be explained by two variates or factors G and E. If you try this in R now nothing will happen. The exact context in which this syntax is used will be given later. For now, note that</p>
<p><code>P~G</code></p>
<p>would represents linear regression of P on G Interactions can also be included:</p>
<p><code>P~G+E+G:E</code></p>
<p>includes an interaction term between G and E (represented by G:E). This can be abbreviated to:</p>
<p><code>P~G*E</code></p>
<p>Yet more complex models, for example with nested factors, can be described by including bracketed terms.</p>
<p>The third equals symbol is <code>=</code>. This is most often used within commands to provide information about specific parameters. For example we shall come across:</p>
<p><code>xlab= "what you put here is used to label the x axis in a graph"</code></p>
<p>used in commands to generate graphs.</p>
<p>The final equals description is <code>==</code>. This truly means “is equal to,” and is used to test relationships: is A equal to B is written as</p>
<p><code>A==B</code></p>
<p>Try this sequence of commands</p>
<p><code>A&lt;-2</code></p>
<p><code>B&lt;-3</code></p>
<p><code>A==B</code></p>
<p>Related to == are:</p>
<p><code>!=</code> not equal to</p>
<p><code>&gt;</code> greater than</p>
<p><code>&lt;</code> less than</p>
<p><code>&gt;=</code> greater than or equal to</p>
<p><code>&lt;=</code> less than or equal to.</p>
<p>These, together with <code>==</code> itself, are entered here for completeness. Try <code>A!=B</code>.</p>
</div>
<div id="continuation-character." class="section level3" number="3.3.3">
<h3><span class="header-section-number">3.3.3</span> Continuation character.</h3>
<p>When the <code>+</code> symbol is shown rather than the <code>&gt;</code> symbol in the console window, it means that your previous command was ‘syntactically incomplete.’ If an R command is incomplete when the carriage return is depressed, you are prompted with a <code>+</code> to continue the command on the next line. Typing <code>sqrt(log10</code> returns: <code>+</code>.</p>
<p>The rest of the command can be entered after the <code>+</code>:</p>
<p><code>sqrt(log10</code></p>
<p><code>+ ((9+1)^2))</code></p>
<p><code>[1] 1.414214</code></p>

<div class="rmdnote">
You might find it hard to generate these errors in Rstudio, as it helpfully inserts a <code>)</code> as soon as you type a <code>(</code>.
</div>
</div>
<div id="text-and-numbers" class="section level3" number="3.3.4">
<h3><span class="header-section-number">3.3.4</span> Text and numbers</h3>
<p>Text is distinguished from numbers and from R commands by the use of quotes. Either single quotes ‘text’ or double quotes “text” will do, but the quotes must match. Again, RStudio will help you with this.</p>
</div>
<div id="summary-of-syntax" class="section level3" number="3.3.5">
<h3><span class="header-section-number">3.3.5</span> Summary of syntax</h3>
<p>Arithmetic: <code>+ - / * ^ ( )</code> just like Excel</p>
<p>Commands use brackets: <code>quit()</code> is correct. <code>quit</code> is wrong.</p>
<p>Equals:</p>
<ul>
<li><code>&lt;-</code> stores a result</li>
<li><code>~</code> is described by</li>
<li><code>=</code> options in commands</li>
<li><code>==</code> logical equivalence</li>
</ul>
<p>For text use quotes: <code>"hello"</code> or <code>'hello'</code>.</p>
<p>This introduction should provide sufficient syntax to get you going. Additional syntax is introduced, as required, in the discussion of specific commands and operations in the remainder of this document.</p>

<div class="rmdquiz">
<p><strong>2</strong> Assign the result of <code>(9.75 * 80) - 34</code> to an object called: <code>my_second_result</code>. Then print out <code>my_second_result</code>. Lastly divide <code>my_second_result</code> by 2.</p>
Assign the word <code>harvester</code> to an object called: <code>Combine</code>. Then run <code>Combine</code>.
</div>
</div>
</div>
<div id="reading-in-data" class="section level2" number="3.4">
<h2><span class="header-section-number">3.4</span> Reading in data</h2>
<p>There are many ways of entering the data you wish to analyse. For example, R can read data from the clipboard, so you can copy data directly from Excel. It will read tab separated files (files in which different columns of data are separated by tabs, and it will read comma separated files (csv files). There are more advanced methods to interface directly with databases, to capture data from the web, and so on. There are also several different formats for reading data in.</p>
<p>Earlier we changed our working directory to the location of our TriticeaeGenome data files, which we saved as a CSV file. If you missed this, do it now:</p>
<p>From the top of the Rstudio window select:</p>
<p><em>Session / Set Working Directory / Choose Directory</em></p>
<p>Then browse to locate your directory (choose the folder where you saved the data as a CSV file). When we exit RStudio, your work will be saved here (as long as you remember to actually save it), making it easier to return later and pick up where you left of. We shall explain this later.</p>
<p>If you follow the above process, in the Console window, you will see echoed the command that you could have typed to change directories. For me, this was:</p>
<p><code>setwd("C:/Users/x991625/OneDrive - NIAB/a). NIAB WORK/QMPB 2021/R_bookdown_local/bookdown-demo-master")</code></p>

<div class="rmdtip">
If you copy this line to the top of your <em>script editing window</em> and if you had to re-run the script this line would set the working directory for you. It can also serve as a reminder on where your files are saved. There may be times when you wish to change directories as part of a larger sequence of commands that you want to run, in which case you would use the typed form. Note - A sequence of commands is commonly called a “script”: in essence it is a small computer program.
</div>
<p>We may not cover Rprojects over these days but an alternative available within RStudio, but not R, is to create a project. From the menu at the top, select:</p>
<p><em>File / New Project / Existing Directory /</em></p>
<p>Then browse to the directory you want to work in and <em>Create project</em></p>
<p>Later, when you quit RStudio, you will find a file with the extension .Rproj has been created in the selected directory. If you click on that, then RStudio will open with all your previous work saved.</p>
<p>Now we have our working directory set we can read in the data. Personally I always think it’s safest to read in the data as a comma-separated values (CSV) file. Earlier I asked you to save the data as a CSV file and that file should be in our working directory. We can check what files are in our working directory by running:</p>
<div class="sourceCode" id="cb14"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb14-1"><a href="R-and-stats.html#cb14-1" aria-hidden="true" tabindex="-1"></a><span class="fu">list.files</span>()</span></code></pre></div>
<pre><code>##  [1] &quot;_book&quot;                                              
##  [2] &quot;_bookdown.yml&quot;                                      
##  [3] &quot;_bookdown_files&quot;                                    
##  [4] &quot;_build.sh&quot;                                          
##  [5] &quot;_deploy.sh&quot;                                         
##  [6] &quot;_output.yml&quot;                                        
##  [7] &quot;_render.R&quot;                                          
##  [8] &quot;02-Acknow.Rmd&quot;                                      
##  [9] &quot;03-R_and_stats.Rmd&quot;                                 
## [10] &quot;04-Trial_design.Rmd&quot;                                
## [11] &quot;05-Trial_analysis.Rmd&quot;                              
## [12] &quot;06-Cross_site_analysis.Rmd&quot;                         
## [13] &quot;07-Population_genetics.Rmd&quot;                         
## [14] &quot;08-Imputation.Rmd&quot;                                  
## [15] &quot;09-Assocation_mapping.Rmd&quot;                          
## [16] &quot;10-Mapping_Markers_and_Quantitative_Trait_Loci.Rmd&quot; 
## [17] &quot;11-Data_visualization.Rmd&quot;                          
## [18] &quot;12-Selection_theory_and_breeding_program_design.Rmd&quot;
## [19] &quot;13-Genomic_prediction.Rmd&quot;                          
## [20] &quot;14-Breeding_simulations.Rmd&quot;                        
## [21] &quot;15-references.Rmd&quot;                                  
## [22] &quot;aug_blocks.csv&quot;                                     
## [23] &quot;AxCmapped.csv&quot;                                      
## [24] &quot;book.bib&quot;                                           
## [25] &quot;bookdown-demo.aux&quot;                                  
## [26] &quot;bookdown-demo.bbl&quot;                                  
## [27] &quot;bookdown-demo.blg&quot;                                  
## [28] &quot;bookdown-demo.idx&quot;                                  
## [29] &quot;bookdown-demo.ilg&quot;                                  
## [30] &quot;bookdown-demo.ind&quot;                                  
## [31] &quot;bookdown-demo.log&quot;                                  
## [32] &quot;bookdown-demo.out&quot;                                  
## [33] &quot;bookdown-demo.pdf&quot;                                  
## [34] &quot;bookdown-demo.Rmd&quot;                                  
## [35] &quot;bookdown-demo.Rproj&quot;                                
## [36] &quot;bookdown-demo.tex&quot;                                  
## [37] &quot;bookdown-demo.toc&quot;                                  
## [38] &quot;bookdown-demo_files&quot;                                
## [39] &quot;cover.jpg&quot;                                          
## [40] &quot;data&quot;                                               
## [41] &quot;delete.csv&quot;                                         
## [42] &quot;DESCRIPTION&quot;                                        
## [43] &quot;Dockerfile&quot;                                         
## [44] &quot;images&quot;                                             
## [45] &quot;index.Rmd&quot;                                          
## [46] &quot;journals.bib&quot;                                       
## [47] &quot;latex&quot;                                              
## [48] &quot;lattice_25.csv&quot;                                     
## [49] &quot;lattice_9.csv&quot;                                      
## [50] &quot;LICENSE&quot;                                            
## [51] &quot;now.json&quot;                                           
## [52] &quot;packages.bib&quot;                                       
## [53] &quot;preamble.tex&quot;                                       
## [54] &quot;README.md&quot;                                          
## [55] &quot;style.css&quot;                                          
## [56] &quot;tables&quot;                                             
## [57] &quot;TG data for day 1.csv&quot;                              
## [58] &quot;this will be the file name.csv&quot;                     
## [59] &quot;toc.css&quot;</code></pre>
<p>In my example there are a number of files in my working directory. That’s because my working directory contains all the files used to build this book. We are interested in reading in the file: <code>TG data for day 1.csv</code>. To read the data in we would use a command like:</p>
<div class="sourceCode" id="cb16"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb16-1"><a href="R-and-stats.html#cb16-1" aria-hidden="true" tabindex="-1"></a>TG_data_for_day_1 <span class="ot">&lt;-</span> <span class="fu">read.csv</span>(<span class="st">&quot;TG data for day 1.csv&quot;</span>,<span class="at">header=</span>T)</span></code></pre></div>
<p>Looks complicated? It’s not really. We are only using the function <code>read.csv</code> which takes the the name of the csv file you want to read in as the first argument. Note this name has to be in speech marks and requires the .csv at the end. The second argument is <code>header=T</code>, this ensures that the top row in the excel file is read in as the column headers in R. The other part of the command was to assign the data to an R object called <code>TG_data_for_day_1</code>.</p>
<p>A key point is to consider how any program you are using handles <code>NA</code> (missing data). For R to treat excel cells with no data as <code>NA</code> those cells should contain <code>NA</code> and not be left blank. This was done for you with this data. However, that won’t always be the case.</p>
<p>It’s always good to make sure that the data was read in correctly and there are no obvious issues. This can be done with some simple functions. I always like to make sure the headers (column headers were read in correctly):</p>
<div class="sourceCode" id="cb17"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb17-1"><a href="R-and-stats.html#cb17-1" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(TG_data_for_day_1)</span></code></pre></div>
<pre><code>##  [1] &quot;variety&quot;     &quot;year&quot;        &quot;ORIGIN&quot;      &quot;Rht2&quot;        &quot;PpdD1&quot;      
##  [6] &quot;yield&quot;       &quot;CALLOW_2011&quot; &quot;FRANCE_2010&quot; &quot;FRANCE_2011&quot; &quot;LGE_2010&quot;   
## [11] &quot;LGE_2011&quot;    &quot;NIAB_2011&quot;   &quot;FT&quot;          &quot;HT&quot;          &quot;AWNS&quot;</code></pre>
<p>Looks good, we have 15 distinct headers. One of the many strengths of R is the help documentation. Throughout this tutorial and your own research, if you come across a function you are unsure about, or want to know what arguments the function uses or want some examples of using the function, you can just enter the function name into <strong>help()</strong>. For example try running:</p>
<p><code>help("names")</code></p>
<p>This opens a help window in the Rstudio pane on your right, where you should see all the required information.</p>
<p>We should also check the dimensions of the data, which can often indicate if there’s an issue. From the excel sheet we have 377 rows and 15 columns. We check the dimensions of our object in R like this:</p>
<div class="sourceCode" id="cb19"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb19-1"><a href="R-and-stats.html#cb19-1" aria-hidden="true" tabindex="-1"></a><span class="fu">dim</span>(TG_data_for_day_1)</span></code></pre></div>
<pre><code>## [1] 376  15</code></pre>
<p>Rows are shown first, followed by columns.</p>

<div class="rmdcaution">
The rows are one less than expected. Why? Because we read in the first row as a column headers using <code>headers=T</code> in the <code>read.csv()</code> function.
</div>
<p>Additionally, you should also see the data in the <em>Environment Window</em> in the top right pane of RStudio. You can see the number of rows (obs.) and columns (variables) there too. You can click on this (or run <code>View(TG_data_for_day_1)</code>) to view the data displayed in a spreadsheet format in the editing window. You can apply filters to the various columns to see subsets of the data if desired. Don’t do this if you are working with large datasets! It will freeze RStudio or takes a very long time to load. R is not designed for viewing large data files in this way.</p>
<p>We now have our data ready and waiting for analysis. To display a variable (a column), say yield, we should type <code>TG_data_for_day_1$yield</code>. However, this would print out 376 numbers and to save paper, we can only request the first five values using <code>[1:5]</code> after our command, like so:</p>
<div class="sourceCode" id="cb21"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb21-1"><a href="R-and-stats.html#cb21-1" aria-hidden="true" tabindex="-1"></a>TG_data_for_day_1<span class="sc">$</span>yield [<span class="dv">1</span><span class="sc">:</span><span class="dv">5</span>]</span></code></pre></div>
<pre><code>## [1] 103.23  88.55  90.76  82.24  96.66</code></pre>
<p>Notice the use of the <code>$</code> to access a variable from your data frame.</p>
<p>Writing out the name of the data object each time is tedious. To speed things up, we can <code>attach</code> our data:</p>
<div class="sourceCode" id="cb23"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb23-1"><a href="R-and-stats.html#cb23-1" aria-hidden="true" tabindex="-1"></a><span class="fu">attach</span>(TG_data_for_day_1)</span></code></pre></div>
<p>R will now assume that when you type yield, you mean <code>TG_data_for_day_1$yield</code>.</p>

<div class="rmdwarning">
This can sometimes be dangerous: if you have another dataset you are working with at the same time which also has a variable called yield, then it can be confusing to know which you are working with.
</div>
<p>To stop working by default with the TG_data_for_day_1 dataset you could type: <code>detach(TG_data_for_day_1)</code>. If you do this now, don’t forget to attach it again!</p>
<p>We could had read our data in using <code>read.table()</code>, this is how we would read in data saved as a .txt file. We will use this function throughout the course. It is very similar to <code>read.csv</code>. Use the help documentation in R, to the read further information on the <code>read.table()</code> command, i.e. run: <code>help(read.table)</code>.</p>
<p>You can also use the <code>read.table()</code> function to read from your clipboard. This is achieved by copying cells in excel (your data table) and then running:</p>
<p><code>read.table(file = "clipboard", header=TRUE)</code></p>
<p>Note that this is not a reproducible method and can lead to some downstream issues if you forget where your data came from.</p>

<div class="rmdquiz">
<p><strong>3</strong> You can also read data from a .txt file using the function <code>read.table()</code>. The function will take the same arguments as <code>read.csv()</code>. Save the data in excel as a tab delimited file (.txt) and then read this file in using <code>read.table()</code>, save it as: my_data2.</p>
Use the functions I showed above to ensure <code>read.table()</code> was read in properly.
</div>
</div>
<div id="understanding-data-types-and-r-objects" class="section level2" number="3.5">
<h2><span class="header-section-number">3.5</span> Understanding data types and R objects</h2>
<p>Now that we have imported our data into R, it becomes important to understand some of the basic data types.</p>
<div class="sourceCode" id="cb24"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb24-1"><a href="R-and-stats.html#cb24-1" aria-hidden="true" tabindex="-1"></a><span class="fu">class</span>(TG_data_for_day_1)</span></code></pre></div>
<pre><code>## [1] &quot;data.frame&quot;</code></pre>
<p>If we run the function <code>class()</code>, R will tell us what type of data object we are working on. The data.frame is a common form of data for plotting, analysis and statistics. The <code>read.csv</code> function will form a data.frame automatically. Within a data.frame the columns are vectors. A vector contains a series of numbers (or characters). A vector is the most common and simple data form of R. You can assign a vector manually, using the function <code>c()</code>. This is a generic function which combines its arguments (‘c’ is short for ‘concatenate’). For example:</p>
<div class="sourceCode" id="cb26"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb26-1"><a href="R-and-stats.html#cb26-1" aria-hidden="true" tabindex="-1"></a>my_first_vector <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="dv">167</span>,<span class="dv">212</span>,<span class="dv">358</span>,<span class="dv">973</span>,<span class="dv">214</span>,<span class="dv">558</span>)</span>
<span id="cb26-2"><a href="R-and-stats.html#cb26-2" aria-hidden="true" tabindex="-1"></a>my_first_vector</span></code></pre></div>
<pre><code>## [1] 167 212 358 973 214 558</code></pre>
<p>Above we have produced a numeric vector of 6 numbers. We can confirm this by using <code>class()</code> function.</p>
<div class="sourceCode" id="cb28"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb28-1"><a href="R-and-stats.html#cb28-1" aria-hidden="true" tabindex="-1"></a><span class="fu">class</span>(my_first_vector)</span></code></pre></div>
<pre><code>## [1] &quot;numeric&quot;</code></pre>
<p>We could also make a vector containing just characters:</p>
<div class="sourceCode" id="cb30"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb30-1"><a href="R-and-stats.html#cb30-1" aria-hidden="true" tabindex="-1"></a>my_second_vector<span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&quot;Tios&quot;</span>, <span class="st">&quot;Paragon&quot;</span>, <span class="st">&quot;Skyfall&quot;</span>, <span class="st">&quot;Grafton&quot;</span>, <span class="st">&quot;Graham&quot;</span>, <span class="st">&quot;Robigus&quot;</span> )</span>
<span id="cb30-2"><a href="R-and-stats.html#cb30-2" aria-hidden="true" tabindex="-1"></a>my_second_vector</span></code></pre></div>
<pre><code>## [1] &quot;Tios&quot;    &quot;Paragon&quot; &quot;Skyfall&quot; &quot;Grafton&quot; &quot;Graham&quot;  &quot;Robigus&quot;</code></pre>
<p>We’ve produced a character vector that contains 6 names.</p>
<div class="sourceCode" id="cb32"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb32-1"><a href="R-and-stats.html#cb32-1" aria-hidden="true" tabindex="-1"></a><span class="fu">class</span>(my_second_vector)</span></code></pre></div>
<pre><code>## [1] &quot;character&quot;</code></pre>
<p>A Vector can also be logical, as in it contains either TRUE or FALSE:</p>
<div class="sourceCode" id="cb34"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb34-1"><a href="R-and-stats.html#cb34-1" aria-hidden="true" tabindex="-1"></a>my_third_vector <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="cn">FALSE</span>, <span class="cn">FALSE</span>, <span class="cn">TRUE</span>, <span class="cn">TRUE</span> , <span class="cn">TRUE</span> , <span class="cn">TRUE</span>)</span>
<span id="cb34-2"><a href="R-and-stats.html#cb34-2" aria-hidden="true" tabindex="-1"></a><span class="fu">class</span>(my_third_vector)</span></code></pre></div>
<pre><code>## [1] &quot;logical&quot;</code></pre>
<p>We can now actually combine these vectors into a data.frame. A data.frame can handle a combination of variables which are of different modes(numeric, logical, ect.).For example if we use the <code>str()</code> function on our previously loaded data.frame, we can see the different types of each variable</p>
<div class="sourceCode" id="cb36"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb36-1"><a href="R-and-stats.html#cb36-1" aria-hidden="true" tabindex="-1"></a><span class="fu">str</span>(TG_data_for_day_1)</span></code></pre></div>
<pre><code>## &#39;data.frame&#39;:    376 obs. of  15 variables:
##  $ variety    : chr  &quot;AARDEN&quot; &quot;AARDVARK&quot; &quot;ABELE&quot; &quot;ABO&quot; ...
##  $ year       : int  2003 1997 1970 1977 1998 2007 2004 2007 2006 1992 ...
##  $ ORIGIN     : chr  &quot;DEU&quot; &quot;GBR&quot; &quot;GBR&quot; &quot;FRA&quot; ...
##  $ Rht2       : int  1 NA 0 0 1 1 1 0 1 1 ...
##  $ PpdD1      : int  0 1 0 1 0 1 1 1 0 0 ...
##  $ yield      : num  103.2 88.5 90.8 82.2 96.7 ...
##  $ CALLOW_2011: num  114.2 96.7 95.3 87.1 102.9 ...
##  $ FRANCE_2010: num  136 122 124 115 132 ...
##  $ FRANCE_2011: num  100.2 96.4 92.9 85.8 101.5 ...
##  $ LGE_2010   : num  93.5 75 82.8 77.5 84.7 ...
##  $ LGE_2011   : num  96.2 75.7 83.2 68.4 87.5 ...
##  $ NIAB_2011  : num  79.4 65.3 66.1 59.5 71.3 ...
##  $ FT         : num  156 153 154 149 157 ...
##  $ HT         : num  67.6 66.3 81.1 69.6 66.7 ...
##  $ AWNS       : int  0 0 0 0 0 1 0 0 0 0 ...</code></pre>
<p>It seems that the variables (columns in the data.frame) are either in the form of characters, numeric or integers, depending on their content. Let’s form a data.frame from our three vectors:</p>
<div class="sourceCode" id="cb38"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb38-1"><a href="R-and-stats.html#cb38-1" aria-hidden="true" tabindex="-1"></a>my_DF<span class="ot">&lt;-</span><span class="fu">data.frame</span>(my_second_vector, my_first_vector, my_third_vector)</span>
<span id="cb38-2"><a href="R-and-stats.html#cb38-2" aria-hidden="true" tabindex="-1"></a>my_DF</span></code></pre></div>
<pre><code>##   my_second_vector my_first_vector my_third_vector
## 1             Tios             167           FALSE
## 2          Paragon             212           FALSE
## 3          Skyfall             358            TRUE
## 4          Grafton             973            TRUE
## 5           Graham             214            TRUE
## 6          Robigus             558            TRUE</code></pre>
<p>Note the length of the variables need to be the same. We can now use <code>class()</code>, <code>str()</code> and <code>dim()</code> to inspect the <code>data.frame</code>.</p>
<div class="sourceCode" id="cb40"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb40-1"><a href="R-and-stats.html#cb40-1" aria-hidden="true" tabindex="-1"></a><span class="fu">class</span>(my_DF)</span></code></pre></div>
<pre><code>## [1] &quot;data.frame&quot;</code></pre>
<div class="sourceCode" id="cb42"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb42-1"><a href="R-and-stats.html#cb42-1" aria-hidden="true" tabindex="-1"></a><span class="fu">str</span>(my_DF)</span></code></pre></div>
<pre><code>## &#39;data.frame&#39;:    6 obs. of  3 variables:
##  $ my_second_vector: chr  &quot;Tios&quot; &quot;Paragon&quot; &quot;Skyfall&quot; &quot;Grafton&quot; ...
##  $ my_first_vector : num  167 212 358 973 214 558
##  $ my_third_vector : logi  FALSE FALSE TRUE TRUE TRUE TRUE</code></pre>
<div class="sourceCode" id="cb44"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb44-1"><a href="R-and-stats.html#cb44-1" aria-hidden="true" tabindex="-1"></a><span class="fu">dim</span>(my_DF)</span></code></pre></div>
<pre><code>## [1] 6 3</code></pre>
<p>So we have a new data.frame, with 6 rows and 3 columns. The first column contains factors (which is typically how text is handled) and the second is numeric. The third is logical. Factors are categorical data and are stored as integer vectors.</p>
<p>We can use a combination of <code>names()</code> and <code>c()</code> to change our column names:</p>
<div class="sourceCode" id="cb46"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb46-1"><a href="R-and-stats.html#cb46-1" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(my_DF)<span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&quot;Line&quot;</span>,<span class="st">&quot;Block&quot;</span>,<span class="st">&quot;Winter_Habit&quot;</span>)</span>
<span id="cb46-2"><a href="R-and-stats.html#cb46-2" aria-hidden="true" tabindex="-1"></a>my_DF</span></code></pre></div>
<pre><code>##      Line Block Winter_Habit
## 1    Tios   167        FALSE
## 2 Paragon   212        FALSE
## 3 Skyfall   358         TRUE
## 4 Grafton   973         TRUE
## 5  Graham   214         TRUE
## 6 Robigus   558         TRUE</code></pre>
<p>We can force R to change the type of a column. For instance if we wanted “Block” to be treated as a factor rather then numeric, we could use:</p>
<div class="sourceCode" id="cb48"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb48-1"><a href="R-and-stats.html#cb48-1" aria-hidden="true" tabindex="-1"></a>my_DF<span class="sc">$</span>Block<span class="ot">&lt;-</span><span class="fu">as.factor</span>(my_DF<span class="sc">$</span>Block)</span>
<span id="cb48-2"><a href="R-and-stats.html#cb48-2" aria-hidden="true" tabindex="-1"></a><span class="fu">str</span>(my_DF)</span></code></pre></div>
<pre><code>## &#39;data.frame&#39;:    6 obs. of  3 variables:
##  $ Line        : chr  &quot;Tios&quot; &quot;Paragon&quot; &quot;Skyfall&quot; &quot;Grafton&quot; ...
##  $ Block       : Factor w/ 6 levels &quot;167&quot;,&quot;212&quot;,&quot;214&quot;,..: 1 2 4 6 3 5
##  $ Winter_Habit: logi  FALSE FALSE TRUE TRUE TRUE TRUE</code></pre>
<p>Note the use of the <code>$</code> sign to specify a column (variable) of a data.frame. We will use this a lot throughout. We could work in reverse now and use this to extract a column from a data.frame and save it as a vector:</p>
<div class="sourceCode" id="cb50"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb50-1"><a href="R-and-stats.html#cb50-1" aria-hidden="true" tabindex="-1"></a>my_V<span class="ot">&lt;-</span>TG_data_for_day_1<span class="sc">$</span>yield</span>
<span id="cb50-2"><a href="R-and-stats.html#cb50-2" aria-hidden="true" tabindex="-1"></a><span class="fu">length</span>(my_V)</span></code></pre></div>
<pre><code>## [1] 376</code></pre>
<p>We have extracted the yield column from the data.frame we loaded earlier and saved it as a variable consisting of 376 numbers. If we run <code>my_V</code> we would see all of these numbers in the console.</p>
<p>To save space, let’s output only the 5th number from the 376 numbers, using <code>[]</code> brackets like so:</p>
<div class="sourceCode" id="cb52"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb52-1"><a href="R-and-stats.html#cb52-1" aria-hidden="true" tabindex="-1"></a>my_V[<span class="dv">5</span>]</span></code></pre></div>
<pre><code>## [1] 96.66</code></pre>
<div class="sourceCode" id="cb54"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb54-1"><a href="R-and-stats.html#cb54-1" aria-hidden="true" tabindex="-1"></a>my_V      [<span class="dv">5</span>]</span></code></pre></div>
<pre><code>## [1] 96.66</code></pre>
<div class="sourceCode" id="cb56"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb56-1"><a href="R-and-stats.html#cb56-1" aria-hidden="true" tabindex="-1"></a>my_V            [<span class="dv">5</span>]</span></code></pre></div>
<pre><code>## [1] 96.66</code></pre>
<p>Note how the variable uses of spaces makes no difference, that’s a nice element of R.</p>
<p>Let’s say we only wanted the first five elements:</p>
<div class="sourceCode" id="cb58"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb58-1"><a href="R-and-stats.html#cb58-1" aria-hidden="true" tabindex="-1"></a>my_V[<span class="dv">1</span><span class="sc">:</span><span class="dv">5</span>]</span></code></pre></div>
<pre><code>## [1] 103.23  88.55  90.76  82.24  96.66</code></pre>
<p>We can reverse the output like so:</p>
<div class="sourceCode" id="cb60"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb60-1"><a href="R-and-stats.html#cb60-1" aria-hidden="true" tabindex="-1"></a>my_V[<span class="dv">5</span><span class="sc">:</span><span class="dv">1</span>]</span></code></pre></div>
<pre><code>## [1]  96.66  82.24  90.76  88.55 103.23</code></pre>
<p>We can also pick and choose using <code>c()</code>:</p>
<div class="sourceCode" id="cb62"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb62-1"><a href="R-and-stats.html#cb62-1" aria-hidden="true" tabindex="-1"></a>my_V[<span class="fu">c</span>(<span class="dv">121</span>, <span class="dv">200</span>, <span class="dv">345</span>)]</span></code></pre></div>
<pre><code>## [1]  98.83 102.75 100.24</code></pre>
<p>The use of square brackets is the same for a data.frame, except a data.frame consists of rows and columns:</p>
<div class="sourceCode" id="cb64"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb64-1"><a href="R-and-stats.html#cb64-1" aria-hidden="true" tabindex="-1"></a><span class="fu">dim</span>(TG_data_for_day_1)</span></code></pre></div>
<pre><code>## [1] 376  15</code></pre>
<p>We have 376 rows and 15 columns and with this any element of a data.frame can be called using: <code>data.frame[row, column]</code>.</p>
<div class="sourceCode" id="cb66"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb66-1"><a href="R-and-stats.html#cb66-1" aria-hidden="true" tabindex="-1"></a>TG_data_for_day_1[<span class="dv">5</span>,<span class="dv">6</span>]</span></code></pre></div>
<pre><code>## [1] 96.66</code></pre>
<p>That’s the data point in the 5th row and the 6th column. We could look at the whole of the 5th row like so:</p>
<div class="sourceCode" id="cb68"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb68-1"><a href="R-and-stats.html#cb68-1" aria-hidden="true" tabindex="-1"></a>TG_data_for_day_1[<span class="dv">5</span>,]</span></code></pre></div>
<pre><code>##   variety year ORIGIN Rht2 PpdD1 yield CALLOW_2011 FRANCE_2010 FRANCE_2011
## 5  ACCESS 1998    GBR    1     0 96.66      102.86      132.06      101.49
##   LGE_2010 LGE_2011 NIAB_2011    FT    HT AWNS
## 5    84.68     87.5     71.34 156.6 66.69    0</code></pre>
<p>We can’t save this as a vector as there are different data types present (factor and numeric), so assigning this row from our data.frame to a new object would create a new data.frame, with 1 row and 15 columns.</p>
<div id="summary" class="section level3" number="3.5.1">
<h3><span class="header-section-number">3.5.1</span> Summary</h3>
<p>I’ve only focused on a subset of data types and classes. Here are some more examples and formal definitions that I have copied from a book called “R in action” by Robert Kabacoff. You can find a free PDF online.</p>
<p>In R, an object is anything that can be assigned to a variable. This includes constants, data structures, functions, and even graphs. An object has a mode (which describes how the object is stored) and a class (which tells generic functions like print how to handle it).</p>
<p>A data frame is a structure in R that holds data and is similar to the datasets found in standard statistical packages (for example, SAS, SPSS, and Stata). The columns are variables, and the rows are observations. You can have variables of different types (for example, numeric or character) in the same data frame. Data frames are the main structures you use to store datasets.</p>
<p><strong>Factors</strong> are nominal or ordinal variables. They’re stored and treated specially in R.</p>
<p><strong>Vectors</strong> are one-dimensional arrays that can hold numeric data, character data, or logical data.</p>
<p><strong>Matrix</strong> is a two-dimensional array in which each element has the same mode
(numeric, character, or logical). Matrices are created with the matrix()function.</p>
<p><strong>Arrays</strong> are similar to matrices but can have more than two dimensions.</p>
<p><strong>Data frame</strong> is more general than a matrix in that different columns can contain different modes of data (numeric, character, and so on). It’s similar to the dataset you’d typically see in SAS, SPSS, and Stata. Data frames are the most common data structure you’ll deal with in R.</p>
<p><strong>Lists</strong> are the most complex of the R data types. Basically, a list is an ordered collection of objects (components). A list allows you to gather a variety of (possibly unrelated) objects under one name. For example, a list may contain a combination of vectors, matrices, data frames, and even other lists. You create a list using the list() function.</p>

<div class="rmdquiz">
<p><strong>4</strong> Form three vectors. The first, a character vector containing the names of each day of the week. The second, a numeric vector containing 7 random numbers (hint: try using <code>runif(7, 1, 10)</code>). The third, a logical vector corresponding to whether you would typically visit work on each day of the week (TRUE or FALSE).</p>
<p>Save the above as a data.frame and change the column names to: day_of_week, coffee_consumed and attendance.</p>
Change the mode of the data in the coffee_consumed from numeric to factor.
</div>
</div>
</div>
<div id="summarising-and-displaying-data" class="section level2" number="3.6">
<h2><span class="header-section-number">3.6</span> Summarising and displaying data</h2>
<p><em>‘Time spent in reconnaissance is never wasted’ - Napoleon (attrib).</em></p>
<p>It is always worthwhile to spend time scanning and summarising new datasets before starting formal statistical analysis. Simple methods such as studying the range, the distribution and the relationships between variables can often reveal unexpected structure or the presence of errors in a dataset.</p>
<p>Let’s look at the TriticeaeGenome data. Try some or all of the commands below.</p>
<p><code>hist(yield)</code></p>
<p><code>plot(yield~year)</code></p>
<p><code>plot(yield~Rht2)</code></p>
<p>It is very easy to produce basic informative graphs in R. Titles, font sizes, colours and so on can all be changed by adding additional options to the basic commands above, but we shall introduce those as we go along.</p>
<p><code>plot(yield~Rht2)</code> is not great. Try <code>boxplot(yield~Rht2)</code> which produces a box plot, or box-and-whisker plot. The central bold line is the median. The boxes show, approximately, the first and third quartiles. The lines extending from the boxes to the horizontal bars then show the distance to the maximum and minimum observations. However, any data viewed as outliers are plotted separately. These plots must not be used as statistical test for outliers. The definition of what is plotted as an outlier can be changed with additional arguments. The default is 1.5 times the interquartile range (the width of the middle 50% of observations).</p>
<div class="sourceCode" id="cb70"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb70-1"><a href="R-and-stats.html#cb70-1" aria-hidden="true" tabindex="-1"></a><span class="fu">boxplot</span>(yield<span class="sc">~</span>PpdD1<span class="sc">+</span>Rht2, <span class="at">varwidth=</span><span class="cn">TRUE</span>)</span></code></pre></div>
<p><img src="bookdown-demo_files/figure-html/unnamed-chunk-44-1.png" width="672" /></p>
<p>We now have boxplots for the four classes of +/- ppd and +/- rht2 alleles. We added <code>varwidth=TRUE</code> to scale the width of each boxplot to the square root of the different samples sizes. The group “0.1” carries the wild-type ppdD1 allele for late flowering and the dwarfing allele at rht2, and would appear to be the highest yielding, and largest, group of lines. Substitute FT (flowering time) for yield in the boxplot to get something a little more exciting.</p>
<p>To find out what options are available for commands, switch from the “Plots” tab to “Help,” then type in the name of the command you are interested in. The help is often cryptic, but with practice, its interpretation becomes easier.</p>
<p>A useful command for displaying relations between all variables in a data set is “pairs,” we’ll introduce the command and then refine it. Try running:</p>
<p><code>pairs(TG_data_for_day_1)</code></p>
<p>It produces an error message? Some of the data are non-numeric. This is being caused by the names in the “variety” column. We’ll remind ourselves what the names of the data columns are:</p>
<div class="sourceCode" id="cb71"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb71-1"><a href="R-and-stats.html#cb71-1" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(TG_data_for_day_1)</span></code></pre></div>
<pre><code>##  [1] &quot;variety&quot;     &quot;year&quot;        &quot;ORIGIN&quot;      &quot;Rht2&quot;        &quot;PpdD1&quot;      
##  [6] &quot;yield&quot;       &quot;CALLOW_2011&quot; &quot;FRANCE_2010&quot; &quot;FRANCE_2011&quot; &quot;LGE_2010&quot;   
## [11] &quot;LGE_2011&quot;    &quot;NIAB_2011&quot;   &quot;FT&quot;          &quot;HT&quot;          &quot;AWNS&quot;</code></pre>
<p>We are only interested, in plotting the yield data, ignoring the other traits for now. We’ll also select the subset of lines which were registered in Germany. These are labeled DEU in the origin column. Finally we’ll add a title:</p>
<div class="sourceCode" id="cb73"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb73-1"><a href="R-and-stats.html#cb73-1" aria-hidden="true" tabindex="-1"></a><span class="fu">pairs</span>(TG_data_for_day_1[(ORIGIN<span class="sc">==</span><span class="st">&quot;DEU&quot;</span>),<span class="dv">6</span><span class="sc">:</span><span class="dv">12</span>],<span class="at">main=</span><span class="st">&quot;German varieties only&quot;</span>)</span></code></pre></div>
<p><img src="bookdown-demo_files/figure-html/unnamed-chunk-46-1.png" width="672" /></p>
<p>This is easier to interpret and we can see that the yield of German varieties is strongly correlated across all the sites. Note the use of the square brackets <code>[ ]</code> to reference rows and columns of our data set. We have seen this in the last section too. As revision:</p>
<ul>
<li><code>[1,2]</code> refers to a single cell - row one, column 2.</li>
<li><code>[2:5,]</code> refers to rows 2 to 5 inclusive, and to all columns.</li>
<li><code>[,-3]</code> refers to all rows and all columns except column 3.</li>
<li><code>[,c(1,4:6)]</code> refers to all rows and to columns 1, 4, 5, and 6.</li>
</ul>
<p>You cannot mix omission of rows or columns by using “-” with inclusion using anything else, but otherwise this provides you with a flexible way of selecting subsets of data.</p>
<p><code>c(... , ... , ... ,)</code> as used above, is a method of concatenating data into a single entity and is used quite extensively. It can also be used as a method of entering small amounts of data directly into R:</p>
<div class="sourceCode" id="cb74"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb74-1"><a href="R-and-stats.html#cb74-1" aria-hidden="true" tabindex="-1"></a>fred<span class="ot">&lt;-</span><span class="fu">c</span>(<span class="dv">1</span><span class="sc">:</span><span class="dv">2</span>, <span class="st">&quot;Buckle my shoe&quot;</span>)</span>
<span id="cb74-2"><a href="R-and-stats.html#cb74-2" aria-hidden="true" tabindex="-1"></a>fred</span></code></pre></div>
<pre><code>## [1] &quot;1&quot;              &quot;2&quot;              &quot;Buckle my shoe&quot;</code></pre>
<p>Another way of selecting a subset of data is to use the subset command</p>
<div class="sourceCode" id="cb76"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb76-1"><a href="R-and-stats.html#cb76-1" aria-hidden="true" tabindex="-1"></a>fred<span class="ot">&lt;-</span><span class="fu">subset</span>(TG_data_for_day_1,<span class="at">select=</span><span class="fu">c</span>(variety,ORIGIN,yield))</span>
<span id="cb76-2"><a href="R-and-stats.html#cb76-2" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(fred)</span></code></pre></div>
<pre><code>##    variety ORIGIN  yield
## 1   AARDEN    DEU 103.23
## 2 AARDVARK    GBR  88.55
## 3    ABELE    GBR  90.76
## 4      ABO    FRA  82.24
## 5   ACCESS    GBR  96.66
## 6    ACCOR    FRA  90.05</code></pre>
<p>The new command <code>head</code>, gives the same result as: <code>fred[1:6,]</code>. It prints the first 6 rows.</p>
<p>In R, the simplest non-graphical method of generating a summary of data is:</p>
<div class="sourceCode" id="cb78"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb78-1"><a href="R-and-stats.html#cb78-1" aria-hidden="true" tabindex="-1"></a><span class="fu">summary</span>(TG_data_for_day_1)[,<span class="dv">1</span><span class="sc">:</span><span class="dv">6</span>]</span></code></pre></div>
<pre><code>##    variety               year         ORIGIN               Rht2       
##  Length:376         Min.   :1946   Length:376         Min.   :0.0000  
##  Class :character   1st Qu.:1988   Class :character   1st Qu.:0.0000  
##  Mode  :character   Median :1997   Mode  :character   Median :1.0000  
##                     Mean   :1994                      Mean   :0.5508  
##                     3rd Qu.:2003                      3rd Qu.:1.0000  
##                     Max.   :2007                      Max.   :1.0000  
##                                                       NA&#39;s   :22      
##      PpdD1            yield       
##  Min.   :0.0000   Min.   : 67.79  
##  1st Qu.:0.0000   1st Qu.: 88.60  
##  Median :0.0000   Median : 92.63  
##  Mean   :0.2261   Mean   : 91.83  
##  3rd Qu.:0.0000   3rd Qu.: 96.32  
##  Max.   :1.0000   Max.   :105.58  
## </code></pre>
<p>Again, to save space, I’ve limited this command to columns 1 to 6, by using: <code>[,1:6]</code>.</p>
<p>For a numeric column such as yield, the output typically shows:</p>
<ul>
<li>Min. : the minimum value</li>
<li>1st Qu.: the first quantile</li>
<li>Median : the median</li>
<li>Mean : the sample average</li>
<li>3rd Qu.: the third quantile</li>
<li>Max. : the maximum value.</li>
<li>NA’s : the number of missing values</li>
</ul>
<p>The first quartile, the median and the third quartile give the values of the observations, 1/4, 1/2, and 3/4 of the way down a sorted list of each variable. These values, together with the minimum, maximum and average, give a simple assessment of the distribution of the traits.
You can also just get the summary information for one variable by using:</p>
<div class="sourceCode" id="cb80"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb80-1"><a href="R-and-stats.html#cb80-1" aria-hidden="true" tabindex="-1"></a><span class="fu">summary</span>(HT)</span></code></pre></div>
<pre><code>##    Min. 1st Qu.  Median    Mean 3rd Qu.    Max. 
##   57.52   69.77   73.79   75.25   80.12   97.15</code></pre>
<p>The summary statistics given collectively by summary are also available as separate commands, listed below (substitute the variable name of your choice for “x”):</p>
<p><code>mean(x)</code></p>
<p><code>median(x)</code></p>
<p><code>quantile(x)</code></p>
<p><code>min(x)</code></p>
<p><code>max(x)</code></p>
<p>Note that <code>quantile</code> returns the quartiles. This is the default for the command, which can be altered.</p>
<p>This is an important quirk of R:</p>
<div class="sourceCode" id="cb82"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb82-1"><a href="R-and-stats.html#cb82-1" aria-hidden="true" tabindex="-1"></a><span class="fu">mean</span>(Rht2)</span></code></pre></div>
<pre><code>## [1] NA</code></pre>
<p>R does not ignore missing values by default (although some functions do). Because Rht2 has some missing data (the NAs) R returns “NA” rather than the mean of the available data. This is often irritating, but it is safe: you will not forget that you have missing data. The rather longwinded way to cope with this is as follows:</p>
<div class="sourceCode" id="cb84"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb84-1"><a href="R-and-stats.html#cb84-1" aria-hidden="true" tabindex="-1"></a><span class="fu">mean</span>(Rht2,<span class="at">na.rm =</span> <span class="cn">TRUE</span>)</span></code></pre></div>
<pre><code>## [1] 0.5508475</code></pre>
<p>The additional logical variable <code>na.rm</code> (meaning: not available, remove) is set to the value TRUE. In English: remove the NA values before calculating the mean. A number of other commands require this option too.</p>
<p>One dangerous exception to the default behavior of R is the command length, which counts the number of entries in a vector:</p>
<div class="sourceCode" id="cb86"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb86-1"><a href="R-and-stats.html#cb86-1" aria-hidden="true" tabindex="-1"></a><span class="fu">length</span>(Rht2)</span></code></pre></div>
<pre><code>## [1] 376</code></pre>
<p>The value of 376 is returned: this includes the missing values, which is logical but often misleading. To count the number of observations, excluding the NAs:</p>
<div class="sourceCode" id="cb88"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb88-1"><a href="R-and-stats.html#cb88-1" aria-hidden="true" tabindex="-1"></a><span class="fu">sum</span>(<span class="sc">!</span><span class="fu">is.na</span>(Rht2))</span></code></pre></div>
<pre><code>## [1] 354</code></pre>
<p>Breaking this down, <code>is.na()</code> is the command to return the logical value TRUE or FALSE depending on whether each value of a variate exists or not.</p>
<div class="sourceCode" id="cb90"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb90-1"><a href="R-and-stats.html#cb90-1" aria-hidden="true" tabindex="-1"></a><span class="fu">is.na</span>(Rht2)[<span class="dv">1</span><span class="sc">:</span><span class="dv">5</span>]</span></code></pre></div>
<pre><code>## [1] FALSE  TRUE FALSE FALSE FALSE</code></pre>
<p>I’ve only shown the first five logical values, but this would continue for every row. The use of <code>!is.na()</code> switches this around to return TRUE if the value is NA (ie does not exist). <code>!</code> is the R symbol to negate the following argument. (We introduced <code>!=</code> to mean “not equal to” earlier.)</p>
<div class="sourceCode" id="cb92"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb92-1"><a href="R-and-stats.html#cb92-1" aria-hidden="true" tabindex="-1"></a><span class="sc">!</span><span class="fu">is.na</span>(Rht2)[<span class="dv">1</span><span class="sc">:</span><span class="dv">5</span>]</span></code></pre></div>
<pre><code>## [1]  TRUE FALSE  TRUE  TRUE  TRUE</code></pre>
<p>For arithmetic purposes, the logical TRUE has a value of 1 and FALSE has a value of 0, <code>sum()</code> returns the sum or total, so <code>sum(!is.na(Rht2))</code> returns the value of 354.</p>
<p>Some other useful summary commands are listed below:</p>
<p><code>sum(yield,na.rm=T)</code> returns the total yield</p>
<p><code>sum(yield,na.rm=T)</code> returns the of yield variance</p>
<p><code>sd(yield,na.rm=T)</code> returns the standard deviation.</p>
<p><code>rowMeans(dataset,na.rm=T)</code> returns means across rows of a dataset</p>
<p><code>colMeans(dataset,na.rm=T)</code> returns means across columns of a dataset</p>
<p><code>rowSums(dataset,na.rm=T)</code> returns totals across rows of a dataset</p>
<p><code>colSums(dataset,na.rm=T)</code> returns totals across columns of a dataset</p>
<p>These commands have all been listed including the <code>na.rm=T</code> option. If the dataset is complete this option need not be included.</p>

<div class="rmdquiz">
<p><strong>5</strong> Let’s say we measured leaf length from 10 plants from one of our plots. For whatever reason one leaf was missing from the measurements:</p>
<p>Create this vector: <code>leaf_sizes&lt;-c(30.1, 32.3, 36.8, 34.1, 32.0, NA, 32.4, 30.1, 29.9, 32.6)</code></p>
<p>Plot a histogram showing the distribution of our leaves. Then add a title and figure legends (measurements are in cm), see <code>help(hist)</code> if you get stuck.</p>
<p>What’s the mean of the 9 leaves?</p>
It’s pointless really to keep the NA in the vector here, produce a vector of 9 numbers without the NA.
</div>
</div>
<div id="calculations-and-data-manipulation" class="section level2" number="3.7">
<h2><span class="header-section-number">3.7</span> Calculations and data manipulation</h2>
<p>Once data have been read into R, the vectors (columns of data) which contain each variable can be manipulated in the same manner as individual numbers. For example:</p>
<div class="sourceCode" id="cb94"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb94-1"><a href="R-and-stats.html#cb94-1" aria-hidden="true" tabindex="-1"></a>harvest_index<span class="ot">&lt;-</span>yield<span class="sc">/</span>HT</span>
<span id="cb94-2"><a href="R-and-stats.html#cb94-2" aria-hidden="true" tabindex="-1"></a><span class="fu">boxplot</span>(harvest_index<span class="sc">~</span>PpdD1)</span></code></pre></div>
<p><img src="bookdown-demo_files/figure-html/unnamed-chunk-58-1.png" width="672" /></p>
<p>The late flowering allele (coded 1) seems to have a slightly higher harvest index. Whether this difference is likely genuine or just due to chance is something shall see how to test shortly.</p>
<div id="sorting-data" class="section level3" number="3.7.1">
<h3><span class="header-section-number">3.7.1</span> Sorting data</h3>
<p>Sorting data and inspection of high and low values is also of assistance in detecting errors. In R this is carried out using the command sort:</p>
<div class="sourceCode" id="cb95"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb95-1"><a href="R-and-stats.html#cb95-1" aria-hidden="true" tabindex="-1"></a><span class="fu">sort</span>(yield) [<span class="dv">1</span><span class="sc">:</span><span class="dv">5</span>]</span></code></pre></div>
<pre><code>## [1] 67.79 71.12 73.67 74.12 74.38</code></pre>
<p>To save paper I’ve added the <code>[1:5]</code> to only produce the first 5 values, or in this case the lowest 5 values.</p>
<p>More generally, it is usual to sort a block of data with respect to one or more columns. Routinely, this may be more easily achieved in Excel. In R: First select yield and some descriptors for ease of display:</p>
<div class="sourceCode" id="cb97"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb97-1"><a href="R-and-stats.html#cb97-1" aria-hidden="true" tabindex="-1"></a>TG.sub<span class="ot">&lt;-</span><span class="fu">subset</span>(TG_data_for_day_1,<span class="at">select=</span><span class="fu">c</span>(variety,ORIGIN,year,yield))</span></code></pre></div>
<p>Then display the top few lines of the new dataframe:</p>
<div class="sourceCode" id="cb98"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb98-1"><a href="R-and-stats.html#cb98-1" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(TG.sub[<span class="fu">order</span>(ORIGIN,year),])</span></code></pre></div>
<pre><code>##          variety ORIGIN year yield
## 190         IBIS    DEU 1963 89.61
## 234 MIRONOVSKAJA    DEU 1963 84.43
## 45        ASTRON    DEU 1965 92.19
## 97        CARIBO    DEU 1968 88.29
## 310      SENATOR    DEU 1970 95.76
## 229       MERKUR    DEU 1973 86.37</code></pre>
<p>We used <code>order(ORIGIN,year)</code> to sort the whole dataframe, indexed by country of origin and then year of release. Alphabetically, the German varieties are first.</p>
</div>
<div id="relationships-between-variates" class="section level3" number="3.7.2">
<h3><span class="header-section-number">3.7.2</span> Relationships between variates</h3>
<p>The correlation coefficient ranges from zero to one and measures the strength of the relationship between two variables:</p>
<div class="sourceCode" id="cb100"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb100-1"><a href="R-and-stats.html#cb100-1" aria-hidden="true" tabindex="-1"></a><span class="fu">cor</span>(yield,year)</span></code></pre></div>
<pre><code>## [1] 0.6218021</code></pre>
<p>We can enter a block of data and generate a table of correlation coefficients in a single command.
There are some traps for the unwary however:</p>
<div class="sourceCode" id="cb102"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb102-1"><a href="R-and-stats.html#cb102-1" aria-hidden="true" tabindex="-1"></a><span class="fu">cor</span>(TG_data_for_day_1[,<span class="dv">4</span><span class="sc">:</span><span class="dv">6</span>])</span></code></pre></div>
<pre><code>##       Rht2      PpdD1      yield
## Rht2     1         NA         NA
## PpdD1   NA  1.0000000 -0.1556021
## yield   NA -0.1556021  1.0000000</code></pre>
<div class="sourceCode" id="cb104"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb104-1"><a href="R-and-stats.html#cb104-1" aria-hidden="true" tabindex="-1"></a><span class="fu">cor</span>(TG_data_for_day_1[,<span class="dv">4</span><span class="sc">:</span><span class="dv">6</span>],<span class="at">use=</span><span class="st">&quot;complete&quot;</span>)</span></code></pre></div>
<pre><code>##             Rht2      PpdD1      yield
## Rht2   1.0000000 -0.2461724  0.5021948
## PpdD1 -0.2461724  1.0000000 -0.1728991
## yield  0.5021948 -0.1728991  1.0000000</code></pre>
<div class="sourceCode" id="cb106"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb106-1"><a href="R-and-stats.html#cb106-1" aria-hidden="true" tabindex="-1"></a><span class="fu">cor</span>(TG_data_for_day_1[,<span class="dv">4</span><span class="sc">:</span><span class="dv">6</span>],<span class="at">use=</span><span class="st">&quot;pairwise.complete&quot;</span>)</span></code></pre></div>
<pre><code>##             Rht2      PpdD1      yield
## Rht2   1.0000000 -0.2461724  0.5021948
## PpdD1 -0.2461724  1.0000000 -0.1556021
## yield  0.5021948 -0.1556021  1.0000000</code></pre>
<p>The first call gives no results for Rht2 because it has missing data. The second method uses “complete observations,” that is to say only records with no missing data for any field. This is the default method for some commercial statistical software systems. In the example here, 22 records of data are discarded. The third method does not discard complete records. It excludes from the analysis only those pairs of observations in which at least one of the pair is NA. In this example, correlations among PpdD1 and yield will be based on all varieties as there are no missing data for these two traits. This option can be particularly useful with extensive sets of genotype data: even if marker calling rates are high, with multiple markers it may be rare for a single individual or line to have no missing data.</p>
<p>In passing, we note that the calculation of a correlation coefficient is not usually a particularly sensible or conventional way to study the relationship between two binary variables (here Rht2 and PpdD1). More conventional would be to tabulate the data in a 2 x 2 contingency table. However, for marker data, the squared correlation coefficient is very commonly used as a measure of linkage disequilibrium between two loci, so the example given here has some justification (although a 2 x 2 table of observations is still informative and is demonstrated later on).</p>
<p>Correlation coefficients are a simple way of quantifying relationships between two variables. However, as we have already commented, it is often better to visualise the data in a scatter plot:</p>
<div class="sourceCode" id="cb108"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb108-1"><a href="R-and-stats.html#cb108-1" aria-hidden="true" tabindex="-1"></a><span class="fu">plot</span>(year,yield)</span></code></pre></div>
<p><img src="bookdown-demo_files/figure-html/unnamed-chunk-66-1.png" width="672" /></p>
<p>R also provides other methods of displaying these correlations graphically. <code>Image</code> creates a coloured grid of a matrix with colours depending on the values in the matrix. First we shall save the correlation matrix, working only on the trait data:</p>
<div class="sourceCode" id="cb109"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb109-1"><a href="R-and-stats.html#cb109-1" aria-hidden="true" tabindex="-1"></a>TG.cor<span class="ot">&lt;-</span><span class="fu">cor</span>(TG_data_for_day_1[,<span class="fu">c</span>(<span class="sc">-</span><span class="dv">1</span><span class="sc">:-</span><span class="dv">3</span>)],<span class="at">use=</span><span class="st">&quot;pairwise.complete&quot;</span>)</span>
<span id="cb109-2"><a href="R-and-stats.html#cb109-2" aria-hidden="true" tabindex="-1"></a><span class="fu">image</span>(TG.cor<span class="sc">^</span><span class="dv">2</span>)</span></code></pre></div>
<p><img src="bookdown-demo_files/figure-html/unnamed-chunk-67-1.png" width="672" /></p>
<p>Various colours are schemes are possible: type help(image)for more detail. Note we have squared the correlation coefficient - so all values are positive - otherwise the interpretation with both large negative and large positive correlations present can become difficult. Here, the block of pale colours in the centre of the plot indicates very high correlations among the yield measurements, with lower correlations among the other traits. The function <code>heatmap()</code> is a useful extension to this which also plots a dendogram showing estimated relationships between entries.</p>
<div class="sourceCode" id="cb110"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb110-1"><a href="R-and-stats.html#cb110-1" aria-hidden="true" tabindex="-1"></a><span class="fu">heatmap</span>(TG.cor<span class="sc">^</span><span class="dv">2</span>,<span class="at">symm=</span>T)</span></code></pre></div>
<p><img src="bookdown-demo_files/figure-html/unnamed-chunk-68-1.png" width="672" />
This plot alters the row and column order. <code>symm=T</code> is required for heatmap to recognize that it is working on a symmetrical matrix.</p>
<p>The interpretation seems clear. The yield measurements correlate highly. PpdD1 and flowering time are associated, as are Rht2 and height. The relationships show up even better using a different colour scheme (which I found someone else using and copied):</p>
<div class="sourceCode" id="cb111"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb111-1"><a href="R-and-stats.html#cb111-1" aria-hidden="true" tabindex="-1"></a><span class="fu">heatmap</span>(TG.cor<span class="sc">^</span><span class="dv">2</span>,<span class="at">symm=</span>T,<span class="at">col=</span><span class="fu">c</span>(<span class="fu">grey</span>(<span class="fl">0.4</span>), <span class="st">&quot;blue&quot;</span>, <span class="st">&quot;green&quot;</span>, <span class="st">&quot;orange&quot;</span>,<span class="st">&quot;red&quot;</span>))</span></code></pre></div>
<p><img src="bookdown-demo_files/figure-html/unnamed-chunk-69-1.png" width="672" /></p>
<p>This highlights a weak relationship between height and yield, at least in some environments.</p>
</div>
<div id="exporting-your-script-and-workspace" class="section level3" number="3.7.3">
<h3><span class="header-section-number">3.7.3</span> Exporting your script and workspace</h3>
<p>This is not as straightforward as you would think as the script and workspace (your data in the Environment window) need to be saved separately.</p>
<p>To save your workspace (the working environment you have created) in your working directory, you can just click:</p>
<p><em>Session / Save Workspace As</em></p>
<p>Or you could run:</p>
<p><code>save.image(file="File_name_for_your_data.RData")</code></p>
<p>Note that this will save an <code>.RData</code> file.</p>
<p>Next save the script you have created in the <em>script editing window</em> as an R file. You can then come back to your results by loading both the R file and the saved workspace. You simply save the script by clicking on <em>File / Save</em>.
Or you could press: <strong>Ctrl + S</strong></p>
<p>To reload your workspace you can go to: <em>Session / Load Workspace</em>. Or use <code>load("myfile.RData")</code> if you are in the correct working directory. To open your script file, you would simple click on it and it will open RStudio (or open in RStudio if you already have the program open).</p>
</div>
<div id="saving-a-data-table" class="section level3" number="3.7.4">
<h3><span class="header-section-number">3.7.4</span> Saving a data table</h3>
<p>There are two simple methods for this, the simplest is to cut and paste from R to Word or Excel, exactly as you would for any other Windows application.</p>
<p>However, a better method would be to use R to do it. If you want to export a table you have created as a csv file (other formats are available):</p>
<div class="sourceCode" id="cb112"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb112-1"><a href="R-and-stats.html#cb112-1" aria-hidden="true" tabindex="-1"></a><span class="co">#create a subset of the main data, in this case we just take varieties with a GBR origin</span></span>
<span id="cb112-2"><a href="R-and-stats.html#cb112-2" aria-hidden="true" tabindex="-1"></a>GBRorigin<span class="ot">&lt;-</span><span class="fu">subset</span>(TG_data_for_day_1, ORIGIN<span class="sc">==</span><span class="st">&quot;GBR&quot;</span>)</span>
<span id="cb112-3"><a href="R-and-stats.html#cb112-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb112-4"><a href="R-and-stats.html#cb112-4" aria-hidden="true" tabindex="-1"></a><span class="co">#save this as a new csv file</span></span>
<span id="cb112-5"><a href="R-and-stats.html#cb112-5" aria-hidden="true" tabindex="-1"></a><span class="fu">write.csv</span>(GBRorigin, <span class="st">&quot;this will be the file name.csv&quot;</span>)</span></code></pre></div>

<div class="rmdtip">
That file will be saved in your directory now. Notice my use of # here. You will see this a lot in R. Putting a # at the start of a line means that R won’t try to run that line, which means that you can add notes to yourself within your scripts.
</div>

<div class="rmdquiz">
<p><strong>6</strong> You now should of seen the use of <code>subset()</code> several times. This is a very useful function which subsets large data.frames to smaller ones based on criteria. Firstly use <code>subset()</code> to form a new data.frame of only the unawned lines. Ask for help if you are stuck.</p>
<p>Re-order your new data.frame by origin.</p>
Using your new data, form a plot showing the relationship between height (HT) and yield.
Use <code>cor()</code> to test the correlation between the two variables.
</div>
</div>
</div>
<div id="basic-statistical-analysis" class="section level2" number="3.8">
<h2><span class="header-section-number">3.8</span> Basic statistical analysis</h2>
<div id="the-t-test" class="section level3" number="3.8.1">
<h3><span class="header-section-number">3.8.1</span> The t-test</h3>
<p>The t-test is a simple and robust method to test if the difference in means between two samples, or the difference between the mean of a sample and a known constant, is statistically significant. In other words, does the difference look too large to have occurred as a result of bad luck in selecting the samples for analysis?</p>
<p>t = difference / standard error of the difference.</p>
<p>For large sample sizes, a value of t &gt; 1.96 will only occur by chance, in the absence of any genuine difference, in about 5% of experiments. A value &gt; 1.96 is therefore judged to be improbably large: the difference in means is declared to be statistically significant at the 5% level.</p>
<p>The t-test assumes that the sampling error of the difference being tested is normally distributed. In real data sets, this condition is often met. Firstly, the trait being measured is itself often normally distributed, and secondly, even if the trait has a non-normal distribution, mean trait values will be close to normal provided the sample size is moderately large (greater than about 10).</p>
<p>For example, the plot below shows the distribution of 1000 numbers. Each number was generated by taking the mean of 10 uniformly distributed random numbers. It is clear that although the distribution of original random numbers was very non-normal, the mean of a sample of 10 such numbers is pretty close to normal. In fact, in the early days of computing, normally distributed random numbers were often generated in this way. The tendency for the distribution of means to be normally distributed is called the Central Limit Theorem. It explains the popularity of the normal distribution in statistics and also the tendency for many traits in nature to be roughly normally distributed - for example if variation in a phenotype results from variation at multiple genes, the phenotype itself will often inevitably be normally distributed.</p>
<div class="figure">
<img src="images/Distribution%20Plot.png" alt="" />
<p class="caption">Illustration of the central limit theorem</p>
</div>
<p>The t-test is very simply invoked in R. To test the difference in means between yield in Germany in the two years of testing:</p>
<div class="sourceCode" id="cb113"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb113-1"><a href="R-and-stats.html#cb113-1" aria-hidden="true" tabindex="-1"></a><span class="fu">t.test</span>(LGE_2010,LGE_2011)</span></code></pre></div>
<pre><code>## 
##  Welch Two Sample t-test
## 
## data:  LGE_2010 and LGE_2011
## t = 9.8055, df = 745.63, p-value &lt; 2.2e-16
## alternative hypothesis: true difference in means is not equal to 0
## 95 percent confidence interval:
##  4.019585 6.032010
## sample estimates:
## mean of x mean of y 
##  85.79497  80.76918</code></pre>
<p>The output provides a value for t and its associated p-value to test the significance of the difference in means, and also the means themselves. In addition, 95% confidence intervals are provided. These refer to the difference between the two means. Statisticians can get quite hot under the collar about what, exactly, 95% confidence intervals actually are. We can state that over a long lifetime of calculating 95 % confidence intervals for parameter estimates, they will have included the true parameter value in 95 % of cases. It is best not to worry too much about this.</p>
<p>Note that the degrees of freedom (df) is 745.63 and not a whole number. This is because the default setting for R is to assume that, whether or not the means of the two groups being tested are different, the variances themselves are different. In accounting for this we end up with fractional degrees of freedom. This is the Welch variant of the t-test - stated in the first line of the output.</p>
<p>To test whether the variances in the two groups are similar, we can use a variance ratio test, or F test - dividing one variance by the other and estimating whether the deviation from the expected value of 1 is attributable to chance or is indicative of something else:</p>
<div class="sourceCode" id="cb115"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb115-1"><a href="R-and-stats.html#cb115-1" aria-hidden="true" tabindex="-1"></a><span class="fu">var.test</span>(LGE_2010,LGE_2011)</span></code></pre></div>
<pre><code>## 
##  F test to compare two variances
## 
## data:  LGE_2010 and LGE_2011
## F = 0.85785, num df = 375, denom df = 375, p-value = 0.1381
## alternative hypothesis: true ratio of variances is not equal to 1
## 95 percent confidence interval:
##  0.700429 1.050643
## sample estimates:
## ratio of variances 
##          0.8578466</code></pre>
<p>The F-ratio of 0.86, with 375 and 375 degrees of freedom is not significant (p-value 0.1381).</p>
<p>If var.test(LGE_2011,LGE_2010) was called (the order of the two trials is reversed), the F-ratio would be 1.17 but the p-value would be unchanged. When testing the ratios of two variances in an F test, care is required because we are not testing whether the numerator variance is significantly greater than the denominator, which is the usual use of the F-ratio in ANOVA. Here we are carrying out a two-tailed test of significance rather than a one-tailed. Fortunately using var.test() takes care of this for you. If you were to do this by hand; dividing the larger variance by the smaller, then looking up the probability associated with that value of F in tables (or you could use an R function, p(f), then the probability would need to be doubled.</p>
<p>Since the variances are not significantly different (i.e. they are homogeneous), <code>t.test</code> can be called in a form to take this into account:</p>
<div class="sourceCode" id="cb117"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb117-1"><a href="R-and-stats.html#cb117-1" aria-hidden="true" tabindex="-1"></a><span class="fu">t.test</span>(LGE_2010,LGE_2011,<span class="at">var.equal =</span> T)</span></code></pre></div>
<pre><code>## 
##  Two Sample t-test
## 
## data:  LGE_2010 and LGE_2011
## t = 9.8055, df = 750, p-value &lt; 2.2e-16
## alternative hypothesis: true difference in means is not equal to 0
## 95 percent confidence interval:
##  4.019595 6.032001
## sample estimates:
## mean of x mean of y 
##  85.79497  80.76918</code></pre>
<p>This is the more usual form for the t test - as given in most text books. It is more powerful test than Welch’s variant, provided the variances are homogeneous. Note that the degrees of freedom is now an integer.</p>
<p>A more biologically interesting comparison is whether Rht2 and PpdD1 have an effect on yield. The problem here is that data for the two groups to be compared (yields for one allele and yields for the other) are no longer in separate variables. To tell R that data to be analysed are in one variate but are described by data in another, we use the tilde operator (~) introduced in the section on syntax:</p>
<div class="sourceCode" id="cb119"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb119-1"><a href="R-and-stats.html#cb119-1" aria-hidden="true" tabindex="-1"></a><span class="fu">t.test</span>(yield<span class="sc">~</span>PpdD1)</span></code></pre></div>
<pre><code>## 
##  Welch Two Sample t-test
## 
## data:  yield by PpdD1
## t = 3.0435, df = 136.67, p-value = 0.002806
## alternative hypothesis: true difference in means is not equal to 0
## 95 percent confidence interval:
##  0.8154943 3.8410701
## sample estimates:
## mean in group 0 mean in group 1 
##        92.35522        90.02694</code></pre>
<p>The difference in mean yield between the two PpdD1 alleles is significant, but quite small. PpdD1 could have a direct effect on yield or it could lie close to a QTL for yield, or the significant result could be due to something we don’t know about concerning the origins of the varieties under test. Disentangling trait-marker associations due to the presence of a closely linked QTL from other spurious causes of association is the challenge of association genetics. We can also see if the effect of PpdD1 is consistent across all sites. There are six sites, in columns 6-12 of the dataset. We could analyse each site by editing the name of the variable before running the command, but we can automate this procedure using a loop.</p>
<p>This would be achieved by running:</p>
<div class="sourceCode" id="cb121"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb121-1"><a href="R-and-stats.html#cb121-1" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span>(i <span class="cf">in</span> <span class="dv">6</span><span class="sc">:</span><span class="dv">12</span>) {</span>
<span id="cb121-2"><a href="R-and-stats.html#cb121-2" aria-hidden="true" tabindex="-1"></a>   <span class="fu">print</span>(<span class="fu">t.test</span>(TG_data_for_day_1[,i]<span class="sc">~</span> PpdD1))</span>
<span id="cb121-3"><a href="R-and-stats.html#cb121-3" aria-hidden="true" tabindex="-1"></a>       }</span></code></pre></div>
<p>Give this a go on your computers.</p>
<p>What do you make of the output? It’s possible to select individual components of the output (such as the means and p-values) so we can summarise many analyses concisely.</p>
<p><code>for(i in 6:12)</code> means repeat the commands which follow a number of times by varying an index variable, here “i,” from 6 to 12 (seven times in total), incrementing it by 1 each repeat. Then, because we are analysing have TG_data_for_day_1[,i], on each iteration [,i] is substituted with [,6] then [,7] … [,12] and all our analyses are done. We are beginning to write scripts!</p>
<p>The commands we want to loop over do not have to be restricted to a single line, provided they are enclosed in curly brackets. Note that as these commands are run within a loop, the output does not get echoed to the screen unless we include the “print” statement as above.</p>
</div>
<div id="linear-regression" class="section level3" number="3.8.2">
<h3><span class="header-section-number">3.8.2</span> Linear Regression</h3>
<p>We have already come across the command to correlate two traits: <code>cor()</code>. To fit a straight line to a data set we use the R command <code>lm()</code>; for linear model. Suppose we want to study the effect of year on yield:</p>
<div class="sourceCode" id="cb122"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb122-1"><a href="R-and-stats.html#cb122-1" aria-hidden="true" tabindex="-1"></a><span class="fu">lm</span>(yield<span class="sc">~</span>year)</span></code></pre></div>
<pre><code>## 
## Call:
## lm(formula = yield ~ year)
## 
## Coefficients:
## (Intercept)         year  
##   -583.2088       0.3386</code></pre>
<p>The output is somewhat sparse. A feature of R, in contrast to many statistical packages is that by default it does not deliver multiple pages of output by default, from which you may only wish to extract a single figure. Here the output gives you the best fitting straight line:</p>
<p><span class="math inline">\(yield = -583.4 + (0.3386 * year)\)</span></p>
<p>More output is available but we need to be explicit that we wish R to produce it. First we shall rerun the analysis, but save the results:</p>
<div class="sourceCode" id="cb124"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb124-1"><a href="R-and-stats.html#cb124-1" aria-hidden="true" tabindex="-1"></a>yield_year_regression<span class="ot">&lt;-</span><span class="fu">lm</span>(yield<span class="sc">~</span>year)</span></code></pre></div>
<p>No output is generated. We can produce some using <code>summary()</code>:</p>
<div class="sourceCode" id="cb125"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb125-1"><a href="R-and-stats.html#cb125-1" aria-hidden="true" tabindex="-1"></a><span class="fu">summary</span>(yield_year_regression)</span></code></pre></div>
<pre><code>## 
## Call:
## lm(formula = yield ~ year)
## 
## Residuals:
##      Min       1Q   Median       3Q      Max 
## -18.1783  -3.2680   0.7123   3.1559  12.6978 
## 
## Coefficients:
##               Estimate Std. Error t value Pr(&gt;|t|)    
## (Intercept) -583.20878   43.96487  -13.27   &lt;2e-16 ***
## year           0.33855    0.02205   15.35   &lt;2e-16 ***
## ---
## Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1
## 
## Residual standard error: 4.915 on 374 degrees of freedom
## Multiple R-squared:  0.3866, Adjusted R-squared:  0.385 
## F-statistic: 235.8 on 1 and 374 DF,  p-value: &lt; 2.2e-16</code></pre>
<p>The most interesting part of the output is given at the end: the F-statistic and p-value for the significance of the regression - hugely significant in this example. The Multiple R-Squared is the proportion of the total sum of squares accounted for by the regression. It is also the square of the correlation coefficient between yield and year of origin. The Adjusted R-squared is the proportional reduction in variance after fitting the regression (i.e. 1-residual variance/total variance). Both these figures give an indication of how effective the regression has been in accounting for the observed variation: a significant regression does not imply that a relationship is particularly important (although it is here). Equally, with very small experiments, large proportions of variation may be accounted for, but the regression is still non-significant. This is generally an indication that you should have designed a larger experiment.</p>
<p>A more conventional display of the regression analysis is given as:</p>
<div class="sourceCode" id="cb127"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb127-1"><a href="R-and-stats.html#cb127-1" aria-hidden="true" tabindex="-1"></a><span class="fu">anova</span>(yield_year_regression)</span></code></pre></div>
<pre><code>## Analysis of Variance Table
## 
## Response: yield
##            Df Sum Sq Mean Sq F value    Pr(&gt;F)    
## year        1 5694.7  5694.7  235.75 &lt; 2.2e-16 ***
## Residuals 374 9034.0    24.2                      
## ---
## Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1</code></pre>
<p>This uses the <code>anova()</code> command. To R, anova is the name given to a form of tabular output. Formally, the analysis of variance itself is just a particular type of multiple regression analysis, and that is exactly how R treats it, as we shall see shortly.</p>
<p>We can look at a plot of the data with our fitted line a follows:</p>
<div class="sourceCode" id="cb129"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb129-1"><a href="R-and-stats.html#cb129-1" aria-hidden="true" tabindex="-1"></a><span class="fu">plot</span>(year,yield)</span>
<span id="cb129-2"><a href="R-and-stats.html#cb129-2" aria-hidden="true" tabindex="-1"></a><span class="fu">abline</span>(yield_year_regression)</span></code></pre></div>
<p><img src="bookdown-demo_files/figure-html/unnamed-chunk-82-1.png" width="672" /></p>
<p>When using plot, the name of the variate you want plotted on the x axis comes first. The additional command, <code>abline()</code>, adds the best fitting straight line. There is a lot of variation around the best fitting line, but as we have varieties from three different countries of origin, and we have tested them in three different countries, this is not surprising.</p>
<p>Finally, we look at how to extract residual values and fitted values from a regression. Large residual values for particular observations are often of use in searching for errors in data. Also, identification of the individuals or varieties responsible for large residuals may sometimes suggest some other factor which needs to be considered in the analysis. Fitted values and residuals are extracted as:</p>
<div class="sourceCode" id="cb130"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb130-1"><a href="R-and-stats.html#cb130-1" aria-hidden="true" tabindex="-1"></a><span class="fu">fitted</span>(yield_year_regression)[<span class="dv">1</span><span class="sc">:</span><span class="dv">5</span>]</span></code></pre></div>
<pre><code>##        1        2        3        4        5 
## 94.90825 92.87695 83.73608 86.10594 93.21550</code></pre>
<div class="sourceCode" id="cb132"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb132-1"><a href="R-and-stats.html#cb132-1" aria-hidden="true" tabindex="-1"></a><span class="fu">resid</span>(yield_year_regression)[<span class="dv">1</span><span class="sc">:</span><span class="dv">5</span>]</span></code></pre></div>
<pre><code>##         1         2         3         4         5 
##  8.321746 -4.326949  7.023919 -3.865936  3.444500</code></pre>
<p>Note I’ve only printed values 1 to 5 (<code>[1:5]</code>) to save space. These could be saved to other variables, or plotted:</p>
<div class="sourceCode" id="cb134"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb134-1"><a href="R-and-stats.html#cb134-1" aria-hidden="true" tabindex="-1"></a><span class="fu">plot</span>(<span class="fu">fitted</span>(yield_year_regression), <span class="fu">resid</span>(yield_year_regression))</span></code></pre></div>
<p><img src="bookdown-demo_files/figure-html/unnamed-chunk-85-1.png" width="672" /></p>
<p>A plot of residuals against fitted values is often informative. If large residuals tend to be associated with large fitted values, for example, this indicates that the error variances are not homogenous and we treat our results with more caution. Residuals which increase in magnitude with increasing fitted values are often an indication that transforming the data to logarithms before analysis may be warranted.</p>
<p>For linear regression in R, however, we can produce four diagnostic plots very easily:</p>
<div class="sourceCode" id="cb135"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb135-1"><a href="R-and-stats.html#cb135-1" aria-hidden="true" tabindex="-1"></a><span class="fu">plot</span>(yield_year_regression)</span></code></pre></div>
<p><img src="bookdown-demo_files/figure-html/unnamed-chunk-86-1.png" width="672" /><img src="bookdown-demo_files/figure-html/unnamed-chunk-86-2.png" width="672" /><img src="bookdown-demo_files/figure-html/unnamed-chunk-86-3.png" width="672" /><img src="bookdown-demo_files/figure-html/unnamed-chunk-86-4.png" width="672" /></p>
<p>The first two plots are the most important and the easiest to interpret:</p>
<p>The first plot is of residuals against fitted values as previously described. There is nothing untoward here.</p>
<p>The QQ plot takes the residuals, sorts them and standardises them by dividing them by the standard deviation, so they have a mean of zero and a variance of 1. It is possible to work out what the expected value of these sorted normal deviates should be if the error distribution was truly normal. Plotting the observed against the expected gives you a QQ plot (QQ for quantile-quantile). For well behaved data, the points should lie on a straight line with a gradient of one. The data here look acceptable.</p>
<p>The Scale-Location plot is essentially looking at the same thing as the plot of residuals against fitted values.</p>
<p>The Residuals vs Leverage shows the strength of data points on the parameters of the regression. Data points with very high or low values tend to have more effect (leverage) on the regression equation. This can be quantified. If a data point with high leverage also has a big residual (positive or negative) it is an indication that the data for that observation wants checking for errors, and maybe even deleting. Cook’s distance, which is also mentioned in this plot, is a measure of the improvement in error sum of squares if a particular data point were dropped. In the plot above, there is nothing to be checked: the data are good.</p>
</div>
<div id="multiple-regression" class="section level3" number="3.8.3">
<h3><span class="header-section-number">3.8.3</span> Multiple regression</h3>
<p>Multiple regression in R requires little more than simple linear regression. The <code>lm()</code> command is still used. All that is required is to specify a more complex model using the syntax described in the Basic R Syntax section of this guide:</p>
<div class="sourceCode" id="cb136"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb136-1"><a href="R-and-stats.html#cb136-1" aria-hidden="true" tabindex="-1"></a>yield_mult_reg<span class="ot">&lt;-</span><span class="fu">lm</span>(yield<span class="sc">~</span>year<span class="sc">+</span>ORIGIN<span class="sc">+</span>PpdD1<span class="sc">+</span>Rht2<span class="sc">+</span>FT<span class="sc">+</span>HT<span class="sc">+</span>AWNS)</span>
<span id="cb136-2"><a href="R-and-stats.html#cb136-2" aria-hidden="true" tabindex="-1"></a>yield_mult_reg</span></code></pre></div>
<pre><code>## 
## Call:
## lm(formula = yield ~ year + ORIGIN + PpdD1 + Rht2 + FT + HT + 
##     AWNS)
## 
## Coefficients:
## (Intercept)         year    ORIGINFRA    ORIGINGBR        PpdD1         Rht2  
##   -460.4045       0.2639      -0.5753       0.9456      -0.2138       2.2100  
##          FT           HT         AWNS  
##      0.2335      -0.1403      -0.4815</code></pre>
<p>Thus yield is predicted as:</p>
<p>-461.2409<br />
+ 0.2645 x year<br />
+ -0.5637 if French origin<br />
+ 0.9510 if UK origin<br />
+ -0.2157 x Pppd1<br />
+ 2.2112 x Rht2<br />
+ 0.2311 x FT<br />
+ -0.1402 x HT<br />
+ -0.6402 x Awns</p>
<p>Note this form of analysis is acceptable for PpdD1 and Rht2, SNPs and other binary markers provided the markers are coded numerically. 0 and 1 are ideal codes, but 0 can sometimes be confused with a missing value, especially if data are to be analysed in packages other than R. 1, 2 coding is also acceptable. Numeric coding cannot be used at all for multiallelic markers since it implies that alleles coded with a higher number have a higher value than those with a lower number.</p>
<p>Note the order in which the variates are supplied to lm does not affect the estimated coefficients:</p>
<div class="sourceCode" id="cb138"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb138-1"><a href="R-and-stats.html#cb138-1" aria-hidden="true" tabindex="-1"></a><span class="fu">lm</span>(yield<span class="sc">~</span> year<span class="sc">+</span>ORIGIN<span class="sc">+</span>FT<span class="sc">+</span>Rht2<span class="sc">+</span>PpdD1<span class="sc">+</span>HT<span class="sc">+</span>AWNS)</span></code></pre></div>
<pre><code>## 
## Call:
## lm(formula = yield ~ year + ORIGIN + FT + Rht2 + PpdD1 + HT + 
##     AWNS)
## 
## Coefficients:
## (Intercept)         year    ORIGINFRA    ORIGINGBR           FT         Rht2  
##   -460.4045       0.2639      -0.5753       0.9456       0.2335       2.2100  
##       PpdD1           HT         AWNS  
##     -0.2138      -0.1403      -0.4815</code></pre>
<p>Will give the same answer. However, this is not the case for estimates of significance:</p>
<div class="sourceCode" id="cb140"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb140-1"><a href="R-and-stats.html#cb140-1" aria-hidden="true" tabindex="-1"></a><span class="fu">anova</span>(yield_mult_reg)</span></code></pre></div>
<pre><code>## Analysis of Variance Table
## 
## Response: yield
##            Df Sum Sq Mean Sq  F value    Pr(&gt;F)    
## year        1 5177.9  5177.9 253.1041 &lt; 2.2e-16 ***
## ORIGIN      2  663.2   331.6  16.2102 1.879e-07 ***
## PpdD1       1  121.4   121.4   5.9356  0.015348 *  
## Rht2        1  554.0   554.0  27.0786 3.375e-07 ***
## FT          1   33.0    33.0   1.6122  0.205051    
## HT          1  172.5   172.5   8.4344  0.003921 ** 
## AWNS        1    5.1     5.1   0.2475  0.619158    
## Residuals 342 6996.4    20.5                       
## ---
## Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1</code></pre>
<div class="sourceCode" id="cb142"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb142-1"><a href="R-and-stats.html#cb142-1" aria-hidden="true" tabindex="-1"></a><span class="fu">anova</span>(<span class="fu">lm</span>(yield<span class="sc">~</span> year<span class="sc">+</span>ORIGIN<span class="sc">+</span>FT<span class="sc">+</span>Rht2<span class="sc">+</span>PpdD1<span class="sc">+</span>HT<span class="sc">+</span>AWNS))</span></code></pre></div>
<pre><code>## Analysis of Variance Table
## 
## Response: yield
##            Df Sum Sq Mean Sq  F value    Pr(&gt;F)    
## year        1 5177.9  5177.9 253.1041 &lt; 2.2e-16 ***
## ORIGIN      2  663.2   331.6  16.2102 1.879e-07 ***
## FT          1  164.7   164.7   8.0498  0.004822 ** 
## Rht2        1  543.2   543.2  26.5504 4.349e-07 ***
## PpdD1       1    0.5     0.5   0.0261  0.871836    
## HT          1  172.5   172.5   8.4344  0.003921 ** 
## AWNS        1    5.1     5.1   0.2475  0.619158    
## Residuals 342 6996.4    20.5                       
## ---
## Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1</code></pre>
<p>In the two analyses we’ve swapped the order in which PpdD1 and FT were fitted. In the first anova table, the sum of squares for PpdD1 is 121.4 and 33.0 for FT. In the second analysis, the sum of squares is 0.5 for PpdD1 and 164.7 for FT. The sums of squares for Rht2 has changed a bit but all the others, and most importantly the Residual SS, are unchanged. In unbalanced experiments such as this, where combinations of PpdD1 and height (and most other things too) are not all equally represented, the results from the analysis of variance depend on the order in which the terms are represented in the model. However, there is an easy way to interpret this table. Taking output from the last analysis a line at a time:</p>
<p>The effect for year is fitted first, and is found to be statistically significant:</p>
<p>year <code>P &lt; 2.2e-16 ***</code></p>
<p>After fitting the year, origin it is fitted next,</p>
<p>ORIGIN <code>P =  1.867e-07 ***</code></p>
<p>then FT:</p>
<p>FT <code>P =  0.004813 **</code></p>
<p>The p-value of 0.0049 for FT is the significance for FT, <em>after</em> accounting for any effect of year and origin. By the time we get to PpdD1, its effect after fitting everything above it in the table is non-significant:</p>
<p>PpdD1 <code>P = 0.871809</code></p>
<p>In designed experiments, where different combinations of treatments and factors are usually equally represented, or balanced, the order in which terms are fitted makes no difference; the terms are said to be <em>orthogonal</em>. Balance not only has the property of making the terms orthogonal, it also makes the arithmetic very much easier. This was very important before the advent of readily available computers. However, the requirement for balance, solely from the point of view of data analysis, is now no longer required and many contemporary designs for variety trials (for example alpha-designs) are not balanced and would be impossible to analyse without a computer. A readable account of a contemporary approach to experimental design is given in the book Mead, Gilmour and Mead: “Statistical Principles for the Design of Experiments.”</p>
<p>In this example, the interpretation is clear: PpdD1 is known to be the major locus determining flowering time in wheat, but it doesn’t control all the variation. If we had to choose between Ppd or FT in the model, we would chose FT because its sum of squares is bigger: we account for more of the variation in the data. But here it makes sense to fit Ppd first and then to test if there is a residual effect of FT on yield. If you have time, try the same thing for Rht2 and HT.</p>
<p>Our prior knowledge isn’t always so good and the interpretation and order of fitting isn’t always so clear. Selecting the order in which terms are fitted, and selecting which terms to include in the final model and which to exclude is something of an art, which we shall not develop here. There are formal methods to assist in this process. These too are not covered here, but are available within R. Generally, with genetic analysis, it is usual to account for variation attributable to causes other than genes or markers first, and then fit the genetic effects. This is a conservative approach. If we had followed it here we would not have detected an effect for PpdD1.</p>
<p>Note that <code>resid()</code>, <code>fitted()</code>, <code>plot()</code> and <code>summary()</code> work for multiple regression exactly as for simple linear regression.
Before we leave regression, compare these two simple analyses of the relationship between yield and PpdD1</p>
<div class="sourceCode" id="cb144"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb144-1"><a href="R-and-stats.html#cb144-1" aria-hidden="true" tabindex="-1"></a><span class="fu">t.test</span>(yield<span class="sc">~</span>PpdD1,<span class="at">var.equal=</span>T)</span></code></pre></div>
<pre><code>## 
##  Two Sample t-test
## 
## data:  yield by PpdD1
## t = 3.0463, df = 374, p-value = 0.002481
## alternative hypothesis: true difference in means is not equal to 0
## 95 percent confidence interval:
##  0.8254243 3.8311401
## sample estimates:
## mean in group 0 mean in group 1 
##        92.35522        90.02694</code></pre>
<div class="sourceCode" id="cb146"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb146-1"><a href="R-and-stats.html#cb146-1" aria-hidden="true" tabindex="-1"></a><span class="fu">anova</span>(<span class="fu">lm</span>(yield<span class="sc">~</span> PpdD1))</span></code></pre></div>
<pre><code>## Analysis of Variance Table
## 
## Response: yield
##            Df  Sum Sq Mean Sq F value   Pr(&gt;F)   
## PpdD1       1   356.6  356.61    9.28 0.002481 **
## Residuals 374 14372.1   38.43                    
## ---
## Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1</code></pre>
<p>The p-values are identical. For a regression analysis on a single variate with only two values or classes, the two tests are equivalent. In fact:</p>
<p><span class="math inline">\(t^2 = F\)</span></p>
<p>In this case</p>
<p><span class="math inline">\(3.0463^2 = 9.280\)</span>.</p>
</div>
<div id="the-analysis-of-variance" class="section level3" number="3.8.4">
<h3><span class="header-section-number">3.8.4</span> The analysis of variance</h3>
<p>In the t-test, we test if the difference between two treatment means is statistically significant. This is a special case of the Analysis of Variance in which we test if differences among multiple treatments are jointly statistically significant. We could compare multiple treatments by carrying out multiple t tests, but this increases the risk that at least one test will be declared significant by chance alone - the so called problem of multiple testing. In addition the interpretation of results becomes increasingly complex. (There is, in fact, an R command that automates this procedure and includes an adjustment for multiple testing: pairwise.t.test). The omnibus test for significance of all means, considered together, that the Analysis of Variance offers is therefore of great value.</p>
<p>The principal of the Analysis of Variance is that, in the absence of any genuine difference among means, the variability among those means can be predicted from the variability from observation to observation within each treatment. This argument is little more than saying that the variance of a mean is just the variance among the observations that contribute to that mean divided by the number of observations contributing to the mean:</p>
<p><span class="math inline">\(V\hat{x} = Vx/n\)</span></p>
<p>We therefore have independent estimates of <span class="math inline">\(\hat{x}\)</span>, one directly from the means, and one from the within treatments variance (/n). If there are no true differences in the means of the treatments, these two estimates of variance are expected to be the same though they will still differ through sampling variation. However, if the means differ, in addition, through differential responses to the treatments we no longer expect them to be the same. Rather:</p>
<p><span class="math inline">\(V\hat{x} &gt; Vx/n\)</span></p>
<p>The test for statistically significant differences among the means is therefore a variance ratio, or F test: the variance among treatment means is divided by the expected variance calculated within treatments. This is an oversimplification: differences in the number of observations within treatments must also be taken into account and with more complicated experimental designs the analysis is also more complicated, but the basic principal remains the same: differences in means inflate the estimate of variance between treatment means compared to the independent estimate of variance from observations within treatments.</p>
<p>We’ll start with a simple example. Does height differ significantly between countries of origin?</p>
<div class="sourceCode" id="cb148"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb148-1"><a href="R-and-stats.html#cb148-1" aria-hidden="true" tabindex="-1"></a><span class="fu">boxplot</span>(HT<span class="sc">~</span>ORIGIN,<span class="at">main=</span><span class="st">&quot;Height by country of origin&quot;</span>)</span></code></pre></div>
<p><img src="bookdown-demo_files/figure-html/unnamed-chunk-93-1.png" width="672" /></p>
<p>It’s clear from this that there is a difference - German lines tend to be taller. The command for an analysis of variance is just like that for linear regression:</p>
<div class="sourceCode" id="cb149"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb149-1"><a href="R-and-stats.html#cb149-1" aria-hidden="true" tabindex="-1"></a><span class="fu">anova</span>(<span class="fu">lm</span>(HT<span class="sc">~</span>ORIGIN))</span></code></pre></div>
<pre><code>## Analysis of Variance Table
## 
## Response: HT
##            Df  Sum Sq Mean Sq F value    Pr(&gt;F)    
## ORIGIN      2  4274.3 2137.17  47.307 &lt; 2.2e-16 ***
## Residuals 371 16760.6   45.18                      
## ---
## Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1</code></pre>
<p>There are three countries of origin: so we have a test with 2 degrees of freedom. That is because if we know a line didn’t come from France or Germany, then we know (for this dataset) that it must have come from the UK: there are only two independent countries of origin. We’ve seen this before: in the earlier multiple regression analysis of yield we included origin and saw that it gave an effect for the UK and an effect for France but not for Germany. The German effect was the default - the mean - to be adjusted by the appropriate effect if a line wasn’t German. In fact if we had two extra columns of data, one with zeros everywhere except if a variety was from the UK, when we had a 1, and the second with zeros except if a variety was French, when we had a 1, then including these two columns in the analysis instead of “origin” would give exactly the same answer. And if we swapped out the “UK column” and inserted a new “German” column, we would still get exactly the same answer. Try it if you don’t believe me!</p>
<p>To get more information from our analysis of variance, we can use all the methods introduced earlier for linear and multiple regressions: as stated, the analysis of variance is just a special case of multiple regression.</p>
<p>Let’s finish off the analysis of variance with something more interesting. We know that the three countries tend to have different requirements for optimum flowering times and height. France tends to require earlier flowering lines and many varieties carry the PpdD1 early allele to achieve this. German lines tend to be taller because German wheat is often treated wheat with a growth regulator which reduces height, so there is less requirement for varieties to be bred for reduced height, commonly by using a dwarfing allele at one of the rht loci. After adjusting HT for country of origin, can we detect the effects of these major phenological loci? Do they interact?</p>
<p>Let’s have a look at the data first:</p>
<div class="sourceCode" id="cb151"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb151-1"><a href="R-and-stats.html#cb151-1" aria-hidden="true" tabindex="-1"></a><span class="fu">boxplot</span>(HT<span class="sc">~</span>PpdD1<span class="sc">*</span>Rht2,<span class="at">main=</span><span class="st">&quot;HT by PpdD1 and rht2&quot;</span>)</span></code></pre></div>
<p><img src="bookdown-demo_files/figure-html/unnamed-chunk-95-1.png" width="672" /></p>
<p>Note the coding within the command to indicate we wanted to see four classes. 0.0 represents the late Ppd allele and the tall Rht2 allele. 1.0 is the Ppd early and Rht2 tall allele, and so on. It’s clear that both loci have a big effect on height, but you can see also that although 1.1 is the shortest class, there is a bigger drop from 0.0 to 0.1 and 1.0 than there is from 1.0 and 0.1 to 1.1</p>
<p>Analysing the data, while taking into account country of origin?</p>
<div class="sourceCode" id="cb152"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb152-1"><a href="R-and-stats.html#cb152-1" aria-hidden="true" tabindex="-1"></a><span class="fu">anova</span>(<span class="fu">lm</span>(HT<span class="sc">~</span>ORIGIN<span class="sc">+</span>Rht2<span class="sc">*</span>PpdD1))</span></code></pre></div>
<pre><code>## Analysis of Variance Table
## 
## Response: HT
##             Df  Sum Sq Mean Sq  F value    Pr(&gt;F)    
## ORIGIN       2  4141.7  2070.8  68.2521 &lt; 2.2e-16 ***
## Rht2         1  3691.8  3691.8 121.6777 &lt; 2.2e-16 ***
## PpdD1        1  1576.7  1576.7  51.9647 3.575e-12 ***
## Rht2:PpdD1   1   119.8   119.8   3.9501   0.04766 *  
## Residuals  346 10498.0    30.3                       
## ---
## Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1</code></pre>
<p>It appears there is some weak evidence that Rht2 and PpdD1are interacting. But we are ignoring other confounding factors such as flowering time itself, and many possible interactions. We can fit a complete model:</p>
<div class="sourceCode" id="cb154"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb154-1"><a href="R-and-stats.html#cb154-1" aria-hidden="true" tabindex="-1"></a><span class="fu">anova</span>(<span class="fu">lm</span>(HT<span class="sc">~</span>ORIGIN<span class="sc">*</span>FT<span class="sc">*</span>Rht2<span class="sc">*</span>PpdD1))</span></code></pre></div>
<pre><code>## Analysis of Variance Table
## 
## Response: HT
##                 Df Sum Sq Mean Sq  F value    Pr(&gt;F)    
## ORIGIN           2 4141.7  2070.8  76.8732 &lt; 2.2e-16 ***
## FT               1 1049.5  1049.5  38.9589 1.312e-09 ***
## Rht2             1 4681.8  4681.8 173.7978 &lt; 2.2e-16 ***
## PpdD1            1  152.2   152.2   5.6487 0.0180309 *  
## ORIGIN:FT        2  307.4   153.7   5.7049 0.0036625 ** 
## ORIGIN:Rht2      2   76.2    38.1   1.4146 0.2444857    
## FT:Rht2          1  308.9   308.9  11.4673 0.0007928 ***
## ORIGIN:PpdD1     2  133.7    66.8   2.4813 0.0851744 .  
## FT:PpdD1         1   24.3    24.3   0.9025 0.3428055    
## Rht2:PpdD1       1    5.5     5.5   0.2033 0.6523338    
## ORIGIN:FT:Rht2   2  113.2    56.6   2.1016 0.1238698    
## FT:Rht2:PpdD1    1   36.2    36.2   1.3455 0.2469017    
## Residuals      334 8997.4    26.9                       
## ---
## Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1</code></pre>
</div>
</div>
<div id="categorical-data---the-chi-squared-test" class="section level2" number="3.9">
<h2><span class="header-section-number">3.9</span> Categorical data - the <em>chi</em>-squared test</h2>
<p>Suppose we wish to test if there is an association between Rht2 and PpdD1. They are not linked, but perhaps breeders have favoured particular combinations of +/- at the two loci. Or an association might arise because of the way the varieties have been selected for inclusion in the panel of lines. The standard method of analysis of such data is the contingency chi-squared test.
First we need to format our data into a table:</p>
<div class="sourceCode" id="cb156"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb156-1"><a href="R-and-stats.html#cb156-1" aria-hidden="true" tabindex="-1"></a>rht.ppd.table<span class="ot">&lt;-</span><span class="fu">table</span>(Rht2,PpdD1) </span>
<span id="cb156-2"><a href="R-and-stats.html#cb156-2" aria-hidden="true" tabindex="-1"></a>rht.ppd.table</span></code></pre></div>
<pre><code>##     PpdD1
## Rht2   0   1
##    0 106  53
##    1 170  25</code></pre>
<p>From this table, we can see that the allele frequency of allele 1 at PpdD1 is:</p>
<p><strong>(53+25)/(106+170+53+25) or 0.22</strong></p>
<p>and similarly the allele frequency of allele 1 at Rht2 is 0.55</p>
<p>If these loci are independent of each other in this sample, the allele carried by a variety at Rht2 will be independent of the allele carried by the same variety at PpdD1. In this case, the expected frequency of (Rht2 allele 1 + PpdD1 allele 1) individuals will be 0.22 x 0.55 or 0.12. The predicted number of “11” haplotypes is therefore 0.12 x (106+170+53+25) or 42. We have observed 25. This same exercise can be carried out for each of the other three combinations of alleles at the two loci. (These combinations are commonly called haplotypes, though this term strictly only applies if the two loci are linked). If deviations between observed and expected numbers are sufficiently large, we draw the conclusion that genotypes at the two loci are not independent of each other. The statistical test is a chi-squared with 1 degree of freedom, calculated as:</p>
<p><span class="math display">\[\chi^2 = \sum \frac {(O - E)^2}{E}\]</span>
O represents the observed numbers and E the expected.</p>
<p>This chi-squared test is simply carried out as:</p>
<div class="sourceCode" id="cb158"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb158-1"><a href="R-and-stats.html#cb158-1" aria-hidden="true" tabindex="-1"></a><span class="fu">chisq.test</span>(rht.ppd.table)</span></code></pre></div>
<pre><code>## 
##  Pearson&#39;s Chi-squared test with Yates&#39; continuity correction
## 
## data:  rht.ppd.table
## X-squared = 20.275, df = 1, p-value = 6.706e-06</code></pre>
<p>The result is obviously hugely significant; the p-value is vanishingly small. What might you conclude about how breeders have selected on these two loci?</p>
<p>Chi squared tests can have a problem with low numbers of observations in some cells. Try
this:</p>
<div class="sourceCode" id="cb160"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb160-1"><a href="R-and-stats.html#cb160-1" aria-hidden="true" tabindex="-1"></a>rht.ppd.table.small<span class="ot">&lt;-</span><span class="fu">round</span>(rht.ppd.table<span class="sc">/</span><span class="dv">10</span>)</span>
<span id="cb160-2"><a href="R-and-stats.html#cb160-2" aria-hidden="true" tabindex="-1"></a>rht.ppd.table.small</span></code></pre></div>
<pre><code>##     PpdD1
## Rht2  0  1
##    0 11  5
##    1 17  2</code></pre>
<p>We’ve created a table 1/10 of the size of the original. <code>round()</code> rounds real numbers to integers in the standard manner (with 1/2 always rounded down).</p>
<div class="sourceCode" id="cb162"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb162-1"><a href="R-and-stats.html#cb162-1" aria-hidden="true" tabindex="-1"></a><span class="fu">chisq.test</span>(rht.ppd.table.small)</span></code></pre></div>
<pre><code>## Warning in chisq.test(rht.ppd.table.small): Chi-squared approximation may be
## incorrect</code></pre>
<pre><code>## 
##  Pearson&#39;s Chi-squared test with Yates&#39; continuity correction
## 
## data:  rht.ppd.table.small
## X-squared = 1.2161, df = 1, p-value = 0.2701</code></pre>
<p>If you run this, you should see this warning message:</p>
<p><strong>Chi-squared approximation may be incorrect</strong></p>
<p>The warning message is given because some cells have expected counts less than 5. Under this threshold, there is a chance that the chi-squared test will give misleading results. Note it is the <em>expected</em> count that matters, not the <em>observed</em> - which can legitimately be zero and often is when we are considering closely linked SNPs for example, where we may observe only two or three of the four possible haplotypes. There will be more on this later in the course when we come to discuss linkage disequilibrium. We can extract and examine the expected values:</p>
<div class="sourceCode" id="cb165"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb165-1"><a href="R-and-stats.html#cb165-1" aria-hidden="true" tabindex="-1"></a><span class="fu">chisq.test</span>(rht.ppd.table.small)<span class="sc">$</span>expected</span></code></pre></div>
<pre><code>## Warning in chisq.test(rht.ppd.table.small): Chi-squared approximation may be
## incorrect</code></pre>
<pre><code>##     PpdD1
## Rht2    0   1
##    0 12.8 3.2
##    1 15.2 3.8</code></pre>
<div class="sourceCode" id="cb168"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb168-1"><a href="R-and-stats.html#cb168-1" aria-hidden="true" tabindex="-1"></a><span class="fu">chisq.test</span>(rht.ppd.table.small)<span class="sc">$</span>observed</span></code></pre></div>
<pre><code>## Warning in chisq.test(rht.ppd.table.small): Chi-squared approximation may be
## incorrect</code></pre>
<pre><code>##     PpdD1
## Rht2  0  1
##    0 11  5
##    1 17  2</code></pre>
<p>The expected count in two cells is less than five. There is an approximate correction, Yates’s correction, that takes into account the potential failure of the test statistic to follow a chi-squared distribution when the expected numbers were low. This has been applied automatically. To see results without Yates’ correction:</p>
<div class="sourceCode" id="cb171"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb171-1"><a href="R-and-stats.html#cb171-1" aria-hidden="true" tabindex="-1"></a><span class="fu">chisq.test</span>(rht.ppd.table.small,<span class="at">correct=</span>F)</span></code></pre></div>
<pre><code>## Warning in chisq.test(rht.ppd.table.small, correct = F): Chi-squared
## approximation may be incorrect</code></pre>
<pre><code>## 
##  Pearson&#39;s Chi-squared test
## 
## data:  rht.ppd.table.small
## X-squared = 2.3314, df = 1, p-value = 0.1268</code></pre>
<p>The <code>correct = F</code> option turns Yates’ correction off and the chi-sq test statistic is larger.</p>
<p>It is increasingly common, and better, to derive the distribution of the test statistic empirically, by repeated randomisation or permutation of the observed data, followed by recalculation of test statistic. The proportion of times the randomised test statistic is greater than or equal to the observed test statistic is then the empirical p-value. Here the randomisation procedure is very simple: the data for Rht2 are randomised over subjects, while the data for PpdD1 are held constant. For a chi-squared test, R automates this procedure:</p>
<div class="sourceCode" id="cb174"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb174-1"><a href="R-and-stats.html#cb174-1" aria-hidden="true" tabindex="-1"></a><span class="fu">chisq.test</span>(rht.ppd.table.small,<span class="at">simulate.p.value =</span> T,<span class="at">B=</span><span class="dv">10000</span>)</span></code></pre></div>
<pre><code>## 
##  Pearson&#39;s Chi-squared test with simulated p-value (based on 10000
##  replicates)
## 
## data:  rht.ppd.table.small
## X-squared = 2.3314, df = NA, p-value = 0.2011</code></pre>
<p>The value of B, the number of randomisations, can be set by the user. The default value is 2000, but I’ve run 10,000 which is still very quick when the number of observations is small.</p>
<p>The contingency chi-squared test will easily accommodate larger tables; a 10 x 10 table for example. In the context of genetic data, this could be used to test for association between genetic regions with multiple haplotype alleles.</p>
</div>
<div id="more-useful-information" class="section level2" number="3.10">
<h2><span class="header-section-number">3.10</span> More useful information</h2>
<div id="plotting-graphs" class="section level3" number="3.10.1">
<h3><span class="header-section-number">3.10.1</span> Plotting Graphs</h3>
<p>Graphical methods were introduced in context in the sections on summary statistics and on basic statistical analysis. The following methods have been used:</p>
<p><code>hist()</code> produces a histogram</p>
<p><code>plot()</code> produces a scatter graph</p>
<p><code>abline()</code> adds a line of best fit to a scatter graph</p>
<p><code>boxplot()</code> produces a box-and-whisker plot.</p>
<p><code>pairs()</code> produces a matrix of scatter plots.</p>
<p>Control of many graphics parameters - font sizes, colours and so on - is given by the command <code>par()</code>. Type <code>help(par)</code> for more details. One useful option is:</p>
<p><code>par(mfrow=c(x,y))</code></p>
<p>This will plot graphs successively in a window x graphs wide by y graphs deep. Once the panel of graphs is complete, the next graph will overwrite the whole window and start filling the panel again. For example, the following commands set up a panel two graphs deep by three wide and fills in the first four slots.</p>
<div class="sourceCode" id="cb176"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb176-1"><a href="R-and-stats.html#cb176-1" aria-hidden="true" tabindex="-1"></a><span class="fu">par</span>(<span class="at">mfrow=</span><span class="fu">c</span>(<span class="dv">2</span>,<span class="dv">3</span>))</span>
<span id="cb176-2"><a href="R-and-stats.html#cb176-2" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="dv">3</span>) <span class="fu">plot</span>(<span class="fu">runif</span>(<span class="dv">100</span>),<span class="fu">runif</span>(<span class="dv">100</span>))</span>
<span id="cb176-3"><a href="R-and-stats.html#cb176-3" aria-hidden="true" tabindex="-1"></a><span class="fu">hist</span>(<span class="fu">rnorm</span>(<span class="dv">1000</span>))</span></code></pre></div>
<p><img src="bookdown-demo_files/figure-html/unnamed-chunk-106-1.png" width="672" /></p>
<p>As you can see, the graphs need not all be of the same type. The related version of the command <code>par(mfcol=c(x,y))</code> will produce a graphical window of the same dimensions but fill it in column order rather than in row order.</p>
<p>A benefit of RStudio over native R is that graphs are saved and you can scroll back and forwards over previous efforts and export them, either as files or to the clipboard. In native R, the export of graphs is no more complicated, but they are overwritten so you cannot scroll to inspect previous plots but must recreate them.</p>
<p>Although R generates graphs very quickly and simply, the labeling and formatting are often not ideal. To alter graph titles, include the options:</p>
<p><code>main="Main title"</code></p>
<p><code>sub="subtitle"</code></p>
<p><code>xlab="x-label"</code></p>
<p><code>ylab="y-label"</code></p>
<p>For example:</p>
<div class="sourceCode" id="cb177"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb177-1"><a href="R-and-stats.html#cb177-1" aria-hidden="true" tabindex="-1"></a><span class="co">#switch back to plotting a single plot</span></span>
<span id="cb177-2"><a href="R-and-stats.html#cb177-2" aria-hidden="true" tabindex="-1"></a><span class="fu">par</span>(<span class="at">mfrow=</span><span class="fu">c</span>(<span class="dv">1</span>,<span class="dv">1</span>))</span>
<span id="cb177-3"><a href="R-and-stats.html#cb177-3" aria-hidden="true" tabindex="-1"></a><span class="co">#make plot</span></span>
<span id="cb177-4"><a href="R-and-stats.html#cb177-4" aria-hidden="true" tabindex="-1"></a><span class="fu">plot</span>(year,yield,<span class="at">main=</span><span class="st">&quot;TG panel yields over time&quot;</span>,<span class="at">sub=</span><span class="st">&quot;example scatterplot&quot;</span>,<span class="at">ylab=</span><span class="st">&quot;yield in quintal / ha&quot;</span>,<span class="at">xlab=</span><span class="st">&quot;year first in trial&quot;</span>)</span></code></pre></div>
<p><img src="bookdown-demo_files/figure-html/unnamed-chunk-107-1.png" width="672" /></p>
<p>We have only used the basic plotting commands which come with the core R package. Very highly rated, and increasingly used is a package called “ggplot2” <a href="http://ggplot2.org/" class="uri">http://ggplot2.org/</a>. This is generally included as part of the tidyverse too.There is no time go into this in detail. Select ggplot2 from the “Packages” tab in the bottom right window to active the package. If not installed, click on the “Packages” tab in the bottom right window, then click “install” and enter ggplot2 as the package you wish to install. Alternatively, you can run this line: <code>install.packages("ggplot2")</code>, then load the package from your library:</p>
<div class="sourceCode" id="cb178"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb178-1"><a href="R-and-stats.html#cb178-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(<span class="st">&quot;ggplot2&quot;</span>)</span></code></pre></div>
<p>Then:</p>
<div class="sourceCode" id="cb179"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb179-1"><a href="R-and-stats.html#cb179-1" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(<span class="at">data=</span>TG_data_for_day_1,<span class="fu">aes</span>(<span class="at">x=</span>year,<span class="at">y=</span>yield)) <span class="sc">+</span> <span class="fu">geom_point</span>(<span class="at">shape=</span><span class="dv">1</span>)</span></code></pre></div>
<p><img src="bookdown-demo_files/figure-html/unnamed-chunk-109-1.png" width="672" /></p>
<p><code>aes</code> stands for aesthetics. If you think this look better and is easier, then ggplot2 is for you. See the web site for more details.</p>
<p>There’s plenty more you can do with ggplot2. There are some good <a href="https://www.rstudio.com/wp-content/uploads/2015/03/ggplot2-cheatsheet.pdf">cheat sheets</a> available too.</p>

<div class="rmdnote">
There are a number of other very useful cheatsheets for R and RStudio, you can take a look at many <a href="https://rstudio.com/resources/cheatsheets/">here.</a> It is always useful to have these handy when you are travelling through the R universe.
</div>
</div>
<div id="probability-distributions" class="section level3" number="3.10.2">
<h3><span class="header-section-number">3.10.2</span> Probability distributions</h3>
<p>In this section we describe how to use R to look up the p-value associated with the most
commonly used test statistics and how to look up test statistics associated with p-values.</p>
<p>Firstly the p-value associated with chi-squared:</p>
<div class="sourceCode" id="cb180"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb180-1"><a href="R-and-stats.html#cb180-1" aria-hidden="true" tabindex="-1"></a><span class="fu">pchisq</span>(<span class="fl">3.84</span>,<span class="dv">1</span>,<span class="at">lower.tail=</span>F)</span></code></pre></div>
<pre><code>## [1] 0.05004352</code></pre>
<p>The parameters given to <code>pchisq</code>, in order, are:</p>
<p><code>3.84</code> value of the test statistic.</p>
<p><code>1</code> the degrees of freedom</p>
<p><code>lower.tail</code> If set to <code>T</code> (the default) the result is the cumulative distribution up to the value of the test statistic <code>- 0.95</code> in the example. For significance testing we require the area of the upper tail: 1-0.95 and so set <code>lower.tail=F</code>.</p>
<p>To calculate a chi squared from a p-value we use the command qchisq:</p>
<div class="sourceCode" id="cb182"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb182-1"><a href="R-and-stats.html#cb182-1" aria-hidden="true" tabindex="-1"></a><span class="fu">qchisq</span>(<span class="fl">0.05</span>,<span class="dv">1</span>,<span class="at">lower.tail=</span>F)</span></code></pre></div>
<pre><code>## [1] 3.841459</code></pre>
<p>The syntax is identical to that for <code>pchisq()</code>, so we are required to include <code>lower.tail=F</code>.
Examples for the F distribution are shown below.</p>
<div class="sourceCode" id="cb184"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb184-1"><a href="R-and-stats.html#cb184-1" aria-hidden="true" tabindex="-1"></a><span class="fu">pf</span>(<span class="fl">3.84</span>,<span class="dv">1</span>,<span class="dv">1000</span>,<span class="at">lower.tail=</span>F)</span></code></pre></div>
<pre><code>## [1] 0.05032099</code></pre>
<div class="sourceCode" id="cb186"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb186-1"><a href="R-and-stats.html#cb186-1" aria-hidden="true" tabindex="-1"></a><span class="fu">qf</span>(<span class="fl">0.05</span>,<span class="dv">1</span>,<span class="dv">1000</span>,<span class="at">lower.tail=</span>F)</span></code></pre></div>
<pre><code>## [1] 3.850775</code></pre>
<p>The examples here are for 1 degree of freedom for the numerator and 1000 degrees of freedom for the denominator. The results are identical to those for a chi-squared test with 1 df. In fact, a chi-squared test with n degrees of freedom is identical to an F test with n degrees of freedom in the numerator and a very large number of degrees of freedom (ideally infinite) in the denominator.</p>
<p>Values for probabilities associated with a normal distribution are:</p>
<div class="sourceCode" id="cb188"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb188-1"><a href="R-and-stats.html#cb188-1" aria-hidden="true" tabindex="-1"></a><span class="fu">pnorm</span>(<span class="fl">1.96</span>,<span class="at">lower.tail=</span>F)</span></code></pre></div>
<pre><code>## [1] 0.0249979</code></pre>
<p>This probability is for a standardised normal distribution: with a mean of zero and a variance of 1. The probability is for a single tail of the distribution. Generally, we would require the result for a two tailed test - the probability of values higher than 1.96 and lower than -1.96. This probability is just double that for a single tail: 0.05 in this case. Again, this is the same value as for chi-squared with 1 degree of freedom. If a variate has a standardised normal distribution, the variate squared has a chi-squared distribution with 1 degree of freedom:</p>
<p><span class="math inline">\(1.96^2 = -1.96^2 = 3.84\)</span></p>
<p>Probabilities associated with normally distributed variables with different means and variances are produced by specifying the mean and variance.</p>
<p><code>pnorm(q,mean=x,sd=y)</code> with the inverse function: <code>qnorm(p,mean=x,sd=y)</code></p>
<p><code>lower.tail=F</code> can be added if required. The values for mean and standard deviation are now user specified (substitute for x and y in pnorm and qnorm). The default values are 0 and 1.</p>
<p>Finally, the t-test:</p>
<div class="sourceCode" id="cb190"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb190-1"><a href="R-and-stats.html#cb190-1" aria-hidden="true" tabindex="-1"></a><span class="fu">pt</span>(<span class="fl">1.96</span>,<span class="dv">1000</span>,<span class="at">lower.tail=</span>F) </span></code></pre></div>
<pre><code>## [1] 0.02513659</code></pre>
<div class="sourceCode" id="cb192"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb192-1"><a href="R-and-stats.html#cb192-1" aria-hidden="true" tabindex="-1"></a><span class="fu">pt</span>(<span class="sc">-</span><span class="fl">1.96</span>,<span class="dv">1000</span>) </span></code></pre></div>
<pre><code>## [1] 0.02513659</code></pre>
<p>We usually carry out a two sided t test - so we require the sum of the lower and upper tail probabilities: equal to two times the single tailed probability. The inverse function follows the usual format and nomenclature:</p>
<div class="sourceCode" id="cb194"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb194-1"><a href="R-and-stats.html#cb194-1" aria-hidden="true" tabindex="-1"></a><span class="fu">qt</span>(<span class="fl">0.025</span>,<span class="dv">10000</span>,<span class="at">lower.tail=</span>F)</span></code></pre></div>
<pre><code>## [1] 1.960201</code></pre>
</div>
<div id="random-number-generation" class="section level3" number="3.10.3">
<h3><span class="header-section-number">3.10.3</span> Random number generation</h3>
<p>R has excellent random number generators for several distributions:</p>
<p><code>rnorm(n, mean=0, sd=1)</code></p>
<p><code>runif(n, min=0, max=1)</code></p>
<p><code>rchisq(n, df, ncp=0)</code></p>
<p><code>rt(n, df, ncp)</code></p>
<p><code>rf(n, df1, df2, ncp)</code>
and there are others. These are self explanatory except for <code>ncp</code>. This is the non-centrality parameter. For most use, this will be zero. It is used in power calculations to provide the distribution of the test statistic under an alternative hypothesis. The <code>ncp</code> is the value you would get if you plug values of parameters or observations you expect under the alternative hypothesis into the formula for the test. For example, with a t test, the value under the null hypothesis is zero. If you expect the true difference between your means to be three, then this is the <code>ncp</code>. For a chi-squared test, you would create expected numbers from expected proportions from an alternative hypothesis and calculate a chi-squared value which will be the <code>ncp</code> for power calculations.</p>
</div>
<div id="loops-second-example" class="section level3" number="3.10.4">
<h3><span class="header-section-number">3.10.4</span> Loops: <em>second example</em></h3>
<p>A block of commands, each on a separate line, can be saved in a file and later pasted into R where it will execute (i.e. run as a program). There is, of course, much more to programming in R than this, but this is enough to get you started. The most common requirement is to repeat an operation multiple times. Below is the format for a simple loop which can be easily edited to fit your own purposes.</p>
<div class="sourceCode" id="cb196"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb196-1"><a href="R-and-stats.html#cb196-1" aria-hidden="true" tabindex="-1"></a>no_times<span class="ot">&lt;-</span><span class="dv">100000</span></span>
<span id="cb196-2"><a href="R-and-stats.html#cb196-2" aria-hidden="true" tabindex="-1"></a>diff<span class="ot">&lt;-</span><span class="cn">NA</span></span>
<span id="cb196-3"><a href="R-and-stats.html#cb196-3" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span>no_times) {</span>
<span id="cb196-4"><a href="R-and-stats.html#cb196-4" aria-hidden="true" tabindex="-1"></a>diff[i]<span class="ot">&lt;-</span><span class="fu">runif</span>(<span class="dv">1</span>)<span class="sc">-</span><span class="fu">runif</span>(<span class="dv">1</span>)</span>
<span id="cb196-5"><a href="R-and-stats.html#cb196-5" aria-hidden="true" tabindex="-1"></a>}</span></code></pre></div>
<p>This code plots the distribution of the difference in value between a pair of uniformly distributed values. You may prefer to reduce the number from 100,000. If you cut and paste the code into R you should get something like this:</p>
<div class="sourceCode" id="cb197"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb197-1"><a href="R-and-stats.html#cb197-1" aria-hidden="true" tabindex="-1"></a><span class="fu">hist</span>(diff,<span class="at">breaks=</span><span class="dv">1000</span>)</span></code></pre></div>
<p><img src="bookdown-demo_files/figure-html/unnamed-chunk-120-1.png" width="672" /></p>
<p>Note that the beginning and end of the loop (or any R program block) is defined by curly brackets. The indentation isn’t obligatory but is good programming practice. The program creates a vector, <code>diff</code> which is then indexed by <code>i</code>. If <code>diff</code> wasn’t first created outside the loop, the program would fail.</p>
<p>Loops in R can run slowly. Often, they can be avoided by the use of the command <code>apply()</code>:</p>
<div class="sourceCode" id="cb198"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb198-1"><a href="R-and-stats.html#cb198-1" aria-hidden="true" tabindex="-1"></a><span class="fu">apply</span>(TG_data_for_day_1,<span class="dv">2</span>,sd)</span></code></pre></div>
<pre><code>## Warning in var(if (is.vector(x) || is.factor(x)) x else as.double(x), na.rm =
## na.rm): NAs introduced by coercion

## Warning in var(if (is.vector(x) || is.factor(x)) x else as.double(x), na.rm =
## na.rm): NAs introduced by coercion</code></pre>
<pre><code>##     variety        year      ORIGIN        Rht2       PpdD1       yield 
##          NA  11.5105231          NA          NA   0.4188383   6.2670999 
## CALLOW_2011 FRANCE_2010 FRANCE_2011    LGE_2010    LGE_2011   NIAB_2011 
##   7.5343555   7.6016464   8.0943785   6.7535158   7.2916383   5.1220707 
##          FT          HT        AWNS 
##   3.2931899   7.5069727          NA</code></pre>
<p>The first argument to apply is the dataset or matrix we want to work on, the second is whether we want to loop over rows (1), or columns (2 - as in this example). The third is the function you wish to loop over. So here we loop over columns to calculate the standard deviation of each.</p>
<p>The missing data for Rht2 has caused the standard deviation to be returned as NA. We need to include <code>na.rm=T</code> as an option. Unfortunately, there is no way I know to do this within apply. Instead we need to define our own function (which also allows us to look at how to define functions in R in general - another useful lesson in learning to program in R).</p>
<div class="sourceCode" id="cb201"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb201-1"><a href="R-and-stats.html#cb201-1" aria-hidden="true" tabindex="-1"></a>sd.na.rm.is.t<span class="ot">&lt;-</span><span class="cf">function</span>(x) <span class="fu">sd</span>(x,<span class="at">na.rm=</span>T)</span></code></pre></div>
<p><code>function(x)</code> defines a set of arguments that our new function <code>sd.na.rm.is.t</code> requires. Here we only need one argument; the name of the dataset, which will be equated to x when we run the function. Following that, on the same line, is the set of commands we want to run. Here we are running an inbuilt function, <code>sd(x,na.rm=T)</code>,which will take the data vector x as the source of data.</p>
<p>Try:
<code>sd(AWNS)</code>
<code>sd(AWNS,na.rm=T)</code>
<code>sd.na.rm.is.t(AWNS)</code></p>
<p>So now:</p>
<div class="sourceCode" id="cb202"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb202-1"><a href="R-and-stats.html#cb202-1" aria-hidden="true" tabindex="-1"></a><span class="fu">apply</span>(TG_data_for_day_1,<span class="dv">2</span>,sd.na.rm.is.t)</span></code></pre></div>
<pre><code>## Warning in var(if (is.vector(x) || is.factor(x)) x else as.double(x), na.rm =
## na.rm): NAs introduced by coercion

## Warning in var(if (is.vector(x) || is.factor(x)) x else as.double(x), na.rm =
## na.rm): NAs introduced by coercion</code></pre>
<pre><code>##     variety        year      ORIGIN        Rht2       PpdD1       yield 
##          NA  11.5105231          NA   0.4981119   0.4188383   6.2670999 
## CALLOW_2011 FRANCE_2010 FRANCE_2011    LGE_2010    LGE_2011   NIAB_2011 
##   7.5343555   7.6016464   8.0943785   6.7535158   7.2916383   5.1220707 
##          FT          HT        AWNS 
##   3.2931899   7.5069727   0.2716556</code></pre>
<p>This returns the desired standard deviation of each column of data. We still get NA’s and warnings for attempting to calculate the standard deviation for non-numeric data (for Variety name and country of origin). We could specify which columns of <code>TG_data_for_day_1</code> to work on to avoid this.</p>
<p><code>apply(TG_data_for_day_1[,-1:-3],2,sd.na.rm.is.t)</code></p>
<p>The <code>[,-1:-3]</code> informs R that we need all rows but to drop the first three columns.
This is long winded, but functions are frequently longer sequences of commands rather than just one as in our example. The commands for a long function, which will run over many lines are bracketed with <code>{</code> and <code>}</code>, just as in loops using <code>for</code>.</p>
</div>
<div id="miscellany" class="section level3" number="3.10.5">
<h3><span class="header-section-number">3.10.5</span> Miscellany</h3>
<p>Included here are some useful commands which have not so far been described and some less frequently used commands which come in handy from time to time.</p>
<p><code>attributes(object)</code></p>
<p>Gives information on the object which is often to help you work out what you’ve done.</p>
<p><code>cbind(datasetA)</code> and <code>rbind(datasetB)</code></p>
<p>Joins up two datasets by rows or columns. For example:</p>
<div class="sourceCode" id="cb205"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb205-1"><a href="R-and-stats.html#cb205-1" aria-hidden="true" tabindex="-1"></a>temp<span class="ot">&lt;-</span>TG_data_for_day_1[,<span class="dv">1</span>] [<span class="dv">1</span><span class="sc">:</span><span class="dv">5</span>]</span>
<span id="cb205-2"><a href="R-and-stats.html#cb205-2" aria-hidden="true" tabindex="-1"></a><span class="fu">cbind</span>(temp,TG_data_for_day_1[<span class="dv">1</span><span class="sc">:</span><span class="dv">5</span>,<span class="dv">1</span><span class="sc">:</span><span class="dv">3</span>]  )</span></code></pre></div>
<pre><code>##       temp  variety year ORIGIN
## 1   AARDEN   AARDEN 2003    DEU
## 2 AARDVARK AARDVARK 1997    GBR
## 3    ABELE    ABELE 1970    GBR
## 4      ABO      ABO 1977    FRA
## 5   ACCESS   ACCESS 1998    GBR</code></pre>
<p>We have created a subset of the data with two copies of the variety names. I’ve limited this to the first 5 rows to save paper. There is no particular reason we would want to do this, it’s just as a demonstration.</p>
<p>Next, we create two copies of the first three columns of the data using rbind:</p>
<div class="sourceCode" id="cb207"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb207-1"><a href="R-and-stats.html#cb207-1" aria-hidden="true" tabindex="-1"></a>temp<span class="ot">&lt;-</span>TG_data_for_day_1[,<span class="dv">1</span><span class="sc">:</span><span class="dv">3</span>]</span>
<span id="cb207-2"><a href="R-and-stats.html#cb207-2" aria-hidden="true" tabindex="-1"></a>temp2<span class="ot">&lt;-</span><span class="fu">rbind</span>(temp,TG_data_for_day_1[,<span class="dv">1</span><span class="sc">:</span><span class="dv">3</span>])</span>
<span id="cb207-3"><a href="R-and-stats.html#cb207-3" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(temp2[<span class="fu">order</span>(temp2<span class="sc">$</span>variety),])</span></code></pre></div>
<pre><code>##      variety year ORIGIN
## 1     AARDEN 2003    DEU
## 377   AARDEN 2003    DEU
## 2   AARDVARK 1997    GBR
## 378 AARDVARK 1997    GBR
## 3      ABELE 1970    GBR
## 379    ABELE 1970    GBR</code></pre>
<p>Note, that in ordering the data we have used <code>order(temp2$variety)</code> rather than <code>order(variety)</code> because we require the version of “variety”" associated with the new structure temp2, not the version of variety that we attached earlier.</p>
<p><code>dim(x)</code></p>
<p>Retrieve the dimensions of an array or dataset.</p>
<div class="sourceCode" id="cb209"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb209-1"><a href="R-and-stats.html#cb209-1" aria-hidden="true" tabindex="-1"></a><span class="fu">dim</span>(TG_data_for_day_1)</span></code></pre></div>
<pre><code>## [1] 376  15</code></pre>
<div class="sourceCode" id="cb211"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb211-1"><a href="R-and-stats.html#cb211-1" aria-hidden="true" tabindex="-1"></a><span class="fu">dim</span>(temp2)</span></code></pre></div>
<pre><code>## [1] 752   3</code></pre>
<p><code>help(command)</code></p>
<p>This opens a new window and provides help on the command. help(lm) for example, will give help on the linear modelling command that we have used extensively. The help is written in a terse and technical style however, which may be hard to understand at first but you get used to it. It is useful to see what options are available with each command - for many of the commands used in this guide more are available than have been described. Often, sufficient of the output from help makes sense to be able to get a command working by trial and error. At the bottom of the output, there are often examples of the command’s use: again not always easy to follow. In Rstudio, help is available (in the same unhelpful style) directly from the windows.</p>
<p><code>history(x)</code></p>
<p>In native R, this opens up a window with a list of the most recently issued x commands. The default number is 25. These can be copied back into the R window and re-executed. The window with the output can be saved from the File menu to keep a record of commands issued during the R session. In RStudio, the history of commands is available directly from a tab in the top right hand window.c</p>
<p><code>ls()  or equivalently objects()</code></p>
<p>Lists all the variables available in the current R session. This command can also be executed from the menu in native R, selecting first “Misc,” then “List objects.” In RStudio it is displayed using the Environment tab in the top right hand window.</p>
<p><code>matrix(data, nrow = x,ncol =y)</code></p>
<p>Defines a matrix and fills it with data. For example:</p>
<div class="sourceCode" id="cb213"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb213-1"><a href="R-and-stats.html#cb213-1" aria-hidden="true" tabindex="-1"></a><span class="fu">matrix</span>(<span class="fu">rnorm</span>(<span class="dv">12</span>),<span class="at">nrow=</span><span class="dv">3</span>, <span class="at">ncol=</span><span class="dv">4</span>)</span></code></pre></div>
<pre><code>##           [,1]       [,2]       [,3]       [,4]
## [1,] -1.438823  1.6780448 -0.8348513 -0.5138228
## [2,] -1.167761 -0.1034786  2.0115816 -1.3274879
## [3,]  1.383999  1.0041599 -0.5920141  0.4396548</code></pre>
<p>The <code>rnorm()</code> part has simulated 12 random numbers which have a specified normal distribution.</p>
<p>Note that there is no need to define the column number; it is fixed by the data and by the row number. More commonly, we would want to name the matrix. By default, all data are initially recorded as missing:</p>
<div class="sourceCode" id="cb215"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb215-1"><a href="R-and-stats.html#cb215-1" aria-hidden="true" tabindex="-1"></a><span class="fu">matrix</span>(<span class="at">nrow=</span><span class="dv">3</span>,<span class="at">ncol=</span><span class="dv">4</span>)</span></code></pre></div>
<pre><code>##      [,1] [,2] [,3] [,4]
## [1,]   NA   NA   NA   NA
## [2,]   NA   NA   NA   NA
## [3,]   NA   NA   NA   NA</code></pre>
<p><code>rep(x,y)</code></p>
<p>Generates a repeated list:</p>
<div class="sourceCode" id="cb217"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb217-1"><a href="R-and-stats.html#cb217-1" aria-hidden="true" tabindex="-1"></a><span class="fu">rep</span>(<span class="dv">1</span><span class="sc">:</span><span class="dv">3</span>,<span class="dv">4</span>)</span></code></pre></div>
<pre><code>##  [1] 1 2 3 1 2 3 1 2 3 1 2 3</code></pre>
<div class="sourceCode" id="cb219"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb219-1"><a href="R-and-stats.html#cb219-1" aria-hidden="true" tabindex="-1"></a><span class="fu">rep</span>(<span class="fu">c</span>(<span class="st">&#39;apples&#39;</span>,<span class="st">&#39;pears&#39;</span>),<span class="dv">2</span>)</span></code></pre></div>
<pre><code>## [1] &quot;apples&quot; &quot;pears&quot;  &quot;apples&quot; &quot;pears&quot;</code></pre>
<p>Remove an object:</p>
<div class="sourceCode" id="cb221"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb221-1"><a href="R-and-stats.html#cb221-1" aria-hidden="true" tabindex="-1"></a><span class="fu">rm</span>(harvest_index)</span></code></pre></div>
<p>Remove those objects that are no longer required.</p>
<p><code>rm(list=ls(all.names=TRUE))</code> Would removes all variables, although this can be more easily achieved from the Session tab which offers more options.</p>
<p>Pull row names from an object (and only show first 5):</p>
<div class="sourceCode" id="cb222"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb222-1"><a href="R-and-stats.html#cb222-1" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(TG_data_for_day_1)  [<span class="dv">1</span><span class="sc">:</span><span class="dv">5</span>]</span></code></pre></div>
<pre><code>## [1] &quot;1&quot; &quot;2&quot; &quot;3&quot; &quot;4&quot; &quot;5&quot;</code></pre>
<p>Pull column names:</p>
<div class="sourceCode" id="cb224"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb224-1"><a href="R-and-stats.html#cb224-1" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(TG_data_for_day_1)</span></code></pre></div>
<pre><code>##  [1] &quot;variety&quot;     &quot;year&quot;        &quot;ORIGIN&quot;      &quot;Rht2&quot;        &quot;PpdD1&quot;      
##  [6] &quot;yield&quot;       &quot;CALLOW_2011&quot; &quot;FRANCE_2010&quot; &quot;FRANCE_2011&quot; &quot;LGE_2010&quot;   
## [11] &quot;LGE_2011&quot;    &quot;NIAB_2011&quot;   &quot;FT&quot;          &quot;HT&quot;          &quot;AWNS&quot;</code></pre>
<p>You can also write new names:</p>
<div class="sourceCode" id="cb226"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb226-1"><a href="R-and-stats.html#cb226-1" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(TG_data_for_day_1)[<span class="dv">1</span><span class="sc">:</span><span class="dv">3</span>]<span class="ot">&lt;-</span><span class="fu">c</span>(<span class="st">&quot;a&quot;</span>,<span class="st">&quot;b&quot;</span>,<span class="st">&quot;c&quot;</span>)</span>
<span id="cb226-2"><a href="R-and-stats.html#cb226-2" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(TG_data_for_day_1)[<span class="dv">1</span><span class="sc">:</span><span class="dv">10</span>]</span></code></pre></div>
<pre><code>##  [1] &quot;a&quot;  &quot;b&quot;  &quot;c&quot;  &quot;4&quot;  &quot;5&quot;  &quot;6&quot;  &quot;7&quot;  &quot;8&quot;  &quot;9&quot;  &quot;10&quot;</code></pre>
<p><code>seq(x)</code></p>
<p>Generates a sequential list.</p>
<pre><code>`seq(10)`                 produces a list from 1:10. 
`seq(10,12,0.25)`       produces a list from 10 to 12 in steps of 1/4</code></pre>
<p><code>sink(filename)</code></p>
<p>Sends output to the given file rather than to the console. <code>sink()</code> with no filename will turn off the previous redirection, Useful if you are expecting lots of output. You can issue this before the command and then turn it off when the command has completed.</p>
<p><code>source(filename)</code></p>
<p>Runs commands from the filename - given as a variable containing the name or as the name itself in quotes.</p>
<p><code>split(x,y)</code></p>
<p>Partitions a variable or whole dataframe <code>x</code> into separate variables or dataframes on the basis of matches to <code>y</code>. Give this a go using:</p>
<p><code>split(yield,ORIGIN) [1:2]</code></p>
<p>In my hands, the following convoluted code will split our data into three separate data frames.</p>
<div class="sourceCode" id="cb229"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb229-1"><a href="R-and-stats.html#cb229-1" aria-hidden="true" tabindex="-1"></a>DEU<span class="ot">&lt;-</span><span class="fu">data.frame</span>(<span class="fu">split</span>(TG_data_for_day_1,ORIGIN)[<span class="dv">1</span>])</span>
<span id="cb229-2"><a href="R-and-stats.html#cb229-2" aria-hidden="true" tabindex="-1"></a>FRA<span class="ot">&lt;-</span><span class="fu">data.frame</span>(<span class="fu">split</span>(TG_data_for_day_1, ORIGIN)[<span class="dv">2</span>])</span>
<span id="cb229-3"><a href="R-and-stats.html#cb229-3" aria-hidden="true" tabindex="-1"></a>UK<span class="ot">&lt;-</span><span class="fu">data.frame</span>(<span class="fu">split</span>(TG_data_for_day_1, ORIGIN)[<span class="dv">3</span>])</span>
<span id="cb229-4"><a href="R-and-stats.html#cb229-4" aria-hidden="true" tabindex="-1"></a><span class="fu">dim</span>(DEU)</span></code></pre></div>
<pre><code>## [1] 89 15</code></pre>
<div class="sourceCode" id="cb231"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb231-1"><a href="R-and-stats.html#cb231-1" aria-hidden="true" tabindex="-1"></a><span class="fu">dim</span>(FRA)</span></code></pre></div>
<pre><code>## [1] 212  15</code></pre>
<div class="sourceCode" id="cb233"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb233-1"><a href="R-and-stats.html#cb233-1" aria-hidden="true" tabindex="-1"></a><span class="fu">dim</span>(UK)</span></code></pre></div>
<pre><code>## [1] 73 15</code></pre>
<div class="sourceCode" id="cb235"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb235-1"><a href="R-and-stats.html#cb235-1" aria-hidden="true" tabindex="-1"></a><span class="fu">dim</span>(TG_data_for_day_1)</span></code></pre></div>
<pre><code>## [1] 376  15</code></pre>
<p>The row numbers don’t add up because for some countries the origin was <code>NA</code>.</p>
</div>
<div id="my-number-one-piece-of-advice" class="section level3" number="3.10.6">
<h3><span class="header-section-number">3.10.6</span> My number one piece of advice</h3>
<p>If you have reached this stage of the tutorial, no doubt you have taken on board a lot of information! It’s inconceivable to remember all that information. While you can refer back to this tutorial whenever you need to for help, I find the best method is usually just to <em>‘google it’</em>.</p>
<p>Whether you are a first time user, an intermediate user, or professional software developer, this is the universal solution to most programming/coding problems.</p>
<p>In terms of R issues, you can normally just google your problem in the following format. Let’s say we wanted to change the position of a plot title for a figure we were forming for a publication/presentation, I would search for:</p>
<p><em>“How to change position of plot title in r”</em></p>
<p>The first hit for me is <a href="https://stackoverflow.com/questions/20355410/adjust-plot-title-main-position">this</a>. That link shows someone asking my question and some fantastic solutions to solving the problem.</p>
<p>The websites <a href="https://stackoverflow.com/">stackoverflow</a> and <a href="https://stackexchange.com/">stackexchange</a> are typically the best options and have helped me a great deal. Normally, the question you are asking has been already addressed on these websites and users have provided more solutions than you need for the question at hand. I would suggest getting to grips with the format and etiquette of these website communities before posting your own questions. The moderators/users like the questions to be asked in a certain way and can be slightly hostile if you don’t meet their criteria. Then your question and their slightly rude responses remain on the internet for everyone to see. I’m speaking from past experience here!</p>
<p>There are some good R reddit communities that accept all type of questions (e.g. <a href="https://www.reddit.com/r/Rlanguage/">try this one</a>).</p>
</div>
<div id="learn-more" class="section level3" number="3.10.7">
<h3><span class="header-section-number">3.10.7</span> Learn more</h3>
<p>Much useful information and documentation is available on the <a href="http://www.r-project.org/">R web site</a>, including the R manual “An Introduction to R.”</p>

<div class="rmdnote">
The manuals, including “An introduction to R,” are available directly from the Help menu from with R.
</div>
<p>The ‘Introductory Statistics with R’ book <span class="citation">(<a href="references.html#ref-dalgaard_introductory_2008" role="doc-biblioref">Dalgaard 2008</a>)</span> is an excellent introduction both to R and to statistical analysis, with many simple examples.</p>
<p>R in a Nutshell <span class="citation">(<a href="references.html#ref-adler_r_2010" role="doc-biblioref">Adler 2010</a>)</span>, by Joseph Adler in the Nutshell series from O’Reilly gives a lot of detail about what and how R does what it does. An excellent reference book, but not for beginners.</p>
</div>
<div id="packages" class="section level3" number="3.10.8">
<h3><span class="header-section-number">3.10.8</span> Packages</h3>
<p>Many are available from the Comprehensive R Archive network (CRAN) web site <a href="http://cran.r-project.org/" class="uri">http://cran.r-project.org/</a>. They can be installed directly from this site, or downloaded as zip files and installed later (useful if you need to install onto multiple machines). We shall be using several of these later.</p>
<p>These packages generally come with their own manual, often detailed. That for the R/QTL package, for example, runs to 96 pages. Although the CRAN website is the first place to search for suitable packages, they are also found elsewhere and are often referred to in methodological publications or the methods sections of paper: programming in R is an expanding industry. The more popular packages (R/QTL again) may have books published about them and often course material and lecture notes can be found on the web.</p>

<div class="rmdwarning">
It is worth reiterating that our guide to the syntax and structures used in R has been very superficial. We have mentioned the data frame and little else. Knowledge of other structures - arrays, matrices, lists - should be acquired at some stage. They are explained in “An Introduction to R” but this book is not a page turner.
</div>
</div>
<div id="the-tidyverse" class="section level3" number="3.10.9">
<h3><span class="header-section-number">3.10.9</span> The Tidyverse</h3>
<p>If you are new to R, it is worth putting the effort into learning about this. The website is: <a href="https://www.tidyverse.org/" class="uri">https://www.tidyverse.org/</a>. The book “R for Data Science” <span class="citation">(<a href="references.html#ref-wickham_r_2016" role="doc-biblioref">Wickham and Grolemund 2016</a>)</span>, which is free online from this website is a very good introduction, including to ggplot.</p>
</div>
<div id="other-resources" class="section level3" number="3.10.10">
<h3><span class="header-section-number">3.10.10</span> Other resources</h3>
<p>For the “Understanding data types and R objects” section in this tutorial, inspiration was taken from <a href="https://datacarpentry.org/R-ecology-lesson/" class="uri">https://datacarpentry.org/R-ecology-lesson/</a>. This is an excellent resource and worth checking out. Additionally, formal definitions were taken from the book “R in Action: Data Analysis and Graphics with R” <span class="citation">(<a href="references.html#ref-kabacoff_r_2011" role="doc-biblioref">Kabacoff 2011</a>)</span>.</p>
</div>
</div>
<div id="list-of-commands-mainly-described-in-this-guide." class="section level2" number="3.11">
<h2><span class="header-section-number">3.11</span> List of commands, mainly described in this guide.</h2>
<div id="general" class="section level4" number="3.11.0.1">
<h4><span class="header-section-number">3.11.0.1</span> General</h4>
<p><code>apply</code> = fast way of looping over rows and columns</p>
<p><code>attach</code> = attaches a dataset to R for subsequent analyses</p>
<p><code>attributes</code> = get information on the attributes of an object</p>
<p><code>cbind</code> = concatenate two tables or vectors by columns</p>
<p><code>colnames</code> = adds column names to a table or reads them from a table</p>
<p><code>detach</code> = attaches a dataset to R for subsequent analyses</p>
<p><code>demo</code> = demonstration a command (not available for most commands)</p>
<p><code>dim</code> = retrieve the dimensions of an object</p>
<p><code>factor</code> = convert a variable or text into a factor</p>
<p><code>for</code> = loop over a a set of commands with different input values</p>
<p><code>function</code> = define a function which can then be run repeatedly</p>
<p><code>getwd</code> = returns the path to the working directory</p>
<p><code>help</code> = returns help on a command</p>
<p><code>history</code> = lists previously issued commands</p>
<p><code>install.packages</code> = select a package to install</p>
<p><code>is.na</code> = returns TRUE if a value is NA.</p>
<p><code>length</code> = returns the number of entries in a variate</p>
<p><code>ls</code> = lists all active data structures and variates</p>
<p><code>matrix</code> = defines a matrix and fills it with data.</p>
<p><code>order</code> = returns the order or a variate for use in a subsequent sort</p>
<p><code>quit</code> = exit R</p>
<p><code>rbind</code> = concatenate two tables or vectors by rows</p>
<p><code>read.csv</code> = reads in data from .csv file</p>
<p><code>read.table</code> = reads in data</p>
<p><code>rep</code> = generates a list with repeating elements</p>
<p><code>rm</code> = delete data structures and variates from R</p>
<p><code>rownames</code> = adds row names to a table or reads them from a table</p>
<p><code>searchpaths</code> = lists attached packages and datasets in the order in which
searched.</p>
<p><code>seq</code> = creates a sequence of values</p>
<p><code>sink</code> = writes output to a file</p>
<p><code>sort</code> = sorts data</p>
<p><code>source</code> = reads commands from a file</p>
<p><code>split</code> = splits data on a factor value</p>
<p><code>subset</code> = selects a subset of data for subsequent analysis</p>
<p><code>table</code> = defines a table: used for input into contingency chi sq tests.</p>
</div>
<div id="graphical" class="section level4" number="3.11.0.2">
<h4><span class="header-section-number">3.11.0.2</span> Graphical</h4>
<p><code>abline</code> = add the best fitting straight line to a scattergram</p>
<p><code>boxplot</code> = produce a Box-and-whisker plot</p>
<p><code>heatmap</code> = plots a heatmap and dendogram</p>
<p><code>curve</code> = add a function to a graph</p>
<p><code>hist</code> = plot a histogram</p>
<p><code>image</code> = colours-in a matrix according to its values.</p>
<p><code>pairs</code> = plot multiple scattergrams in a matrix format</p>
<p><code>par</code> = set graphical parameters</p>
<p><code>pdf</code> = writes graphical output to a pdf file</p>
<p><code>plot</code> = produce a scattergram</p>
</div>
<div id="statistics" class="section level4" number="3.11.0.3">
<h4><span class="header-section-number">3.11.0.3</span> Statistics</h4>
<p><code>anova</code> = return an anova table from a linear model</p>
<p><code>apply</code> = applies a function to rows and/or columns of a matrix</p>
<p><code>chisq.test</code> = carry out a contingency chi-squared test</p>
<p><code>colMeans</code> = returns means across columns of a dataset</p>
<p><code>colSums</code> = returns totals across columns of a dataset</p>
<p><code>cor</code> = returns the correlation coefficient</p>
<p><code>fitted</code> = returns the fitted values from a linear model</p>
<p><code>lm</code> = define and execute a linear model</p>
<p><code>mean</code> = returns the mean of a variate</p>
<p><code>median</code> = returns the median of a variate</p>
<p><code>minimum</code> = returns the minimum of a variate</p>
<p><code>maximum</code> = returns the maximum of a variate</p>
<p><code>pchisq</code> = returns the p-value of a chi-squared statistic</p>
<p><code>pf</code> = returns the p-value of a F (variance ratio) statistic</p>
<p><code>pnorm</code> = returns the p-value for a normally distributed variate</p>
<p><code>pt</code> = returns the p-value of a t-test</p>
<p><code>qchisq</code> = returns a chi-squared statistic for a given probability</p>
<p><code>qnorm</code> = returns a normality distributed variate for a given probability</p>
<p><code>quantile</code> = returns the quantiles of a variate</p>
<p><code>qf</code> = returns a F (variance ratio) statistic for a given probability</p>
<p><code>qt</code> = returns the t-test statistic for a given probability</p>
<p><code>resid</code> = returns the residuals from a linear model</p>
<p><code>rchisq</code> = returns random numbers from the chisq distribution</p>
<p><code>rf</code> = returns random numbers from the F distribution</p>
<p><code>rnorm</code> = returns random numbers from the normal distribution</p>
<p><code>rowMeans</code> = returns means across rows of a dataset</p>
<p><code>rowSums</code> = returns totals across rows of a dataset</p>
<p><code>runif</code> = returns random numbers from the uniform distribution</p>
<p><code>rt</code> = returns random numbers from the t distribution</p>
<p><code>sd</code> = returns the standard deviation of a variate</p>
<p><code>sum</code> = returns the sum of a variate</p>
<p><code>summary</code> = summarise data</p>
<p><code>t.test</code> = one and two sample t-test</p>
<p><code>var</code> = returns the mean of a variate</p>
<p><code>var.test</code> = compare two variances by an F test</p>

</div>
</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="Acknow.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="Trial-design.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": "https://github.com/rstudio/bookdown-demo/edit/master/03-R_and_stats.Rmd",
"text": "Edit"
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": ["bookdown-demo.pdf", "bookdown-demo.epub"],
"toc": {
"collapse": "subsection"
}
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
