<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>References | Quantitative Methods in Plant Breeding: The Digital Tutorials.</title>
  <meta name="description" content="This is the online tutorial booklet for the QMPB course run at NIAB. The online book was formed using R bookdown." />
  <meta name="generator" content="bookdown 0.21 and GitBook 2.6.7" />

  <meta property="og:title" content="References | Quantitative Methods in Plant Breeding: The Digital Tutorials." />
  <meta property="og:type" content="book" />
  
  <meta property="og:image" content="images/cover.jpg" />
  <meta property="og:description" content="This is the online tutorial booklet for the QMPB course run at NIAB. The online book was formed using R bookdown." />
  <meta name="github-repo" content="rstudio/bookdown" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="References | Quantitative Methods in Plant Breeding: The Digital Tutorials." />
  
  <meta name="twitter:description" content="This is the online tutorial booklet for the QMPB course run at NIAB. The online book was formed using R bookdown." />
  <meta name="twitter:image" content="images/cover.jpg" />

<meta name="author" content="NIAB 2021" />


<meta name="date" content="2021-02-27" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="Breeding-simulations.html"/>

<script src="libs/header-attrs-2.6/header-attrs.js"></script>
<script src="libs/jquery-2.2.3/jquery.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />











<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">The QMPB digital tutorials</a></li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> Introduction</a></li>
<li class="chapter" data-level="2" data-path="Acknow.html"><a href="Acknow.html"><i class="fa fa-check"></i><b>2</b> Acknowledgements and Contents</a>
<ul>
<li class="chapter" data-level="2.1" data-path="Acknow.html"><a href="Acknow.html#acknowledgements"><i class="fa fa-check"></i><b>2.1</b> Acknowledgements</a></li>
<li class="chapter" data-level="2.2" data-path="Acknow.html"><a href="Acknow.html#digital-tutorial-chapters"><i class="fa fa-check"></i><b>2.2</b> Digital Tutorial Chapters:</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="R-and-stats.html"><a href="R-and-stats.html"><i class="fa fa-check"></i><b>3</b> R and Rstudio: a general introduction with statistics.</a>
<ul>
<li class="chapter" data-level="3.1" data-path="R-and-stats.html"><a href="R-and-stats.html#introduction-section-on-r"><i class="fa fa-check"></i><b>3.1</b> Introduction section on R</a>
<ul>
<li class="chapter" data-level="3.1.1" data-path="R-and-stats.html"><a href="R-and-stats.html#getting-the-data-ready"><i class="fa fa-check"></i><b>3.1.1</b> Getting the data ready</a></li>
</ul></li>
<li class="chapter" data-level="3.2" data-path="R-and-stats.html"><a href="R-and-stats.html#the-rstudio-layout-workspace-and-environment."><i class="fa fa-check"></i><b>3.2</b> The Rstudio layout, workspace and environment.</a>
<ul>
<li class="chapter" data-level="3.2.1" data-path="R-and-stats.html"><a href="R-and-stats.html#working-directory"><i class="fa fa-check"></i><b>3.2.1</b> Working directory</a></li>
</ul></li>
<li class="chapter" data-level="3.3" data-path="R-and-stats.html"><a href="R-and-stats.html#basic-r-syntx"><i class="fa fa-check"></i><b>3.3</b> Basic R Syntx</a>
<ul>
<li class="chapter" data-level="3.3.1" data-path="R-and-stats.html"><a href="R-and-stats.html#brackets"><i class="fa fa-check"></i><b>3.3.1</b> Brackets</a></li>
<li class="chapter" data-level="3.3.2" data-path="R-and-stats.html"><a href="R-and-stats.html#equals-symbols"><i class="fa fa-check"></i><b>3.3.2</b> Equals symbols</a></li>
<li class="chapter" data-level="3.3.3" data-path="R-and-stats.html"><a href="R-and-stats.html#continuation-character."><i class="fa fa-check"></i><b>3.3.3</b> Continuation character.</a></li>
<li class="chapter" data-level="3.3.4" data-path="R-and-stats.html"><a href="R-and-stats.html#text-and-numbers"><i class="fa fa-check"></i><b>3.3.4</b> Text and numbers</a></li>
<li class="chapter" data-level="3.3.5" data-path="R-and-stats.html"><a href="R-and-stats.html#summary-of-syntax"><i class="fa fa-check"></i><b>3.3.5</b> Summary of syntax</a></li>
</ul></li>
<li class="chapter" data-level="3.4" data-path="R-and-stats.html"><a href="R-and-stats.html#reading-in-data"><i class="fa fa-check"></i><b>3.4</b> Reading in data</a></li>
<li class="chapter" data-level="3.5" data-path="R-and-stats.html"><a href="R-and-stats.html#understanding-data-types-and-r-objects"><i class="fa fa-check"></i><b>3.5</b> Understanding data types and R objects</a>
<ul>
<li class="chapter" data-level="3.5.1" data-path="R-and-stats.html"><a href="R-and-stats.html#summary"><i class="fa fa-check"></i><b>3.5.1</b> Summary</a></li>
</ul></li>
<li class="chapter" data-level="3.6" data-path="R-and-stats.html"><a href="R-and-stats.html#summarising-and-displaying-data"><i class="fa fa-check"></i><b>3.6</b> Summarising and displaying data</a></li>
<li class="chapter" data-level="3.7" data-path="R-and-stats.html"><a href="R-and-stats.html#calculations-and-data-manipulation"><i class="fa fa-check"></i><b>3.7</b> Calculations and data manipulation</a>
<ul>
<li class="chapter" data-level="3.7.1" data-path="R-and-stats.html"><a href="R-and-stats.html#sorting-data"><i class="fa fa-check"></i><b>3.7.1</b> Sorting data</a></li>
<li class="chapter" data-level="3.7.2" data-path="R-and-stats.html"><a href="R-and-stats.html#relationships-between-variates"><i class="fa fa-check"></i><b>3.7.2</b> Relationships between variates</a></li>
<li class="chapter" data-level="3.7.3" data-path="R-and-stats.html"><a href="R-and-stats.html#exporting-your-script-and-workspace"><i class="fa fa-check"></i><b>3.7.3</b> Exporting your script and workspace</a></li>
<li class="chapter" data-level="3.7.4" data-path="R-and-stats.html"><a href="R-and-stats.html#saving-a-data-table"><i class="fa fa-check"></i><b>3.7.4</b> Saving a data table</a></li>
</ul></li>
<li class="chapter" data-level="3.8" data-path="R-and-stats.html"><a href="R-and-stats.html#basic-statistical-analysis"><i class="fa fa-check"></i><b>3.8</b> Basic statistical analysis</a>
<ul>
<li class="chapter" data-level="3.8.1" data-path="R-and-stats.html"><a href="R-and-stats.html#the-t-test"><i class="fa fa-check"></i><b>3.8.1</b> The t-test</a></li>
<li class="chapter" data-level="3.8.2" data-path="R-and-stats.html"><a href="R-and-stats.html#linear-regression"><i class="fa fa-check"></i><b>3.8.2</b> Linear Regression</a></li>
<li class="chapter" data-level="3.8.3" data-path="R-and-stats.html"><a href="R-and-stats.html#multiple-regression"><i class="fa fa-check"></i><b>3.8.3</b> Multiple regression</a></li>
<li class="chapter" data-level="3.8.4" data-path="R-and-stats.html"><a href="R-and-stats.html#the-analysis-of-variance"><i class="fa fa-check"></i><b>3.8.4</b> The analysis of variance</a></li>
</ul></li>
<li class="chapter" data-level="3.9" data-path="R-and-stats.html"><a href="R-and-stats.html#categorical-data---the-chi-squared-test"><i class="fa fa-check"></i><b>3.9</b> Categorical data - the <em>chi</em>-squared test</a></li>
<li class="chapter" data-level="3.10" data-path="R-and-stats.html"><a href="R-and-stats.html#more-useful-information"><i class="fa fa-check"></i><b>3.10</b> More useful information</a>
<ul>
<li class="chapter" data-level="3.10.1" data-path="R-and-stats.html"><a href="R-and-stats.html#plotting-graphs"><i class="fa fa-check"></i><b>3.10.1</b> Plotting Graphs</a></li>
<li class="chapter" data-level="3.10.2" data-path="R-and-stats.html"><a href="R-and-stats.html#probability-distributions"><i class="fa fa-check"></i><b>3.10.2</b> Probability distributions</a></li>
<li class="chapter" data-level="3.10.3" data-path="R-and-stats.html"><a href="R-and-stats.html#random-number-generation"><i class="fa fa-check"></i><b>3.10.3</b> Random number generation</a></li>
<li class="chapter" data-level="3.10.4" data-path="R-and-stats.html"><a href="R-and-stats.html#loops-second-example"><i class="fa fa-check"></i><b>3.10.4</b> Loops: <em>second example</em></a></li>
<li class="chapter" data-level="3.10.5" data-path="R-and-stats.html"><a href="R-and-stats.html#miscellany"><i class="fa fa-check"></i><b>3.10.5</b> Miscellany</a></li>
<li class="chapter" data-level="3.10.6" data-path="R-and-stats.html"><a href="R-and-stats.html#my-number-one-piece-of-advice"><i class="fa fa-check"></i><b>3.10.6</b> My number one piece of advice</a></li>
<li class="chapter" data-level="3.10.7" data-path="R-and-stats.html"><a href="R-and-stats.html#learn-more"><i class="fa fa-check"></i><b>3.10.7</b> Learn more</a></li>
<li class="chapter" data-level="3.10.8" data-path="R-and-stats.html"><a href="R-and-stats.html#packages"><i class="fa fa-check"></i><b>3.10.8</b> Packages</a></li>
<li class="chapter" data-level="3.10.9" data-path="R-and-stats.html"><a href="R-and-stats.html#the-tidyverse"><i class="fa fa-check"></i><b>3.10.9</b> The Tidyverse</a></li>
<li class="chapter" data-level="3.10.10" data-path="R-and-stats.html"><a href="R-and-stats.html#other-resources"><i class="fa fa-check"></i><b>3.10.10</b> Other resources</a></li>
</ul></li>
<li class="chapter" data-level="3.11" data-path="R-and-stats.html"><a href="R-and-stats.html#list-of-commands-mainly-described-in-this-guide."><i class="fa fa-check"></i><b>3.11</b> List of commands, mainly described in this guide.</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="Trial-design.html"><a href="Trial-design.html"><i class="fa fa-check"></i><b>4</b> Trial Design.</a>
<ul>
<li class="chapter" data-level="4.1" data-path="Trial-design.html"><a href="Trial-design.html#randomised-complete-block-designs"><i class="fa fa-check"></i><b>4.1</b> Randomised complete block designs</a></li>
<li class="chapter" data-level="4.2" data-path="Trial-design.html"><a href="Trial-design.html#trial-design-in-r"><i class="fa fa-check"></i><b>4.2</b> Trial design in R</a>
<ul>
<li class="chapter" data-level="4.2.1" data-path="Trial-design.html"><a href="Trial-design.html#partially-replicated-designs"><i class="fa fa-check"></i><b>4.2.1</b> Partially replicated designs</a></li>
<li class="chapter" data-level="4.2.2" data-path="Trial-design.html"><a href="Trial-design.html#two-dimensional-designs"><i class="fa fa-check"></i><b>4.2.2</b> Two-dimensional designs</a></li>
</ul></li>
<li class="chapter" data-level="4.3" data-path="Trial-design.html"><a href="Trial-design.html#farm-scale-experiments"><i class="fa fa-check"></i><b>4.3</b> Farm scale experiments</a></li>
<li class="chapter" data-level="4.4" data-path="Trial-design.html"><a href="Trial-design.html#alpha-designs."><i class="fa fa-check"></i><b>4.4</b> Alpha designs.</a>
<ul>
<li class="chapter" data-level="4.4.1" data-path="Trial-design.html"><a href="Trial-design.html#creation-of-augmented-and-p-rep-designs-from-2-replicate-alpha-designs"><i class="fa fa-check"></i><b>4.4.1</b> Creation of augmented and p-rep designs from 2-replicate alpha designs</a></li>
<li class="chapter" data-level="4.4.2" data-path="Trial-design.html"><a href="Trial-design.html#final-message"><i class="fa fa-check"></i><b>4.4.2</b> Final message</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="5" data-path="Trial-analysis.html"><a href="Trial-analysis.html"><i class="fa fa-check"></i><b>5</b> Trial analysis (sugar beet example).</a>
<ul>
<li class="chapter" data-level="5.1" data-path="Trial-analysis.html"><a href="Trial-analysis.html#the-layout-of-the-sugar-beet-trial"><i class="fa fa-check"></i><b>5.1</b> The layout of the sugar beet trial</a></li>
<li class="chapter" data-level="5.2" data-path="Trial-analysis.html"><a href="Trial-analysis.html#fixed-effects-random-effects-and-reml"><i class="fa fa-check"></i><b>5.2</b> Fixed effects, random effects and REML</a>
<ul>
<li class="chapter" data-level="5.2.1" data-path="Trial-analysis.html"><a href="Trial-analysis.html#preparing-the-data"><i class="fa fa-check"></i><b>5.2.1</b> Preparing the data</a></li>
<li class="chapter" data-level="5.2.2" data-path="Trial-analysis.html"><a href="Trial-analysis.html#randomised-complete-block-design"><i class="fa fa-check"></i><b>5.2.2</b> Randomised complete block design</a></li>
<li class="chapter" data-level="5.2.3" data-path="Trial-analysis.html"><a href="Trial-analysis.html#randomised-compelete-block-design---with-mixed-models"><i class="fa fa-check"></i><b>5.2.3</b> Randomised compelete block design - with mixed models</a></li>
<li class="chapter" data-level="5.2.4" data-path="Trial-analysis.html"><a href="Trial-analysis.html#plotting-residuals-against-field-structure."><i class="fa fa-check"></i><b>5.2.4</b> Plotting residuals against field structure.</a></li>
<li class="chapter" data-level="5.2.5" data-path="Trial-analysis.html"><a href="Trial-analysis.html#alpha-design-include-block-structure"><i class="fa fa-check"></i><b>5.2.5</b> Alpha design (include block structure)</a></li>
<li class="chapter" data-level="5.2.6" data-path="Trial-analysis.html"><a href="Trial-analysis.html#rows-and-column-analysis"><i class="fa fa-check"></i><b>5.2.6</b> Rows and column analysis</a></li>
</ul></li>
<li class="chapter" data-level="5.3" data-path="Trial-analysis.html"><a href="Trial-analysis.html#spatial-analysis"><i class="fa fa-check"></i><b>5.3</b> Spatial analysis</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="Cross-site-analysis.html"><a href="Cross-site-analysis.html"><i class="fa fa-check"></i><b>6</b> Cross sites analysis in R.</a>
<ul>
<li class="chapter" data-level="6.1" data-path="Cross-site-analysis.html"><a href="Cross-site-analysis.html#blues-blups-and-one-stage-analysis"><i class="fa fa-check"></i><b>6.1</b> BLUEs, BLUPs and one stage analysis</a>
<ul>
<li class="chapter" data-level="6.1.1" data-path="Cross-site-analysis.html"><a href="Cross-site-analysis.html#one-stage-analysis-with-weights"><i class="fa fa-check"></i><b>6.1.1</b> One stage analysis with weights</a></li>
</ul></li>
<li class="chapter" data-level="6.2" data-path="Cross-site-analysis.html"><a href="Cross-site-analysis.html#weighted-two-stage-analysis"><i class="fa fa-check"></i><b>6.2</b> Weighted two-stage analysis</a></li>
<li class="chapter" data-level="6.3" data-path="Cross-site-analysis.html"><a href="Cross-site-analysis.html#ammi"><i class="fa fa-check"></i><b>6.3</b> AMMI</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="Population-genetics.html"><a href="Population-genetics.html"><i class="fa fa-check"></i><b>7</b> Population genetics.</a>
<ul>
<li class="chapter" data-level="7.1" data-path="Population-genetics.html"><a href="Population-genetics.html#hardy-weinberg-equilibrium-and-population-structure"><i class="fa fa-check"></i><b>7.1</b> Hardy-Weinberg equilibrium and population structure</a></li>
<li class="chapter" data-level="7.2" data-path="Population-genetics.html"><a href="Population-genetics.html#other-ways-of-analysing-population-structure"><i class="fa fa-check"></i><b>7.2</b> Other ways of analysing population structure</a>
<ul>
<li class="chapter" data-level="7.2.1" data-path="Population-genetics.html"><a href="Population-genetics.html#distance-trees"><i class="fa fa-check"></i><b>7.2.1</b> Distance Trees</a></li>
</ul></li>
<li class="chapter" data-level="7.3" data-path="Population-genetics.html"><a href="Population-genetics.html#structure"><i class="fa fa-check"></i><b>7.3</b> Structure</a>
<ul>
<li class="chapter" data-level="7.3.1" data-path="Population-genetics.html"><a href="Population-genetics.html#what-structure-does"><i class="fa fa-check"></i><b>7.3.1</b> What STRUCTURE does</a></li>
<li class="chapter" data-level="7.3.2" data-path="Population-genetics.html"><a href="Population-genetics.html#limitations-of-structure"><i class="fa fa-check"></i><b>7.3.2</b> Limitations of STRUCTURE</a></li>
<li class="chapter" data-level="7.3.3" data-path="Population-genetics.html"><a href="Population-genetics.html#using-structure"><i class="fa fa-check"></i><b>7.3.3</b> Using Structure</a></li>
<li class="chapter" data-level="7.3.4" data-path="Population-genetics.html"><a href="Population-genetics.html#the-exercise"><i class="fa fa-check"></i><b>7.3.4</b> The exercise</a></li>
</ul></li>
<li class="chapter" data-level="7.4" data-path="Population-genetics.html"><a href="Population-genetics.html#linkage-disequilibrium-analysis"><i class="fa fa-check"></i><b>7.4</b> Linkage disequilibrium analysis</a>
<ul>
<li class="chapter" data-level="7.4.1" data-path="Population-genetics.html"><a href="Population-genetics.html#linkage-disequilibrium-in-the-teosinte-and-barley-data-sets"><i class="fa fa-check"></i><b>7.4.1</b> Linkage disequilibrium in the teosinte and barley data sets</a></li>
</ul></li>
<li class="chapter" data-level="7.5" data-path="Population-genetics.html"><a href="Population-genetics.html#recombination-and-linkage-disequilibrium-applications"><i class="fa fa-check"></i><b>7.5</b> Recombination and linkage disequilibrium: applications</a></li>
<li class="chapter" data-level="7.6" data-path="Population-genetics.html"><a href="Population-genetics.html#pedigree-analysis"><i class="fa fa-check"></i><b>7.6</b> Pedigree analysis</a></li>
</ul></li>
<li class="chapter" data-level="8" data-path="Imputation.html"><a href="Imputation.html"><i class="fa fa-check"></i><b>8</b> Imputation.</a>
<ul>
<li class="chapter" data-level="8.1" data-path="Imputation.html"><a href="Imputation.html#introduction"><i class="fa fa-check"></i><b>8.1</b> Introduction</a></li>
<li class="chapter" data-level="8.2" data-path="Imputation.html"><a href="Imputation.html#the-exercise-1"><i class="fa fa-check"></i><b>8.2</b> The exercise</a></li>
<li class="chapter" data-level="8.3" data-path="Imputation.html"><a href="Imputation.html#data-preparation-for-all-methods"><i class="fa fa-check"></i><b>8.3</b> Data preparation (for all methods)</a></li>
<li class="chapter" data-level="8.4" data-path="Imputation.html"><a href="Imputation.html#svd-imputation"><i class="fa fa-check"></i><b>8.4</b> SVD imputation</a></li>
<li class="chapter" data-level="8.5" data-path="Imputation.html"><a href="Imputation.html#mean-imputation"><i class="fa fa-check"></i><b>8.5</b> Mean imputation</a></li>
<li class="chapter" data-level="8.6" data-path="Imputation.html"><a href="Imputation.html#random-forest-imputation"><i class="fa fa-check"></i><b>8.6</b> Random Forest Imputation</a></li>
<li class="chapter" data-level="8.7" data-path="Imputation.html"><a href="Imputation.html#summary-1"><i class="fa fa-check"></i><b>8.7</b> Summary</a></li>
<li class="chapter" data-level="8.8" data-path="Imputation.html"><a href="Imputation.html#conclusions-and-comments"><i class="fa fa-check"></i><b>8.8</b> Conclusions and comments</a></li>
</ul></li>
<li class="chapter" data-level="9" data-path="Association-mapping.html"><a href="Association-mapping.html"><i class="fa fa-check"></i><b>9</b> Association Mapping.</a>
<ul>
<li class="chapter" data-level="9.1" data-path="Association-mapping.html"><a href="Association-mapping.html#introduction-1"><i class="fa fa-check"></i><b>9.1</b> Introduction</a></li>
<li class="chapter" data-level="9.2" data-path="Association-mapping.html"><a href="Association-mapping.html#preparing-the-data-set"><i class="fa fa-check"></i><b>9.2</b> Preparing the data set</a></li>
<li class="chapter" data-level="9.3" data-path="Association-mapping.html"><a href="Association-mapping.html#running-gwaspoly"><i class="fa fa-check"></i><b>9.3</b> Running GWASpoly</a></li>
<li class="chapter" data-level="9.4" data-path="Association-mapping.html"><a href="Association-mapping.html#the-exercise-2"><i class="fa fa-check"></i><b>9.4</b> The exercise</a>
<ul>
<li class="chapter" data-level="9.4.1" data-path="Association-mapping.html"><a href="Association-mapping.html#calculation-and-comparison-of-kinship"><i class="fa fa-check"></i><b>9.4.1</b> Calculation and comparison of kinship</a></li>
<li class="chapter" data-level="9.4.2" data-path="Association-mapping.html"><a href="Association-mapping.html#gwas"><i class="fa fa-check"></i><b>9.4.2</b> GWAS</a></li>
<li class="chapter" data-level="9.4.3" data-path="Association-mapping.html"><a href="Association-mapping.html#analysis-with-covariates"><i class="fa fa-check"></i><b>9.4.3</b> Analysis with covariates</a></li>
<li class="chapter" data-level="9.4.4" data-path="Association-mapping.html"><a href="Association-mapping.html#accounting-for-population-structure"><i class="fa fa-check"></i><b>9.4.4</b> Accounting for population structure</a></li>
<li class="chapter" data-level="9.4.5" data-path="Association-mapping.html"><a href="Association-mapping.html#further-analysis"><i class="fa fa-check"></i><b>9.4.5</b> Further analysis</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="10" data-path="Mapping-Markers.html"><a href="Mapping-Markers.html"><i class="fa fa-check"></i><b>10</b> Mapping Markers and Quantitative Trait Loci.</a>
<ul>
<li class="chapter" data-level="10.1" data-path="Mapping-Markers.html"><a href="Mapping-Markers.html#introduction-2"><i class="fa fa-check"></i><b>10.1</b> Introduction</a></li>
<li class="chapter" data-level="10.2" data-path="Mapping-Markers.html"><a href="Mapping-Markers.html#constructing-a-genetic-map"><i class="fa fa-check"></i><b>10.2</b> Constructing a genetic map</a></li>
<li class="chapter" data-level="10.3" data-path="Mapping-Markers.html"><a href="Mapping-Markers.html#formatting-and-reading-in-the-data"><i class="fa fa-check"></i><b>10.3</b> Formatting and reading in the data</a></li>
<li class="chapter" data-level="10.4" data-path="Mapping-Markers.html"><a href="Mapping-Markers.html#a-quick-map"><i class="fa fa-check"></i><b>10.4</b> A quick map!</a></li>
<li class="chapter" data-level="10.5" data-path="Mapping-Markers.html"><a href="Mapping-Markers.html#estimating-recombination-fractions"><i class="fa fa-check"></i><b>10.5</b> Estimating recombination fractions</a></li>
<li class="chapter" data-level="10.6" data-path="Mapping-Markers.html"><a href="Mapping-Markers.html#forming-linkage-groups"><i class="fa fa-check"></i><b>10.6</b> Forming Linkage groups</a></li>
<li class="chapter" data-level="10.7" data-path="Mapping-Markers.html"><a href="Mapping-Markers.html#marker-ordering"><i class="fa fa-check"></i><b>10.7</b> Marker ordering</a></li>
<li class="chapter" data-level="10.8" data-path="Mapping-Markers.html"><a href="Mapping-Markers.html#improving-the-map"><i class="fa fa-check"></i><b>10.8</b> Improving the map</a></li>
<li class="chapter" data-level="10.9" data-path="Mapping-Markers.html"><a href="Mapping-Markers.html#quality-control-commands-for-marker-qc-and-mapping"><i class="fa fa-check"></i><b>10.9</b> Quality Control commands for marker QC and mapping</a></li>
<li class="chapter" data-level="10.10" data-path="Mapping-Markers.html"><a href="Mapping-Markers.html#asmap"><i class="fa fa-check"></i><b>10.10</b> ASMap</a></li>
</ul></li>
<li class="chapter" data-level="11" data-path="Data-Visualization.html"><a href="Data-Visualization.html"><i class="fa fa-check"></i><b>11</b> Data Visualization.</a>
<ul>
<li class="chapter" data-level="11.1" data-path="Data-Visualization.html"><a href="Data-Visualization.html#flapjack"><i class="fa fa-check"></i><b>11.1</b> Flapjack</a>
<ul>
<li class="chapter" data-level="11.1.1" data-path="Data-Visualization.html"><a href="Data-Visualization.html#genotype-data-.geno-file."><i class="fa fa-check"></i><b>11.1.1</b> Genotype data: “.geno” file.</a></li>
<li class="chapter" data-level="11.1.2" data-path="Data-Visualization.html"><a href="Data-Visualization.html#map-data-.map-file."><i class="fa fa-check"></i><b>11.1.2</b> Map data : “.MAP” file.</a></li>
<li class="chapter" data-level="11.1.3" data-path="Data-Visualization.html"><a href="Data-Visualization.html#trait-data"><i class="fa fa-check"></i><b>11.1.3</b> Trait data</a></li>
<li class="chapter" data-level="11.1.4" data-path="Data-Visualization.html"><a href="Data-Visualization.html#the-exercise-3"><i class="fa fa-check"></i><b>11.1.4</b> The Exercise</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="12" data-path="selection-theory.html"><a href="selection-theory.html"><i class="fa fa-check"></i><b>12</b> Selection theory and breeding program design.</a></li>
<li class="chapter" data-level="13" data-path="Genomic-prediction.html"><a href="Genomic-prediction.html"><i class="fa fa-check"></i><b>13</b> Genomic prediction.</a>
<ul>
<li class="chapter" data-level="13.1" data-path="Genomic-prediction.html"><a href="Genomic-prediction.html#ridge-regression-and-genomic-prediction-in-excel"><i class="fa fa-check"></i><b>13.1</b> Ridge regression and Genomic Prediction in Excel</a>
<ul>
<li class="chapter" data-level="13.1.1" data-path="Genomic-prediction.html"><a href="Genomic-prediction.html#introduction-3"><i class="fa fa-check"></i><b>13.1.1</b> Introduction</a></li>
<li class="chapter" data-level="13.1.2" data-path="Genomic-prediction.html"><a href="Genomic-prediction.html#the-exercise-4"><i class="fa fa-check"></i><b>13.1.2</b> The Exercise</a></li>
</ul></li>
<li class="chapter" data-level="13.2" data-path="Genomic-prediction.html"><a href="Genomic-prediction.html#genomic-prediction-with-rrblup"><i class="fa fa-check"></i><b>13.2</b> Genomic Prediction with rrBLUP</a>
<ul>
<li class="chapter" data-level="13.2.1" data-path="Genomic-prediction.html"><a href="Genomic-prediction.html#introduction-4"><i class="fa fa-check"></i><b>13.2.1</b> Introduction</a></li>
<li class="chapter" data-level="13.2.2" data-path="Genomic-prediction.html"><a href="Genomic-prediction.html#the-dataset"><i class="fa fa-check"></i><b>13.2.2</b> The dataset</a></li>
<li class="chapter" data-level="13.2.3" data-path="Genomic-prediction.html"><a href="Genomic-prediction.html#loading-and-preparing-the-data"><i class="fa fa-check"></i><b>13.2.3</b> Loading and preparing the data</a></li>
<li class="chapter" data-level="13.2.4" data-path="Genomic-prediction.html"><a href="Genomic-prediction.html#examining-the-data-structure"><i class="fa fa-check"></i><b>13.2.4</b> Examining the data structure</a></li>
<li class="chapter" data-level="13.2.5" data-path="Genomic-prediction.html"><a href="Genomic-prediction.html#ordinary-least-squares"><i class="fa fa-check"></i><b>13.2.5</b> Ordinary least squares</a></li>
<li class="chapter" data-level="13.2.6" data-path="Genomic-prediction.html"><a href="Genomic-prediction.html#rrblup"><i class="fa fa-check"></i><b>13.2.6</b> rrBLUP</a></li>
<li class="chapter" data-level="13.2.7" data-path="Genomic-prediction.html"><a href="Genomic-prediction.html#prediction-forward-in-time"><i class="fa fa-check"></i><b>13.2.7</b> Prediction forward in time</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="14" data-path="Breeding-simulations.html"><a href="Breeding-simulations.html"><i class="fa fa-check"></i><b>14</b> Breeding simulations.</a></li>
<li class="chapter" data-level="" data-path="references.html"><a href="references.html"><i class="fa fa-check"></i>References</a></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Quantitative Methods in Plant Breeding: The Digital Tutorials.</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="references" class="section level1 unnumbered">
<h1>References</h1>

<div id="refs" class="references csl-bib-body hanging-indent">
<div id="ref-adler_r_2010" class="csl-entry">
Adler, Joseph. 2010. <em>R in a Nutshell</em>. 1st ed. Sebastopol, CA: O’Reilly.
</div>
<div id="ref-R-lme4" class="csl-entry">
Bates, Douglas, Martin Maechler, Ben Bolker, and Steven Walker. 2020. <em>Lme4: Linear Mixed-Effects Models Using Eigen and S4</em>. <a href="https://github.com/lme4/lme4/">https://github.com/lme4/lme4/</a>.
</div>
<div id="ref-Bentley2014" class="csl-entry">
Bentley, Alison R, Marco Scutari, Nicolas Gosman, Sebastien Faure, Felicity Bedford, Phil Howell, James Cockram, et al. 2014. <span>“<span class="nocase">Applying association mapping and genomic selection to the dissection of key traits in elite European wheat</span>.”</span> <em>Theoretical and Applied Genetics</em> 127 (12): 2619–33. <a href="https://doi.org/10.1007/s00122-014-2403-y">https://doi.org/10.1007/s00122-014-2403-y</a>.
</div>
<div id="ref-Bernardo2007" class="csl-entry">
Bernardo, Rex, and Jianming Yu. 2007. <span>“Prospects for Genomewide Selection for Quantitative Traits in Maize.”</span> <em>Crop Science</em> 47 (3): 1082–90. https://doi.org/<a href="https://doi.org/10.2135/cropsci2006.11.0690">https://doi.org/10.2135/cropsci2006.11.0690</a>.
</div>
<div id="ref-broman_guide_2009" class="csl-entry">
Broman, Karl W., and Saunak Sen. 2009. <em>A <span>Guide</span> to <span>QTL</span> <span>Mapping</span> with <span>R</span>/Qtl</em>. Statistics for <span>Biology</span> and <span>Health</span>. New York, NY: Springer New York. <a href="https://doi.org/10.1007/978-0-387-92125-9">https://doi.org/10.1007/978-0-387-92125-9</a>.
</div>
<div id="ref-dalgaard_introductory_2008" class="csl-entry">
Dalgaard, Peter. 2008. <em>Introductory Statistics with <span>R</span></em>. 2nd ed. Statistics and Computing. New York: Springer.
</div>
<div id="ref-R-agricolae" class="csl-entry">
de Mendiburu, Felipe. 2020. <em>Agricolae: Statistical Procedures for Agricultural Research</em>. <a href="http://tarwi.lamolina.edu.pe/ fmendiburu">http://tarwi.lamolina.edu.pe/ fmendiburu</a>.
</div>
<div id="ref-STRUCTURE2012" class="csl-entry">
Earl, Dent A., and Bridgett Marie vonHoldt. 2012. <span>“STRUCTURE HARVESTER: A Website and Program for Visualizing STRUCTURE Output and Implementing the Evanno Method.”</span> <em>Conservation Genetics Resources</em> 4 (2): 359–61. <a href="https://doi.org/10.1007/s12686-011-9548-7">https://doi.org/10.1007/s12686-011-9548-7</a>.
</div>
<div id="ref-R-blocksdesign" class="csl-entry">
Edmondson., R. N. 2021. <em>Blocksdesign: Nested and Crossed Block Designs for Factorial and Unstructured Treatment Sets</em>. <a href="https://&lt;doi:10.1007/s13253-020-00416-0&gt;">&lt;doi:10.1007/s13253-020-00416-0&gt;</a>.
</div>
<div id="ref-Endelman2011" class="csl-entry">
Endelman, Jeffrey B. 2011. <span>“Ridge Regression and Other Kernels for Genomic Selection with r Package rrBLUP.”</span> <em>The Plant Genome</em> 4 (3). https://doi.org/<a href="https://doi.org/10.3835/plantgenome2011.08.0024">https://doi.org/10.3835/plantgenome2011.08.0024</a>.
</div>
<div id="ref-Evanno2005" class="csl-entry">
Evanno, G., S. Regnaut, and J. Goudet. 2005. <span>“Detecting the Number of Clusters of Individuals Using the Software Structure: A Simulation Study.”</span> <em>Molecular Ecology</em> 14 (8): 2611–20. https://doi.org/<a href="https://doi.org/10.1111/j.1365-294X.2005.02553.x">https://doi.org/10.1111/j.1365-294X.2005.02553.x</a>.
</div>
<div id="ref-Gardner2016" class="csl-entry">
Gardner, Keith A., Lukas M. Wittern, and Ian J. Mackay. 2016. <span>“A Highly Recombined, High-Density, Eight-Founder Wheat MAGIC Map Reveals Extensive Segregation Distortion and Genomic Locations of Introgression Segments.”</span> <em>Plant Biotechnology Journal</em> 14 (6): 1406–17. https://doi.org/<a href="https://doi.org/10.1111/pbi.12504">https://doi.org/10.1111/pbi.12504</a>.
</div>
<div id="ref-R-hierfstat" class="csl-entry">
Goudet, Jerome, and Thibaut Jombart. 2020. <em>Hierfstat: Estimation and Tests of Hierarchical f-Statistics</em>. <a href="https://CRAN.R-project.org/package=hierfstat">https://CRAN.R-project.org/package=hierfstat</a>.
</div>
<div id="ref-Heffner2009" class="csl-entry">
Heffner, Elliot L., Mark E. Sorrells, and Jean-Luc Jannink. 2009. <span>“Genomic Selection for Crop Improvement.”</span> <em>Crop Science</em> 49 (1): 1–12. https://doi.org/<a href="https://doi.org/10.2135/cropsci2008.08.0512">https://doi.org/10.2135/cropsci2008.08.0512</a>.
</div>
<div id="ref-james_2013" class="csl-entry">
James, Gareth, Daniela Witten, Trevor Hastie, and Robert Tibshirani. 2013. <em>An <span>Introduction</span> to <span>Statistical</span> <span>Learning</span></em>. Vol. 103. Springer <span>Texts</span> in <span>Statistics</span>. New York, NY: Springer New York. <a href="https://doi.org/10.1007/978-1-4614-7138-7">https://doi.org/10.1007/978-1-4614-7138-7</a>.
</div>
<div id="ref-Jannink2010" class="csl-entry">
Jannink, Jean-Luc, Aaron J. Lorenz, and Hiroyoshi Iwata. 2010. <span>“<span class="nocase">Genomic selection in plant breeding: from theory to practice</span>.”</span> <em>Briefings in Functional Genomics</em> 9 (2): 166–77. <a href="https://doi.org/10.1093/bfgp/elq001">https://doi.org/10.1093/bfgp/elq001</a>.
</div>
<div id="ref-adegenet2008" class="csl-entry">
Jombart, T. 2008. <span>“Adegenet: A r Package for the Multivariate Analysis of Genetic Markers.”</span> <em>Bioinformatics</em> 24: 1403–5. <a href="https://doi.org/10.1093/bioinformatics/btn129">https://doi.org/10.1093/bioinformatics/btn129</a>.
</div>
<div id="ref-kabacoff_r_2011" class="csl-entry">
Kabacoff, Robert. 2011. <em>R in Action: Data Analysis and Graphics with <span>R</span></em>. Shelter Island, NY: Manning.
</div>
<div id="ref-kang_variance_2010" class="csl-entry">
Kang, Hyun Min, Jae Hoon Sul, Susan K Service, Noah A Zaitlen, Sit-yee Kong, Nelson B Freimer, Chiara Sabatti, and Eleazar Eskin. 2010. <span>“Variance Component Model to Account for Sample Structure in Genome-Wide Association Studies.”</span> <em>Nature Genetics</em> 42 (4): 348–54. <a href="https://doi.org/10.1038/ng.548">https://doi.org/10.1038/ng.548</a>.
</div>
<div id="ref-R-emmeans" class="csl-entry">
Lenth, Russell V. 2020. <em>Emmeans: Estimated Marginal Means, Aka Least-Squares Means</em>. <a href="https://github.com/rvlenth/emmeans">https://github.com/rvlenth/emmeans</a>.
</div>
<div id="ref-R-predictmeans" class="csl-entry">
Luo, Dongwen, Siva Ganesh, and John Koolaard. 2020. <em>Predictmeans: Calculate Predicted Means for Linear Models</em>. <a href="https://CRAN.R-project.org/package=predictmeans">https://CRAN.R-project.org/package=predictmeans</a>.
</div>
<div id="ref-Mackay2011" class="csl-entry">
Mackay, Ian, A Horwell, J Garner, Jon White, J McKee, and H Philpott. 2011. <span>“Reanalyses of the Historical Series of UK Variety Trials to Quantify the Contributions of Genetic and Environmental Factors to Trends and Variability in Yield over Time.”</span> <em>TAG. Theoretical and Applied Genetics. Theoretische Und Angewandte Genetik</em> 122 (September): 225–38. <a href="https://doi.org/10.1007/s00122-010-1438-y">https://doi.org/10.1007/s00122-010-1438-y</a>.
</div>
<div id="ref-mead_design_1994" class="csl-entry">
Mead, Roger. 1994. <em>The Design of Experiments: Statistical Principles for Practical Applications</em>. 1. paperback ed. (with corr.), reprinted. Cambridge: Cambridge Univ. Press.
</div>
<div id="ref-Meuwissen1819" class="csl-entry">
Meuwissen, T. H. E., B. J. Hayes, and M. E. Goddard. 2001. <span>“Prediction of Total Genetic Value Using Genome-Wide Dense Marker Maps.”</span> <em>Genetics</em> 157 (4): 1819–29. <a href="https://www.genetics.org/content/157/4/1819">https://www.genetics.org/content/157/4/1819</a>.
</div>
<div id="ref-R-pegas" class="csl-entry">
Paradis, Emmanuel, Thibaut Jombart, Zhian N. Kamvar, Brian Knaus, Klaus Schliep, Alastair Potts, and David Winter. 2020. <em>Pegas: Population and Evolutionary Genetics Analysis System</em>. <a href="http://ape-package.ird.fr/pegas.html">http://ape-package.ird.fr/pegas.html</a>.
</div>
<div id="ref-patterson_new_1976" class="csl-entry">
Patterson, H. D., and E. R. Williams. 1976. <span>“A New Class of Resolvable Incomplete Block Designs.”</span> <em>Biometrika</em> 63 (1): 83–92. <a href="https://doi.org/10.1093/biomet/63.1.83">https://doi.org/10.1093/biomet/63.1.83</a>.
</div>
<div id="ref-Piepho2009" class="csl-entry">
Piepho, H. P. 2009. <span>“Ridge Regression and Extensions for Genomewide Selection in Maize.”</span> <em>Crop Science</em> 49 (4): 1165–76. https://doi.org/<a href="https://doi.org/10.2135/cropsci2008.10.0595">https://doi.org/10.2135/cropsci2008.10.0595</a>.
</div>
<div id="ref-rodríguezálvarez2016" class="csl-entry">
Rodríguez-Álvarez, María Xosé, Martin P. Boer, Fred A. van Eeuwijk, and Paul H. C. Eilers. 2016. <span>“Spatial Models for Field Trials.”</span> <a href="http://arxiv.org/abs/1607.08255">http://arxiv.org/abs/1607.08255</a>.
</div>
<div id="ref-SpATS2017" class="csl-entry">
Rodríguez-Álvarez, María Xosé, Martin P. Boer, Fred A. van Eeuwijk, and Paul H. C. Eilers. 2017. <span>“Correcting for Spatial Heterogeneity in Plant Breeding Experiments with p-Splines.”</span> <em>Spatial Statistics</em> 23: 52–71. <a href="https://doi.org/10.1016/j.spasta.2017.10.003">https://doi.org/10.1016/j.spasta.2017.10.003</a>.
</div>
<div id="ref-Rutkoski_2013" class="csl-entry">
Rutkoski, Jessica E, Jesse Poland, Jean-Luc Jannink, and Mark E Sorrells. 2013. <span>“<span class="nocase">Imputation of Unordered Markers and the Impact on Genomic Selection Accuracy</span>.”</span> <em>G3 Genes|Genomes|Genetics</em> 3 (3): 427–39. <a href="https://doi.org/10.1534/g3.112.005363">https://doi.org/10.1534/g3.112.005363</a>.
</div>
<div id="ref-speed2012" class="csl-entry">
Speed, Doug, Gibran Hemani, Michael R. Johnson, and David J. Balding. 2012. <span>“Improved Heritability Estimation from Genome-Wide SNPs.”</span> <em>The American Journal of Human Genetics</em> 91 (6): 1011–21. https://doi.org/<a href="https://doi.org/10.1016/j.ajhg.2012.10.010">https://doi.org/10.1016/j.ajhg.2012.10.010</a>.
</div>
<div id="ref-vanraden_efficient_2008" class="csl-entry">
VanRaden, P. M. 2008. <span>“Efficient <span>Methods</span> to <span>Compute</span> <span>Genomic</span> <span>Predictions</span>.”</span> <em>Journal of Dairy Science</em> 91 (11): 4414–23. <a href="https://doi.org/10.3168/jds.2007-0980">https://doi.org/10.3168/jds.2007-0980</a>.
</div>
<div id="ref-velazco_modelling_2017" class="csl-entry">
Velazco, Julio G., María Xosé Rodríguez-Álvarez, Martin P. Boer, David R. Jordan, Paul H. C. Eilers, Marcos Malosetti, and Fred A. van Eeuwijk. 2017. <span>“Modelling Spatial Trends in Sorghum Breeding Field Trials Using a Two-Dimensional <span>P</span>-Spline Mixed Model.”</span> <em>Theoretical and Applied Genetics</em> 130 (7): 1375–92. <a href="https://doi.org/10.1007/s00122-017-2894-4">https://doi.org/10.1007/s00122-017-2894-4</a>.
</div>
<div id="ref-weber_major_2007" class="csl-entry">
Weber, Allison, Richard M. Clark, Laura Vaughn, José de Jesús Sánchez-Gonzalez, Jianming Yu, Brian S. Yandell, Peter Bradbury, and John Doebley. 2007. <span>“Major <span>Regulatory</span> <span>Genes</span> in <span>Maize</span> <span>Contribute</span> to <span>Standing</span> <span>Variation</span> in <span>Teosinte</span> ( <em><span>Zea</span> Mays</em> Ssp. <em>Parviglumis</em> ).”</span> <em>Genetics</em> 177 (4): 2349–59. <a href="https://doi.org/10.1534/genetics.107.080424">https://doi.org/10.1534/genetics.107.080424</a>.
</div>
<div id="ref-wickham_r_2016" class="csl-entry">
Wickham, Hadley, and Garrett Grolemund. 2016. <em>R for Data Science: Import, Tidy, Transform, Visualize, and Model Data</em>. First edition. Sebastopol, CA: O’Reilly.
</div>
<div id="ref-R-desplot" class="csl-entry">
Wright, Kevin. 2020. <em>Desplot: Plotting Field Plans for Agricultural Experiments</em>. <a href="http://kwstat.github.io/desplot/">http://kwstat.github.io/desplot/</a>.
</div>
<div id="ref-xie2015" class="csl-entry">
Xie, Yihui. 2015. <em>Dynamic Documents with <span>R</span> and Knitr</em>. 2nd ed. Boca Raton, Florida: Chapman; Hall/CRC. <a href="http://yihui.name/knitr/">http://yihui.name/knitr/</a>.
</div>
<div id="ref-R-bookdown" class="csl-entry">
———. 2020. <em>Bookdown: Authoring Books and Technical Documents with r Markdown</em>. <a href="https://github.com/rstudio/bookdown">https://github.com/rstudio/bookdown</a>.
</div>
<div id="ref-zhang_mixed_2010" class="csl-entry">
Zhang, Zhiwu, Elhan Ersoz, Chao-Qiang Lai, Rory J Todhunter, Hemant K Tiwari, Michael A Gore, Peter J Bradbury, et al. 2010. <span>“Mixed Linear Model Approach Adapted for Genome-Wide Association Studies.”</span> <em>Nature Genetics</em> 42 (4): 355–60. <a href="https://doi.org/10.1038/ng.546">https://doi.org/10.1038/ng.546</a>.
</div>
</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="Breeding-simulations.html" class="navigation navigation-prev navigation-unique" aria-label="Previous page"><i class="fa fa-angle-left"></i></a>

    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": "https://github.com/rstudio/bookdown-demo/edit/master/15-references.Rmd",
"text": "Edit"
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": ["bookdown-demo.pdf", "bookdown-demo.epub"],
"toc": {
"collapse": "subsection"
}
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
