<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 4 Trial Design. | Quantitative Methods in Plant Breeding: The Digital Tutorials.</title>
  <meta name="description" content="This is the online tutorial booklet for the QMPB course run at NIAB. The online book was formed using R bookdown." />
  <meta name="generator" content="bookdown 0.21 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 4 Trial Design. | Quantitative Methods in Plant Breeding: The Digital Tutorials." />
  <meta property="og:type" content="book" />
  
  <meta property="og:image" content="images/cover.jpg" />
  <meta property="og:description" content="This is the online tutorial booklet for the QMPB course run at NIAB. The online book was formed using R bookdown." />
  <meta name="github-repo" content="rstudio/bookdown" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 4 Trial Design. | Quantitative Methods in Plant Breeding: The Digital Tutorials." />
  
  <meta name="twitter:description" content="This is the online tutorial booklet for the QMPB course run at NIAB. The online book was formed using R bookdown." />
  <meta name="twitter:image" content="images/cover.jpg" />

<meta name="author" content="NIAB 2021" />


<meta name="date" content="2021-02-27" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="R-and-stats.html"/>
<link rel="next" href="Trial-analysis.html"/>
<script src="libs/header-attrs-2.6/header-attrs.js"></script>
<script src="libs/jquery-2.2.3/jquery.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />











<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">The QMPB digital tutorials</a></li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> Introduction</a></li>
<li class="chapter" data-level="2" data-path="Acknow.html"><a href="Acknow.html"><i class="fa fa-check"></i><b>2</b> Acknowledgements and Contents</a>
<ul>
<li class="chapter" data-level="2.1" data-path="Acknow.html"><a href="Acknow.html#acknowledgements"><i class="fa fa-check"></i><b>2.1</b> Acknowledgements</a></li>
<li class="chapter" data-level="2.2" data-path="Acknow.html"><a href="Acknow.html#digital-tutorial-chapters"><i class="fa fa-check"></i><b>2.2</b> Digital Tutorial Chapters:</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="R-and-stats.html"><a href="R-and-stats.html"><i class="fa fa-check"></i><b>3</b> R and Rstudio: a general introduction with statistics.</a>
<ul>
<li class="chapter" data-level="3.1" data-path="R-and-stats.html"><a href="R-and-stats.html#introduction-section-on-r"><i class="fa fa-check"></i><b>3.1</b> Introduction section on R</a>
<ul>
<li class="chapter" data-level="3.1.1" data-path="R-and-stats.html"><a href="R-and-stats.html#getting-the-data-ready"><i class="fa fa-check"></i><b>3.1.1</b> Getting the data ready</a></li>
</ul></li>
<li class="chapter" data-level="3.2" data-path="R-and-stats.html"><a href="R-and-stats.html#the-rstudio-layout-workspace-and-environment."><i class="fa fa-check"></i><b>3.2</b> The Rstudio layout, workspace and environment.</a>
<ul>
<li class="chapter" data-level="3.2.1" data-path="R-and-stats.html"><a href="R-and-stats.html#working-directory"><i class="fa fa-check"></i><b>3.2.1</b> Working directory</a></li>
</ul></li>
<li class="chapter" data-level="3.3" data-path="R-and-stats.html"><a href="R-and-stats.html#basic-r-syntx"><i class="fa fa-check"></i><b>3.3</b> Basic R Syntx</a>
<ul>
<li class="chapter" data-level="3.3.1" data-path="R-and-stats.html"><a href="R-and-stats.html#brackets"><i class="fa fa-check"></i><b>3.3.1</b> Brackets</a></li>
<li class="chapter" data-level="3.3.2" data-path="R-and-stats.html"><a href="R-and-stats.html#equals-symbols"><i class="fa fa-check"></i><b>3.3.2</b> Equals symbols</a></li>
<li class="chapter" data-level="3.3.3" data-path="R-and-stats.html"><a href="R-and-stats.html#continuation-character."><i class="fa fa-check"></i><b>3.3.3</b> Continuation character.</a></li>
<li class="chapter" data-level="3.3.4" data-path="R-and-stats.html"><a href="R-and-stats.html#text-and-numbers"><i class="fa fa-check"></i><b>3.3.4</b> Text and numbers</a></li>
<li class="chapter" data-level="3.3.5" data-path="R-and-stats.html"><a href="R-and-stats.html#summary-of-syntax"><i class="fa fa-check"></i><b>3.3.5</b> Summary of syntax</a></li>
</ul></li>
<li class="chapter" data-level="3.4" data-path="R-and-stats.html"><a href="R-and-stats.html#reading-in-data"><i class="fa fa-check"></i><b>3.4</b> Reading in data</a></li>
<li class="chapter" data-level="3.5" data-path="R-and-stats.html"><a href="R-and-stats.html#understanding-data-types-and-r-objects"><i class="fa fa-check"></i><b>3.5</b> Understanding data types and R objects</a>
<ul>
<li class="chapter" data-level="3.5.1" data-path="R-and-stats.html"><a href="R-and-stats.html#summary"><i class="fa fa-check"></i><b>3.5.1</b> Summary</a></li>
</ul></li>
<li class="chapter" data-level="3.6" data-path="R-and-stats.html"><a href="R-and-stats.html#summarising-and-displaying-data"><i class="fa fa-check"></i><b>3.6</b> Summarising and displaying data</a></li>
<li class="chapter" data-level="3.7" data-path="R-and-stats.html"><a href="R-and-stats.html#calculations-and-data-manipulation"><i class="fa fa-check"></i><b>3.7</b> Calculations and data manipulation</a>
<ul>
<li class="chapter" data-level="3.7.1" data-path="R-and-stats.html"><a href="R-and-stats.html#sorting-data"><i class="fa fa-check"></i><b>3.7.1</b> Sorting data</a></li>
<li class="chapter" data-level="3.7.2" data-path="R-and-stats.html"><a href="R-and-stats.html#relationships-between-variates"><i class="fa fa-check"></i><b>3.7.2</b> Relationships between variates</a></li>
<li class="chapter" data-level="3.7.3" data-path="R-and-stats.html"><a href="R-and-stats.html#exporting-your-script-and-workspace"><i class="fa fa-check"></i><b>3.7.3</b> Exporting your script and workspace</a></li>
<li class="chapter" data-level="3.7.4" data-path="R-and-stats.html"><a href="R-and-stats.html#saving-a-data-table"><i class="fa fa-check"></i><b>3.7.4</b> Saving a data table</a></li>
</ul></li>
<li class="chapter" data-level="3.8" data-path="R-and-stats.html"><a href="R-and-stats.html#basic-statistical-analysis"><i class="fa fa-check"></i><b>3.8</b> Basic statistical analysis</a>
<ul>
<li class="chapter" data-level="3.8.1" data-path="R-and-stats.html"><a href="R-and-stats.html#the-t-test"><i class="fa fa-check"></i><b>3.8.1</b> The t-test</a></li>
<li class="chapter" data-level="3.8.2" data-path="R-and-stats.html"><a href="R-and-stats.html#linear-regression"><i class="fa fa-check"></i><b>3.8.2</b> Linear Regression</a></li>
<li class="chapter" data-level="3.8.3" data-path="R-and-stats.html"><a href="R-and-stats.html#multiple-regression"><i class="fa fa-check"></i><b>3.8.3</b> Multiple regression</a></li>
<li class="chapter" data-level="3.8.4" data-path="R-and-stats.html"><a href="R-and-stats.html#the-analysis-of-variance"><i class="fa fa-check"></i><b>3.8.4</b> The analysis of variance</a></li>
</ul></li>
<li class="chapter" data-level="3.9" data-path="R-and-stats.html"><a href="R-and-stats.html#categorical-data---the-chi-squared-test"><i class="fa fa-check"></i><b>3.9</b> Categorical data - the <em>chi</em>-squared test</a></li>
<li class="chapter" data-level="3.10" data-path="R-and-stats.html"><a href="R-and-stats.html#more-useful-information"><i class="fa fa-check"></i><b>3.10</b> More useful information</a>
<ul>
<li class="chapter" data-level="3.10.1" data-path="R-and-stats.html"><a href="R-and-stats.html#plotting-graphs"><i class="fa fa-check"></i><b>3.10.1</b> Plotting Graphs</a></li>
<li class="chapter" data-level="3.10.2" data-path="R-and-stats.html"><a href="R-and-stats.html#probability-distributions"><i class="fa fa-check"></i><b>3.10.2</b> Probability distributions</a></li>
<li class="chapter" data-level="3.10.3" data-path="R-and-stats.html"><a href="R-and-stats.html#random-number-generation"><i class="fa fa-check"></i><b>3.10.3</b> Random number generation</a></li>
<li class="chapter" data-level="3.10.4" data-path="R-and-stats.html"><a href="R-and-stats.html#loops-second-example"><i class="fa fa-check"></i><b>3.10.4</b> Loops: <em>second example</em></a></li>
<li class="chapter" data-level="3.10.5" data-path="R-and-stats.html"><a href="R-and-stats.html#miscellany"><i class="fa fa-check"></i><b>3.10.5</b> Miscellany</a></li>
<li class="chapter" data-level="3.10.6" data-path="R-and-stats.html"><a href="R-and-stats.html#my-number-one-piece-of-advice"><i class="fa fa-check"></i><b>3.10.6</b> My number one piece of advice</a></li>
<li class="chapter" data-level="3.10.7" data-path="R-and-stats.html"><a href="R-and-stats.html#learn-more"><i class="fa fa-check"></i><b>3.10.7</b> Learn more</a></li>
<li class="chapter" data-level="3.10.8" data-path="R-and-stats.html"><a href="R-and-stats.html#packages"><i class="fa fa-check"></i><b>3.10.8</b> Packages</a></li>
<li class="chapter" data-level="3.10.9" data-path="R-and-stats.html"><a href="R-and-stats.html#the-tidyverse"><i class="fa fa-check"></i><b>3.10.9</b> The Tidyverse</a></li>
<li class="chapter" data-level="3.10.10" data-path="R-and-stats.html"><a href="R-and-stats.html#other-resources"><i class="fa fa-check"></i><b>3.10.10</b> Other resources</a></li>
</ul></li>
<li class="chapter" data-level="3.11" data-path="R-and-stats.html"><a href="R-and-stats.html#list-of-commands-mainly-described-in-this-guide."><i class="fa fa-check"></i><b>3.11</b> List of commands, mainly described in this guide.</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="Trial-design.html"><a href="Trial-design.html"><i class="fa fa-check"></i><b>4</b> Trial Design.</a>
<ul>
<li class="chapter" data-level="4.1" data-path="Trial-design.html"><a href="Trial-design.html#randomised-complete-block-designs"><i class="fa fa-check"></i><b>4.1</b> Randomised complete block designs</a></li>
<li class="chapter" data-level="4.2" data-path="Trial-design.html"><a href="Trial-design.html#trial-design-in-r"><i class="fa fa-check"></i><b>4.2</b> Trial design in R</a>
<ul>
<li class="chapter" data-level="4.2.1" data-path="Trial-design.html"><a href="Trial-design.html#partially-replicated-designs"><i class="fa fa-check"></i><b>4.2.1</b> Partially replicated designs</a></li>
<li class="chapter" data-level="4.2.2" data-path="Trial-design.html"><a href="Trial-design.html#two-dimensional-designs"><i class="fa fa-check"></i><b>4.2.2</b> Two-dimensional designs</a></li>
</ul></li>
<li class="chapter" data-level="4.3" data-path="Trial-design.html"><a href="Trial-design.html#farm-scale-experiments"><i class="fa fa-check"></i><b>4.3</b> Farm scale experiments</a></li>
<li class="chapter" data-level="4.4" data-path="Trial-design.html"><a href="Trial-design.html#alpha-designs."><i class="fa fa-check"></i><b>4.4</b> Alpha designs.</a>
<ul>
<li class="chapter" data-level="4.4.1" data-path="Trial-design.html"><a href="Trial-design.html#creation-of-augmented-and-p-rep-designs-from-2-replicate-alpha-designs"><i class="fa fa-check"></i><b>4.4.1</b> Creation of augmented and p-rep designs from 2-replicate alpha designs</a></li>
<li class="chapter" data-level="4.4.2" data-path="Trial-design.html"><a href="Trial-design.html#final-message"><i class="fa fa-check"></i><b>4.4.2</b> Final message</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="5" data-path="Trial-analysis.html"><a href="Trial-analysis.html"><i class="fa fa-check"></i><b>5</b> Trial analysis (sugar beet example).</a>
<ul>
<li class="chapter" data-level="5.1" data-path="Trial-analysis.html"><a href="Trial-analysis.html#the-layout-of-the-sugar-beet-trial"><i class="fa fa-check"></i><b>5.1</b> The layout of the sugar beet trial</a></li>
<li class="chapter" data-level="5.2" data-path="Trial-analysis.html"><a href="Trial-analysis.html#fixed-effects-random-effects-and-reml"><i class="fa fa-check"></i><b>5.2</b> Fixed effects, random effects and REML</a>
<ul>
<li class="chapter" data-level="5.2.1" data-path="Trial-analysis.html"><a href="Trial-analysis.html#preparing-the-data"><i class="fa fa-check"></i><b>5.2.1</b> Preparing the data</a></li>
<li class="chapter" data-level="5.2.2" data-path="Trial-analysis.html"><a href="Trial-analysis.html#randomised-complete-block-design"><i class="fa fa-check"></i><b>5.2.2</b> Randomised complete block design</a></li>
<li class="chapter" data-level="5.2.3" data-path="Trial-analysis.html"><a href="Trial-analysis.html#randomised-compelete-block-design---with-mixed-models"><i class="fa fa-check"></i><b>5.2.3</b> Randomised compelete block design - with mixed models</a></li>
<li class="chapter" data-level="5.2.4" data-path="Trial-analysis.html"><a href="Trial-analysis.html#plotting-residuals-against-field-structure."><i class="fa fa-check"></i><b>5.2.4</b> Plotting residuals against field structure.</a></li>
<li class="chapter" data-level="5.2.5" data-path="Trial-analysis.html"><a href="Trial-analysis.html#alpha-design-include-block-structure"><i class="fa fa-check"></i><b>5.2.5</b> Alpha design (include block structure)</a></li>
<li class="chapter" data-level="5.2.6" data-path="Trial-analysis.html"><a href="Trial-analysis.html#rows-and-column-analysis"><i class="fa fa-check"></i><b>5.2.6</b> Rows and column analysis</a></li>
</ul></li>
<li class="chapter" data-level="5.3" data-path="Trial-analysis.html"><a href="Trial-analysis.html#spatial-analysis"><i class="fa fa-check"></i><b>5.3</b> Spatial analysis</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="Cross-site-analysis.html"><a href="Cross-site-analysis.html"><i class="fa fa-check"></i><b>6</b> Cross sites analysis in R.</a>
<ul>
<li class="chapter" data-level="6.1" data-path="Cross-site-analysis.html"><a href="Cross-site-analysis.html#blues-blups-and-one-stage-analysis"><i class="fa fa-check"></i><b>6.1</b> BLUEs, BLUPs and one stage analysis</a>
<ul>
<li class="chapter" data-level="6.1.1" data-path="Cross-site-analysis.html"><a href="Cross-site-analysis.html#one-stage-analysis-with-weights"><i class="fa fa-check"></i><b>6.1.1</b> One stage analysis with weights</a></li>
</ul></li>
<li class="chapter" data-level="6.2" data-path="Cross-site-analysis.html"><a href="Cross-site-analysis.html#weighted-two-stage-analysis"><i class="fa fa-check"></i><b>6.2</b> Weighted two-stage analysis</a></li>
<li class="chapter" data-level="6.3" data-path="Cross-site-analysis.html"><a href="Cross-site-analysis.html#ammi"><i class="fa fa-check"></i><b>6.3</b> AMMI</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="Population-genetics.html"><a href="Population-genetics.html"><i class="fa fa-check"></i><b>7</b> Population genetics.</a>
<ul>
<li class="chapter" data-level="7.1" data-path="Population-genetics.html"><a href="Population-genetics.html#hardy-weinberg-equilibrium-and-population-structure"><i class="fa fa-check"></i><b>7.1</b> Hardy-Weinberg equilibrium and population structure</a></li>
<li class="chapter" data-level="7.2" data-path="Population-genetics.html"><a href="Population-genetics.html#other-ways-of-analysing-population-structure"><i class="fa fa-check"></i><b>7.2</b> Other ways of analysing population structure</a>
<ul>
<li class="chapter" data-level="7.2.1" data-path="Population-genetics.html"><a href="Population-genetics.html#distance-trees"><i class="fa fa-check"></i><b>7.2.1</b> Distance Trees</a></li>
</ul></li>
<li class="chapter" data-level="7.3" data-path="Population-genetics.html"><a href="Population-genetics.html#structure"><i class="fa fa-check"></i><b>7.3</b> Structure</a>
<ul>
<li class="chapter" data-level="7.3.1" data-path="Population-genetics.html"><a href="Population-genetics.html#what-structure-does"><i class="fa fa-check"></i><b>7.3.1</b> What STRUCTURE does</a></li>
<li class="chapter" data-level="7.3.2" data-path="Population-genetics.html"><a href="Population-genetics.html#limitations-of-structure"><i class="fa fa-check"></i><b>7.3.2</b> Limitations of STRUCTURE</a></li>
<li class="chapter" data-level="7.3.3" data-path="Population-genetics.html"><a href="Population-genetics.html#using-structure"><i class="fa fa-check"></i><b>7.3.3</b> Using Structure</a></li>
<li class="chapter" data-level="7.3.4" data-path="Population-genetics.html"><a href="Population-genetics.html#the-exercise"><i class="fa fa-check"></i><b>7.3.4</b> The exercise</a></li>
</ul></li>
<li class="chapter" data-level="7.4" data-path="Population-genetics.html"><a href="Population-genetics.html#linkage-disequilibrium-analysis"><i class="fa fa-check"></i><b>7.4</b> Linkage disequilibrium analysis</a>
<ul>
<li class="chapter" data-level="7.4.1" data-path="Population-genetics.html"><a href="Population-genetics.html#linkage-disequilibrium-in-the-teosinte-and-barley-data-sets"><i class="fa fa-check"></i><b>7.4.1</b> Linkage disequilibrium in the teosinte and barley data sets</a></li>
</ul></li>
<li class="chapter" data-level="7.5" data-path="Population-genetics.html"><a href="Population-genetics.html#recombination-and-linkage-disequilibrium-applications"><i class="fa fa-check"></i><b>7.5</b> Recombination and linkage disequilibrium: applications</a></li>
<li class="chapter" data-level="7.6" data-path="Population-genetics.html"><a href="Population-genetics.html#pedigree-analysis"><i class="fa fa-check"></i><b>7.6</b> Pedigree analysis</a></li>
</ul></li>
<li class="chapter" data-level="8" data-path="Imputation.html"><a href="Imputation.html"><i class="fa fa-check"></i><b>8</b> Imputation.</a>
<ul>
<li class="chapter" data-level="8.1" data-path="Imputation.html"><a href="Imputation.html#introduction"><i class="fa fa-check"></i><b>8.1</b> Introduction</a></li>
<li class="chapter" data-level="8.2" data-path="Imputation.html"><a href="Imputation.html#the-exercise-1"><i class="fa fa-check"></i><b>8.2</b> The exercise</a></li>
<li class="chapter" data-level="8.3" data-path="Imputation.html"><a href="Imputation.html#data-preparation-for-all-methods"><i class="fa fa-check"></i><b>8.3</b> Data preparation (for all methods)</a></li>
<li class="chapter" data-level="8.4" data-path="Imputation.html"><a href="Imputation.html#svd-imputation"><i class="fa fa-check"></i><b>8.4</b> SVD imputation</a></li>
<li class="chapter" data-level="8.5" data-path="Imputation.html"><a href="Imputation.html#mean-imputation"><i class="fa fa-check"></i><b>8.5</b> Mean imputation</a></li>
<li class="chapter" data-level="8.6" data-path="Imputation.html"><a href="Imputation.html#random-forest-imputation"><i class="fa fa-check"></i><b>8.6</b> Random Forest Imputation</a></li>
<li class="chapter" data-level="8.7" data-path="Imputation.html"><a href="Imputation.html#summary-1"><i class="fa fa-check"></i><b>8.7</b> Summary</a></li>
<li class="chapter" data-level="8.8" data-path="Imputation.html"><a href="Imputation.html#conclusions-and-comments"><i class="fa fa-check"></i><b>8.8</b> Conclusions and comments</a></li>
</ul></li>
<li class="chapter" data-level="9" data-path="Association-mapping.html"><a href="Association-mapping.html"><i class="fa fa-check"></i><b>9</b> Association Mapping.</a>
<ul>
<li class="chapter" data-level="9.1" data-path="Association-mapping.html"><a href="Association-mapping.html#introduction-1"><i class="fa fa-check"></i><b>9.1</b> Introduction</a></li>
<li class="chapter" data-level="9.2" data-path="Association-mapping.html"><a href="Association-mapping.html#preparing-the-data-set"><i class="fa fa-check"></i><b>9.2</b> Preparing the data set</a></li>
<li class="chapter" data-level="9.3" data-path="Association-mapping.html"><a href="Association-mapping.html#running-gwaspoly"><i class="fa fa-check"></i><b>9.3</b> Running GWASpoly</a></li>
<li class="chapter" data-level="9.4" data-path="Association-mapping.html"><a href="Association-mapping.html#the-exercise-2"><i class="fa fa-check"></i><b>9.4</b> The exercise</a>
<ul>
<li class="chapter" data-level="9.4.1" data-path="Association-mapping.html"><a href="Association-mapping.html#calculation-and-comparison-of-kinship"><i class="fa fa-check"></i><b>9.4.1</b> Calculation and comparison of kinship</a></li>
<li class="chapter" data-level="9.4.2" data-path="Association-mapping.html"><a href="Association-mapping.html#gwas"><i class="fa fa-check"></i><b>9.4.2</b> GWAS</a></li>
<li class="chapter" data-level="9.4.3" data-path="Association-mapping.html"><a href="Association-mapping.html#analysis-with-covariates"><i class="fa fa-check"></i><b>9.4.3</b> Analysis with covariates</a></li>
<li class="chapter" data-level="9.4.4" data-path="Association-mapping.html"><a href="Association-mapping.html#accounting-for-population-structure"><i class="fa fa-check"></i><b>9.4.4</b> Accounting for population structure</a></li>
<li class="chapter" data-level="9.4.5" data-path="Association-mapping.html"><a href="Association-mapping.html#further-analysis"><i class="fa fa-check"></i><b>9.4.5</b> Further analysis</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="10" data-path="Mapping-Markers.html"><a href="Mapping-Markers.html"><i class="fa fa-check"></i><b>10</b> Mapping Markers and Quantitative Trait Loci.</a>
<ul>
<li class="chapter" data-level="10.1" data-path="Mapping-Markers.html"><a href="Mapping-Markers.html#introduction-2"><i class="fa fa-check"></i><b>10.1</b> Introduction</a></li>
<li class="chapter" data-level="10.2" data-path="Mapping-Markers.html"><a href="Mapping-Markers.html#constructing-a-genetic-map"><i class="fa fa-check"></i><b>10.2</b> Constructing a genetic map</a></li>
<li class="chapter" data-level="10.3" data-path="Mapping-Markers.html"><a href="Mapping-Markers.html#formatting-and-reading-in-the-data"><i class="fa fa-check"></i><b>10.3</b> Formatting and reading in the data</a></li>
<li class="chapter" data-level="10.4" data-path="Mapping-Markers.html"><a href="Mapping-Markers.html#a-quick-map"><i class="fa fa-check"></i><b>10.4</b> A quick map!</a></li>
<li class="chapter" data-level="10.5" data-path="Mapping-Markers.html"><a href="Mapping-Markers.html#estimating-recombination-fractions"><i class="fa fa-check"></i><b>10.5</b> Estimating recombination fractions</a></li>
<li class="chapter" data-level="10.6" data-path="Mapping-Markers.html"><a href="Mapping-Markers.html#forming-linkage-groups"><i class="fa fa-check"></i><b>10.6</b> Forming Linkage groups</a></li>
<li class="chapter" data-level="10.7" data-path="Mapping-Markers.html"><a href="Mapping-Markers.html#marker-ordering"><i class="fa fa-check"></i><b>10.7</b> Marker ordering</a></li>
<li class="chapter" data-level="10.8" data-path="Mapping-Markers.html"><a href="Mapping-Markers.html#improving-the-map"><i class="fa fa-check"></i><b>10.8</b> Improving the map</a></li>
<li class="chapter" data-level="10.9" data-path="Mapping-Markers.html"><a href="Mapping-Markers.html#quality-control-commands-for-marker-qc-and-mapping"><i class="fa fa-check"></i><b>10.9</b> Quality Control commands for marker QC and mapping</a></li>
<li class="chapter" data-level="10.10" data-path="Mapping-Markers.html"><a href="Mapping-Markers.html#asmap"><i class="fa fa-check"></i><b>10.10</b> ASMap</a></li>
</ul></li>
<li class="chapter" data-level="11" data-path="Data-Visualization.html"><a href="Data-Visualization.html"><i class="fa fa-check"></i><b>11</b> Data Visualization.</a>
<ul>
<li class="chapter" data-level="11.1" data-path="Data-Visualization.html"><a href="Data-Visualization.html#flapjack"><i class="fa fa-check"></i><b>11.1</b> Flapjack</a>
<ul>
<li class="chapter" data-level="11.1.1" data-path="Data-Visualization.html"><a href="Data-Visualization.html#genotype-data-.geno-file."><i class="fa fa-check"></i><b>11.1.1</b> Genotype data: “.geno” file.</a></li>
<li class="chapter" data-level="11.1.2" data-path="Data-Visualization.html"><a href="Data-Visualization.html#map-data-.map-file."><i class="fa fa-check"></i><b>11.1.2</b> Map data : “.MAP” file.</a></li>
<li class="chapter" data-level="11.1.3" data-path="Data-Visualization.html"><a href="Data-Visualization.html#trait-data"><i class="fa fa-check"></i><b>11.1.3</b> Trait data</a></li>
<li class="chapter" data-level="11.1.4" data-path="Data-Visualization.html"><a href="Data-Visualization.html#the-exercise-3"><i class="fa fa-check"></i><b>11.1.4</b> The Exercise</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="12" data-path="selection-theory.html"><a href="selection-theory.html"><i class="fa fa-check"></i><b>12</b> Selection theory and breeding program design.</a></li>
<li class="chapter" data-level="13" data-path="Genomic-prediction.html"><a href="Genomic-prediction.html"><i class="fa fa-check"></i><b>13</b> Genomic prediction.</a>
<ul>
<li class="chapter" data-level="13.1" data-path="Genomic-prediction.html"><a href="Genomic-prediction.html#ridge-regression-and-genomic-prediction-in-excel"><i class="fa fa-check"></i><b>13.1</b> Ridge regression and Genomic Prediction in Excel</a>
<ul>
<li class="chapter" data-level="13.1.1" data-path="Genomic-prediction.html"><a href="Genomic-prediction.html#introduction-3"><i class="fa fa-check"></i><b>13.1.1</b> Introduction</a></li>
<li class="chapter" data-level="13.1.2" data-path="Genomic-prediction.html"><a href="Genomic-prediction.html#the-exercise-4"><i class="fa fa-check"></i><b>13.1.2</b> The Exercise</a></li>
</ul></li>
<li class="chapter" data-level="13.2" data-path="Genomic-prediction.html"><a href="Genomic-prediction.html#genomic-prediction-with-rrblup"><i class="fa fa-check"></i><b>13.2</b> Genomic Prediction with rrBLUP</a>
<ul>
<li class="chapter" data-level="13.2.1" data-path="Genomic-prediction.html"><a href="Genomic-prediction.html#introduction-4"><i class="fa fa-check"></i><b>13.2.1</b> Introduction</a></li>
<li class="chapter" data-level="13.2.2" data-path="Genomic-prediction.html"><a href="Genomic-prediction.html#the-dataset"><i class="fa fa-check"></i><b>13.2.2</b> The dataset</a></li>
<li class="chapter" data-level="13.2.3" data-path="Genomic-prediction.html"><a href="Genomic-prediction.html#loading-and-preparing-the-data"><i class="fa fa-check"></i><b>13.2.3</b> Loading and preparing the data</a></li>
<li class="chapter" data-level="13.2.4" data-path="Genomic-prediction.html"><a href="Genomic-prediction.html#examining-the-data-structure"><i class="fa fa-check"></i><b>13.2.4</b> Examining the data structure</a></li>
<li class="chapter" data-level="13.2.5" data-path="Genomic-prediction.html"><a href="Genomic-prediction.html#ordinary-least-squares"><i class="fa fa-check"></i><b>13.2.5</b> Ordinary least squares</a></li>
<li class="chapter" data-level="13.2.6" data-path="Genomic-prediction.html"><a href="Genomic-prediction.html#rrblup"><i class="fa fa-check"></i><b>13.2.6</b> rrBLUP</a></li>
<li class="chapter" data-level="13.2.7" data-path="Genomic-prediction.html"><a href="Genomic-prediction.html#prediction-forward-in-time"><i class="fa fa-check"></i><b>13.2.7</b> Prediction forward in time</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="14" data-path="Breeding-simulations.html"><a href="Breeding-simulations.html"><i class="fa fa-check"></i><b>14</b> Breeding simulations.</a></li>
<li class="chapter" data-level="" data-path="references.html"><a href="references.html"><i class="fa fa-check"></i>References</a></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Quantitative Methods in Plant Breeding: The Digital Tutorials.</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="Trial-design" class="section level1" number="4">
<h1><span class="header-section-number">Chapter 4</span> Trial Design.</h1>
<div id="randomised-complete-block-designs" class="section level2" number="4.1">
<h2><span class="header-section-number">4.1</span> Randomised complete block designs</h2>
<p>Firstly, we shall use Excel to create a randomised complete block design for 10 varieties in 6 replicates. Start a new workbook. You will need three column headings: “replicate,” “variety code,” and “random no.” First, we need to enter the trial design:</p>
<p>In the “replicate” column, enter 1 in the first 10 rows, 2 in the second 10 rows …. 6 in rows 52-61.</p>
<p>In the “variety” column, enter 1…10 in the first 10 rows, 1…10 in the second 10 rows…….1 to 10 in rows 52-61.</p>
<p>Next, we want to randomise the position of the varieties within replicates; each replicate must end up still containing every variety once only.</p>
<p>In the “random no” column, enter “=rand()” in each of the 60 rows below the header. Sort on replicate, then on random number within replicate.</p>
<p>Job done!</p>
</div>
<div id="trial-design-in-r" class="section level2" number="4.2">
<h2><span class="header-section-number">4.2</span> Trial design in R</h2>
<p>We will use the R package <code>blocksdesign</code><span class="citation">(<a href="references.html#ref-R-blocksdesign" role="doc-biblioref">Edmondson. 2021</a>)</span> to create partially replicated and other designs. This package originated as a stand-alone program and website, a mirror of which was hosted at NIAB, but the developer, Rodney Edmondson, now only maintains the R version.</p>
<p>To start this tutorial, open a new RStudio screen and start a new script window. If you don’t have the package already in your library you can install it by running:</p>
<p><code>install.packages("blocksdesign")</code></p>
<p>Regardless of if you’ve got the package installed in your library already, you will still need to load it to your current working session. You can do this with:</p>
<div class="sourceCode" id="cb237"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb237-1"><a href="Trial-design.html#cb237-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(blocksdesign)</span></code></pre></div>
<p>I always start my scripts with a list of library commands, stating the packages I will want to use in each analysis.</p>
<p>The command blocks is used to create designs:</p>
<p><code>blocks(treatments, replicates, blocks = NULL, searches = NULL, seed = NULL, jumps = 1)</code></p>
<ul>
<li><code>treatments</code> is the number of varieties</li>
<li><code>replicates</code> is the number of replicates</li>
<li><code>blocks</code> defines the number of (incomplete) row blocks</li>
</ul>
<p>The other options can be ignored for now, and omitted. <code>blocks</code> can be ignored too, for simple designs:</p>
<div class="sourceCode" id="cb238"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb238-1"><a href="Trial-design.html#cb238-1" aria-hidden="true" tabindex="-1"></a><span class="fu">blocks</span>(<span class="at">treatments=</span><span class="dv">10</span>, <span class="at">replicates =</span> <span class="dv">6</span>)</span></code></pre></div>
<pre><code>## $Replication
##    Treatments Freq
## 1           1    6
## 2           2    6
## 3           3    6
## 4           4    6
## 5           5    6
## 6           6    6
## 7           7    6
## 8           8    6
## 9           9    6
## 10         10    6
## 
## $Blocks_model
##   Level Blocks D-Efficiency A-Efficiency A-Bound
## 1     1      1            1            1       1
## 
## $Design
##    Level_1 plots treatments
## 1       B1     1          2
## 2       B1     2          8
## 3       B1     3          3
## 4       B1     4          3
## 5       B1     5          9
## 6       B1     6          4
## 7       B1     7          4
## 8       B1     8          5
## 9       B1     9          7
## 10      B1    10          8
## 11      B1    11          9
## 12      B1    12          5
## 13      B1    13          9
## 14      B1    14          6
## 15      B1    15          5
## 16      B1    16          2
## 17      B1    17         10
## 18      B1    18         10
## 19      B1    19          6
## 20      B1    20          8
## 21      B1    21          1
## 22      B1    22         10
## 23      B1    23          7
## 24      B1    24          7
## 25      B1    25          1
## 26      B1    26          7
## 27      B1    27          2
## 28      B1    28          7
## 29      B1    29          9
## 30      B1    30          2
## 31      B1    31          1
## 32      B1    32          2
## 33      B1    33          5
## 34      B1    34          3
## 35      B1    35          6
## 36      B1    36          4
## 37      B1    37          6
## 38      B1    38          9
## 39      B1    39          3
## 40      B1    40          6
## 41      B1    41          4
## 42      B1    42          5
## 43      B1    43          1
## 44      B1    44          1
## 45      B1    45          8
## 46      B1    46          8
## 47      B1    47          4
## 48      B1    48          3
## 49      B1    49          4
## 50      B1    50         10
## 51      B1    51          6
## 52      B1    52         10
## 53      B1    53          8
## 54      B1    54          2
## 55      B1    55          1
## 56      B1    56          5
## 57      B1    57          9
## 58      B1    58          3
## 59      B1    59         10
## 60      B1    60          7
## 
## $Plan
##   Level_1 Blocks.Plots: 1 2 3 4 5 6 7 8 9 10 11 12 13 14 15 16 17 18 19 20 21
## 1      B1               2 8 3 3 9 4 4 5 7  8  9  5  9  6  5  2 10 10  6  8  1
##   22 23 24 25 26 27 28 29 30 31 32 33 34 35 36 37 38 39 40 41 42 43 44 45 46 47
## 1 10  7  7  1  7  2  7  9  2  1  2  5  3  6  4  6  9  3  6  4  5  1  1  8  8  4
##   48 49 50 51 52 53 54 55 56 57 58 59 60
## 1  3  4 10  6 10  8  2  1  5  9  3 10  7
## 
## $seed
## NULL
## 
## $searches
## [1] 84
## 
## $jumps
## [1] 1</code></pre>
<p>This will produce a version of the randomised complete block design design we produced above. That was much faster than the excel approach! It’s better for us to save the result. We will save it as <code>RCB.save</code>:</p>
<div class="sourceCode" id="cb240"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb240-1"><a href="Trial-design.html#cb240-1" aria-hidden="true" tabindex="-1"></a>RCB.save<span class="ot">&lt;-</span> <span class="fu">blocks</span>(<span class="at">treatments=</span><span class="dv">10</span>, <span class="at">replicates =</span> <span class="dv">6</span>)</span></code></pre></div>
<p>The package saves the results in a list object. You can determine what type of object you’ve saved by running:</p>
<div class="sourceCode" id="cb241"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb241-1"><a href="Trial-design.html#cb241-1" aria-hidden="true" tabindex="-1"></a><span class="fu">class</span>(RCB.save)</span></code></pre></div>
<pre><code>## [1] &quot;list&quot;</code></pre>
<p>In this case, you can use <code>$</code> to access the useful output you saw printed when we ran the command before saving.</p>

<div class="rmdquiz">
<p><strong>1</strong> Try running these on your own computer:</p>
<p><code>RCB.save$Treatments</code></p>
<p><code>RCB.save$Design</code></p>
<code>RCB.save$Plan</code>
</div>
<div id="partially-replicated-designs" class="section level3" number="4.2.1">
<h3><span class="header-section-number">4.2.1</span> Partially replicated designs</h3>
<p>P-rep (partially replicated) designs in which not all entries are replicated, and augmented designs in which a small number of control varieties may be highly replicated but candidate entries are not, are useful in the early stages of plant breeding when only limited seed is available. We still require some replication both to provide an estimate of error and to give some local control of fertility effects through blocking and/or spatial analysis. The <code>blocksdesign</code> package can create these.</p>
<div class="sourceCode" id="cb243"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb243-1"><a href="Trial-design.html#cb243-1" aria-hidden="true" tabindex="-1"></a>p.rep<span class="fl">.1</span><span class="ot">&lt;-</span> <span class="fu">blocks</span>(<span class="at">treatments=</span><span class="fu">c</span>(<span class="dv">2</span>,<span class="dv">8</span>), <span class="at">replicates =</span> <span class="fu">c</span>(<span class="dv">10</span>,<span class="dv">1</span>))</span></code></pre></div>
<p>The above will produce a design with 2 varieties in 10 replicates and 8 in 1. Note that the order in treatments and replicates must agree. Although this works, you should find it has produced a completely randomised design with no blocking. That is not too bad in this case, as the number of entries is so low, but it is not usually what we usually require.</p>
<div class="sourceCode" id="cb244"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb244-1"><a href="Trial-design.html#cb244-1" aria-hidden="true" tabindex="-1"></a>p.rep<span class="fl">.2</span><span class="ot">&lt;-</span> <span class="fu">blocks</span>(<span class="at">treatments=</span><span class="fu">c</span>(<span class="dv">2</span>,<span class="dv">8</span>), <span class="at">replicates =</span> <span class="fu">c</span>(<span class="dv">10</span>,<span class="dv">1</span>), <span class="at">blocks=</span><span class="dv">2</span>)</span></code></pre></div>
<p>If we run the above we now have two blocks.</p>

<div class="rmdcaution">
A cautionary note: many people (including me sometimes) use ‘blocks’ and ‘replicates’ interchangeably. But this only true for an randomised complete block design. In other cases it can lead to confusion when, as here, there are variable replicate numbers so that none of the block structures corresponds to ‘replicate blocks’ or ‘complete blocks.’
</div>

<div class="rmdquiz">
<strong>2</strong> What do you find about the allocation of the two control varieties over the blocks? As the design is small you can just look and count.
</div>
<p>If we wanted to count control number across block in larger trial we could run:</p>
<div class="sourceCode" id="cb245"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb245-1"><a href="Trial-design.html#cb245-1" aria-hidden="true" tabindex="-1"></a><span class="fu">table</span>(p.rep<span class="fl">.2</span><span class="sc">$</span>Design<span class="sc">$</span>Level_1, p.rep<span class="fl">.2</span><span class="sc">$</span>Design<span class="sc">$</span>treatments)</span></code></pre></div>
<pre><code>##     
##      1 2 3 4 5 6 7 8 9 10
##   B1 5 5 1 0 1 1 0 1 0  0
##   B2 5 5 0 1 0 0 1 0 1  1</code></pre>
<p>This uses the function <code>table</code>. Perhaps we think that a block size of 10 is too great, but that 5 would be better.</p>
<div class="sourceCode" id="cb247"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb247-1"><a href="Trial-design.html#cb247-1" aria-hidden="true" tabindex="-1"></a>p.rep<span class="fl">.3</span><span class="ot">&lt;-</span> <span class="fu">blocks</span>(<span class="at">treatments=</span><span class="fu">c</span>(<span class="dv">2</span>,<span class="dv">8</span>), <span class="at">replicates =</span> <span class="fu">c</span>(<span class="dv">10</span>,<span class="dv">1</span>),<span class="at">blocks=</span><span class="dv">4</span>)</span></code></pre></div>
<p>Running the above will achieve a block size of 5.</p>

<div class="rmdquiz">
<strong>3</strong> Either use <code>table</code> or inspect the <code>design</code> to examine the allocation of the control varieties over blocks. Does it look sensible?
</div>
<p>If we run:</p>
<div class="sourceCode" id="cb248"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb248-1"><a href="Trial-design.html#cb248-1" aria-hidden="true" tabindex="-1"></a>p.rep<span class="fl">.4</span><span class="ot">&lt;-</span> <span class="fu">blocks</span>(<span class="at">treatments=</span><span class="fu">c</span>(<span class="dv">2</span>,<span class="dv">8</span>), <span class="at">replicates =</span> <span class="fu">c</span>(<span class="dv">10</span>,<span class="dv">1</span>),<span class="at">blocks=</span><span class="fu">c</span>(<span class="dv">2</span>,<span class="dv">2</span>))</span>
<span id="cb248-2"><a href="Trial-design.html#cb248-2" aria-hidden="true" tabindex="-1"></a>p.rep<span class="fl">.4</span><span class="sc">$</span>Plan</span></code></pre></div>
<pre><code>##   Level_1 Level_2 Blocks.Plots:  1 2 3 4 5 6 7
## 1      B1   B1.B1                7 1 2 2 1 6 1
## 2      B1   B1.B2                5 1 1 2 2 2 3
## 3      B2   B2.B1               10 1 8 1 2 1 2
## 4      B2   B2.B2                1 9 4 2 1 2 2</code></pre>
<p>We have introduced an additional constraint. There are two levels of blocking: one with two blocks, each of size 10, and then one with two blocks of size five nested within each of the larger blocks. In this case, there is no difference in design between p.rep.3 and p.rep.4, but this is not usually the case.</p>
<p>The first level of blocking is defining complete replicates and the second incomplete blocks within replicates. The second level of blocking is labelled 1, 2 within each of the first levels. This is confusing as there are now two separate blocks both labeled 1 (and 2). It is also a potential cause of error when we come to analyse the data - we’ll look at that tomorrow.</p>
<p>The <code>blocks</code> command is great for blocking in one dimension, which is very often all we want to do. The facility to nest blocks within one dimension has advantages: it removes much of the angst about selecting block size since you can have two or three nested blocks with little loss of precision compared to choosing the best size in the first case and a potential gain in precision compared to using the wrong block size.</p>
</div>
<div id="two-dimensional-designs" class="section level3" number="4.2.2">
<h3><span class="header-section-number">4.2.2</span> Two-dimensional designs</h3>
<p><code>blocks</code> will not construct two-dimensional designs however. For this, we need to use the more flexible command <code>design</code>. We shall use this to create a classical two-dimensional model: the balanced lattice square design has a treatment number which is a square (9, 16, 25 etc.) and each treatment occurs once with every other treatment in both a row block and a column block. The block size, k, is the square root of the number of treatments. There are k+1 complete replicates and there are k blocks in each replicate. These are powerful designs, controlling error variation in two dimensions. They are also very restrictive in acceptable variety and replicate numbers.</p>
<p>We shall create the design for 9 varieties in 4 replicates. <code>design</code> works on the complete block structure and list of <code>treatments</code> and then allocates the <code>treatments</code> to the <code>blocks</code> optimally. This is very flexible but you need to create the list of blocks and treatments, I created the blocks structure in Excel. Read it in from the file “lattice_9.csv.”</p>

<div class="rmdimportant">
For the next steps to work you will either need to change your RStudio working directory to the folder containing the downloaded data files for Chapter 4 (which you should have downloaded to your computer from the course website). Or move the datafiles to your current working directory outside of R.
</div>
<p>Read in the block structure from the existing file:</p>
<div class="sourceCode" id="cb250"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb250-1"><a href="Trial-design.html#cb250-1" aria-hidden="true" tabindex="-1"></a>lattice<span class="fl">.9</span><span class="ot">&lt;-</span><span class="fu">read.csv</span>(<span class="st">&quot;lattice_9.csv&quot;</span>,<span class="at">header=</span>T,<span class="at">colClasses =</span> <span class="st">&quot;factor&quot;</span>)</span></code></pre></div>
<p>In each replicate there are three row blocks and three column blocks.</p>
<p>The <code>colClasses = "factor"</code> option ensures all the columns are treated as factors rather than numbers. It is possible to create this in R with a few short commands; see examples under <code>help(design)</code> but in this case it is as easy to create the design in Excel.</p>
<p>We also need a factor for nine varieties replicated four times:</p>
<div class="sourceCode" id="cb251"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb251-1"><a href="Trial-design.html#cb251-1" aria-hidden="true" tabindex="-1"></a>var.code<span class="ot">&lt;-</span><span class="fu">factor</span>(<span class="fu">rep</span>(<span class="dv">1</span><span class="sc">:</span><span class="dv">9</span>,<span class="dv">4</span>))</span></code></pre></div>
<p>This is slight overkill in this case: there is no need to provide the replicate number as it is equal for all varieties and R just repeats the list, but it introduces the easy extension for unequal replicate numbers. For example:</p>
<div class="sourceCode" id="cb252"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb252-1"><a href="Trial-design.html#cb252-1" aria-hidden="true" tabindex="-1"></a><span class="fu">factor</span>(<span class="fu">c</span>(<span class="fu">rep</span>(<span class="dv">1</span><span class="sc">:</span><span class="dv">7</span>,<span class="dv">4</span>),<span class="fu">rep</span>(<span class="dv">8</span>,<span class="dv">8</span>)))</span></code></pre></div>
<pre><code>##  [1] 1 2 3 4 5 6 7 1 2 3 4 5 6 7 1 2 3 4 5 6 7 1 2 3 4 5 6 7 8 8 8 8 8 8 8 8
## Levels: 1 2 3 4 5 6 7 8</code></pre>
<p>The above command defines a treatment structure for seven varieties in 4 replicates and one in 8 replicates.</p>
<p>Returning to the 3 x 3 lattice, to produce the design:</p>
<div class="sourceCode" id="cb254"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb254-1"><a href="Trial-design.html#cb254-1" aria-hidden="true" tabindex="-1"></a>latt.sq<span class="ot">&lt;-</span><span class="fu">design</span>(<span class="at">treatments=</span>var.code,<span class="at">blocks=</span>lattice<span class="fl">.9</span>)</span></code></pre></div>
<p>That takes slightly longer to run than the <code>blocks</code> command. The syntax is clear. Check the output makes sense.</p>
<p>We want to check if each variety pair occurs once in each row-block and once in each column-block. We can use <code>interaction</code> to give each of these blocks a unique code over replicates and <code>table</code> to tabulate variety codes with blocks. We shall use a new command <code>crossprod</code> to work out the number of concurrences of each variety with every other in these blocks. [In matrix terminology, crossprod pre-multiplies a matrix by its transpose.]</p>
<div class="sourceCode" id="cb255"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb255-1"><a href="Trial-design.html#cb255-1" aria-hidden="true" tabindex="-1"></a><span class="fu">crossprod</span>(<span class="fu">table</span>(<span class="fu">interaction</span>(latt.sq<span class="sc">$</span>Design<span class="sc">$</span>reps,latt.sq<span class="sc">$</span>Design<span class="sc">$</span>rows),</span>
<span id="cb255-2"><a href="Trial-design.html#cb255-2" aria-hidden="true" tabindex="-1"></a>latt.sq<span class="sc">$</span>Design<span class="sc">$</span>treatments))</span></code></pre></div>
<pre><code>##    
##     1 2 3 4 5 6 7 8 9
##   1 4 1 1 1 1 1 1 1 1
##   2 1 4 1 1 1 1 1 1 1
##   3 1 1 4 1 1 1 1 1 1
##   4 1 1 1 4 1 1 1 1 1
##   5 1 1 1 1 4 1 1 1 1
##   6 1 1 1 1 1 4 1 1 1
##   7 1 1 1 1 1 1 4 1 1
##   8 1 1 1 1 1 1 1 4 1
##   9 1 1 1 1 1 1 1 1 4</code></pre>
<p>Row blocks look good. What about column blocks?</p>
<div class="sourceCode" id="cb257"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb257-1"><a href="Trial-design.html#cb257-1" aria-hidden="true" tabindex="-1"></a><span class="fu">crossprod</span>(<span class="fu">table</span>(<span class="fu">interaction</span>(latt.sq<span class="sc">$</span>Design<span class="sc">$</span>reps,latt.sq<span class="sc">$</span>Design<span class="sc">$</span>cols),</span>
<span id="cb257-2"><a href="Trial-design.html#cb257-2" aria-hidden="true" tabindex="-1"></a>latt.sq<span class="sc">$</span>Design<span class="sc">$</span>treatments))</span></code></pre></div>
<pre><code>##    
##     1 2 3 4 5 6 7 8 9
##   1 4 1 1 1 1 1 1 1 1
##   2 1 4 1 1 1 1 1 1 1
##   3 1 1 4 1 1 1 1 1 1
##   4 1 1 1 4 1 1 1 1 1
##   5 1 1 1 1 4 1 1 1 1
##   6 1 1 1 1 1 4 1 1 1
##   7 1 1 1 1 1 1 4 1 1
##   8 1 1 1 1 1 1 1 4 1
##   9 1 1 1 1 1 1 1 1 4</code></pre>
<p>blocksdesign searches for the best design. We can make it search harder by increasing the number of <code>searches</code>:</p>

<div class="rmdquiz">
<strong>4</strong> Increase search number to 1000 in the design command. You can do this by adding <code>searches = 1000</code>. Does that improve the design?
</div>

<div class="rmdnote">
<p>Increasing search number will mean the function takes longer to run. It can be hard to know if R is still running something and you can think your computer has frozen.</p>
When something is running, a red stop sign will appear in the toolbar of the console window (top righthand side). If you press this it will stop the run. It’s also a very usefull indicator to know if something is running still.
</div>
<div id="a-larger-balanced-lattice-square" class="section level4" number="4.2.2.1">
<h4><span class="header-section-number">4.2.2.1</span> A larger balanced lattice square</h4>
<p>Read in the following file from your working directory:</p>
<div class="sourceCode" id="cb259"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb259-1"><a href="Trial-design.html#cb259-1" aria-hidden="true" tabindex="-1"></a>lattice<span class="fl">.25</span><span class="ot">&lt;-</span><span class="fu">read.csv</span>(<span class="st">&quot;lattice_25.csv&quot;</span>,<span class="at">header=</span>T,<span class="at">colClasses =</span> <span class="st">&quot;factor&quot;</span>)</span></code></pre></div>
<p>This file contains a design for 25 varieties in six replicates. Each replicate is arranged with 5 row blocks and 5 column blocks.</p>
<p>Define the <code>treatments</code>:</p>
<div class="sourceCode" id="cb260"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb260-1"><a href="Trial-design.html#cb260-1" aria-hidden="true" tabindex="-1"></a>var.code<span class="fl">.25</span><span class="ot">&lt;-</span><span class="fu">factor</span>(<span class="fu">rep</span>(<span class="dv">1</span><span class="sc">:</span><span class="dv">25</span>,<span class="dv">6</span>))</span></code></pre></div>
<p>The target is that each pair of varieties should occur together once in a row block and once in a column block. (With a block size of 5, a variety can occur with 4 other varieties in a block. There are 6 replicates of each variety so there are 6 x 4 = 24 concurrences: hopefully one for every variety pair). As accuracy of comparison between varieties should be greatest when varieties occur in the same block, all pairs of varieties should be compared with equal precision if we achieve the design objective.</p>
<div class="sourceCode" id="cb261"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb261-1"><a href="Trial-design.html#cb261-1" aria-hidden="true" tabindex="-1"></a>latt.sq<span class="fl">.25</span><span class="ot">&lt;-</span> <span class="fu">design</span>(var.code<span class="fl">.25</span>,lattice<span class="fl">.25</span>)</span></code></pre></div>
<p>Efficiencies (or <code>efficiency</code> factors) can be examined as:</p>
<div class="sourceCode" id="cb262"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb262-1"><a href="Trial-design.html#cb262-1" aria-hidden="true" tabindex="-1"></a>latt.sq<span class="sc">$</span>Treatments_model</span></code></pre></div>
<pre><code>##   Treatment.model Model.DF D.Efficiency
## 1    ~ treatments        8            1</code></pre>
<div class="sourceCode" id="cb264"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb264-1"><a href="Trial-design.html#cb264-1" aria-hidden="true" tabindex="-1"></a>latt.sq<span class="fl">.25</span><span class="sc">$</span>Blocks_model</span></code></pre></div>
<pre><code>##        First_order effects  D.Effic  A.Effic       Second_order effects D.Effic
## 1           (reps)       5 1.000000 1.000000           (reps)^2       5      NA
## 2      (reps+rows)      29 0.833333 0.833333      (reps+rows)^2      29      NA
## 3 (reps+rows+cols)      53 0.660474 0.654234 (reps+rows+cols)^2     149      NA
##   A.Effic
## 1      NA
## 2      NA
## 3      NA</code></pre>
<p><code>D-efficiency</code> and <code>A-efficiency</code> are measures of the average precision with which the parameters in the model (for us the variety effects) are estimated. More information can be found on this <a href="http://homepage.divms.uiowa.edu/~gwoodwor/AdvancedDesign/KuhfeldTobiasGarratt.pdf">here</a>.</p>
<p>These are hard to interpret. <code>A-efficiency</code> is related to the average of the variance of the treatment estimates and is usually what we are interested in.</p>

<div class="rmdquiz">
<strong>5</strong> Easier is to examine the design as before. Edit the command for the 3 x 3 lattice to look at the balance of variety concurrences in row and column blocks. Have we reached the optimum design?
</div>
<p>If you think we should improve out attempt:</p>
<div class="sourceCode" id="cb266"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb266-1"><a href="Trial-design.html#cb266-1" aria-hidden="true" tabindex="-1"></a>latt.sq<span class="fl">.25</span><span class="sc">$</span>searches</span></code></pre></div>
<pre><code>## [1] 34</code></pre>
<p>That shows that the computer algorithm had 34 attempts at improving the design.</p>
<p>Overriding this:</p>
<div class="sourceCode" id="cb268"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb268-1"><a href="Trial-design.html#cb268-1" aria-hidden="true" tabindex="-1"></a>latt.sq<span class="fl">.25</span><span class="ot">&lt;-</span> <span class="fu">design</span>(var.code<span class="fl">.25</span>,lattice<span class="fl">.25</span>,<span class="at">searches =</span> <span class="dv">1000</span>)</span></code></pre></div>

<div class="rmdquiz">
<strong>6</strong> Inspect the output. How did we do by adding more searches?
</div>
<p>We have a very good design, particularly if we are mainly concerned in controlling variation in one dimension. But if you need a balanced design for 25 varieties in 6 replicates, with two dimensional blocks, you are better off referring to a published catalogue of optimal designs; though you are in danger or reverting to “Procrustean design” <span class="citation">(<a href="references.html#ref-mead_design_1994" role="doc-biblioref">Mead 1994</a>)</span>. Or run <code>blocksdesign</code> for many more iterations; I eventually succeeded in recreating the balanced lattice.</p>
</div>
<div id="a-partially-replicated-design" class="section level4" number="4.2.2.2">
<h4><span class="header-section-number">4.2.2.2</span> A partially replicated design</h4>
<p>As a final example, we shall return to p-rep and augmented designs: assume we have 36 experimental varieties, but only 12 of them have enough seed for 2 replicates. We want to test all 36. We also have a control variety which we want to test in 12 replications.</p>
<p>We therefore have (12 x 2) + (24 x 1) + (12 x 1) = 60 plots in total. We shall create a nested design with two main blocks of size 30, and 6 sub blocks of size 5.
We are going to use the <code>design</code> function for this. Therefore, we need to form our <code>treatments</code>.</p>

<div class="rmdquiz">
<p><strong>7</strong> Have a go.</p>
<p>Hint = I’ve filled in some of the options below. You have to substitute the correct numbers or parameters for x, y and z:</p>
<code>aug.treats&lt;-factor(c(rep(1:12,x),rep(y:36,1),rep(37,z)))</code>
</div>
<p>Check your answer. (e.g. is it the right length). If you are ever doing this on your own and struggle, just do it in Excel and then read it in. I’ve already created the blocks structure in excel, so you can read it straight in:</p>
<div class="sourceCode" id="cb269"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb269-1"><a href="Trial-design.html#cb269-1" aria-hidden="true" tabindex="-1"></a>aug.blocks<span class="ot">&lt;-</span><span class="fu">read.csv</span>(<span class="st">&quot;aug_blocks.csv&quot;</span>,<span class="at">header=</span>T,<span class="at">colClasses =</span> <span class="st">&quot;factor&quot;</span>)</span></code></pre></div>

<div class="rmdquiz">
<strong>8</strong> Now run the <code>design</code> command:
<code>aug.expt&lt;-design(a,b)</code> # You substitute for a and b
</div>
<p>Extract and examine the block structures again. How does that look?
You should find there is a control occurs in every block. What is the distribution of 2-rep varieties over the sub blocks?</p>
<div class="sourceCode" id="cb270"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb270-1"><a href="Trial-design.html#cb270-1" aria-hidden="true" tabindex="-1"></a><span class="fu">rowMeans</span>(<span class="fu">table</span>(<span class="fu">interaction</span>(aug.expt<span class="sc">$</span>Design<span class="sc">$</span>main_blocks,aug.expt<span class="sc">$</span>Design<span class="sc">$</span>sub_blocks),aug.expt<span class="sc">$</span>Design<span class="sc">$</span>treatments)[,<span class="dv">1</span><span class="sc">:</span><span class="dv">12</span>])<span class="sc">*</span><span class="dv">12</span></span></code></pre></div>
<pre><code>##  1.1  2.1 1.10 2.10 1.11 2.11 1.12 2.12  1.2  2.2  1.3  2.3  1.4  2.4  1.5  2.5 
##    2    0    0    2    0    2    0    2    2    0    2    0    2    0    2    0 
##  1.6  2.6  1.7  2.7  1.8  2.8  1.9  2.9 
##    2    0    0    2    0    2    0    2</code></pre>
<p>The combinations of main-blocks and sub-blocks that have none of the 2-rep varieties don’t exist in the trial. For the ones that do, the 2-rep varieties are spread out well.</p>
<p>It is always worth checking the distribution of entries of blocks appears sensible; there is a risk of error in entering the design information and also a risk that the software you are using has not found the allocation of varieties to blocks that you expected.</p>
<p>You could produce a similar design by hand; you could start with a design for 12 varieties in 2 replicates with 6 sub-blocks of size 2 in each replicate. Then add the control to each sub block. Then add two unreplicated entries to each sub block. This should produce a very similar design. Alternatively, you could produce an alpha design for 30 entries in two replicates and then allocate varieties to codes to end up with the same even distribution.</p>
</div>
<div id="an-augmented-paritially-replicated-design-complicated" class="section level4" number="4.2.2.3">
<h4><span class="header-section-number">4.2.2.3</span> An augmented paritially replicated design: complicated</h4>
<p>The vignette for <code>blocksdesign</code> suggests that if you have unreplicated entries, it may be better to create a design for the replicated entries first, then fill in the unreplicated entries by hand. For example, I tried to create a design with 4 controls in 10 replicates, 55 candidates in 2 replicates and 250 entries in 1 replicate, with two big blocks of size 400 and 20 nested blocks of size 10 in each big block. The target design would have a single control in each of the 40 small blocks – slightly increasing the accuracy of the comparison between controls and candidates at the expense of reducing the accuracy among the four controls.</p>
<p>Running:</p>
<p><code>aug.p.rep&lt;-blocks(treatments = c(4,55,250),replicates = c(10,2,1),blocks=c(2,20),searches=1000)</code></p>
<p>Produced a design in which the four controls concurred once or twice in the small blocks and in which two controls concurred with the two replicate entries 17 times and the other two controls concurred 18 times.</p>
<p>However, re-running the design as:</p>
<p><code>aug.p.rep.2&lt;-blocks(treatments = c(4,55),replicates = c(10,2),blocks=c(2,20),searches=1000)</code></p>
<p>Created a design in which the controls never concurred, but concurred 28, 27, 28 and 29 times with the 2-rep candidates. 30 blocks had 4 entries and 10 had 3. Allocating the 1 replicate entries round up all blocks to size ten would give the desired design. (Entries within blocks would require re-randomising too.) This design will have slightly reduced the accuracy of comparisons among the four controls and increased the accuracy among candidates and between candidates and the controls. There is nothing wrong with the first deign, however, though the second is aesthetically more pleasing.</p>

<div class="rmdnote">
For a multi-location trials series, you can treat each location as a block, and then have nested blocks within locations as before. The locations need not be complete blocks – not all varieties are tested at all locations. This is becoming common in large plant breeding programmes. An advantage of this approach is that genomic prediction methods (discussed next week) can be used to predict the performance of the varieties in the missing locations.
</div>
</div>
</div>
</div>
<div id="farm-scale-experiments" class="section level2" number="4.3">
<h2><span class="header-section-number">4.3</span> Farm scale experiments</h2>
<p>On farm experiments, using farm equipment, are experiencing a revival, partly promoted by better farming equipment with on-harvester yield assessment, GPS, remote sensing and so on. The principles of good experimental design, the three R’s, still apply however. Generally, the experimental unit is large: a fields or half a field, but the number of available experimental units available on a single farm is low. Experimental units within a farm can be treated as a single block. The size of the blocks may vary from farm to farm, however.</p>
<p>As an example, consider the imaginary design in this table:</p>
<table>
<caption><span id="tab:unnamed-chunk-175">Table 4.1: </span>Farm Blocks</caption>
<thead>
<tr class="header">
<th align="left">farm</th>
<th align="left">fields</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">1</td>
<td align="left">1</td>
</tr>
<tr class="even">
<td align="left">1</td>
<td align="left">1</td>
</tr>
<tr class="odd">
<td align="left">1</td>
<td align="left">2</td>
</tr>
<tr class="even">
<td align="left">2</td>
<td align="left">1</td>
</tr>
<tr class="odd">
<td align="left">2</td>
<td align="left">1</td>
</tr>
<tr class="even">
<td align="left">2</td>
<td align="left">2</td>
</tr>
<tr class="odd">
<td align="left">2</td>
<td align="left">2</td>
</tr>
<tr class="even">
<td align="left">2</td>
<td align="left">3</td>
</tr>
<tr class="odd">
<td align="left">2</td>
<td align="left">3</td>
</tr>
<tr class="even">
<td align="left">3</td>
<td align="left">1</td>
</tr>
<tr class="odd">
<td align="left">3</td>
<td align="left">1</td>
</tr>
<tr class="even">
<td align="left">3</td>
<td align="left">2</td>
</tr>
<tr class="odd">
<td align="left">4</td>
<td align="left">1</td>
</tr>
<tr class="even">
<td align="left">4</td>
<td align="left">1</td>
</tr>
<tr class="odd">
<td align="left">4</td>
<td align="left">2</td>
</tr>
<tr class="even">
<td align="left">4</td>
<td align="left">2</td>
</tr>
<tr class="odd">
<td align="left">5</td>
<td align="left">1</td>
</tr>
<tr class="even">
<td align="left">5</td>
<td align="left">1</td>
</tr>
<tr class="odd">
<td align="left">5</td>
<td align="left">2</td>
</tr>
<tr class="even">
<td align="left">5</td>
<td align="left">2</td>
</tr>
<tr class="odd">
<td align="left">6</td>
<td align="left">1</td>
</tr>
<tr class="even">
<td align="left">6</td>
<td align="left">1</td>
</tr>
<tr class="odd">
<td align="left">6</td>
<td align="left">2</td>
</tr>
<tr class="even">
<td align="left">7</td>
<td align="left">1</td>
</tr>
<tr class="odd">
<td align="left">7</td>
<td align="left">1</td>
</tr>
<tr class="even">
<td align="left">7</td>
<td align="left">2</td>
</tr>
<tr class="odd">
<td align="left">7</td>
<td align="left">2</td>
</tr>
<tr class="even">
<td align="left">8</td>
<td align="left">1</td>
</tr>
<tr class="odd">
<td align="left">8</td>
<td align="left">1</td>
</tr>
<tr class="even">
<td align="left">8</td>
<td align="left">2</td>
</tr>
<tr class="odd">
<td align="left">8</td>
<td align="left">2</td>
</tr>
<tr class="even">
<td align="left">8</td>
<td align="left">3</td>
</tr>
<tr class="odd">
<td align="left">9</td>
<td align="left">1</td>
</tr>
<tr class="even">
<td align="left">9</td>
<td align="left">1</td>
</tr>
<tr class="odd">
<td align="left">9</td>
<td align="left">2</td>
</tr>
<tr class="even">
<td align="left">9</td>
<td align="left">2</td>
</tr>
<tr class="odd">
<td align="left">9</td>
<td align="left">3</td>
</tr>
<tr class="even">
<td align="left">10</td>
<td align="left">1</td>
</tr>
<tr class="odd">
<td align="left">10</td>
<td align="left">1</td>
</tr>
<tr class="even">
<td align="left">10</td>
<td align="left">2</td>
</tr>
<tr class="odd">
<td align="left">10</td>
<td align="left">2</td>
</tr>
</tbody>
</table>
<p>There are 10 farms, with between two and three fields available per farm. If a field is big enough, it is split in half – two ‘plots’ per field. This number could be larger for bigger fields, but I’ve kept a maximum of two. There are 41 plots in total: 18 fields with 2 plots and 5 with only 1.</p>
<p>Following the same process as before to create the design:</p>
<p>Suppose we have six varieties to test. Five of these can be replicated seven times and one six times.</p>
<div class="sourceCode" id="cb272"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb272-1"><a href="Trial-design.html#cb272-1" aria-hidden="true" tabindex="-1"></a>farm.var<span class="ot">&lt;-</span><span class="fu">factor</span>(<span class="fu">c</span>(<span class="fu">rep</span>(<span class="dv">1</span><span class="sc">:</span><span class="dv">5</span>,<span class="dv">7</span>),<span class="fu">rep</span>(<span class="dv">6</span>,<span class="dv">6</span>)))</span>
<span id="cb272-2"><a href="Trial-design.html#cb272-2" aria-hidden="true" tabindex="-1"></a>farm.expt<span class="ot">&lt;-</span><span class="fu">design</span>(<span class="at">treatments=</span>farm.var,<span class="at">blocks=</span>farm.blocks,<span class="at">searches=</span><span class="dv">1000</span>)</span>
<span id="cb272-3"><a href="Trial-design.html#cb272-3" aria-hidden="true" tabindex="-1"></a>farm.expt<span class="sc">$</span>Design</span></code></pre></div>
<pre><code>##    farm fields treatments
## 1     1      1          3
## 2     1      1          5
## 3     1      2          4
## 4    10      1          2
## 5    10      1          1
## 6    10      2          5
## 7    10      2          3
## 8     2      1          1
## 9     2      1          4
## 10    2      2          2
## 11    2      2          3
## 12    2      3          6
## 13    2      3          5
## 14    3      1          5
## 15    3      1          4
## 16    3      2          2
## 17    4      1          2
## 18    4      1          6
## 19    4      2          4
## 20    4      2          1
## 21    5      1          6
## 22    5      1          3
## 23    5      2          2
## 24    5      2          4
## 25    6      1          4
## 26    6      1          3
## 27    6      2          1
## 28    7      1          4
## 29    7      1          6
## 30    7      2          5
## 31    7      2          1
## 32    8      1          5
## 33    8      1          2
## 34    8      2          6
## 35    8      2          3
## 36    8      3          1
## 37    9      1          3
## 38    9      1          1
## 39    9      2          5
## 40    9      2          6
## 41    9      3          2</code></pre>

<div class="rmdnote">
If you want to have a go at this yourself, there is a csv file called “farm_trials_design.csv” in your data folder. Or you can inspect the output below and take our word for it.
</div>
<p>Let’s take a look at the results:</p>
<div class="sourceCode" id="cb274"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb274-1"><a href="Trial-design.html#cb274-1" aria-hidden="true" tabindex="-1"></a><span class="fu">table</span>(farm.expt<span class="sc">$</span>Design<span class="sc">$</span>treatments,farm.expt<span class="sc">$</span>Design<span class="sc">$</span>farm)</span></code></pre></div>
<pre><code>##    
##     1 10 2 3 4 5 6 7 8 9
##   1 0  1 1 0 1 0 1 1 1 1
##   2 0  1 1 1 1 1 0 0 1 1
##   3 1  1 1 0 0 1 1 0 1 1
##   4 1  0 1 1 1 1 1 1 0 0
##   5 1  1 1 1 0 0 0 1 1 1
##   6 0  0 1 0 1 1 0 1 1 1</code></pre>
<div class="sourceCode" id="cb276"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb276-1"><a href="Trial-design.html#cb276-1" aria-hidden="true" tabindex="-1"></a><span class="fu">crossprod</span>(<span class="fu">table</span>(farm.expt<span class="sc">$</span>Design<span class="sc">$</span>farm,farm.expt<span class="sc">$</span>Design<span class="sc">$</span>treatments))</span></code></pre></div>
<pre><code>##    
##     1 2 3 4 5 6
##   1 7 5 5 4 5 5
##   2 5 7 5 4 5 5
##   3 5 5 7 4 5 4
##   4 4 4 4 7 4 4
##   5 5 5 5 4 7 4
##   6 5 5 4 4 4 6</code></pre>
<p>The allocation of varieties across farms looks reasonable. What about the allocation within fields?</p>
<div class="sourceCode" id="cb278"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb278-1"><a href="Trial-design.html#cb278-1" aria-hidden="true" tabindex="-1"></a><span class="fu">crossprod</span>(<span class="fu">table</span>(<span class="fu">interaction</span>(farm.expt<span class="sc">$</span>Design<span class="sc">$</span>farm,farm.expt<span class="sc">$</span>Design<span class="sc">$</span>fields),farm.expt<span class="sc">$</span>Design<span class="sc">$</span>treatments))</span></code></pre></div>
<pre><code>##    
##     1 2 3 4 5 6
##   1 7 1 1 2 1 0
##   2 1 7 1 1 1 1
##   3 1 1 7 1 2 2
##   4 2 1 1 7 1 1
##   5 1 1 2 1 7 2
##   6 0 1 2 1 2 6</code></pre>

<div class="rmdquiz">
<strong>9</strong> What do you think? I increased searches to 1,000 because one of my test runs did not do so well. It is quite likely that plot sizes will vary from farm to farm. What effect would this have? Will it invalidate the design?
</div>
</div>
<div id="alpha-designs." class="section level2" number="4.4">
<h2><span class="header-section-number">4.4</span> Alpha designs.</h2>
<div id="creation-of-augmented-and-p-rep-designs-from-2-replicate-alpha-designs" class="section level3" number="4.4.1">
<h3><span class="header-section-number">4.4.1</span> Creation of augmented and p-rep designs from 2-replicate alpha designs</h3>
<p>Alpha designs are forms of incomplete block designs which have proven popular since their introduction in the late 1970’s. They remain the standard design for NL and RL trials in the UK. They are resolvable incomplete block designs. “Resolvable” means that the incomplete blocks can be grouped into complete replicates. This is not necessary from the statistical point of view - non-resolvable incomplete block designs can be very effective - but does make management and scoring of the trial easier. The incomplete blocks give better control of field effects. This is particularly true of very large experiments, as often used in the early stages of testing in breeding programmes. The computer programs to create alpha designs no longer seem to be freely available. A catalogue of designs for number of varieties up to 100 has been published, and can be accessed from within GenStat (and I think within <a href="http://www.edgarweb.org.uk/">EDGAR</a>). For larger number of varieties, the packages <a href="http://designcomputing.net/gendex/alpha/">Gendex</a> or
<a href="https://www.vsni.co.uk/software/cycdesign/">CycDesigN</a> will create the designs, but these are not free, though you may be able to download a test copy. The R package agricolae <span class="citation">(<a href="references.html#ref-R-agricolae" role="doc-biblioref">de Mendiburu 2020</a>)</span> will also produce designs. However, designs for only two replicates are easy to create. These also lend themselves to simple creation of partially replicated designs. For the very large numbers of varieties found in breeders’ trials, it is unlikely that designs with greater than two replicates would be required.</p>
<p><span class="citation"><a href="references.html#ref-patterson_new_1976" role="doc-biblioref">Patterson and Williams</a> (<a href="references.html#ref-patterson_new_1976" role="doc-biblioref">1976</a>)</span> presented simple methods of generating alpha designs in two, three and four replicates, with some constraints. The method for two replicates is that followed described below. The designs are not necessarily optimal but are good and will allow large numbers of varieties. I used to use these extensively.</p>
<p>Alpha designs are not balanced. This is in contrast to many classical incomplete block designs with numbers of varieties which are a complete square (25, 36 and so on). Lack of balance, in this sense, means that not all possible pairs of varieties occur together within an incomplete block. Alpha designs are described as 0,1 designs if pairs of varieties concur once or never. They are described as 0,1,2 designs if, in addition, some pairs of varieties concur twice. 0,1 designs are preferred since these give more even estimation of differences between varieties.</p>
<p>The procedure below will create a 0,1 design for two replicates for any number of varieties <span class="math inline">\(v\)</span>, for which <span class="math inline">\(v = sk\)</span> where <span class="math inline">\(k\)</span> is the block size (number of entries per block) and <span class="math inline">\(s\)</span> is the number of blocks per replicate. Thus, one could produce a design for 200 varieties in <span class="math inline">\(s\)</span>=20 blocks per replicate, each with <span class="math inline">\(k\)</span>=10 entries. There is an additional constraint: the number of entries per block must be less than or equal to the square root of the number of varieties. These constraints are not generally restrictive for large breeders’ trials with block sizes of around 10; a reasonable starting point for large trials. The method below does not create the best possible design: rearrangements of which pairs of varieties concur may make improvements, but it will be pretty close.</p>
<p>The small example below is for 12 varieties in block sizes of 3.</p>
<p>First write down the variety codes for the first replicate as <span class="math inline">\(s\)</span> rows of <span class="math inline">\(k\)</span> entries:</p>
<pre><code>  1 5 9
  2 6 10
  3 7 11
  4 8 12</code></pre>
<p>Thus, the third block in the first replicate contains entries 3, 7 and 11.</p>
<p>Alpha designs are cyclical designs. The meaning of the term will be apparent in the creation of the second replicate. Imagine the columns in the first replicate are numbers on a loop. We leave the first column alone but nudge the next column by one position, the third column by two positions, the fourth (if we had one) by three and so on. This gives:</p>
<pre><code>  1 6 11
  2 7 12
  3 8 9
  4 5 10</code></pre>
<p>You can check that over the eight blocks in the two replicates varieties occur together once (e.g. 1 and 6) or never (e.g. 6 and 7).</p>
<p>You must randomise the order of the blocks within replicates, then the order of the entries within blocks. This is then a perfectly reasonable incomplete block design.
You must also allocate the varieties to the code numbers used in the design at random.</p>
<p>There are some restraints that can be put on this randomisation process however, which can be useful. Note that varieties occurring within the same column never occur in the same block. If we have both control and candidate varieties but our greatest interest is in the candidates, then if we allocate the controls (up to four varieties in the example) to code numbers 1 to 4 (at random) then they never concur. Differences between the controls are then measured with less precision, but consequently the precision of comparisons between controls and candidates is increased.</p>
<p>Equally, suppose we had only one control, but we wished this to have greater replication, then this could be allocated to up to four code numbers (numbers 1 to 4) to give up to eight replicates. Or we could create a design with two controls in four replicates by allocating each control to two of the codes numbers 1 to 4.</p>
<p>Now suppose we have some candidate varieties for which we have seed for only a single replicate. For two such candidates, we allocate both to the same code number, but one goes into replicate 1 and the second into replicate 2. This acts to reduce the variation in comparisons among all the candidates. For successive pairs of un-replicated candidates, we should continue with this allocation procedure by working down the columns of the design matrix. Thus, six un-replicated candidates could be allocated to entries 5, 6 and 7. In an extreme case, we might have a single control in eight replicates and 16 un-replicated candidates. From the design above, this would give:</p>
<pre><code>  1 5a 9a 
  1 6a 10a
  1 7a 11a
  1 8a 12a    

  1 6b 11b
  1 7b 12b
  1 8b 9b
  1 5b 10b</code></pre>
<p>Here, code 1 is the control, 5a and 5b are completely different candidates, as are 6a and 6b etc.</p>
<p>This design should be randomised: first blocks within replicates then position within blocks. An example randomisation gave me:</p>
<pre><code>  10a 1  6a
  1  11a 7a
  12a 1  8a
  1  5a  9a
    
  7b 12b 1
  5b 10b 1
  1  11b 6b
  9b  1  8b </code></pre>
<p>For this example, we didn’t need to go through the alpha-design procedure: the only alternative is to arrange the controls systematically. Note we should allocate our candidates to the codes at random: they are almost certainly grouped in some sort of order, by cross or by breeder perhaps, and we should disrupt this patterns to avoid risk of bias.</p>
<p>With larger numbers of candidates and larger block sizes, other arrangements are possible. For example, with block sizes of 12 (requiring a minimum of 144 entries for a 0,1 design), then there will be 12 columns in the design matrix. All entries in the first column could be allocated to one control variety and all entries in the next column to another. This would ensure that each block contained each control once. The candidates could then be allocated to the remaining codes in one or two replicates as described above. A similar end point could be obtained by designing an alpha design with a block size of 10 and then adding the two controls to each block independently of the design. Some caution is required. With no control varieties, if only a small number of candidates are present in two replicates, then the incomplete block analysis may fail: insufficient replicated entries may concur to allow estimation of all block effects. In such a case, analysis as a randomised complete block or by spatial analysis would still be possible.</p>
<p>Regardless of the number of candidates and controls, the candidates must be allocated at random to the candidate codes and the controls at random to the control codes. Then blocks must be randomised within replicates and plot positions within blocks.</p>
</div>
<div id="final-message" class="section level3" number="4.4.2">
<h3><span class="header-section-number">4.4.2</span> Final message</h3>
<p>Four important reminders:</p>
<ol style="list-style-type: decimal">
<li><p>Always save your design and back it up. Keep notes: don’t assume you will remember how you created the design months later, when you have some data to analyse.</p></li>
<li><p><code>blocksdesign</code> and other software, will carry out the randomisation for you. However, most software does not allocate varieties to the code numbers used in the design at random. You should do this yourself if there is any systematic pattern to the list of varieties: for example varieties are listed by parentage. This could, for example, result in the difference between varieties with the same parents being estimated with greater parentage than those with different parents, which may not be what you want. If in doubt <strong>randomise</strong>.</p></li>
<li><p>It is perfectly acceptable to use the same design for multiple experiments, but not the same randomisation. Without additional randomisation, you may end up, again, with the difference between some pairs of varieties being estimated more precisely than others. Moreover, the analysis will not take account of this: it will happen without you knowing.</p></li>
<li><p>Please: always randomise the first replicate. I don’t get around much anymore, but many breeders and official testing authorities used to lay out the first replicate in variety order. This makes it easy of visitors and breeders to inspect varieties, but invalidates the design for the same reasons given in (2) and (3) above. Pity poor variety number one. It will always be in one corner of the first replicate; nearest the gate, and more likely to get trampled on and run-over than the others. It was probably drilled first and will be harvested first; when human and machine errors are more likely. If you want demonstration plots, plant or sow them in a separate block.</p></li>
</ol>

</div>
</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="R-and-stats.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="Trial-analysis.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": "https://github.com/rstudio/bookdown-demo/edit/master/04-Trial_design.Rmd",
"text": "Edit"
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": ["bookdown-demo.pdf", "bookdown-demo.epub"],
"toc": {
"collapse": "subsection"
}
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
